{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9836beda",
   "metadata": {},
   "source": [
    "---\n",
    "title: \"BERTopic on Philosophy of Biology (from Philpapers)\"\n",
    "date: 2025-04-28\n",
    "author:\n",
    "  - name: Jacob Hamel-Mottiez\n",
    "    id: jc\n",
    "    orcid: 0009-0007-3666-908X\n",
    "    email: jacob.hamel-mottiez.1@ulaval.ca\n",
    "    affiliation: \n",
    "      - name: Laval University\n",
    "\n",
    "execute: \n",
    "  enabled: true # This is so that Plotly is rendered. \n",
    "format: \n",
    "  html:\n",
    "    code-fold: false\n",
    "keywords:\n",
    "  - Philosophy of Biology\n",
    "  - Biology\n",
    "  - Bibliometrics\n",
    "  - Topic modeling\n",
    "  - BERTopic\n",
    "\n",
    "theme:\n",
    "  #light: flatly\n",
    "  dark : darkly\n",
    "\n",
    "\n",
    "license: \"CC BY\"\n",
    "copyright: \n",
    "  holder: Jacob Hamel-Mottiez\n",
    "  year: 2024\n",
    "funding: \"The author received funding from the Social Sciences and Humanities Canadian Reseach Council (SSHCRC) as well as from the Fonds de recherche du Qu√©bec - Soci√©t√© et culture.\"\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1e08704d-ac10-497b-9af0-7ee192f0669b",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Directory\n",
    "PATH_TO_DATA = r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\DATA\\\\\"\n",
    "PATH_TO_VIZ = r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\biophilo\\Visualisation\\\\\"\n",
    "# Packages to import. \n",
    "import pandas as pd \n",
    "import numpy as np\n",
    "import datamapplot\n",
    "\n",
    "\n",
    "from sentence_transformers import SentenceTransformer\n",
    "from bertopic import BERTopic\n",
    "from umap import UMAP"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "688725a8",
   "metadata": {},
   "source": [
    "## üìÑ Data\n",
    "This data is from *Biology and Philosophy* journal from 1986 to 2023. I removed all the entries with no abstract. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "df7204a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = pd.read_csv(\"C:\\\\Users\\\\jacob\\\\OneDrive - Universit√© Laval\\\\biophilo\\\\Data\\\\PHILOSOPHY_OF_BIOLOGY_ALL.csv\")\n",
    "#df = df.fillna(\"NULL\")\n",
    "\n",
    "#test\n",
    "df_0 = pd.read_csv(r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\biophilo\\Data\\2025-03-25_archive\\PHILPAPERS_PHILO_DATA\\philpapers_data_pyblio.csv\")\n",
    "df_0 = df_0.fillna(\"NULL\")\n",
    "\n",
    "df_01 = pd.read_csv(r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\biophilo\\Data\\2025-03-25_archive\\PHILPAPERS_PHILO_DATA\\unclassified_philpapers_data_pyblio.csv\")\n",
    "df_01 = df_01.fillna(\"NULL\")\n",
    "\n",
    "df = pd.concat([df_0, df_01])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "4c8c0053",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Unnamed: 0",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "doi",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pii",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pubmed_id",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "title",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtype",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtypeDescription",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "creator",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "afid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affilname",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_city",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_country",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_count",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "author_names",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_ids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_afids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "coverDate",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "coverDisplayDate",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "publicationName",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "source_id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eIssn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "aggregationType",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "volume",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issueIdentifier",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "article_number",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pageRange",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "description",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "authkeywords",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "citedby_count",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "openaccess",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "freetoread",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "freetoreadLabel",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_acr",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_no",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_sponsor",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "e083e202-a561-42a2-8c40-1d423a9c93f8",
       "rows": [
        [
         "0",
         "0",
         "2-s2.0-85209763600",
         "10.1007/s11229-024-04815-5",
         "NULL",
         "NULL",
         "Driftability and niche construction",
         "ar",
         "Article",
         "F√°bregas-Tejeda A.",
         "60025063",
         "KU Leuven",
         "Leuven",
         "Belgium",
         "2.0",
         "F√°bregas-Tejeda, Alejandro;Ramsey, Grant",
         "57196375680;14521522000",
         "60025063;60025063",
         "2024-12-01",
         "December 2024",
         "Synthese",
         "00397857",
         "27699",
         "15730964",
         "Journal",
         "204",
         "6",
         "162",
         "NULL",
         "Niche construction is the process of organisms changing themselves or their environment‚Äîor their relationship with their environment‚Äîin ways that affect the evolutionary trajectory of their population. These evolutionary trajectory changes are traditionally understood to be triggered by changes in selection pressures. Niche construction thus necessarily involves organisms altering selection pressures. In this paper, we argue that changes in selection pressures is not the only way organisms can influence the evolutionary futures of their population. We propose that organisms can also affect drift probabilities, and that such changes should be considered niche construction. Drift probabilities can be modulated by altering population size or by affecting driftability (individual variance in possible reproductive outcomes). We consider both and provide examples of how niche construction can stabilize, increase, or dampen drift probabilities. Finally, we revisit and broaden the traditional definition of niche construction. We hold that organismic activities that modify drift probabilities should count as niche construction, even if selection pressures remain unaltered.",
         "Drift | Driftability | Evolutionary causation | Niche construction | Selection",
         "0",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "FWO",
         "G070122N",
         "Fonds Wetenschappelijk Onderzoek"
        ],
        [
         "1",
         "1",
         "2-s2.0-85202899987",
         "10.1007/s10539-024-09957-x",
         "NULL",
         "NULL",
         "Explanatory gaps in evolutionary theory",
         "ar",
         "Article",
         "Aaby B.H.",
         "60025063;60010348",
         "KU Leuven;Universitetet i Oslo",
         "Leuven;Oslo",
         "Belgium;Norway",
         "3.0",
         "Aaby, Bendik Hellem;Dani, Gianmaria;Ramsey, Grant",
         "57262975700;59309831600;14521522000",
         "60010348;60025063;60025063",
         "2024-10-01",
         "October 2024",
         "Biology and Philosophy",
         "01693867",
         "13699",
         "15728404",
         "Journal",
         "39",
         "5",
         "22",
         "NULL",
         "Proponents of the extended evolutionary synthesis have argued that there are explanatory gaps in evolutionary biology that cannot be bridged by standard evolutionary theory. In this paper, we consider what sort of explanatory gaps they are referring to. We outline three possibilities: data-based gaps, implementation-based gaps, and framework-based gaps. We then examine the purported evolutionary gaps and attempt to classify them using this taxonomy. From there we reconsider the significance of the gaps and what they imply for the proposed need for an extended evolutionary synthesis.",
         "Evolutionary theory | Explanatory gap | Natural selection | Selection explanation | The active role of the organism | The extended synthesis",
         "0",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "3H210777",
         "KU Leuven"
        ],
        [
         "2",
         "2",
         "2-s2.0-85175425123",
         "10.1016/j.shpsa.2023.10.006",
         "S0039368123001449",
         "37907020.0",
         "A Wolf in Sheep's Clothing: Idealisations and the aims of polygenic scores",
         "ar",
         "Article",
         "Serpico D.",
         "60021361;60015986",
         "Uniwersytet Jagiello≈Ñski;Universit√† di Trento",
         "Krakow;Trento",
         "Poland;Italy",
         "1.0",
         "Serpico, Davide",
         "57197799705",
         "60015986-60021361",
         "2023-12-01",
         "December 2023",
         "Studies in History and Philosophy of Science",
         "00393681",
         "11600154632",
         "18792510",
         "Journal",
         "102",
         "NULL",
         "NULL",
         "72-83",
         "Research in pharmacogenomics and precision medicine has recently introduced the concept of Polygenic Scores (PGSs), namely, indexes that aggregate the effects that many genetic variants are predicted to have on individual disease risk. The popularity of PGSs is increasing rapidly, but surprisingly little attention has been paid to the idealisations they make about phenotypic development. Indeed, PGSs rely on quantitative genetics models and methods, which involve considerable theoretical assumptions that have been questioned on various grounds. This comes with epistemological and ethical concerns about the use of PGSs in clinical decision-making. In this paper, I investigate to what extent idealisations in genetics models can impact the data gathering and clinical interpretation of genomics findings, particularly the calculation and predictive accuracy of PGSs. Although idealisations are considered ineliminable components of scientific models, they may be legitimate or not depending on the epistemic aims of a model. I thus analyse how various idealisations have been introduced in classical models and progressively readapted throughout the history of genetic theorising. Notably, this process involved important changes in the epistemic purpose of such idealisations, which raises the question of whether they are legitimate in the context of contemporary genomics.",
         "Genetic prediction | GWAS | Idealisations | Polygenic scores | Predictive accuracy | Quantitative genetics",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "ERC",
         "805498",
         "European Research Council"
        ],
        [
         "3",
         "3",
         "2-s2.0-85152302150",
         "10.1016/j.shpsa.2023.03.002",
         "S0039368123000705",
         "37068423.0",
         "Joint representation: Modeling a phenomenon with multiple biological systems",
         "ar",
         "Article",
         "Yoshida Y.",
         "60029445",
         "University of Minnesota Twin Cities",
         "Minneapolis",
         "United States",
         "1.0",
         "Yoshida, Yoshinari",
         "57211241255",
         "60029445",
         "2023-06-01",
         "June 2023",
         "Studies in History and Philosophy of Science",
         "00393681",
         "11600154632",
         "18792510",
         "Journal",
         "99",
         "NULL",
         "NULL",
         "67-76",
         "Biologists often study particular biological systems as models of a phenomenon of interest even if they already know that the phenomenon is produced by diverse mechanisms and hence none of those systems alone can sufficiently represent it. To understand this modeling practice, the present paper provides an account of how multiple model systems can be used to study a phenomenon that is produced by diverse mechanisms. Even if generalizability of results from a single model system is significantly limited, generalizations concerning specific aspects of mechanisms often hold across certain ranges of biological systems, which enables multiple model systems to jointly represent such a phenomenon. Comparing mechanisms that operate in different biological systems as examples of the same phenomenon also facilitates characterization and investigation of individual mechanisms. I also compare my account with two existing accounts of the use of multiple model systems and argue that my account is distinct from and complementary to them.",
         "Collective cell migration | Generalization | Model organisms | Model systems | Phenomena | Representation",
         "0",
         "0",
         "repositoryvor",
         "Green",
         "UMN",
         "undefined",
         "University of Minnesota"
        ],
        [
         "4",
         "4",
         "2-s2.0-85148606416",
         "10.3390/philosophies8010008",
         "NULL",
         "NULL",
         "Turing‚Äôs Biological Philosophy: Morphogenesis, Mechanisms and Organicism",
         "ar",
         "Article",
         "Greif H.",
         "60003675",
         "Politechnika Warszawska",
         "Warsaw",
         "Poland",
         "3.0",
         "Greif, Hajo;Kubiak, Adam;Stacewicz, Pawe≈Ç",
         "35487486100;57222530470;55497157400",
         "60003675;60003675;60003675",
         "2023-02-01",
         "February 2023",
         "Philosophies",
         "NULL",
         "21101055468",
         "24099287",
         "Journal",
         "8",
         "1",
         "8",
         "NULL",
         "Alan M. Turing‚Äôs last published work and some posthumously published manuscripts were dedicated to the development of his theory of organic pattern formation. In ‚ÄúThe Chemical Basis of Morphogenesis‚Äù (1952), he provided an elaborated mathematical formulation of the theory of the origins of biological form that had been first proposed by Sir D‚ÄôArcy Wendworth Thompson in On Growth and Form (1917/1942). While arguably his most mathematically detailed and his systematically most ambitious effort, Turing‚Äôs morphogenetical writings also form the most thematically self-contained and least philosophically explored part of his work. We dedicate our inquiry to the reasons and the implications of Turing‚Äôs choice of biological topic and viewpoint. We will probe for possible factors in Turing‚Äôs choice that go beyond availability and acquaintance with On Growth and Form. On these grounds, we will explore how and to what extent his theory of morphogenesis actually ties in with his concept of mechanistic computation. Notably, Thompson‚Äôs pioneering work in biological ‚Äòstructuralism‚Äô was organicist in outlook and explicitly critical of the Darwinian approaches that were popular with Turing‚Äôs cyberneticist contemporaries‚Äîand partly used by Turing himself in his proto-connectionist models of learning. Resolving this apparent dichotomy, we demonstrate how Turing‚Äôs quest for mechanistic explanations of how organisation emerges in nature leaves room for a non-mechanist view of nature.",
         "Alan M. Turing | Darwinian evolution | D‚ÄôArcy Wendworth Thompson | explanations in science | mechanism | modern synthesis | morphogenesis | organicism",
         "1",
         "1",
         "publisherfullgold",
         "Gold",
         "NCN",
         "2020/37/B/HS1/01809",
         "Narodowe Centrum Nauki"
        ],
        [
         "5",
         "5",
         "2-s2.0-85091151894",
         "10.1007/s11245-020-09722-1",
         "NULL",
         "NULL",
         "From Obesity to Energy Metabolism: Ontological Perspectives on the Metrics of Human Bodies",
         "ar",
         "Article",
         "Serpico D.",
         "60030318;60025153",
         "Universit√† degli Studi di Milano;Universit√† degli Studi di Genova",
         "Milan;Genoa",
         "Italy;Italy",
         "2.0",
         "Serpico, Davide;Borghini, Andrea",
         "57197799705;57045736900",
         "60025153;60030318",
         "2021-07-01",
         "July 2021",
         "Topoi",
         "01677411",
         "4700152866",
         "15728749",
         "Journal",
         "40",
         "3",
         "NULL",
         "577-586",
         "In this paper, we aim at rethinking the concept of obesity in a way that better captures the connection between underlying medical aspects, on the one hand, and an individual‚Äôs developmental history, on the other. Our proposal rests on the idea that obesity is not to be understood as a phenotypic trait or character; rather, obesity represents one of the many possible states of a more complex phenotypic trait that we call ‚Äòenergy metabolism.‚Äô We argue that this apparently simple conceptual shift can help solve important theoretical misconceptions regarding the genetics, epigenetics, and development of obesity. In addition, we show that our proposal can be fruitfully paired with the concept of developmental channeling of a trait, which connects to the study of the plasticity and canalization of complex traits. Finally, we discuss the potential impact of our approach on the assessment, treatment, and social narratives of obesity.",
         "Definition of obesity | Developmental canalization | Epigenetics of obesity | Genetics of obesity | Obesity | Obesity and public health",
         "9",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "UniMi",
         "undefined",
         "Universit√† degli Studi di Milano"
        ],
        [
         "6",
         "6",
         "2-s2.0-85100567668",
         "10.5944/ENDOXA.46.2020.28849",
         "NULL",
         "NULL",
         "The war of nature, famine, and death as design source for living beings",
         "re",
         "Review",
         "Airoldi G.",
         "60028711",
         "Universidad Nacional de Educacion a Distancia",
         "Madrid",
         "Spain",
         "2.0",
         "Airoldi, Giorgio;Saborido, Cristian",
         "57221908301;35273296700",
         "60028711;60028711",
         "2021-01-01",
         "2021",
         "Endoxa",
         "11335351",
         "21100457408",
         "21745676",
         "Journal",
         "2020",
         "46",
         "NULL",
         "123-164",
         "Alan Grafen's 'Formal Darwinism project' proposes a mathematical formulation of Darwin's theory that seeks to demonstrate that natural selection shapes phenotypic traits through the maximization of fitness. Grafen's project is based on three premises: (1) natural selection is the only force that shapes phenotypes; (2) fitness is the only measure of evolution; and (3) biological design arises as a result of a process of selective optimization. In this paper, we argue that these three premises limit the applicability of the model to the simplest evolutionary facts. This is because Grafen implicitly presupposes a concept of fitness that is both cause and effect of phenotypic innovations; which, on the one hand, empties Darwin's theory of explanatory power and, on the other hand, leads to overlooking the potential contribution of non-selective evolutionary forces in the shaping of biological complexity. To overcome these pitfalls of the Formal Darwinism Project - which are also common to all formal adaptationist models - we propose to replace the causal concept of fitness with the concept of robustness, thus defining biological design as form and function rather than as maximization of fitness.",
         "Biological design | Evolution | Grafen | Natural selection | Philosophy of biology",
         "0",
         "1",
         "publisherfullgold",
         "Gold",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "7",
         "7",
         "2-s2.0-85084479260",
         "10.1007/s10539-020-09750-6",
         "NULL",
         "NULL",
         "Beyond quantitative and qualitative traits: three telling cases in the life sciences",
         "ar",
         "Article",
         "Serpico D.",
         "60025153",
         "Universit√† degli Studi di Genova",
         "Genoa",
         "Italy",
         "1.0",
         "Serpico, Davide",
         "57197799705",
         "60025153",
         "2020-06-01",
         "1 June 2020",
         "Biology and Philosophy",
         "01693867",
         "13699",
         "15728404",
         "Journal",
         "35",
         "3",
         "34",
         "NULL",
         "This paper challenges the common assumption that some phenotypic traits are quantitative while others are qualitative. The distinction between these two kinds of traits is widely influential in biological and biomedical research as well as in scientific education and communication. This is probably due to both historical and epistemological reasons. However, the quantitative/qualitative distinction involves a variety of simplifications on the genetic causes of phenotypic variability and on the development of complex traits. Here, I examine three cases from the life sciences that show inconsistencies in the distinction: Mendelian traits (dwarfism and pigmentation in plant and animal models), Mendelian diseases (phenylketonuria), and polygenic mental disorders (schizophrenia). I show that these traits can be framed both quantitatively and qualitatively depending, for instance, on the methods through which they are investigated and on specific epistemic purposes (e.g., clinical diagnosis versus causal explanation). This suggests that the received view of quantitative and qualitative traits has a limited heuristic power‚Äîlimited to some local contexts or to the specific methodologies adopted. Throughout the paper, I provide directions for framing phenotypes beyond the quantitative/qualitative distinction. I conclude by pointing at the necessity of developing a principled characterisation of what phenotypic traits, in general, are.",
         "Character states | Genetic liability | Mendelian diseases | Qualitative traits | Quantitative traits | Thresholds",
         "16",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "Universit√† degli Studi di Genova"
        ],
        [
         "8",
         "8",
         "2-s2.0-85064435478",
         "10.1007/s10441-019-09344-0",
         "NULL",
         "30949871.0",
         "Control by Viability in a Chemotherapy Cancer Model",
         "ar",
         "Article",
         "Serhani M.",
         "60025457;60015943",
         "Hassan II University of Casablanca;Universit√© Moulay Isma√Øl",
         "Casablanca;Meknes",
         "Morocco;Morocco",
         "4.0",
         "Serhani, M.;Essaadi, H.;Kassara, K.;Boutoulout, A.",
         "6506389979;57208322026;6603249996;6506165183",
         "60015943;60015943;60025457;60015943",
         "2019-09-15",
         "15 September 2019",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "67",
         "3",
         "NULL",
         "177-200",
         "The aim of this study is to provide a feedback control, called the Chemotherapy Protocol Law, with the purpose to keep the density of tumor cells that are treated by chemotherapy below a ‚Äútolerance level‚Äù Lc, while retaining the density of normal cells above a ‚Äúhealthy level‚Äù Nc. The mathematical model is a controlled dynamical system involving three nonlinear differential equations, based on a Gompertzian law of cell growth. By evoking viability and set-valued theories, we derive sufficient conditions for the existence of a Chemotherapy Protocol Law. Thereafter, on a suitable viability domain, we build a multifunction whose selections are the required Chemotherapy Protocol Laws. Finally, we propose a design of selection that generates a Chemotherapy Protocol Law.",
         "Cancer model | Feedback control | Set-valued analysis | Viability",
         "5",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "9",
         "9",
         "2-s2.0-85131712913",
         "10.1007/s13752-019-00317-7",
         "NULL",
         "NULL",
         "Multicellular Individuality: The Case of Bacteria",
         "ar",
         "Article",
         "Ventura R.",
         "60014808;60008724",
         "Bilkent √úniversitesi;Duke University",
         "Ankara;Durham",
         "Turkey;United States",
         "1.0",
         "Ventura, Rafael",
         "57195947053",
         "60008724-60014808",
         "2019-06-15",
         "15 June 2019",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "14",
         "2",
         "NULL",
         "131-140",
         "Recent attention to complex group-level behavior amongst bacteria has led some to conceive of multicellular clusters of bacteria as individuals. In this article, I assess these recent claims by first drawing a distinction between two concepts of individuality: physiological and evolutionary. I then survey cases that are representative of three different modes of growth: myxobacteria (surface-attached agglomerative growth), Bacillus subtilis (agglomerative growth not attached to a surface), and cyanobacteria (filamentous growth). A closer look at these cases indicates that multicellular individuality among bacteria is remarkably complex. Physiologically, the three cases of multicellular clusters do not form physiological individuals. But matters are different when it comes to evolutionary individuality; although multicellular clusters that grow by agglomeration are not highly individuated, filament-forming groups achieve a relatively high degree of individuality. I also suggest that debates about bacterial multicellular individuality may have been obscured by a failure to see that selection on highly individuated groups is by no means the only mechanism to bring about the complex group-level behaviors that have led some to view bacteria as multicellular individuals.",
         "B. subtilis | Bacteria | Cyanobacteria | Individuality | Multicellularity | Myxobacteria",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "NSFC",
         "12171212",
         "National Natural Science Foundation of China"
        ],
        [
         "10",
         "10",
         "2-s2.0-85064948392",
         "10.1002/bies.201800251",
         "NULL",
         "30970170.0",
         "Why the Lipid Divide? Membrane Proteins as Drivers of the Split between the Lipids of the Three Domains of Life",
         "ar",
         "Article",
         "Sojo V.",
         "112777647",
         "Institute for Advanced Study (Wissenschaftskolleg) Berlin",
         "Berlin",
         "Germany",
         "1.0",
         "Sojo, Victor",
         "57091567800",
         "112777647",
         "2019-05-01",
         "May 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "5",
         "1800251",
         "NULL",
         "Recent results from engineered and natural samples show that the starkly different lipids of archaea and bacteria can form stable hybrid membranes. But if the two types can mix, why don't they? That is, why do most bacteria and all eukaryotes have only typically bacterial lipids, and archaea archaeal lipids? It is suggested here that the reason may lie on the other main component of cellular membranes: membrane proteins, and their close adaptation to the lipids. Archaeal lipids in modern bacteria could suggest that the last universal common ancestor (LUCA) had both lipid types. However, this would imply a rather elaborate evolutionary scenario, while negating simpler alternatives. In light of widespread horizontal gene transfer across the prokaryotic domains, hybrid membranes reveal that the lipid divide did not just occur once at the divergence of archaea and bacteria from LUCA. Instead, it continues to occur actively to this day.",
         "archaea | bacteria | divergence | eukaryotes | evolution | horizontal gene transfer | last universal common ancestor (LUCA)",
         "26",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "ALTF-1455-2015",
         "Simons Foundation"
        ],
        [
         "11",
         "11",
         "2-s2.0-85064907987",
         "10.1002/bies.201800260",
         "NULL",
         "30970156.0",
         "How Does Inflammation-Induced Hyperglycemia Cause Mitochondrial Dysfunction in Immune Cells?",
         "re",
         "Review",
         "van Niekerk G.",
         "60001565",
         "Stellenbosch University",
         "Stellenbosch",
         "South Africa",
         "4.0",
         "van Niekerk, Gustav;Davis, Tanja;Patterton, Hugh George;Engelbrecht, Anna Mart",
         "55212609500;56377885400;6701836615;7004364572",
         "60001565;60001565;60001565;60001565",
         "2019-05-01",
         "May 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "5",
         "1800260",
         "NULL",
         "Inflammatory mediators have an established role in inducing insulin resistance and promoting hyperglycemia. In turn, hyperglycemia has been argued to drive immune cell dysfunction as a result of mitochondrial dysfunction. Here, the authors review the evidence challenging this view. First, it is pointed out that inflammatory mediators are known to induce altered mitochondrial function. In this regard, critical care patients suffer both an elevated inflammatory tone as well as hyperglycemia, rendering it difficult to distinguish between the effects of inflammation and hyperglycemia. Second, emerging evidence indicates that a decrease in mitochondrial respiration and an increase in reactive oxygen species (ROS) production are not necessarily manifestations of pathology, but adaptations taking shape as the mitochondria is abdicating its adenosine triphosphate (ATP)-producing function (which is taken over by glycolysis) and instead becomes ‚Äúretooled‚Äù for an immunological role. Collectively, these observations challenge the commonly held belief that acute¬†hyperglycemia induces mitochondrial damage leading to immune cell dysfunction.",
         "glucose toxicity | hyperglycemia | immunometabolism | inflammation | insulin resistance | mitochondria",
         "18",
         "0",
         "NULL",
         "NULL",
         "SAMRC",
         "undefined",
         "South African Medical Research Council"
        ],
        [
         "12",
         "12",
         "2-s2.0-85064907924",
         "10.1002/bies.201800234",
         "NULL",
         "30970171.0",
         "Targeting the Spleen as an Alternative Site for Hematopoiesis",
         "ar",
         "Article",
         "Short C.",
         "60031602",
         "Bond University",
         "Gold Coast",
         "Australia",
         "4.0",
         "Short, Christie;Lim, Hong K.;Tan, Jonathan;O'Neill, Helen C.",
         "57208484540;57207580388;56328599000;7102156014",
         "60031602;60031602;60031602;60031602",
         "2019-05-01",
         "May 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "5",
         "1800234",
         "NULL",
         "Bone marrow is the main site for hematopoiesis in adults. It acts as a niche for hematopoietic stem cells (HSCs) and contains non-hematopoietic cells that contribute to stem cell dormancy, quiescence, self-renewal, and differentiation. HSC also exist in resting spleen of several species, although their contribution to hematopoiesis under steady-state conditions is unknown. The spleen can however undergo extramedullary hematopoiesis (EMH) triggered by physiological stress or disease. With the loss of bone marrow niches in aging and disease, the spleen as an alternative tissue site for hematopoiesis is an important consideration for future therapy, particularly during HSC transplantation. In terms of harnessing the spleen as a site for hematopoiesis, here the remarkable regenerative capacity of the spleen is considered with a view to forming additional or ectopic spleen tissue through cell engraftment. Studies in mice indicate the potential for such grafts to support the influx of hematopoietic cells leading to the development of normal spleen architecture. An important goal will be the formation of functional ectopic spleen tissue as an aid to hematopoietic recovery following clinical treatments that impact bone marrow. For example, expansion or replacement of niches could be considered where myeloablation ahead of HSC transplantation compromises treatment outcomes.",
         "extramedullary hematopoiesis | hematopoietic stem cells | myelopoiesis | spleen | stem cell niche | tissue regeneration | transplantation",
         "34",
         "0",
         "repositoryam",
         "Green",
         "ARC",
         "130101703",
         "Australian Research Council"
        ],
        [
         "13",
         "13",
         "2-s2.0-85064391368",
         "10.1007/s13752-018-0311-0",
         "NULL",
         "NULL",
         "The Nature of Programmed Cell Death",
         "ar",
         "Article",
         "Durand P.M.",
         "60025063;60016218",
         "KU Leuven;University of the Witwatersrand, Johannesburg",
         "Leuven;Johannesburg",
         "Belgium;South Africa",
         "2.0",
         "Durand, Pierre M.;Ramsey, Grant",
         "24343664500;14521522000",
         "60016218;60025063",
         "2019-03-15",
         "15 March 2019",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "14",
         "1",
         "NULL",
         "30-41",
         "In multicellular organisms, cells are frequently programmed to die. This makes good sense: cells that fail to, or are no longer playing important roles are eliminated. From the cell‚Äôs perspective, this also makes sense, since somatic cells in multicellular organisms require the cooperation of clonal relatives. In unicellular organisms, however, programmed cell death (PCD) poses a difficult and unresolved evolutionary problem. The empirical evidence for PCD in diverse microbial taxa has spurred debates about what precisely PCD means in the case of unicellular organisms (how it should be defined). In this article, we survey the concepts of PCD in the literature and the selective pressures associated with its evolution. We show that definitions of PCD have been almost entirely mechanistic and fail to separate questions concerning what PCD fundamentally is from questions about the kinds of mechanisms that realize PCD. We conclude that an evolutionary definition is best able to distinguish PCD from closely related phenomena. Specifically, we define ‚Äútrue‚Äù PCD as an adaptation for death triggered by abiotic or biotic environmental stresses. True PCD is thus not only an evolutionary product but must also have been a target of selection. Apparent PCD resulting from pleiotropy, genetic drift, or trade-offs is not true PCD. We call this ‚Äúersatz PCD‚Äù.",
         "Adaptation | Aging | Apoptosis | Price equation | Programmed cell death | Selection | Unicellular organisms",
         "33",
         "0",
         "repositoryam",
         "Green",
         "NRF",
         "21772155",
         "National Research Foundation"
        ],
        [
         "14",
         "14",
         "2-s2.0-85061982513",
         "10.1002/bies.201800222",
         "NULL",
         "30793782.0",
         "Cell Fate and Developmental Regulation Dynamics by Polycomb Proteins and 3D Genome Architecture",
         "re",
         "Review",
         "Loubiere V.",
         "60108488",
         "Universit√© de Montpellier",
         "Montpellier",
         "France",
         "3.0",
         "Loubiere, Vincent;Martinez, Anne Marie;Cavalli, Giacomo",
         "56084980400;57198919744;56233803600",
         "60108488;60108488;60108488",
         "2019-03-01",
         "March 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "3",
         "1800222",
         "NULL",
         "Targeted transitions in chromatin states at thousands of genes are essential drivers of eukaryotic development. Therefore, understanding the in vivo dynamics of epigenetic regulators is crucial for deciphering the mechanisms underpinning cell fate decisions. This review illustrates how, in addition to its cell memory function, the Polycomb group of transcriptional regulators orchestrates temporal, cell and tissue-specific expression of master genes during development. These highly sophisticated developmental transitions are dependent on the context- and tissue-specific assembly of the different types of Polycomb Group (PcG) complexes, which regulates their targeting and/or activities on chromatin. Here, an overview is provided of how PcG complexes function at multiple scales to regulate transcription, local chromatin environment, and higher order structures that support normal differentiation and are perturbed in tumorigenesis.",
         "3D genome | chromatin | Polycomb",
         "37",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "H2020",
         "232947",
         "Horizon 2020 Framework Programme"
        ],
        [
         "15",
         "15",
         "2-s2.0-85051028218",
         "10.1002/bies.201800028",
         "NULL",
         "30058076.0",
         "Developmental Roles and Evolutionary Significance of AMPA-Type Glutamate Receptors",
         "re",
         "Review",
         "Hirai S.",
         "60025997;60002783",
         "Keio University;Tokyo Metropolitan Institute of Medical Science",
         "Tokyo;Tokyo",
         "Japan;Japan",
         "3.0",
         "Hirai, Shinobu;Hotta, Kohji;Okado, Haruo",
         "23488802700;7202730545;7006007995",
         "60002783;60025997;60002783",
         "2018-09-01",
         "September 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "9",
         "1800028",
         "NULL",
         "Organogenesis and metamorphosis require the intricate orchestration of multiple types of cellular interactions and signaling pathways. Glutamate (Glu) is an excitatory extracellular signaling molecule in the nervous system, while Ca 2+ is a major intracellular signaling molecule. The first Glu receptors to be cloned are Ca 2+ -permeable receptors in mammalian brains. Although recent studies have focused on Glu signaling in synaptic mechanisms of the mammalian central nervous system, it is unclear how this signaling functions in development. Our recent article demonstrated that Ca 2+ -permeable AMPA-type Glu receptors (GluAs) are essential for formation of a photosensitive organ, development of some neurons, and metamorphosis, including tail absorption and body axis rotation, in ascidian embryos. Based on findings in these embryos and mammalian brains, we formed several hypotheses regarding the evolution of GluAs, the non-synaptic function of Glu, the origin of GluA-positive neurons, and the neuronal network that controls metamorphosis in ascidians.",
         "AMPA-type receptor | ascidian development | calcium permeability | evolution | glutamate receptor | metamorphosis | organ formation",
         "3",
         "0",
         "NULL",
         "NULL",
         "JSPS",
         "16H01451",
         "Japan Society for the Promotion of Science"
        ],
        [
         "16",
         "16",
         "2-s2.0-85050402571",
         "10.1002/bies.201800005",
         "NULL",
         "29901833.0",
         "The Energy Maintenance Theory of Aging: Maintaining Energy Metabolism to Allow Longevity",
         "re",
         "Review",
         "Chaudhari S.",
         "60029747",
         "University of Georgia",
         "Athens",
         "United States",
         "2.0",
         "Chaudhari, Snehal N.;Kipreos, Edward T.",
         "56536140500;35586915000",
         "60029747;60029747",
         "2018-08-01",
         "August 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "8",
         "1800005",
         "NULL",
         "Fused, elongated mitochondria are more efficient in generating ATP than fragmented mitochondria. In diverse C. elegans longevity pathways, increased levels of fused mitochondria are associated with lifespan extension. Blocking mitochondrial fusion in these animals abolishes their extended longevity. The long-lived C. elegans vhl-1 mutant is an exception that does not have increased fused mitochondria, and is not dependent on fusion for longevity. Loss of mammalian VHL upregulates alternate energy generating pathways. This suggests that mitochondrial fusion facilitates longevity in C. elegans by increasing energy metabolism. In diverse animals, ATP levels broadly decreases with age. Substantial evidence supports the theory that increasing or maintaining energy metabolism promotes the survival of older animals. Increased ATP levels in older animals allow energy-intensive repair and homeostatic mechanisms such as proteostasis that act to prevent cellular aging. These observations support the emerging paradigm that maintaining energy metabolism promotes the survival of older animals.",
         "C. elegans | energy metabolism | lifespan | mitochondria | mitochondrial fusion",
         "25",
         "0",
         "repositoryam",
         "Green",
         "NIGMS",
         "R01-GM074212",
         "National Institute of General Medical Sciences"
        ],
        [
         "17",
         "17",
         "2-s2.0-85038079138",
         "10.1002/bies.201700148",
         "NULL",
         "29251357.0",
         "Epigenetic and Transcriptional Variability Shape Phenotypic Plasticity",
         "re",
         "Review",
         "Ecker S.",
         "60109234;60097745;60032907;60031101;60005517",
         "UCL Cancer Institute;Centro Nacional de Supercomputaci√≥n;Instituci√≥ Catalana de Recerca i Estudis Avan√ßats;University of Cambridge;Wellcome Sanger Institute",
         "London;Barcelona;Barcelona;Cambridge;Hinxton",
         "United Kingdom;Spain;Spain;United Kingdom;United Kingdom",
         "5.0",
         "Ecker, Simone;Pancaldi, Vera;Valencia, Alfonso;Beck, Stephan;Paul, Dirk S.",
         "55383063200;16043223200;16308427600;57193033325;42062149600",
         "60109234;60097745;60097745-60032907;60109234;60031101-60005517",
         "2018-02-01",
         "February 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "2",
         "1700148",
         "NULL",
         "Epigenetic and transcriptional variability contribute to the vast diversity of cellular and organismal phenotypes and are key in human health and disease. In this review, we describe different types, sources, and determinants of epigenetic and transcriptional variability, enabling cells and organisms to adapt and evolve to a changing environment. We highlight the latest research and hypotheses on how chromatin structure and the epigenome influence gene expression variability. Further, we provide an overview of challenges in the analysis of biological variability. An improved understanding of the molecular mechanisms underlying epigenetic and transcriptional variability, at both the intra- and inter-individual level, provides great opportunity for disease prevention, better therapeutic approaches, and personalized medicine.",
         "3D chromatin structure | biological noise | chromatin | DNA methylation | epigenetics | gene expression | genome architecture | heterogeneity | transcription | variability",
         "58",
         "1",
         "repositoryvor",
         "Green",
         "H2020",
         "733161",
         "Horizon 2020 Framework Programme"
        ],
        [
         "18",
         "18",
         "2-s2.0-85035356179",
         "10.1002/bies.201700126",
         "NULL",
         "29058339.0",
         "Linking Mitochondria and Synaptic Transmission: The CB1 Receptor",
         "re",
         "Review",
         "Djeungoue-Petga M.A.",
         "60009972",
         "Universit√© de Moncton",
         "Moncton",
         "Canada",
         "2.0",
         "Djeungoue-Petga, Marie Ange;Hebert-Chatelain, Etienne",
         "57197852383;36613183500",
         "60009972;60009972",
         "2017-12-01",
         "December 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "12",
         "1700126",
         "NULL",
         "CB1 receptors are functionally present within brain mitochondria (mtCB1), although they are usually considered specifically targeted to plasma membrane. Acute activation of mtCB1 alters mitochondrial ATP generation, synaptic transmission, and memory performance. However, the detailed mechanism linking disrupted mitochondrial metabolism and synaptic transmission is still uncharacterized. CB1 receptors are among the most abundant G protein-coupled receptors in the brain and impact on several processes, including fear coping, anxiety, stress, learning, and memory. Mitochondria perform several key physiological processes for neuronal homeostasis, including production of ATP and reactive oxygen species, calcium buffering, metabolism of neurotransmitters, and apoptosis. It is therefore possible that acute activation of mtCB1 impacts on these different mitochondrial functions to modulate synaptic transmission. In reviewing and integrating across the literature in this area, we describe the possible mechanisms involved in the regulation of brain physiology by mtCB1 receptors.",
         "cannabinoids | CB1 receptors | mitochondria | synaptic transmission",
         "34",
         "0",
         "NULL",
         "NULL",
         "NSERC",
         "undefined",
         "Fondation Brain Canada"
        ],
        [
         "19",
         "19",
         "2-s2.0-84912025857",
         "10.1016/j.shpsc.2014.08.003",
         "S1369848614001083",
         "25220402.0",
         "The machine conception of the organism in development andevolution: A critical analysis",
         "ar",
         "Article",
         "Nicholson D.J.",
         "60026479",
         "University of Exeter",
         "Exeter",
         "United Kingdom",
         "1.0",
         "Nicholson, Daniel J.",
         "36189968300",
         "60026479",
         "2014-12-01",
         "December 01, 2014",
         "Studies in History and Philosophy of Science Part C :Studies in History and Philosophy of Biological and Biomedical Sciences",
         "13698486",
         "14281",
         "NULL",
         "Journal",
         "48",
         "PB",
         "NULL",
         "162-174",
         "This article critically examines one of the most prevalent metaphors in contemporary biology, namely the machine conception of the organism (MCO). Although the fundamental differences between organisms and machines make the MCO an inadequate metaphor for conceptualizing living systems, many biologists and philosophers continue to draw upon the MCO or tacitly accept it as the standard model of the organism. The analysis presented here focuses on the specific difficulties that arise when the MCO is invoked in the contexts of development and evolution. In developmental biology the MCO underlies a logically incoherent model of ontogeny, the genetic program, which serves to legitimate three problematic theses about development: genetic animism, neo-preformationism, and developmental computability. In evolutionary biology the MCO is responsible for grounding unwarranted theoretical appeals to the concept of design as well as to the interpretation of natural selection as an engineer, which promote a distorted understanding of the process and products of evolutionary change. Overall, it is argued that, despite its heuristic value, the MCO today is impeding rather than enabling further progress in our comprehension of living systems.",
         "Design | Engineering | Genetic program | Machine | Metaphor | Organism",
         "54",
         "0",
         "repositoryvor",
         "Green",
         "FP7",
         "324186",
         "Seventh Framework Programme"
        ],
        [
         "20",
         "20",
         "2-s2.0-85062109763",
         "10.1002/bies.201800198",
         "NULL",
         "30805984.0",
         "Veins and Arteries Build Hierarchical Branching Patterns Differently: Bottom-Up versus Top-Down",
         "ar",
         "Article",
         "Red-Horse K.",
         "60012708;60003711",
         "Stanford University;University of Pennsylvania Perelman School of Medicine",
         "Stanford;Philadelphia",
         "United States;United States",
         "2.0",
         "Red-Horse, Kristy;Siekmann, Arndt F.",
         "6505998386;15923667100",
         "60012708;60003711",
         "2019-03-01",
         "March 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "3",
         "1800198",
         "NULL",
         "A tree-like hierarchical branching structure is present in many biological systems, such as the kidney, lung, mammary gland, and blood vessels. Most of these organs form through branching morphogenesis, where outward growth results in smaller and smaller branches. However, the blood vasculature is unique in that it exists as two trees (arterial and venous) connected at their tips. Obtaining this organization might therefore require unique developmental mechanisms. As reviewed here, recent data indicate that arterial trees often form in reverse order. Accordingly, initial arterial endothelial cell differentiation occurs outside of arterial vessels. These pre-artery cells then build trees by following a migratory path from smaller into larger arteries, a process guided by the forces imparted by blood flow. Thus, in comparison to other branched organs, arteries can obtain their structure through inward growth and coalescence. Here, new information on the underlying mechanisms is discussed, and how defects can lead to pathologies, such as hypoplastic arteries and arteriovenous malformations.",
         "angiogenesis | arteries | blood flow | branching morphogenesis | coronary vasculature | cxcr4 | notch | vein",
         "52",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "EXC 1003-CIM",
         "National Institutes of Health"
        ],
        [
         "21",
         "21",
         "2-s2.0-85057869591",
         "10.1002/bies.201800165",
         "NULL",
         "30485463.0",
         "Cell-Cycle-Dependent Regulation of Cell Adhesions: Adhering to the Schedule: Three papers reveal unexpected properties of adhesion structures as cells progress through the cell cycle",
         "re",
         "Review",
         "Li Y.",
         "60020469",
         "UNC School of Medicine",
         "Chapel Hill",
         "United States",
         "2.0",
         "Li, Yitong;Burridge, Keith",
         "57204923786;7006564391",
         "60020469;60020469",
         "2019-01-01",
         "January 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "1",
         "1800165",
         "NULL",
         "Focal adhesions disassemble during mitosis, but surprisingly little is known about how these structures respond to other phases of the cell cycle. Three recent papers reveal unexpected results as they examine adhesions through the cell cycle. A biphasic response is detected where focal adhesions grow during S phase before disassembly begins early in G2. In M phase, activated integrins at the tips of retraction fibers anchor mitotic cells, but these adhesions lack the defining components of focal adhesions, such as talin, paxillin, and zyxin. Re-examining cell-matrix adhesion reveals reticular adhesions, a new class of adhesion. These Œ±VŒ≤5 integrin-mediated adhesions also lack conventional focal adhesion components and anchor mitotic cells to the extracellular matrix. As reviewed here, these studies present insight into how adhesion complexes vary through the cell cycle, and how unconventional adhesions maintain attachment during mitosis while providing spatial memory to guide daughter cell re-spreading after cell division.",
         "CDK1 | cell cycle | focal adhesions | integrin | mitosis | reticular adhesions",
         "16",
         "1",
         "NULL",
         "NULL",
         "NIH",
         "GM029860",
         "National Institutes of Health"
        ],
        [
         "23",
         "23",
         "2-s2.0-85053250759",
         "10.1002/bies.201800063",
         "NULL",
         "30159904.0",
         "TGF-Œ≤ Control of Adaptive Immune Tolerance: A Break From Treg Cells",
         "re",
         "Review",
         "Liu M.",
         "60009343",
         "Memorial Sloan-Kettering Cancer Center",
         "New York",
         "United States",
         "3.0",
         "Liu, Ming;Li, Shun;Li, Ming O.",
         "57192271252;57203866886;37069515500",
         "60009343;60009343;60009343",
         "2018-11-01",
         "November 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "11",
         "1800063",
         "NULL",
         "The vertebrate adaptive immune system has well defined functions in maintaining tolerance to self-tissues. Suppression of autoreactive T cells is dependent on the regulatory cytokine transforming growth factor-Œ≤ (TGF-Œ≤) and regulatory T (Treg) cells, a distinct T cell lineage specified by the transcription factor Foxp3. Although TGF-Œ≤ promotes thymic Treg (tTreg) cell development by repressing T cell clonal deletion and peripheral Treg cell differentiation by inducing Foxp3 expression, a recent study shows that TGF-Œ≤ suppresses autoreactive T cells independent of Foxp3+ Treg cells. These findings imply that as an ancestral growth factor family member, TGF-Œ≤ may have been co-opted as a T cell-intrinsic mechanism of self-tolerance control to assist the evolutionary transition of vertebrate adaptive immunity. Later, perhaps in placental mammals upon their acquisition of a TGF-Œ≤ regulatory element in the Foxp3 locus, the TGF-Œ≤ pathway is further engaged to induce peripheral Treg cell differentiation and expand the scope of T cell tolerance control to innocuous foreign antigens.",
         "adaptive immunity | autoimmunity | evolution | Foxp3 | immune tolerance | TGF-Œ≤ | Treg",
         "73",
         "0",
         "repositoryam",
         "Green",
         "HHMI",
         "P30 CA008748",
         "Howard Hughes Medical Institute"
        ],
        [
         "24",
         "24",
         "2-s2.0-85053191157",
         "10.1002/bies.201800107",
         "NULL",
         "30151860.0",
         "An Emerging System to Study Photosymbiosis, Brain Regeneration, Chronobiology, and Behavior: The Marine Acoel Symsagittifera roscoffensis",
         "re",
         "Review",
         "Arboleda E.",
         "60032907;60027550;60026803;60024631;60009470",
         "Instituci√≥ Catalana de Recerca i Estudis Avan√ßats;University of California, Los Angeles;Station Biologique de Roscoff;University of Fribourg;Howard Hughes Medical Institute",
         "Barcelona;Los Angeles;Roscoff;Fribourg;Chevy Chase",
         "Spain;United States;France;Switzerland;United States",
         "7.0",
         "Arboleda, Enrique;Hartenstein, Volker;Martinez, Pedro;Reichert, Heinrich;Sen, Sonia;Sprecher, Simon;Bailly, Xavier",
         "15062382400;7005813125;24431870500;7006118438;57203857935;57214445626;57195594575",
         "60026803;60027550;60032907;60024631;60009470;60024631;60026803",
         "2018-10-01",
         "October 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "10",
         "1800107",
         "NULL",
         "The acoel worm Symsagittifera roscoffensis, an early offshoot of the Bilateria and the only well-studied marine acoel that lives in a photosymbiotic relationship, exhibits a centralized nervous system, brain regeneration, and a wide repertoire of complex behaviors such as circatidal rhythmicity, photo/geotaxis, and social interactions. While this animal can be collected by the thousands and is studied historically, significant progress is made over the last decade to develop it as an emerging marine model. The authors here present the feasibility of culturing it in the laboratory and describe the progress made on different areas, including genomic and tissue architectures, highlighting the associated challenges. In light of these developments, and on the ability to access abundant synchronized embryos, the authors put forward S. roscoffensis as a marine system to revisit questions in the areas of photosymbiosis, regeneration, chronobiology, and the study of complex behaviors from a molecular and evolutionary perspective.",
         "emerging model system | endosymbiosis | holobiome | microinjection | neoblasts | photosymbiosis | Tetraselmis convolutae",
         "24",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "UMR 7144",
         "R√©gion Bretagne"
        ],
        [
         "25",
         "25",
         "2-s2.0-85051870330",
         "10.1002/bies.201800032",
         "NULL",
         "30080263.0",
         "The Actomyosin Cytoskeleton Drives Micron-Scale Membrane Remodeling In Vivo Via the Generation of Mechanical Forces to Balance Membrane Tension Gradients",
         "re",
         "Review",
         "Ebrahim S.",
         "60013409;60006577;60005281",
         "National Cancer Institute (NCI);National Institutes of Health (NIH);National Heart, Lung, and Blood Institute (NHLBI)",
         "Rockville;Bethesda;Bethesda",
         "United States;United States;United States",
         "3.0",
         "Ebrahim, Seham;Liu, Jian;Weigert, Roberto",
         "55636954500;55189387900;55949380100",
         "60013409-60006577;60005281;60013409",
         "2018-09-01",
         "September 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "9",
         "1800032",
         "NULL",
         "The remodeling of biological membranes is crucial for a vast number of cellular activities and is an inherently multiscale process in both time and space. Seminal work has provided important insights into nanometer-scale membrane deformations, and highlighted the remarkable variation and complexity in the underlying molecular machineries and mechanisms. However, how membranes are remodeled at the micron-scale, particularly in vivo, remains poorly understood. Here, we discuss how using regulated exocytosis of large (1.5‚Äì2.0 Œºm) membrane-bound secretory granules in the salivary gland of live mice as a model system, has provided evidence for the importance of the actomyosin cytoskeleton in micron-scale membrane remodeling in physiological conditions. We highlight some of these advances, and present mechanistic hypotheses for how the various biochemical and biophysical properties of distinct actomyosin networks may drive this process.",
         "actomyosin | cytoskeleton | in vivo | intravital microscopy | membrane remodeling | micron-scale | regulated exocytosis",
         "10",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "ZIA BC 011682",
         "National Institutes of Health"
        ],
        [
         "26",
         "26",
         "2-s2.0-85051856791",
         "10.1002/bies.201800073",
         "NULL",
         "29943439.0",
         "Sex Differences in Early Embryogenesis: Inter-Chromosomal Regulation Sets the Stage for Sex-Biased Gene Networks: The dialogue between the sex chromosomes and autosomes imposes sexual identity soon after fertilization",
         "re",
         "Review",
         "Engel N.",
         "60013936",
         "Lewis Katz School of Medicine",
         "Philadelphia",
         "United States",
         "1.0",
         "Engel, Nora",
         "8355663600",
         "60013936",
         "2018-09-01",
         "September 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "9",
         "1800073",
         "NULL",
         "Sex-specific transcriptional and epigenomic profiles are detectable in the embryo very soon after fertilization. I propose that in male (XY) and female (XX) pre-implantation embryos sex chromosomes establish sexually dimorphic interactions with the autosomes, before overt differences become apparent and long before gonadogenesis. Lineage determination restricts expression biases between the sexes, but the epigenetic differences are less constrained and can be perpetuated, accounting for dimorphisms that arise later in life. In this way, sexual identity is registered in the epigenome very early in development. As development progresses, sex-specific regulatory modules are harbored within shared transcriptional networks that delineate common traits. In reviewing this field, I propose that analyzing the mechanisms for sexual dimorphisms at the molecular and biochemical level and incorporating developmental and environmental factors will lead to a greater understanding of sex differences in health and disease. Also see the video abstract here: https://youtu.be/9BPlbrHtkHQ.",
         "determination | epigenetics | lineage | preimplantation | sex chromosomes | sexual dimorphisms | transcriptional regulation",
         "28",
         "0",
         "NULL",
         "NULL",
         "TU",
         "2017",
         "Temple University"
        ],
        [
         "27",
         "27",
         "2-s2.0-85050404954",
         "10.1002/bies.201800017",
         "NULL",
         "29938810.0",
         "Cells in the Non-Uniform Magnetic World: How Cells Respond to High-Gradient Magnetic Fields",
         "re",
         "Review",
         "Zablotskii V.",
         "60016576",
         "Institute of Physics of the Czech Academy of Sciences",
         "Prague",
         "Czech Republic",
         "3.0",
         "Zablotskii, Vitalii;Polyakova, Tatyana;Dejneka, Alexandr",
         "7003428180;7003596710;25924871900",
         "60016576;60016576;60016576",
         "2018-08-01",
         "August 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "8",
         "1800017",
         "NULL",
         "Imagine cells that live in a high-gradient magnetic field (HGMF). Through what mechanisms do the cells sense a non-uniform magnetic field and how such a field changes the cell fate? We show that magnetic forces generated by HGMFs can be comparable to intracellular forces and therefore may be capable of altering the functionality of an individual cell and tissues in unprecedented ways. We identify the cellular effectors of such fields and propose novel routes in cell biology predicting new biological effects such as magnetic control of cell-to-cell communication and vesicle transport, magnetic control of intracellular ROS levels, magnetically induced differentiation of stem cells, magnetically assisted cell division, or prevention of cells from dividing. On the basis of experimental facts and theoretical modeling we reveal timescales of cellular responses to high-gradient magnetic fields and suggest an explicit dependence of the cell response time on the magnitude of the magnetic field gradient.",
         "cell differentiation | cell fate | cell signaling | intracellular forces | magnetic fields | magnetic gradient",
         "42",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "28",
         "28",
         "2-s2.0-85050402614",
         "10.1002/bies.201800036",
         "NULL",
         "29944192.0",
         "Archaea-First and the Co-Evolutionary Diversification of Domains of Life",
         "re",
         "Review",
         "Staley J.T.",
         "60015481;60000745",
         "University of Washington;University of Illinois Urbana-Champaign",
         "Seattle;Urbana",
         "United States;United States",
         "2.0",
         "Staley, James T.;Caetano-Anoll√©s, Gustavo",
         "35584784400;56269658000",
         "60015481;60000745",
         "2018-08-01",
         "August 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "8",
         "1800036",
         "NULL",
         "The origins and evolution of the Archaea, Bacteria, and Eukarya remain controversial. Phylogenomic-wide studies of molecular features that are evolutionarily conserved, such as protein structural domains, suggest Archaea is the first domain of life to diversify from a stem line of descent. This line embodies the last universal common ancestor of cellular life. Here, we propose that ancestors of Euryarchaeota co-evolved with those of Bacteria prior to the diversification of Eukarya. This co-evolutionary scenario is supported by comparative genomic and phylogenomic analyses of the distributions of fold families of domains in the proteomes of free-living organisms, which show horizontal gene recruitments and informational process homologies. It also benefits from the molecular study of cell physiologies responsible for membrane phospholipids, methanogenesis, methane oxidation, cell division, gas vesicles, and the cell wall. Our theory however challenges popular cell fusion and two-domain of life scenarios derived from sequence analysis, demanding phylogenetic reconciliation. Also see the video abstract here: https://youtu.be/9yVWn_Q9faY.",
         "Crenarchaeota | Euryarchaota | fold families | horizontal transfer | molecular evolution | protein structure | stem line",
         "14",
         "0",
         "NULL",
         "NULL",
         "NIFA",
         "1014249",
         "National Institute of Food and Agriculture"
        ],
        [
         "29",
         "29",
         "2-s2.0-85050402130",
         "10.1002/bies.201800038",
         "NULL",
         "29975812.0",
         "Asymmetric Segregation of Aged Spindle Pole Bodies During Cell Division: Mechanisms and Relevance Beyond Budding Yeast?",
         "re",
         "Review",
         "Lengefeld J.",
         "60025858;60022195",
         "ETH Z√ºrich;Massachusetts Institute of Technology",
         "Zurich;Cambridge",
         "Switzerland;United States",
         "2.0",
         "Lengefeld, Jette;Barral, Yves",
         "55178282100;6603668360",
         "60025858-60022195;60025858",
         "2018-08-01",
         "August 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "8",
         "1800038",
         "NULL",
         "Asymmetric cell division generates cell diversity and contributes to cellular aging and rejuvenation. Here, we review the molecular mechanisms enabling budding yeast to recognize spindle pole bodies (SPB, centrosome equivalent) based on their age, and guide their non-random mitotic segregation: SPB inheritance requires the distinction of old from new SPBs and is regulated by the SPB-inheritance network (SPIN) and the mitotic exit network (MEN). The SPIN marks the pre-existing SPB as old and the MEN recognizes these marks translating them into spindle orientation. We next revisit other molecules and structures that partition depending on their age rather than their abundance at mitosis as, for example, DNA, centrosomes, mitochondria, and histones in yeast and other systems. The recurrence of this differential behavior suggests a functional significance for numerous cell types, which we then discuss. We conclude that non-random segregation may facilitate asymmetric cell fate determination and thereby indirectly aging and rejuvenation. Also see the video abstract here: https://youtu.be/1sQ4rAomnWY.",
         "age-dependent inheritance | aging and rejuvenation | asymmetric cell division | centrosome | SPB | SPB-Inheritance Network (SPIN)",
         "14",
         "1",
         "repositoryvor",
         "Green",
         "ERC",
         "172142",
         "H2020 European Research Council"
        ],
        [
         "30",
         "30",
         "2-s2.0-85050388561",
         "10.1002/bies.201700132",
         "NULL",
         "29882973.0",
         "Primary Cilia Reconsidered in the Context of Ciliopathies: Extraciliary and Ciliary Functions of Cilia Proteins Converge on a Polarity theme?",
         "re",
         "Review",
         "Hua K.",
         "60007977",
         "Albany Medical College",
         "Albany",
         "United States",
         "2.0",
         "Hua, Kiet;Ferland, Russell J.",
         "55966076900;7004332316",
         "60007977;60007977",
         "2018-08-01",
         "August 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "8",
         "1700132",
         "NULL",
         "Once dismissed as vestigial organelles, primary cilia have garnered the interest of scientists, given their importance in development/signaling, and for their implication in a new disease category known as ciliopathies. However, many, if not all, ‚Äúcilia‚Äù proteins also have locations/functions outside of the primary cilium. These extraciliary functions can complicate the interpretation of a particular ciliopathy phenotype: it may be a result of defects at the cilium and/or at extraciliary locations, and it could be broadly related to a unifying cellular process for these proteins, such as polarity. Assembly of a cilium has many similarities to the development of other polarized structures. This evolutionarily preserved process for the assembly of polarized cell structures offers a perspective on how the cilium may have evolved. We hypothesize that cilia proteins are critical for cell polarity, and that core polarity proteins may have been specialized to form various cellular protrusions, including primary cilia.",
         "ciliopathy | cytoskeleton | development | extraciliary | polarity | primary cilia | trafficking",
         "9",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "R01NS092062",
         "National Institutes of Health"
        ],
        [
         "31",
         "31",
         "2-s2.0-85048795662",
         "10.1002/bies.201800031",
         "NULL",
         "29869337.0",
         "Synapse Pruning: Mitochondrial ROS with Their Hands on the Shears",
         "ar",
         "Article",
         "Cobley J.",
         "60105228",
         "University of the Highlands and Islands",
         "Inverness",
         "United Kingdom",
         "1.0",
         "Cobley, James N.",
         "54583377000",
         "60105228",
         "2018-07-01",
         "July 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "7",
         "1800031",
         "NULL",
         "No overarching hypotheses tie the basic mechanisms of mitochondrial reactive oxygen species (ROS) production to activity dependent synapse pruning‚Äîa fundamental biological process in health and disease. Neuronal activity divergently regulates mitochondrial ROS: activity decreases whereas inactivity increases their production, respectively. Placing mitochondrial ROS as innate synaptic activity sentinels informs the novel hypothesis that: (1) at an inactive synapse, increased mitochondrial ROS production initiates intrinsic apoptosis dependent pruning; and (2) at an active synapse, decreased mitochondrial ROS production masks intrinsic apoptosis dependent pruning. Immature antioxidant defense may enable the developing brain to harness mitochondrial ROS to prune weak synapses. Beyond development, endogenous antioxidant defense constrains mitochondrial (ROS) to mask pruning. Unwanted age-related synapse loss may arise when mitochondrial ROS aberrantly recapitulate developmental pruning. Placing mitochondrial ROS with their hands on the shears is beneficial in early but deleterious in later life.",
         "development | hydrogen peroxide | mitochondria | neurodegeneration | superoxide | synapse pruning",
         "24",
         "0",
         "NULL",
         "NULL",
         "HIE",
         "JNC007",
         "Highlands and Islands Enterprise"
        ],
        [
         "32",
         "32",
         "2-s2.0-85046087503",
         "10.1002/bies.201800013",
         "NULL",
         "29709074.0",
         "Mitochondrial Outer Membrane Channels: Emerging Diversity in Transport Processes",
         "re",
         "Review",
         "Becker T.",
         "60101437;60025641;60016458",
         "Albert Ludwigs Universit√§t Freiburg, Medizinische Fakult√§t;Universit√§t Freiburg;Constructor University",
         "Freiburg im Breisgau;Freiburg im Breisgau;Bremen",
         "Germany;Germany;Germany",
         "2.0",
         "Becker, Thomas;Wagner, Richard",
         "58310706500;57192995019",
         "60101437-60025641;60016458",
         "2018-07-01",
         "July 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "7",
         "1800013",
         "NULL",
         "Mitochondrial function and biogenesis depend on the transport of a large variety of proteins, ions, and metabolites across the two surrounding membranes. While several specific transporters are present in the inner membrane, transport processes across the outer membrane are less understood. Recent studies reveal that the number of outer membrane channels and their transport mechanisms are more diverse than originally thought. Four protein-conducting channels promote transport of distinct sets of precursor proteins across and into the outer membrane. The voltage-dependent anion channel (VDAC) forms the major channel for small hydrophilic molecules. In addition, three channels with yet unknown substrate specificity exist in the outer membrane. In this review, we outline the emerging functional diversity, selectivity, and regulation of mitochondrial outer membrane channels. The presence of several channel-forming proteins challenges the traditional view that the outer membrane forms an unspecific size-exclusion filter for the flux of small hydrophilic molecules.",
         "MIM complex | mitochondria | SAM complex | TOM complex | VDAC",
         "49",
         "0",
         "NULL",
         "NULL",
         "DFG",
         "BE4679/2-1",
         "Deutsche Forschungsgemeinschaft"
        ],
        [
         "33",
         "33",
         "2-s2.0-85048711745",
         "10.1002/bies.201800008",
         "NULL",
         "29708272.0",
         "Three Distinct Types of Microautophagy Based on Membrane Dynamics and Molecular Machineries",
         "ar",
         "Article",
         "Oku M.",
         "60120432;60011001",
         "Center for the Promotion of Interdisciplinary Education and Research;Kyoto University",
         "Kyoto;Kyoto",
         "Japan;Japan",
         "2.0",
         "Oku, Masahide;Sakai, Yasuyoshi",
         "7005419880;7403476299",
         "60011001;60011001-60120432",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1800008",
         "NULL",
         "Microautophagy is originally defined as lysosomal (vacuolar) membrane dynamics to directly enwrap and transport cytosolic components into the lumen of the lytic organelle. Molecular details of microautophagy had remained unknown until genetic studies in yeast identified a set of proteins required for the process. Subsequent studies with other experimental model organisms resulted in a series of discoveries that accompanied an expansion of the definition of microautophagy to also encompass endosomal membrane dynamics. These findings, however, still impose puzzling, non-integrated images as to the molecular mechanism of microautophagy. By reviewing recent studies on microautophagy in various experimental systems, we propose the classification of microautophagy into three types, as the basis for developing a comprehensive view of the process.",
         "autophagy | lysosome | macroautophagy | microautophagy | vacuole",
         "190",
         "1",
         "repositoryvor",
         "Green",
         "JSPS",
         "16K07689",
         "Japan Society for the Promotion of Science"
        ],
        [
         "34",
         "34",
         "2-s2.0-85048710809",
         "10.1002/bies.201700232",
         "NULL",
         "29682771.0",
         "Novel Channels of the Outer Membrane of Mitochondria: Recent Discoveries Change Our View",
         "ar",
         "Article",
         "Checchetto V.",
         "60000481",
         "Universit√† degli Studi di Padova",
         "Padua",
         "Italy",
         "2.0",
         "Checchetto, Vanessa;Szabo, Ildiko",
         "36459522800;35595280300",
         "60000481;60000481",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1700232",
         "NULL",
         "Ion channels mediate ion flux across biological membranes and regulate important organellar and cellular tasks. A recent study revealed the presence of four new proteins, the MIM complex (composed by Mim1 and Mim2), Ayr1, OMC7, and OMC8, that are able to form ion-conducting channels in the outer mitochondria membrane (OMM). These findings strongly indicate that the OMM is endowed with many solute-specific channels, in addition to porins and known channels mediating protein import into mitochondria. These solute-specific channels provide essential pathways for the controlled transport of ions and metabolites and may thus add a further layer of specificity to the regulation of mitochondrial function at the organelle-cytosol and/or inter-organellar interface. Future studies will be required to fully understand the way(s) of regulation of these new channels and to integrate them into signaling pathways within the cells.",
         "in vitro expression | ion channels | mitochondria | outer membrane | planar lipid bilayer | Œ≤-barrel proteins",
         "12",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "35",
         "35",
         "2-s2.0-85048702395",
         "10.1002/bies.201700245",
         "NULL",
         "29750374.0",
         "Endothelial Metabolic Control of Lymphangiogenesis",
         "ar",
         "Article",
         "Yu P.",
         "60069758;60017994",
         "Changhai Hospital;Yale School of Medicine",
         "Shanghai;New Haven",
         "China;United States",
         "4.0",
         "Yu, Pengchun;Wu, Guosheng;Lee, Heon Woo;Simons, Michael",
         "35799246300;56342554600;13002676700;7201538867",
         "60017994;60017994-60069758;60017994;60017994",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1700245",
         "NULL",
         "Lymphangiogenesis is an important developmental process that is critical to regulation of fluid homeostasis, immune surveillance and response as well as pathogenesis of a number of diseases, among them cancer, inflammation, and heart failure. Specification, formation, and maturation of lymphatic blood vessels involves an interplay between a series of events orchestrated by various transcription factors that determine expression of key genes involved in lymphangiogenesis. These are traditionally thought to be under control of several key growth factors including vascular growth factor-C (VEGF-C) and fibroblast growth factors (FGFs). Recent insights into VEGF and FGF signaling point to their role in control of endothelial metabolic processes such as glycolysis and fatty acid oxidation that, in turn, play a major role in regulation of lymphangiogenesis. These advances have significantly increased our understanding of lymphatic biology and opened new therapeutic vistas. Here we review our current understanding of metabolic controls in the lymphatic vasculature.",
         "endothelial cells | fatty acid Œ≤-oxidation | glycolysis | growth factor signaling | lymphangiogenesis | lymphatics | metabolism",
         "12",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "HL053793",
         "National Institutes of Health"
        ],
        [
         "36",
         "36",
         "2-s2.0-85044604721",
         "10.1002/bies.201700231",
         "NULL",
         "29603284.0",
         "Demarcation of Viral Shelters Results in Destruction by Membranolytic GTPases: Antiviral Function of Autophagy Proteins and Interferon-Inducible GTPases",
         "re",
         "Review",
         "Brown H.M.",
         "60029278",
         "The University of Chicago",
         "Chicago",
         "United States",
         "5.0",
         "Brown, Hailey M.;Biering, Scott B.;Zhu, Allen;Choi, Jayoung;Hwang, Seungmin",
         "57194634980;57190299860;59436101100;56126661000;7404626305",
         "60029278;60029278;60029278;60029278;60029278",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1700231",
         "NULL",
         "A hallmark of positive-sense RNA viruses is the formation of membranous shelters for safe replication in the cytoplasm. Once considered invisible to the immune system, these viral shelters are now found to be antagonized through the cooperation of autophagy proteins and anti-microbial GTPases. This coordinated effort of autophagy proteins guiding GTPases functions against not only the shelters of viruses but also cytoplasmic vacuoles containing bacteria or protozoa, suggesting a broad immune-defense mechanism against disparate vacuolar pathogens. Fundamental questions regarding this process remain: how the host recognizes these membranous structures as a target, how the autophagy proteins bring the GTPases to the shelters, and how the recruited GTPases disrupt these shelters. In this review, these questions are discussed, the answers to which will significantly advance our understanding of the response to vacuole-like structures of pathogens, thereby paving the way for the development of broadly effective anti-microbial strategies for public health.",
         "autophagy | interferon-inducible GTPases | LC3 conjugation | pathogen-containing vacuoles | positive-sense RNA virus | replication compartment | vacuolar pathogen",
         "7",
         "1",
         "repositoryam",
         "Green",
         "NIH",
         "P30 CA14599",
         "National Institutes of Health"
        ],
        [
         "37",
         "37",
         "2-s2.0-85035339592",
         "10.1002/bies.201700086",
         "NULL",
         "28977693.0",
         "Nanog Expression in Embryonic Stem Cells ‚Äì An Ideal Model System to Dissect Enhancer Function",
         "re",
         "Review",
         "Blinka S.",
         "60013099;60012553",
         "BloodCenter of Wisconsin;Medical College of Wisconsin",
         "Milwaukee;Milwaukee",
         "United States;United States",
         "2.0",
         "Blinka, Steven;Rao, Sridhar",
         "55272520300;32267539900",
         "60012553-60013099;60012553-60013099",
         "2017-12-01",
         "December 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "12",
         "1700086",
         "NULL",
         "Embryonic stem cells (ESCs) are derived from the preimplantation embryo and can differentiate into virtually any other cell type (termed pluripotency), which is governed by lineage specific transcriptions factors (TFs) binding to cis regulatory elements (CREs) to mediate changes in gene expression. The reliance on transcriptional regulation to maintain pluripotency makes ESCs a valuable model to study the role of distal CREs such as enhancers in modulating gene expression to affect cell fate decisions. This review will highlight recent advance on transcriptional enhancers, focusing on studies performed in ESCs. In addition, we argue that the Nanog locus, which encodes for an ESC-critical TF, is particularly informative because it contains multiple co-regulated genes and enhancers in close proximity to one another. The unique landscape at Nanog permits the study of ongoing questions including whether multiple enhancers function additively versus synergistically, determinants of gene specificity, and cell-to-cell variability in gene expression.",
         "embryonic stem cells | eRNAs | higher order chromosome structure | looping | super-enhancers | transcriptional regulation",
         "18",
         "0",
         "repositoryam",
         "Green",
         "NIGMS",
         "GM080202",
         "National Institute of General Medical Sciences"
        ],
        [
         "38",
         "38",
         "2-s2.0-85030466349",
         "10.1002/bies.201700122",
         "NULL",
         "28960439.0",
         "Physical Limits on the Precision of Mitotic Spindle Positioning by Microtubule Pushing forces: Mechanics of mitotic spindle positioning",
         "re",
         "Review",
         "Howard J.",
         "60012708;60005455",
         "Stanford University;Yale University",
         "Stanford;New Haven",
         "United States;United States",
         "2.0",
         "Howard, Jonathon;Garzon-Coral, Carlos",
         "23966634100;33167521000",
         "60005455;60012708",
         "2017-11-01",
         "November 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "11",
         "1700122",
         "NULL",
         "Tissues are shaped and patterned by mechanical and chemical processes. A key mechanical process is the positioning of the mitotic spindle, which determines the size and location of the daughter cells within the tissue. Recent force and position-fluctuation measurements indicate that pushing forces, mediated by the polymerization of astral microtubules against¬≠ the cell cortex, maintain the mitotic spindle at the cell center in Caenorhabditis elegans embryos. The magnitude of the centering forces suggests that the physical limit on the accuracy and precision of this centering mechanism is determined by the number of pushing microtubules rather than by thermally driven fluctuations. In cells that divide asymmetrically, anti-centering, pulling forces generated by cortically located dyneins, in conjunction with microtubule depolymerization, oppose the pushing forces to drive spindle displacements away from the center. Thus, a balance of centering pushing forces and anti-centering pulling forces localize the mitotic spindles within dividing C. elegans cells.",
         "buckling | C. elegans | cell division | microtubules | mitosis | precision | spindle",
         "30",
         "1",
         "repositoryam",
         "Green",
         "NIH",
         "R01GM110386",
         "National Institutes of Health"
        ],
        [
         "39",
         "39",
         "2-s2.0-85107490355",
         "10.1007/s13752-017-0268-4",
         "NULL",
         "NULL",
         "Reconsidering Morphology Through an Experimental Case Study",
         "ar",
         "Article",
         "Albertazzi L.",
         "60015986;60000481",
         "Universit√† di Trento;Universit√† degli Studi di Padova",
         "Trento;Padua",
         "Italy;Italy",
         "6.0",
         "Albertazzi, Liliana;Canal, Luisa;Chist√®, Paolo;De Rosa, Mara;Micciolo, Rocco;Minelli, Alessandro",
         "56365267800;55900401400;57203202727;57324928300;7005868993;7004539407",
         "60015986;60015986;60015986;60015986;60015986;60000481",
         "2017-09-01",
         "September 2017",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "12",
         "3",
         "NULL",
         "131-141",
         "This study analyzes shells of marine gastropods of a zoological museum and the Latin epithets expressing perceptual and connotative attributes that they have received in the standard, Linnaean nomenclature. Making use of the Osgood semantic differential, we presented the subjects with digital 3-D reproductions of the shell specimens to be subjectively evaluated according to 17 pairs of attributes. The results show that, overall, the subjective evaluations given by the subjects are consistent, which suggests that an intersubjective characterization of the shells as wholes according to their qualitative and morphological traits is possible and scientifically explicable. This may apply to other biological items as well. The results also show that the epithets given to the shells by taxonomists do not always reflect the perceptual evaluation of the general population, being a product of the well-known ambiguity in taxonomic categorization and naming.",
         "Linnaean nomenclature | Morphology | Shape invariants | Taxonomic categorization",
         "2",
         "0",
         "NULL",
         "NULL",
         "UNIBO",
         "undefined",
         "Universit√† di Bologna"
        ],
        [
         "40",
         "40",
         "2-s2.0-84995616598",
         "10.1007/s10441-016-9301-4",
         "NULL",
         "27853896.0",
         "Discrete Mesh Approach in Morphogenesis Modelling: the Example of Gastrulation",
         "ar",
         "Article",
         "Demongeot J.",
         "60108275;60104653",
         "Techniques for Biomedical Engineering and Complexity Management Informatics, Mathematics and Applications Grenoble (TIMC-IMAG);Universit√© Grenoble Alpes",
         "La Tronche;Saint Martin d'Heres",
         "France;France",
         "3.0",
         "Demongeot, J.;Lontos, A.;Promayon, E.",
         "7005372567;38561637100;6507228109",
         "60104653;60108275;60108275",
         "2016-12-01",
         "1 December 2016",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "64",
         "4",
         "NULL",
         "427-446",
         "Morphogenesis is a general concept in biology including all the processes which generate tissue shapes and cellular organizations in a living organism. Many hybrid formalizations (i.e., with both discrete and continuous parts) have been proposed for modelling morphogenesis in embryonic or adult animals, like gastrulation. We propose first to study the ventral furrow invagination as the initial step of gastrulation, early stage of embryogenesis. We focus on the study of the connection between the apical constriction of the ventral cells and the initiation of the invagination. For that, we have created a 3D biomechanical model of the embryo of the Drosophila melanogaster based on the finite element method. Each cell is modelled by an elastic hexahedron contour and is firmly attached to its neighbouring cells. A uniform initial distribution of elastic and contractile forces is applied to cells along the model. Numerical simulations show that invagination starts at ventral curved extremities of the embryo and then propagates to the ventral medial layer. Then, this observation already made in some experiments can be attributed uniquely to the specific shape of the embryo and we provide mechanical evidence to support it. Results of the simulations of the ‚Äúpill-shaped‚Äù geometry of the Drosophila melanogaster embryo are compared with those of a spherical geometry corresponding to the Xenopus l√¶vis embryo. Eventually, we propose to study the influence of cell proliferation on the end of the process of invagination represented by the closure of the ventral furrow.",
         "Discrete mesh modelling | Gastrulation | Morphogenesis | Ventral furrow",
         "0",
         "0",
         "NULL",
         "NULL",
         "EC",
         "undefined",
         "European Commission"
        ],
        [
         "41",
         "41",
         "2-s2.0-0027977128",
         "10.1002/bies.950160105",
         "NULL",
         "8141805.0",
         "Radical solutions and cultural problems: Could free oxygen radicals be responsible for the impaired development of preimplantation mammalian embryos in vitro?",
         "re",
         "Review",
         "Johnson M.H.",
         "NULL",
         "NULL",
         "NULL",
         "NULL",
         "2.0",
         "Johnson, Martin H.;Nasresfahani, Mohammad H.",
         "36071272900;57190655374",
         "NULL",
         "1994-01-01",
         "January 1994",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "16",
         "1",
         "NULL",
         "31-38",
         "A major obstacel to the study of mammalian development, and to the practical application of knowledge gained from it in the clinic during therapeutic in vitro fertilisation and embryo transfer (IVF‚ÄêET), is the propensity of embryos to become retarded or arrested during their culture in vitro. The precise developmental cell cycle in which embryos arrest or delay is characteristic for the species and coincides with the earliest period of embryonic gene expression. Much evidence reviewed here implicates free oxygen radicals (FORs) in the process of arrest. Thus, studies on the development of mouse preimplantation embryos in vitro have shown that (i) FORs are elevated in vitro, but not in vivo, at the time at which embryos become arrested or delayed, (ii) systems for removing reactive oxygen species to limit the formation of hydroxy radicals are present, although they have not yet been assessed quantitatively and may differ qualitatively from those in adult cells, (iii) metabolic and possibly genetic adaptations to oxidative damage are evident, (iv) published procedures for overcoming in vitro arrest are explicable in terms of FOR‚Äêmediated damage or responses and (v) the arrest or delay of most embryos in vitro can be reduced or prevented experimentally by addition of metal chelators to limit hydroxy radical formation and lipid hydroperoxidation. Copyright ¬© 1994 Cambridge University Press",
         "NULL",
         "322",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "42",
         "42",
         "2-s2.0-0023504894",
         "10.1007/BF00052064",
         "NULL",
         "3128032.0",
         "Mathematical model and simulation of retina and tectum opticum of lower vertebrates",
         "ar",
         "Article",
         "an der Heiden U.",
         "60022113;60008293",
         "Universit√§t Witten/Herdecke;Universit√§t Bremen",
         "Witten;Bremen",
         "Germany;Germany",
         "2.0",
         "an der Heiden, U.;Roth, G.",
         "6701417263;7202210916",
         "60022113;60008293",
         "1987-09-01",
         "September 1987",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "36",
         "3",
         "NULL",
         "179-212",
         "The processing of information within the retino-tectal visual system of amphibians is decomposed into five major operational stages, three of them taking place in the retina and two in the optic tectum. The stages in the retina involve (i) a spatially local high-pass filtering in connection to the perception of moving objects, (ii) separation of the receptor activity into ON- and OFF-channels regarding the distinction of objects on both light and dark backgrounds, (iii) spatial integration via near excitation and far-reaching inhibition. Variation of the spatial range of excitation and inhibition allows to account for typical activities observed in a variety of classes of retina ganglion cells. Mathematical description of the operations in the tectum opticum include (i) spatial summation of retinal output (mainly of class-2 and class-3 retina ganglion cells), and (ii) direct or indirect lateral inhibition between tectal cells. In the computer simulation, first the output of the mathematical retina model is computed which, then, is used as the input to the tectum model. The full spatio-temporal dynamics is taken into account. The simulations show that different combinations of strength of lateral inhibition on the one side and the response properties of the retina ganglion cells on the other side determine the response properties of tectal cell types involved in object recognition. ¬© 1987 Martinus Nijhoff/Dr W. Junk Publishers.",
         "NULL",
         "15",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "43",
         "43",
         "2-s2.0-85044358499",
         "10.1002/bies.201800029",
         "NULL",
         "29575018.0",
         "The Diversification of Early Emerging Metazoans: A Window into the Evolution of Animal Multicellularity",
         "ar",
         "Article",
         "Revilla-i-Domingo R.",
         "60025988",
         "Universit√§t Wien",
         "Vienna",
         "Austria",
         "2.0",
         "Revilla-i-Domingo, Roger;Simakov, Oleg",
         "6508022715;12783132000",
         "60025988;60025988",
         "2018-05-01",
         "May 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "5",
         "1800029",
         "NULL",
         "The biannual international workshop entitled ‚ÄúThe diversification of early emerging metazoans: A window into animal evolution?‚Äù took place at the Evangelische Akademie Tutzing, Germany, 11‚Äì14. September 2017. It was organized by Thomas Bosch (Kiel), Thomas Holstein (Heidelberg), and Ulrich Technau (Vienna), and it was sponsored by the Deutsche Forschungsgemeinschaft (DFG). The meeting gathered over 140 researchers to discuss the contribution of non-bilaterian metazoan models (Porifera, Ctenophora, Placozoa, and Cnidaria) to our understanding of: a. The evolution of metazoan developmental processes; b. Fundamental molecular mechanisms underlying metazoan features; and c. The complex interactions that animals establish with their environment.",
         "NULL",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "44",
         "44",
         "2-s2.0-85044354981",
         "10.1002/bies.201700239",
         "NULL",
         "29574793.0",
         "Identifying Human Na√Øve Pluripotent Stem Cells ‚àí Evaluating State-Specific Reporter Lines and Cell-Surface Markers",
         "re",
         "Review",
         "Collier A.",
         "60000661",
         "The Babraham Institute",
         "Cambridge",
         "United Kingdom",
         "2.0",
         "Collier, Amanda J.;Rugg-Gunn, Peter J.",
         "57189229032;57197724801",
         "60000661;60000661",
         "2018-05-01",
         "May 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "5",
         "1700239",
         "NULL",
         "Recent reports that human pluripotent stem cells can be captured in a spectrum of states with variable properties has prompted a re-evaluation of how pluripotency is acquired and stabilised. The latest additions to the stem cell hierarchy open up opportunities for understanding human development, reprogramming, and cell state transitions more generally. Many of the new cell lines have been collectively termed ‚Äòna√Øve‚Äô human pluripotent stem cells to distinguish them from the conventional ‚Äòprimed‚Äô cells. Here, several transcriptional and epigenetic hallmarks of human pluripotent states in the recently described cell lines are reviewed and evaluated. Methods to derive and identify human na√Øve pluripotent stem cells are also discussed, with a focus on the uses and future developments of state-specific reporter cell lines and cell-surface proteins. Finally, opportunities and uncertainties in na√Øve stem cell biology are highlighted, and the current limitations of human na√Øve pluripotent stem cells considered, particularly in the context of differentiation.",
         "development | embryonic stem cells | epigenetics | pluripotency | reprogramming | X-chromosome inactivation",
         "25",
         "1",
         "repositoryvor",
         "Green",
         "MRC",
         "MC_PC_12009",
         "Medical Research Council"
        ],
        [
         "45",
         "45",
         "2-s2.0-85043717154",
         "10.1002/bies.201700242",
         "NULL",
         "29543982.0",
         "Demystifying Eukaryote Lateral Gene Transfer (Response to Martin 2017 DOI: 10.1002/bies.201700115)",
         "re",
         "Review",
         "Leger M.M.",
         "60103738;60023750;60003858",
         "CSIC-UPF - Instituto de Biologia Evolutiva (IBE);Dalhousie University, Faculty of Medicine;Uppsala Universitet",
         "Barcelona;Halifax;Uppsala",
         "Spain;Canada;Sweden",
         "4.0",
         "Leger, Michelle M.;Eme, Laura;Stairs, Courtney W.;Roger, Andrew J.",
         "54936029500;26433309900;54416433400;7004844245",
         "60103738;60003858;60003858;60023750",
         "2018-05-01",
         "May 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "5",
         "1700242",
         "NULL",
         "In a recent BioEssays paper [W. F. Martin, BioEssays 2017, 39, 1700115], William Martin sharply criticizes evolutionary interpretations that involve lateral gene transfer (LGT) into eukaryotic genomes. Most published examples of LGTs in eukaryotes, he suggests, are in fact contaminants, ancestral genes that have been lost from other extant lineages, or the result of artefactual phylogenetic inferences. Martin argues that, except for transfers that occurred from endosymbiotic organelles, eukaryote LGT is insignificant. Here, in reviewing this field, we seek to correct some of the misconceptions presented therein with regard to the evidence for LGT in eukaryotes.",
         "eukaryote genomes | horizontal gene transfer | lateral gene transfer",
         "58",
         "0",
         "NULL",
         "NULL",
         "EMBO",
         "ALTF-997-2015",
         "European Molecular Biology Organization"
        ],
        [
         "46",
         "46",
         "2-s2.0-85043275933",
         "10.1002/bies.201700241",
         "NULL",
         "29508910.0",
         "The Rise of the Cartwheel: Seeding the Centriole Organelle",
         "ar",
         "Article",
         "Guichard P.",
         "60028186;60004718",
         "√âcole Polytechnique F√©d√©rale de Lausanne;Universit√© de Gen√®ve",
         "Lausanne;Geneva",
         "Switzerland;Switzerland",
         "3.0",
         "Guichard, Paul;Hamel, Virginie;G√∂nczy, Pierre",
         "25936173500;8971305600;35446142600",
         "60004718;60004718;60028186",
         "2018-04-01",
         "April 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "4",
         "1700241",
         "NULL",
         "The cartwheel is a striking structure critical for building the centriole, a microtubule-based organelle fundamental for organizing centrosomes, cilia, and flagella. Over the last 50 years, the cartwheel has been described in many systems using electron microscopy, but the molecular nature of its constituent building blocks and their assembly mechanisms have long remained mysterious. Here, we review discoveries that led to the current understanding of cartwheel structure, assembly, and function. We focus on the key role of SAS-6 protein self-organization, both for building the signature ring-like structure with hub and spokes, as well as for their vertical stacking. The resemblance of assembly intermediates in vitro and in vivo leads us to propose a novel registration step in cartwheel biogenesis, whereby stacked SAS-6-containing rings are put in register through interactions with peripheral elements anchored to microtubules. We conclude by evoking some avenues for further uncovering cartwheel and centriole assembly mechanisms.",
         "basal bodies | cartwheel | centrioles | cilia | SAS-6",
         "37",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "ERC",
         "340227",
         "European Research Council"
        ],
        [
         "47",
         "47",
         "2-s2.0-85038865269",
         "10.1002/bies.201700018",
         "NULL",
         "29266341.0",
         "Microbiome-Germline Interactions and Their Transgenerational Implications",
         "ar",
         "Article",
         "Elgart M.",
         "119075263",
         "Biomolecular Sciences",
         "Rehovot",
         "Israel",
         "2.0",
         "Elgart, Michael;Soen, Yoav",
         "55964995000;6602347180",
         "119075263;119075263",
         "2018-04-01",
         "April 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "4",
         "1700018",
         "NULL",
         "It is becoming increasingly clear that most, if not all, animals and plants are associated with a diverse array of resident gut microbiota. This symbiosis is regulated by host-microbiome interactions which influence the development, homeostasis, adaptation and evolution of the host. Recent evidence indicated that these interactions can also affect the host germline and have a potential of supporting transgenerational effects, including inheritance of acquired characteristics. Taken together, the influence of gut bacteria on the host soma and germline could potentially give rise to emergent phenotypes, which may be partially inherited by three distinguishable modes of transgenerational influence of gut bacteria: 1) ‚Äúsoma-to-soma‚Äù 2) ‚Äúsoma-to-germline‚Äù and 3) ‚Äúsoma-germline-soma‚Äù. Here, we discuss these possibilities in light of evidence supporting bacterial-mediated modes of transgenerational inheritance.",
         "bacteria | germline | holobiont | influence | inheritance | soma | transgenerational",
         "9",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "48",
         "48",
         "2-s2.0-85041576154",
         "10.1002/bies.201700191",
         "NULL",
         "29405334.0",
         "Mitonuclear Mate Choice: A Missing Component of Sexual Selection Theory?",
         "ar",
         "Article",
         "Hill G.",
         "60011754",
         "Auburn University",
         "Auburn",
         "United States",
         "1.0",
         "Hill, Geoffrey E.",
         "35500077500",
         "60011754",
         "2018-03-01",
         "March 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "3",
         "1700191",
         "NULL",
         "The fitness of a eukaryote hinges on the coordinated function of the products of its nuclear and mitochondrial genomes in achieving oxidative phosphorylation (OXPHOS). I propose that sexual selection plays a key role in the maintenance of mitonuclear coadaptation across generations because it enables pre-zygotic sorting for coadapted mitonuclear genotypes. At each new generation, sexual reproduction creates new combinations of nuclear and mitochondrial genes, and the potential arises for mitonuclear incompatibilities and reduced fitness. In reviewing the literature, I hypothesize that individuals engaged in mate choice select partners with correct species-typical mitochondrial and nuclear genotypes as well as individuals with highly functional cellular respiration. The implication is that mate choice for compatible nuclear and mitochondrial genes can play a significant role in generating the patterns of ornamentation and preferences observed in animals. A number of testable predictions emerge from this mitonuclear compatibility hypothesis of sexual selection.",
         "indicator model | mitochondria | OXPHOS | reinforcement | sexual selection",
         "24",
         "0",
         "NULL",
         "NULL",
         "AU",
         "undefined",
         "Auburn University"
        ],
        [
         "49",
         "49",
         "2-s2.0-85039162962",
         "10.1002/bies.201700184",
         "NULL",
         "29283187.0",
         "Cell Size Control via an Unstable Accumulating Activator and the Phenomenon of Excess Mitotic Delay",
         "re",
         "Review",
         "Rhind N.",
         "60000986",
         "University of Massachusetts Chan Medical School",
         "Worcester",
         "United States",
         "1.0",
         "Rhind, Nicholas",
         "6603138796",
         "60000986",
         "2018-02-01",
         "February 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "2",
         "1700184",
         "NULL",
         "Unstable Accumulating Activator models for cellular size control propose an activator that accumulates in a size-dependent manner and triggers cell cycle progression once it has reached a certain threshold. Having a short half life makes such an activator responsive to changes in cell size and makes specific predictions for how cells respond to perturbation. In particular, it explains the curious phenomenon of excess mitotic delay. Excess mitotic delay, first observed in Tetrahymena in the '50s, is a phenomenon in which a pulse of protein synthesis inhibition causes a delay in mitotic entry that is longer than the pulse and that gets longer the later in the cell cycle the pulse is delivered. The interpretation of this phenomenon championed by Zeuthen and Mitchison in the '60s and '70s is that an unstable activator of mitosis is degraded during the pulse and has to be resynthesized to a threshold level to trigger mitosis; small cells have more time to resynthesize the activator before mitosis and so suffer less excess delay, whereas, large cells have less time thus suffer greater excess delay. Fifty years later, with our detailed understanding of cell cycle biochemistry, we can identify and test candidate Unstable Accumulating Activators. Here I review the field and further develop this concept.",
         "accumulating activator | cell cycle | cell size control | excess mitotic delay | sizer proteins | unstable activator",
         "5",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "50",
         "50",
         "2-s2.0-85035345390",
         "10.1002/bies.201700115",
         "NULL",
         "29068466.0",
         "Too Much Eukaryote LGT",
         "re",
         "Review",
         "Martin W.F.",
         "60025310",
         "Heinrich-Heine-Universit√§t D√ºsseldorf",
         "Dusseldorf",
         "Germany",
         "1.0",
         "Martin, William F.",
         "7402718622",
         "60025310",
         "2017-12-01",
         "December 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "12",
         "1700115",
         "NULL",
         "The realization that prokaryotes naturally and frequently disperse genes across steep taxonomic boundaries via lateral gene transfer (LGT) gave wings to the idea that eukaryotes might do the same. Eukaryotes do acquire genes from mitochondria and plastids and they do transfer genes during the process of secondary endosymbiosis, the spread of plastids via eukaryotic algal endosymbionts. From those observations it, however, does not follow that eukaryotes transfer genes either in the same ways as prokaryotes do, or to a quantitatively similar degree. An important illustration of the difference is that eukaryotes do not exhibit pangenomes, though prokaryotes do. Eukaryotes reveal no detectable cumulative effects of LGT, though prokaryotes do. A critical analysis suggests that something is deeply amiss with eukaryote LGT theories.",
         "genome analysis | horizontal gene transfer | Lamarckian evolution | lateral gene transfer | phylogenetic artefact",
         "90",
         "0",
         "NULL",
         "NULL",
         "H2020",
         "666053",
         "Horizon 2020 Framework Programme"
        ]
       ],
       "shape": {
        "columns": 37,
        "rows": 7913
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>eid</th>\n",
       "      <th>doi</th>\n",
       "      <th>pii</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>title</th>\n",
       "      <th>subtype</th>\n",
       "      <th>subtypeDescription</th>\n",
       "      <th>creator</th>\n",
       "      <th>afid</th>\n",
       "      <th>...</th>\n",
       "      <th>pageRange</th>\n",
       "      <th>description</th>\n",
       "      <th>authkeywords</th>\n",
       "      <th>citedby_count</th>\n",
       "      <th>openaccess</th>\n",
       "      <th>freetoread</th>\n",
       "      <th>freetoreadLabel</th>\n",
       "      <th>fund_acr</th>\n",
       "      <th>fund_no</th>\n",
       "      <th>fund_sponsor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>2-s2.0-85209763600</td>\n",
       "      <td>10.1007/s11229-024-04815-5</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Driftability and niche construction</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>F√°bregas-Tejeda A.</td>\n",
       "      <td>60025063</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Niche construction is the process of organisms...</td>\n",
       "      <td>Drift | Driftability | Evolutionary causation ...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherhybridgold</td>\n",
       "      <td>Hybrid Gold</td>\n",
       "      <td>FWO</td>\n",
       "      <td>G070122N</td>\n",
       "      <td>Fonds Wetenschappelijk Onderzoek</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2-s2.0-85202899987</td>\n",
       "      <td>10.1007/s10539-024-09957-x</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Explanatory gaps in evolutionary theory</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Aaby B.H.</td>\n",
       "      <td>60025063;60010348</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Proponents of the extended evolutionary synthe...</td>\n",
       "      <td>Evolutionary theory | Explanatory gap | Natura...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherhybridgold</td>\n",
       "      <td>Hybrid Gold</td>\n",
       "      <td>NULL</td>\n",
       "      <td>3H210777</td>\n",
       "      <td>KU Leuven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2-s2.0-85175425123</td>\n",
       "      <td>10.1016/j.shpsa.2023.10.006</td>\n",
       "      <td>S0039368123001449</td>\n",
       "      <td>37907020.0</td>\n",
       "      <td>A Wolf in Sheep's Clothing: Idealisations and ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Serpico D.</td>\n",
       "      <td>60021361;60015986</td>\n",
       "      <td>...</td>\n",
       "      <td>72-83</td>\n",
       "      <td>Research in pharmacogenomics and precision med...</td>\n",
       "      <td>Genetic prediction | GWAS | Idealisations | Po...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>ERC</td>\n",
       "      <td>805498</td>\n",
       "      <td>European Research Council</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>2-s2.0-85152302150</td>\n",
       "      <td>10.1016/j.shpsa.2023.03.002</td>\n",
       "      <td>S0039368123000705</td>\n",
       "      <td>37068423.0</td>\n",
       "      <td>Joint representation: Modeling a phenomenon wi...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Yoshida Y.</td>\n",
       "      <td>60029445</td>\n",
       "      <td>...</td>\n",
       "      <td>67-76</td>\n",
       "      <td>Biologists often study particular biological s...</td>\n",
       "      <td>Collective cell migration | Generalization | M...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>UMN</td>\n",
       "      <td>undefined</td>\n",
       "      <td>University of Minnesota</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>2-s2.0-85148606416</td>\n",
       "      <td>10.3390/philosophies8010008</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Turing‚Äôs Biological Philosophy: Morphogenesis,...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Greif H.</td>\n",
       "      <td>60003675</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Alan M. Turing‚Äôs last published work and some ...</td>\n",
       "      <td>Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherfullgold</td>\n",
       "      <td>Gold</td>\n",
       "      <td>NCN</td>\n",
       "      <td>2020/37/B/HS1/01809</td>\n",
       "      <td>Narodowe Centrum Nauki</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3159</th>\n",
       "      <td>3159</td>\n",
       "      <td>2-s2.0-84898444008</td>\n",
       "      <td>10.1080/21550085.2014.885490</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>A Constructivist Approach Toward a General Def...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Meinard Y.</td>\n",
       "      <td>60068952;60012614;114099298</td>\n",
       "      <td>...</td>\n",
       "      <td>88-104</td>\n",
       "      <td>Biodiversity sciences witness a double dynamic...</td>\n",
       "      <td>economics | ecosystem management | philosophy ...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>repositoryam</td>\n",
       "      <td>Green</td>\n",
       "      <td>FRJ</td>\n",
       "      <td>undefined</td>\n",
       "      <td>Fondation pour la recherche juridique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3160</th>\n",
       "      <td>3160</td>\n",
       "      <td>2-s2.0-27844440845</td>\n",
       "      <td>10.3197/096327105774462683</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The commons, game theory and aspects of human ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Dodds W.K.</td>\n",
       "      <td>60000689</td>\n",
       "      <td>...</td>\n",
       "      <td>411-425</td>\n",
       "      <td>Fundamental aspects of human use of the enviro...</td>\n",
       "      <td>Commons | Game theory | Global environment | H...</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3161</th>\n",
       "      <td>3161</td>\n",
       "      <td>2-s2.0-0036297383</td>\n",
       "      <td>10.3197/096327102129341073</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The role of customary institutions in the cons...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Virtanen P.</td>\n",
       "      <td>60011170</td>\n",
       "      <td>...</td>\n",
       "      <td>227-241</td>\n",
       "      <td>Recently the role of customary local instituti...</td>\n",
       "      <td>Africa | Biodiversity | Conservation areas | L...</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3162</th>\n",
       "      <td>3162</td>\n",
       "      <td>2-s2.0-0034962075</td>\n",
       "      <td>10.3197/096327101129340804</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Exotic species, naturalisation, and biological...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Hettinger N.</td>\n",
       "      <td>60013340</td>\n",
       "      <td>...</td>\n",
       "      <td>193-224</td>\n",
       "      <td>Contrary to frequent characterisations, exotic...</td>\n",
       "      <td>Exotics | Native | Nativism | Naturalisation |...</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3163</th>\n",
       "      <td>3163</td>\n",
       "      <td>2-s2.0-0343729540</td>\n",
       "      <td>10.3197/096327100129341976</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Does the convention on biodiversity safeguard ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>M√ºller F.G.</td>\n",
       "      <td>60033154</td>\n",
       "      <td>...</td>\n",
       "      <td>55-80</td>\n",
       "      <td>This paper attempts to assess and evaluate som...</td>\n",
       "      <td>Biological diversity | Commercialisation | Eco...</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7913 rows √ó 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0                 eid                           doi  \\\n",
       "0              0  2-s2.0-85209763600    10.1007/s11229-024-04815-5   \n",
       "1              1  2-s2.0-85202899987    10.1007/s10539-024-09957-x   \n",
       "2              2  2-s2.0-85175425123   10.1016/j.shpsa.2023.10.006   \n",
       "3              3  2-s2.0-85152302150   10.1016/j.shpsa.2023.03.002   \n",
       "4              4  2-s2.0-85148606416   10.3390/philosophies8010008   \n",
       "...          ...                 ...                           ...   \n",
       "3159        3159  2-s2.0-84898444008  10.1080/21550085.2014.885490   \n",
       "3160        3160  2-s2.0-27844440845    10.3197/096327105774462683   \n",
       "3161        3161   2-s2.0-0036297383    10.3197/096327102129341073   \n",
       "3162        3162   2-s2.0-0034962075    10.3197/096327101129340804   \n",
       "3163        3163   2-s2.0-0343729540    10.3197/096327100129341976   \n",
       "\n",
       "                    pii   pubmed_id  \\\n",
       "0                  NULL        NULL   \n",
       "1                  NULL        NULL   \n",
       "2     S0039368123001449  37907020.0   \n",
       "3     S0039368123000705  37068423.0   \n",
       "4                  NULL        NULL   \n",
       "...                 ...         ...   \n",
       "3159               NULL        NULL   \n",
       "3160               NULL        NULL   \n",
       "3161               NULL        NULL   \n",
       "3162               NULL        NULL   \n",
       "3163               NULL        NULL   \n",
       "\n",
       "                                                  title subtype  \\\n",
       "0                   Driftability and niche construction      ar   \n",
       "1               Explanatory gaps in evolutionary theory      ar   \n",
       "2     A Wolf in Sheep's Clothing: Idealisations and ...      ar   \n",
       "3     Joint representation: Modeling a phenomenon wi...      ar   \n",
       "4     Turing‚Äôs Biological Philosophy: Morphogenesis,...      ar   \n",
       "...                                                 ...     ...   \n",
       "3159  A Constructivist Approach Toward a General Def...      ar   \n",
       "3160  The commons, game theory and aspects of human ...      ar   \n",
       "3161  The role of customary institutions in the cons...      ar   \n",
       "3162  Exotic species, naturalisation, and biological...      ar   \n",
       "3163  Does the convention on biodiversity safeguard ...      ar   \n",
       "\n",
       "     subtypeDescription             creator                         afid  ...  \\\n",
       "0               Article  F√°bregas-Tejeda A.                     60025063  ...   \n",
       "1               Article           Aaby B.H.            60025063;60010348  ...   \n",
       "2               Article          Serpico D.            60021361;60015986  ...   \n",
       "3               Article          Yoshida Y.                     60029445  ...   \n",
       "4               Article            Greif H.                     60003675  ...   \n",
       "...                 ...                 ...                          ...  ...   \n",
       "3159            Article          Meinard Y.  60068952;60012614;114099298  ...   \n",
       "3160            Article          Dodds W.K.                     60000689  ...   \n",
       "3161            Article         Virtanen P.                     60011170  ...   \n",
       "3162            Article        Hettinger N.                     60013340  ...   \n",
       "3163            Article         M√ºller F.G.                     60033154  ...   \n",
       "\n",
       "     pageRange                                        description  \\\n",
       "0         NULL  Niche construction is the process of organisms...   \n",
       "1         NULL  Proponents of the extended evolutionary synthe...   \n",
       "2        72-83  Research in pharmacogenomics and precision med...   \n",
       "3        67-76  Biologists often study particular biological s...   \n",
       "4         NULL  Alan M. Turing‚Äôs last published work and some ...   \n",
       "...        ...                                                ...   \n",
       "3159    88-104  Biodiversity sciences witness a double dynamic...   \n",
       "3160   411-425  Fundamental aspects of human use of the enviro...   \n",
       "3161   227-241  Recently the role of customary local instituti...   \n",
       "3162   193-224  Contrary to frequent characterisations, exotic...   \n",
       "3163     55-80  This paper attempts to assess and evaluate som...   \n",
       "\n",
       "                                           authkeywords citedby_count  \\\n",
       "0     Drift | Driftability | Evolutionary causation ...             0   \n",
       "1     Evolutionary theory | Explanatory gap | Natura...             0   \n",
       "2     Genetic prediction | GWAS | Idealisations | Po...             0   \n",
       "3     Collective cell migration | Generalization | M...             0   \n",
       "4     Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...             1   \n",
       "...                                                 ...           ...   \n",
       "3159  economics | ecosystem management | philosophy ...             7   \n",
       "3160  Commons | Game theory | Global environment | H...            15   \n",
       "3161  Africa | Biodiversity | Conservation areas | L...            66   \n",
       "3162  Exotics | Native | Nativism | Naturalisation |...            55   \n",
       "3163  Biological diversity | Commercialisation | Eco...            12   \n",
       "\n",
       "     openaccess           freetoread freetoreadLabel fund_acr  \\\n",
       "0             1  publisherhybridgold     Hybrid Gold      FWO   \n",
       "1             1  publisherhybridgold     Hybrid Gold     NULL   \n",
       "2             1        repositoryvor           Green      ERC   \n",
       "3             0        repositoryvor           Green      UMN   \n",
       "4             1    publisherfullgold            Gold      NCN   \n",
       "...         ...                  ...             ...      ...   \n",
       "3159          0         repositoryam           Green      FRJ   \n",
       "3160          0                 NULL            NULL     NULL   \n",
       "3161          0                 NULL            NULL     NULL   \n",
       "3162          0                 NULL            NULL     NULL   \n",
       "3163          0                 NULL            NULL     NULL   \n",
       "\n",
       "                  fund_no                           fund_sponsor  \n",
       "0                G070122N       Fonds Wetenschappelijk Onderzoek  \n",
       "1                3H210777                              KU Leuven  \n",
       "2                  805498              European Research Council  \n",
       "3               undefined                University of Minnesota  \n",
       "4     2020/37/B/HS1/01809                 Narodowe Centrum Nauki  \n",
       "...                   ...                                    ...  \n",
       "3159            undefined  Fondation pour la recherche juridique  \n",
       "3160            undefined                                   NULL  \n",
       "3161            undefined                                   NULL  \n",
       "3162            undefined                                   NULL  \n",
       "3163            undefined                                   NULL  \n",
       "\n",
       "[7913 rows x 37 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cleaned  = df[df['description'] != 'NULL']\n",
    "df_cleaned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "ca38a4f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table id=\"itables_86e111e2_04d2_4ebc_831d_633a074709e5\" class=\"display nowrap\" data-quarto-disable-processing=\"true\" style=\"table-layout:auto;width:auto;margin:auto;caption-side:bottom\">\n",
       "<thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Document Type</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead><tbody><tr>\n",
       "<td style=\"vertical-align:middle; text-align:left\">\n",
       "<a href=https://mwouts.github.io/itables/><svg class=\"main-svg\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"\n",
       "width=\"64\" viewBox=\"0 0 500 400\" style=\"font-family: 'Droid Sans', sans-serif;\">\n",
       "    <g style=\"fill:#d9d7fc\">\n",
       "        <path d=\"M100,400H500V357H100Z\" />\n",
       "        <path d=\"M100,300H400V257H100Z\" />\n",
       "        <path d=\"M0,200H400V157H0Z\" />\n",
       "        <path d=\"M100,100H500V57H100Z\" />\n",
       "        <path d=\"M100,350H500V307H100Z\" />\n",
       "        <path d=\"M100,250H400V207H100Z\" />\n",
       "        <path d=\"M0,150H400V107H0Z\" />\n",
       "        <path d=\"M100,50H500V7H100Z\" />\n",
       "    </g>\n",
       "    <g style=\"fill:#1a1366;stroke:#1a1366;\">\n",
       "   <rect x=\"100\" y=\"7\" width=\"400\" height=\"43\">\n",
       "    <animate\n",
       "      attributeName=\"width\"\n",
       "      values=\"0;400;0\"\n",
       "      dur=\"5s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "      <animate\n",
       "      attributeName=\"x\"\n",
       "      values=\"100;100;500\"\n",
       "      dur=\"5s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "  </rect>\n",
       "        <rect x=\"0\" y=\"107\" width=\"400\" height=\"43\">\n",
       "    <animate\n",
       "      attributeName=\"width\"\n",
       "      values=\"0;400;0\"\n",
       "      dur=\"3.5s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "    <animate\n",
       "      attributeName=\"x\"\n",
       "      values=\"0;0;400\"\n",
       "      dur=\"3.5s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "  </rect>\n",
       "        <rect x=\"100\" y=\"207\" width=\"300\" height=\"43\">\n",
       "    <animate\n",
       "      attributeName=\"width\"\n",
       "      values=\"0;300;0\"\n",
       "      dur=\"3s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "    <animate\n",
       "      attributeName=\"x\"\n",
       "      values=\"100;100;400\"\n",
       "      dur=\"3s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "  </rect>\n",
       "        <rect x=\"100\" y=\"307\" width=\"400\" height=\"43\">\n",
       "    <animate\n",
       "      attributeName=\"width\"\n",
       "      values=\"0;400;0\"\n",
       "      dur=\"4s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "      <animate\n",
       "      attributeName=\"x\"\n",
       "      values=\"100;100;500\"\n",
       "      dur=\"4s\"\n",
       "      repeatCount=\"indefinite\" />\n",
       "  </rect>\n",
       "        <g style=\"fill:transparent;stroke-width:8; stroke-linejoin:round\" rx=\"5\">\n",
       "            <g transform=\"translate(45 50) rotate(-45)\">\n",
       "                <circle r=\"33\" cx=\"0\" cy=\"0\" />\n",
       "                <rect x=\"-8\" y=\"32\" width=\"16\" height=\"30\" />\n",
       "            </g>\n",
       "\n",
       "            <g transform=\"translate(450 152)\">\n",
       "                <polyline points=\"-15,-20 -35,-20 -35,40 25,40 25,20\" />\n",
       "                <rect x=\"-15\" y=\"-40\" width=\"60\" height=\"60\" />\n",
       "            </g>\n",
       "\n",
       "            <g transform=\"translate(50 352)\">\n",
       "                <polygon points=\"-35,-5 0,-40 35,-5\" />\n",
       "                <polygon points=\"-35,10 0,45 35,10\" />\n",
       "            </g>\n",
       "\n",
       "            <g transform=\"translate(75 250)\">\n",
       "                <polyline points=\"-30,30 -60,0 -30,-30\" />\n",
       "                <polyline points=\"0,30 -30,0 0,-30\" />\n",
       "            </g>\n",
       "\n",
       "            <g transform=\"translate(425 250) rotate(180)\">\n",
       "                <polyline points=\"-30,30 -60,0 -30,-30\" />\n",
       "                <polyline points=\"0,30 -30,0 0,-30\" />\n",
       "            </g>\n",
       "        </g>\n",
       "    </g>\n",
       "</svg>\n",
       "</a>\n",
       "Loading ITables v2.2.4 from the internet...\n",
       "(need <a href=https://mwouts.github.io/itables/troubleshooting.html>help</a>?)</td>\n",
       "</tr></tbody>\n",
       "</table>\n",
       "<link href=\"https://www.unpkg.com/dt_for_itables@2.0.13/dt_bundle.css\" rel=\"stylesheet\">\n",
       "<script type=\"module\">\n",
       "    import {DataTable, jQuery as $} from 'https://www.unpkg.com/dt_for_itables@2.0.13/dt_bundle.js';\n",
       "\n",
       "    document.querySelectorAll(\"#itables_86e111e2_04d2_4ebc_831d_633a074709e5:not(.dataTable)\").forEach(table => {\n",
       "        if (!(table instanceof HTMLTableElement))\n",
       "            return;\n",
       "\n",
       "        // Define the table data\n",
       "        const data = [[0, \"Article\", 6272], [1, \"Review\", 1140], [2, \"Note\", 249], [3, \"Conference Paper\", 147], [4, \"Erratum\", 41], [5, \"Short Survey\", 40], [7, \"Editorial\", 12], [6, \"Letter\", 9], [8, \"Book Chapter\", 2], [9, \"Retracted\", 1]];\n",
       "\n",
       "        // Define the dt_args\n",
       "        let dt_args = {\"layout\": {\"topStart\": null, \"topEnd\": null, \"bottomStart\": null, \"bottomEnd\": null}, \"order\": [], \"warn_on_selected_rows_not_rendered\": true};\n",
       "        dt_args[\"data\"] = data;\n",
       "\n",
       "        \n",
       "        new DataTable(table, dt_args);\n",
       "    });\n",
       "</script>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from itables import show\n",
    "unique_values = df_cleaned['subtypeDescription'].unique()\n",
    "\n",
    "# Create a new DataFrame with unique values of 'citing_year'\n",
    "unique_values_df = pd.DataFrame(unique_values, columns=['subtypeDescription'])\n",
    "\n",
    "# If you want to count how many times each unique value appears in the original DataFrame\n",
    "value_counts = df_cleaned['subtypeDescription'].value_counts()\n",
    "\n",
    "# Optionally, you can merge the unique values with the count if needed\n",
    "unique_values_df['Count'] = unique_values_df['subtypeDescription'].map(value_counts)\n",
    "\n",
    "unique_values_df.rename(columns={'subtypeDescription': 'Document Type'}, inplace=True)\n",
    "unique_values_df = unique_values_df.sort_values(by='Count', ascending=False)\n",
    "\n",
    "# Display the new table (DataFrame)\n",
    "show(unique_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13d14931",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Unnamed: 0",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "doi",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pii",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pubmed_id",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "title",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtype",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtypeDescription",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "creator",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "afid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affilname",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_city",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_country",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_count",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "author_names",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_ids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_afids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "coverDate",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "coverDisplayDate",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "publicationName",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "source_id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eIssn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "aggregationType",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "volume",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issueIdentifier",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "article_number",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pageRange",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "description",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "authkeywords",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "citedby_count",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "openaccess",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "freetoread",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "freetoreadLabel",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_acr",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_no",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_sponsor",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "165cd84c-81fb-41ba-83a4-d861d434b762",
       "rows": [
        [
         "0",
         "0",
         "2-s2.0-85209763600",
         "10.1007/s11229-024-04815-5",
         "NULL",
         "NULL",
         "Driftability and niche construction",
         "ar",
         "Article",
         "F√°bregas-Tejeda A.",
         "60025063",
         "KU Leuven",
         "Leuven",
         "Belgium",
         "2.0",
         "F√°bregas-Tejeda, Alejandro;Ramsey, Grant",
         "57196375680;14521522000",
         "60025063;60025063",
         "2024-12-01",
         "December 2024",
         "Synthese",
         "00397857",
         "27699",
         "15730964",
         "Journal",
         "204",
         "6",
         "162",
         "NULL",
         "Niche construction is the process of organisms changing themselves or their environment‚Äîor their relationship with their environment‚Äîin ways that affect the evolutionary trajectory of their population. These evolutionary trajectory changes are traditionally understood to be triggered by changes in selection pressures. Niche construction thus necessarily involves organisms altering selection pressures. In this paper, we argue that changes in selection pressures is not the only way organisms can influence the evolutionary futures of their population. We propose that organisms can also affect drift probabilities, and that such changes should be considered niche construction. Drift probabilities can be modulated by altering population size or by affecting driftability (individual variance in possible reproductive outcomes). We consider both and provide examples of how niche construction can stabilize, increase, or dampen drift probabilities. Finally, we revisit and broaden the traditional definition of niche construction. We hold that organismic activities that modify drift probabilities should count as niche construction, even if selection pressures remain unaltered.",
         "Drift | Driftability | Evolutionary causation | Niche construction | Selection",
         "0",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "FWO",
         "G070122N",
         "Fonds Wetenschappelijk Onderzoek"
        ],
        [
         "1",
         "1",
         "2-s2.0-85202899987",
         "10.1007/s10539-024-09957-x",
         "NULL",
         "NULL",
         "Explanatory gaps in evolutionary theory",
         "ar",
         "Article",
         "Aaby B.H.",
         "60025063;60010348",
         "KU Leuven;Universitetet i Oslo",
         "Leuven;Oslo",
         "Belgium;Norway",
         "3.0",
         "Aaby, Bendik Hellem;Dani, Gianmaria;Ramsey, Grant",
         "57262975700;59309831600;14521522000",
         "60010348;60025063;60025063",
         "2024-10-01",
         "October 2024",
         "Biology and Philosophy",
         "01693867",
         "13699",
         "15728404",
         "Journal",
         "39",
         "5",
         "22",
         "NULL",
         "Proponents of the extended evolutionary synthesis have argued that there are explanatory gaps in evolutionary biology that cannot be bridged by standard evolutionary theory. In this paper, we consider what sort of explanatory gaps they are referring to. We outline three possibilities: data-based gaps, implementation-based gaps, and framework-based gaps. We then examine the purported evolutionary gaps and attempt to classify them using this taxonomy. From there we reconsider the significance of the gaps and what they imply for the proposed need for an extended evolutionary synthesis.",
         "Evolutionary theory | Explanatory gap | Natural selection | Selection explanation | The active role of the organism | The extended synthesis",
         "0",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "3H210777",
         "KU Leuven"
        ],
        [
         "2",
         "2",
         "2-s2.0-85175425123",
         "10.1016/j.shpsa.2023.10.006",
         "S0039368123001449",
         "37907020.0",
         "A Wolf in Sheep's Clothing: Idealisations and the aims of polygenic scores",
         "ar",
         "Article",
         "Serpico D.",
         "60021361;60015986",
         "Uniwersytet Jagiello≈Ñski;Universit√† di Trento",
         "Krakow;Trento",
         "Poland;Italy",
         "1.0",
         "Serpico, Davide",
         "57197799705",
         "60015986-60021361",
         "2023-12-01",
         "December 2023",
         "Studies in History and Philosophy of Science",
         "00393681",
         "11600154632",
         "18792510",
         "Journal",
         "102",
         "NULL",
         "NULL",
         "72-83",
         "Research in pharmacogenomics and precision medicine has recently introduced the concept of Polygenic Scores (PGSs), namely, indexes that aggregate the effects that many genetic variants are predicted to have on individual disease risk. The popularity of PGSs is increasing rapidly, but surprisingly little attention has been paid to the idealisations they make about phenotypic development. Indeed, PGSs rely on quantitative genetics models and methods, which involve considerable theoretical assumptions that have been questioned on various grounds. This comes with epistemological and ethical concerns about the use of PGSs in clinical decision-making. In this paper, I investigate to what extent idealisations in genetics models can impact the data gathering and clinical interpretation of genomics findings, particularly the calculation and predictive accuracy of PGSs. Although idealisations are considered ineliminable components of scientific models, they may be legitimate or not depending on the epistemic aims of a model. I thus analyse how various idealisations have been introduced in classical models and progressively readapted throughout the history of genetic theorising. Notably, this process involved important changes in the epistemic purpose of such idealisations, which raises the question of whether they are legitimate in the context of contemporary genomics.",
         "Genetic prediction | GWAS | Idealisations | Polygenic scores | Predictive accuracy | Quantitative genetics",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "ERC",
         "805498",
         "European Research Council"
        ],
        [
         "3",
         "3",
         "2-s2.0-85152302150",
         "10.1016/j.shpsa.2023.03.002",
         "S0039368123000705",
         "37068423.0",
         "Joint representation: Modeling a phenomenon with multiple biological systems",
         "ar",
         "Article",
         "Yoshida Y.",
         "60029445",
         "University of Minnesota Twin Cities",
         "Minneapolis",
         "United States",
         "1.0",
         "Yoshida, Yoshinari",
         "57211241255",
         "60029445",
         "2023-06-01",
         "June 2023",
         "Studies in History and Philosophy of Science",
         "00393681",
         "11600154632",
         "18792510",
         "Journal",
         "99",
         "NULL",
         "NULL",
         "67-76",
         "Biologists often study particular biological systems as models of a phenomenon of interest even if they already know that the phenomenon is produced by diverse mechanisms and hence none of those systems alone can sufficiently represent it. To understand this modeling practice, the present paper provides an account of how multiple model systems can be used to study a phenomenon that is produced by diverse mechanisms. Even if generalizability of results from a single model system is significantly limited, generalizations concerning specific aspects of mechanisms often hold across certain ranges of biological systems, which enables multiple model systems to jointly represent such a phenomenon. Comparing mechanisms that operate in different biological systems as examples of the same phenomenon also facilitates characterization and investigation of individual mechanisms. I also compare my account with two existing accounts of the use of multiple model systems and argue that my account is distinct from and complementary to them.",
         "Collective cell migration | Generalization | Model organisms | Model systems | Phenomena | Representation",
         "0",
         "0",
         "repositoryvor",
         "Green",
         "UMN",
         "undefined",
         "University of Minnesota"
        ],
        [
         "4",
         "4",
         "2-s2.0-85148606416",
         "10.3390/philosophies8010008",
         "NULL",
         "NULL",
         "Turing‚Äôs Biological Philosophy: Morphogenesis, Mechanisms and Organicism",
         "ar",
         "Article",
         "Greif H.",
         "60003675",
         "Politechnika Warszawska",
         "Warsaw",
         "Poland",
         "3.0",
         "Greif, Hajo;Kubiak, Adam;Stacewicz, Pawe≈Ç",
         "35487486100;57222530470;55497157400",
         "60003675;60003675;60003675",
         "2023-02-01",
         "February 2023",
         "Philosophies",
         "NULL",
         "21101055468",
         "24099287",
         "Journal",
         "8",
         "1",
         "8",
         "NULL",
         "Alan M. Turing‚Äôs last published work and some posthumously published manuscripts were dedicated to the development of his theory of organic pattern formation. In ‚ÄúThe Chemical Basis of Morphogenesis‚Äù (1952), he provided an elaborated mathematical formulation of the theory of the origins of biological form that had been first proposed by Sir D‚ÄôArcy Wendworth Thompson in On Growth and Form (1917/1942). While arguably his most mathematically detailed and his systematically most ambitious effort, Turing‚Äôs morphogenetical writings also form the most thematically self-contained and least philosophically explored part of his work. We dedicate our inquiry to the reasons and the implications of Turing‚Äôs choice of biological topic and viewpoint. We will probe for possible factors in Turing‚Äôs choice that go beyond availability and acquaintance with On Growth and Form. On these grounds, we will explore how and to what extent his theory of morphogenesis actually ties in with his concept of mechanistic computation. Notably, Thompson‚Äôs pioneering work in biological ‚Äòstructuralism‚Äô was organicist in outlook and explicitly critical of the Darwinian approaches that were popular with Turing‚Äôs cyberneticist contemporaries‚Äîand partly used by Turing himself in his proto-connectionist models of learning. Resolving this apparent dichotomy, we demonstrate how Turing‚Äôs quest for mechanistic explanations of how organisation emerges in nature leaves room for a non-mechanist view of nature.",
         "Alan M. Turing | Darwinian evolution | D‚ÄôArcy Wendworth Thompson | explanations in science | mechanism | modern synthesis | morphogenesis | organicism",
         "1",
         "1",
         "publisherfullgold",
         "Gold",
         "NCN",
         "2020/37/B/HS1/01809",
         "Narodowe Centrum Nauki"
        ],
        [
         "5",
         "5",
         "2-s2.0-85091151894",
         "10.1007/s11245-020-09722-1",
         "NULL",
         "NULL",
         "From Obesity to Energy Metabolism: Ontological Perspectives on the Metrics of Human Bodies",
         "ar",
         "Article",
         "Serpico D.",
         "60030318;60025153",
         "Universit√† degli Studi di Milano;Universit√† degli Studi di Genova",
         "Milan;Genoa",
         "Italy;Italy",
         "2.0",
         "Serpico, Davide;Borghini, Andrea",
         "57197799705;57045736900",
         "60025153;60030318",
         "2021-07-01",
         "July 2021",
         "Topoi",
         "01677411",
         "4700152866",
         "15728749",
         "Journal",
         "40",
         "3",
         "NULL",
         "577-586",
         "In this paper, we aim at rethinking the concept of obesity in a way that better captures the connection between underlying medical aspects, on the one hand, and an individual‚Äôs developmental history, on the other. Our proposal rests on the idea that obesity is not to be understood as a phenotypic trait or character; rather, obesity represents one of the many possible states of a more complex phenotypic trait that we call ‚Äòenergy metabolism.‚Äô We argue that this apparently simple conceptual shift can help solve important theoretical misconceptions regarding the genetics, epigenetics, and development of obesity. In addition, we show that our proposal can be fruitfully paired with the concept of developmental channeling of a trait, which connects to the study of the plasticity and canalization of complex traits. Finally, we discuss the potential impact of our approach on the assessment, treatment, and social narratives of obesity.",
         "Definition of obesity | Developmental canalization | Epigenetics of obesity | Genetics of obesity | Obesity | Obesity and public health",
         "9",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "UniMi",
         "undefined",
         "Universit√† degli Studi di Milano"
        ],
        [
         "7",
         "7",
         "2-s2.0-85084479260",
         "10.1007/s10539-020-09750-6",
         "NULL",
         "NULL",
         "Beyond quantitative and qualitative traits: three telling cases in the life sciences",
         "ar",
         "Article",
         "Serpico D.",
         "60025153",
         "Universit√† degli Studi di Genova",
         "Genoa",
         "Italy",
         "1.0",
         "Serpico, Davide",
         "57197799705",
         "60025153",
         "2020-06-01",
         "1 June 2020",
         "Biology and Philosophy",
         "01693867",
         "13699",
         "15728404",
         "Journal",
         "35",
         "3",
         "34",
         "NULL",
         "This paper challenges the common assumption that some phenotypic traits are quantitative while others are qualitative. The distinction between these two kinds of traits is widely influential in biological and biomedical research as well as in scientific education and communication. This is probably due to both historical and epistemological reasons. However, the quantitative/qualitative distinction involves a variety of simplifications on the genetic causes of phenotypic variability and on the development of complex traits. Here, I examine three cases from the life sciences that show inconsistencies in the distinction: Mendelian traits (dwarfism and pigmentation in plant and animal models), Mendelian diseases (phenylketonuria), and polygenic mental disorders (schizophrenia). I show that these traits can be framed both quantitatively and qualitatively depending, for instance, on the methods through which they are investigated and on specific epistemic purposes (e.g., clinical diagnosis versus causal explanation). This suggests that the received view of quantitative and qualitative traits has a limited heuristic power‚Äîlimited to some local contexts or to the specific methodologies adopted. Throughout the paper, I provide directions for framing phenotypes beyond the quantitative/qualitative distinction. I conclude by pointing at the necessity of developing a principled characterisation of what phenotypic traits, in general, are.",
         "Character states | Genetic liability | Mendelian diseases | Qualitative traits | Quantitative traits | Thresholds",
         "16",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "Universit√† degli Studi di Genova"
        ],
        [
         "8",
         "8",
         "2-s2.0-85064435478",
         "10.1007/s10441-019-09344-0",
         "NULL",
         "30949871.0",
         "Control by Viability in a Chemotherapy Cancer Model",
         "ar",
         "Article",
         "Serhani M.",
         "60025457;60015943",
         "Hassan II University of Casablanca;Universit√© Moulay Isma√Øl",
         "Casablanca;Meknes",
         "Morocco;Morocco",
         "4.0",
         "Serhani, M.;Essaadi, H.;Kassara, K.;Boutoulout, A.",
         "6506389979;57208322026;6603249996;6506165183",
         "60015943;60015943;60025457;60015943",
         "2019-09-15",
         "15 September 2019",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "67",
         "3",
         "NULL",
         "177-200",
         "The aim of this study is to provide a feedback control, called the Chemotherapy Protocol Law, with the purpose to keep the density of tumor cells that are treated by chemotherapy below a ‚Äútolerance level‚Äù Lc, while retaining the density of normal cells above a ‚Äúhealthy level‚Äù Nc. The mathematical model is a controlled dynamical system involving three nonlinear differential equations, based on a Gompertzian law of cell growth. By evoking viability and set-valued theories, we derive sufficient conditions for the existence of a Chemotherapy Protocol Law. Thereafter, on a suitable viability domain, we build a multifunction whose selections are the required Chemotherapy Protocol Laws. Finally, we propose a design of selection that generates a Chemotherapy Protocol Law.",
         "Cancer model | Feedback control | Set-valued analysis | Viability",
         "5",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "9",
         "9",
         "2-s2.0-85131712913",
         "10.1007/s13752-019-00317-7",
         "NULL",
         "NULL",
         "Multicellular Individuality: The Case of Bacteria",
         "ar",
         "Article",
         "Ventura R.",
         "60014808;60008724",
         "Bilkent √úniversitesi;Duke University",
         "Ankara;Durham",
         "Turkey;United States",
         "1.0",
         "Ventura, Rafael",
         "57195947053",
         "60008724-60014808",
         "2019-06-15",
         "15 June 2019",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "14",
         "2",
         "NULL",
         "131-140",
         "Recent attention to complex group-level behavior amongst bacteria has led some to conceive of multicellular clusters of bacteria as individuals. In this article, I assess these recent claims by first drawing a distinction between two concepts of individuality: physiological and evolutionary. I then survey cases that are representative of three different modes of growth: myxobacteria (surface-attached agglomerative growth), Bacillus subtilis (agglomerative growth not attached to a surface), and cyanobacteria (filamentous growth). A closer look at these cases indicates that multicellular individuality among bacteria is remarkably complex. Physiologically, the three cases of multicellular clusters do not form physiological individuals. But matters are different when it comes to evolutionary individuality; although multicellular clusters that grow by agglomeration are not highly individuated, filament-forming groups achieve a relatively high degree of individuality. I also suggest that debates about bacterial multicellular individuality may have been obscured by a failure to see that selection on highly individuated groups is by no means the only mechanism to bring about the complex group-level behaviors that have led some to view bacteria as multicellular individuals.",
         "B. subtilis | Bacteria | Cyanobacteria | Individuality | Multicellularity | Myxobacteria",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "NSFC",
         "12171212",
         "National Natural Science Foundation of China"
        ],
        [
         "10",
         "10",
         "2-s2.0-85064948392",
         "10.1002/bies.201800251",
         "NULL",
         "30970170.0",
         "Why the Lipid Divide? Membrane Proteins as Drivers of the Split between the Lipids of the Three Domains of Life",
         "ar",
         "Article",
         "Sojo V.",
         "112777647",
         "Institute for Advanced Study (Wissenschaftskolleg) Berlin",
         "Berlin",
         "Germany",
         "1.0",
         "Sojo, Victor",
         "57091567800",
         "112777647",
         "2019-05-01",
         "May 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "5",
         "1800251",
         "NULL",
         "Recent results from engineered and natural samples show that the starkly different lipids of archaea and bacteria can form stable hybrid membranes. But if the two types can mix, why don't they? That is, why do most bacteria and all eukaryotes have only typically bacterial lipids, and archaea archaeal lipids? It is suggested here that the reason may lie on the other main component of cellular membranes: membrane proteins, and their close adaptation to the lipids. Archaeal lipids in modern bacteria could suggest that the last universal common ancestor (LUCA) had both lipid types. However, this would imply a rather elaborate evolutionary scenario, while negating simpler alternatives. In light of widespread horizontal gene transfer across the prokaryotic domains, hybrid membranes reveal that the lipid divide did not just occur once at the divergence of archaea and bacteria from LUCA. Instead, it continues to occur actively to this day.",
         "archaea | bacteria | divergence | eukaryotes | evolution | horizontal gene transfer | last universal common ancestor (LUCA)",
         "26",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "ALTF-1455-2015",
         "Simons Foundation"
        ],
        [
         "12",
         "12",
         "2-s2.0-85064907924",
         "10.1002/bies.201800234",
         "NULL",
         "30970171.0",
         "Targeting the Spleen as an Alternative Site for Hematopoiesis",
         "ar",
         "Article",
         "Short C.",
         "60031602",
         "Bond University",
         "Gold Coast",
         "Australia",
         "4.0",
         "Short, Christie;Lim, Hong K.;Tan, Jonathan;O'Neill, Helen C.",
         "57208484540;57207580388;56328599000;7102156014",
         "60031602;60031602;60031602;60031602",
         "2019-05-01",
         "May 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "5",
         "1800234",
         "NULL",
         "Bone marrow is the main site for hematopoiesis in adults. It acts as a niche for hematopoietic stem cells (HSCs) and contains non-hematopoietic cells that contribute to stem cell dormancy, quiescence, self-renewal, and differentiation. HSC also exist in resting spleen of several species, although their contribution to hematopoiesis under steady-state conditions is unknown. The spleen can however undergo extramedullary hematopoiesis (EMH) triggered by physiological stress or disease. With the loss of bone marrow niches in aging and disease, the spleen as an alternative tissue site for hematopoiesis is an important consideration for future therapy, particularly during HSC transplantation. In terms of harnessing the spleen as a site for hematopoiesis, here the remarkable regenerative capacity of the spleen is considered with a view to forming additional or ectopic spleen tissue through cell engraftment. Studies in mice indicate the potential for such grafts to support the influx of hematopoietic cells leading to the development of normal spleen architecture. An important goal will be the formation of functional ectopic spleen tissue as an aid to hematopoietic recovery following clinical treatments that impact bone marrow. For example, expansion or replacement of niches could be considered where myeloablation ahead of HSC transplantation compromises treatment outcomes.",
         "extramedullary hematopoiesis | hematopoietic stem cells | myelopoiesis | spleen | stem cell niche | tissue regeneration | transplantation",
         "34",
         "0",
         "repositoryam",
         "Green",
         "ARC",
         "130101703",
         "Australian Research Council"
        ],
        [
         "13",
         "13",
         "2-s2.0-85064391368",
         "10.1007/s13752-018-0311-0",
         "NULL",
         "NULL",
         "The Nature of Programmed Cell Death",
         "ar",
         "Article",
         "Durand P.M.",
         "60025063;60016218",
         "KU Leuven;University of the Witwatersrand, Johannesburg",
         "Leuven;Johannesburg",
         "Belgium;South Africa",
         "2.0",
         "Durand, Pierre M.;Ramsey, Grant",
         "24343664500;14521522000",
         "60016218;60025063",
         "2019-03-15",
         "15 March 2019",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "14",
         "1",
         "NULL",
         "30-41",
         "In multicellular organisms, cells are frequently programmed to die. This makes good sense: cells that fail to, or are no longer playing important roles are eliminated. From the cell‚Äôs perspective, this also makes sense, since somatic cells in multicellular organisms require the cooperation of clonal relatives. In unicellular organisms, however, programmed cell death (PCD) poses a difficult and unresolved evolutionary problem. The empirical evidence for PCD in diverse microbial taxa has spurred debates about what precisely PCD means in the case of unicellular organisms (how it should be defined). In this article, we survey the concepts of PCD in the literature and the selective pressures associated with its evolution. We show that definitions of PCD have been almost entirely mechanistic and fail to separate questions concerning what PCD fundamentally is from questions about the kinds of mechanisms that realize PCD. We conclude that an evolutionary definition is best able to distinguish PCD from closely related phenomena. Specifically, we define ‚Äútrue‚Äù PCD as an adaptation for death triggered by abiotic or biotic environmental stresses. True PCD is thus not only an evolutionary product but must also have been a target of selection. Apparent PCD resulting from pleiotropy, genetic drift, or trade-offs is not true PCD. We call this ‚Äúersatz PCD‚Äù.",
         "Adaptation | Aging | Apoptosis | Price equation | Programmed cell death | Selection | Unicellular organisms",
         "33",
         "0",
         "repositoryam",
         "Green",
         "NRF",
         "21772155",
         "National Research Foundation"
        ],
        [
         "19",
         "19",
         "2-s2.0-84912025857",
         "10.1016/j.shpsc.2014.08.003",
         "S1369848614001083",
         "25220402.0",
         "The machine conception of the organism in development andevolution: A critical analysis",
         "ar",
         "Article",
         "Nicholson D.J.",
         "60026479",
         "University of Exeter",
         "Exeter",
         "United Kingdom",
         "1.0",
         "Nicholson, Daniel J.",
         "36189968300",
         "60026479",
         "2014-12-01",
         "December 01, 2014",
         "Studies in History and Philosophy of Science Part C :Studies in History and Philosophy of Biological and Biomedical Sciences",
         "13698486",
         "14281",
         "NULL",
         "Journal",
         "48",
         "PB",
         "NULL",
         "162-174",
         "This article critically examines one of the most prevalent metaphors in contemporary biology, namely the machine conception of the organism (MCO). Although the fundamental differences between organisms and machines make the MCO an inadequate metaphor for conceptualizing living systems, many biologists and philosophers continue to draw upon the MCO or tacitly accept it as the standard model of the organism. The analysis presented here focuses on the specific difficulties that arise when the MCO is invoked in the contexts of development and evolution. In developmental biology the MCO underlies a logically incoherent model of ontogeny, the genetic program, which serves to legitimate three problematic theses about development: genetic animism, neo-preformationism, and developmental computability. In evolutionary biology the MCO is responsible for grounding unwarranted theoretical appeals to the concept of design as well as to the interpretation of natural selection as an engineer, which promote a distorted understanding of the process and products of evolutionary change. Overall, it is argued that, despite its heuristic value, the MCO today is impeding rather than enabling further progress in our comprehension of living systems.",
         "Design | Engineering | Genetic program | Machine | Metaphor | Organism",
         "54",
         "0",
         "repositoryvor",
         "Green",
         "FP7",
         "324186",
         "Seventh Framework Programme"
        ],
        [
         "20",
         "20",
         "2-s2.0-85062109763",
         "10.1002/bies.201800198",
         "NULL",
         "30805984.0",
         "Veins and Arteries Build Hierarchical Branching Patterns Differently: Bottom-Up versus Top-Down",
         "ar",
         "Article",
         "Red-Horse K.",
         "60012708;60003711",
         "Stanford University;University of Pennsylvania Perelman School of Medicine",
         "Stanford;Philadelphia",
         "United States;United States",
         "2.0",
         "Red-Horse, Kristy;Siekmann, Arndt F.",
         "6505998386;15923667100",
         "60012708;60003711",
         "2019-03-01",
         "March 2019",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "41",
         "3",
         "1800198",
         "NULL",
         "A tree-like hierarchical branching structure is present in many biological systems, such as the kidney, lung, mammary gland, and blood vessels. Most of these organs form through branching morphogenesis, where outward growth results in smaller and smaller branches. However, the blood vasculature is unique in that it exists as two trees (arterial and venous) connected at their tips. Obtaining this organization might therefore require unique developmental mechanisms. As reviewed here, recent data indicate that arterial trees often form in reverse order. Accordingly, initial arterial endothelial cell differentiation occurs outside of arterial vessels. These pre-artery cells then build trees by following a migratory path from smaller into larger arteries, a process guided by the forces imparted by blood flow. Thus, in comparison to other branched organs, arteries can obtain their structure through inward growth and coalescence. Here, new information on the underlying mechanisms is discussed, and how defects can lead to pathologies, such as hypoplastic arteries and arteriovenous malformations.",
         "angiogenesis | arteries | blood flow | branching morphogenesis | coronary vasculature | cxcr4 | notch | vein",
         "52",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "EXC 1003-CIM",
         "National Institutes of Health"
        ],
        [
         "31",
         "31",
         "2-s2.0-85048795662",
         "10.1002/bies.201800031",
         "NULL",
         "29869337.0",
         "Synapse Pruning: Mitochondrial ROS with Their Hands on the Shears",
         "ar",
         "Article",
         "Cobley J.",
         "60105228",
         "University of the Highlands and Islands",
         "Inverness",
         "United Kingdom",
         "1.0",
         "Cobley, James N.",
         "54583377000",
         "60105228",
         "2018-07-01",
         "July 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "7",
         "1800031",
         "NULL",
         "No overarching hypotheses tie the basic mechanisms of mitochondrial reactive oxygen species (ROS) production to activity dependent synapse pruning‚Äîa fundamental biological process in health and disease. Neuronal activity divergently regulates mitochondrial ROS: activity decreases whereas inactivity increases their production, respectively. Placing mitochondrial ROS as innate synaptic activity sentinels informs the novel hypothesis that: (1) at an inactive synapse, increased mitochondrial ROS production initiates intrinsic apoptosis dependent pruning; and (2) at an active synapse, decreased mitochondrial ROS production masks intrinsic apoptosis dependent pruning. Immature antioxidant defense may enable the developing brain to harness mitochondrial ROS to prune weak synapses. Beyond development, endogenous antioxidant defense constrains mitochondrial (ROS) to mask pruning. Unwanted age-related synapse loss may arise when mitochondrial ROS aberrantly recapitulate developmental pruning. Placing mitochondrial ROS with their hands on the shears is beneficial in early but deleterious in later life.",
         "development | hydrogen peroxide | mitochondria | neurodegeneration | superoxide | synapse pruning",
         "24",
         "0",
         "NULL",
         "NULL",
         "HIE",
         "JNC007",
         "Highlands and Islands Enterprise"
        ],
        [
         "33",
         "33",
         "2-s2.0-85048711745",
         "10.1002/bies.201800008",
         "NULL",
         "29708272.0",
         "Three Distinct Types of Microautophagy Based on Membrane Dynamics and Molecular Machineries",
         "ar",
         "Article",
         "Oku M.",
         "60120432;60011001",
         "Center for the Promotion of Interdisciplinary Education and Research;Kyoto University",
         "Kyoto;Kyoto",
         "Japan;Japan",
         "2.0",
         "Oku, Masahide;Sakai, Yasuyoshi",
         "7005419880;7403476299",
         "60011001;60011001-60120432",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1800008",
         "NULL",
         "Microautophagy is originally defined as lysosomal (vacuolar) membrane dynamics to directly enwrap and transport cytosolic components into the lumen of the lytic organelle. Molecular details of microautophagy had remained unknown until genetic studies in yeast identified a set of proteins required for the process. Subsequent studies with other experimental model organisms resulted in a series of discoveries that accompanied an expansion of the definition of microautophagy to also encompass endosomal membrane dynamics. These findings, however, still impose puzzling, non-integrated images as to the molecular mechanism of microautophagy. By reviewing recent studies on microautophagy in various experimental systems, we propose the classification of microautophagy into three types, as the basis for developing a comprehensive view of the process.",
         "autophagy | lysosome | macroautophagy | microautophagy | vacuole",
         "190",
         "1",
         "repositoryvor",
         "Green",
         "JSPS",
         "16K07689",
         "Japan Society for the Promotion of Science"
        ],
        [
         "34",
         "34",
         "2-s2.0-85048710809",
         "10.1002/bies.201700232",
         "NULL",
         "29682771.0",
         "Novel Channels of the Outer Membrane of Mitochondria: Recent Discoveries Change Our View",
         "ar",
         "Article",
         "Checchetto V.",
         "60000481",
         "Universit√† degli Studi di Padova",
         "Padua",
         "Italy",
         "2.0",
         "Checchetto, Vanessa;Szabo, Ildiko",
         "36459522800;35595280300",
         "60000481;60000481",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1700232",
         "NULL",
         "Ion channels mediate ion flux across biological membranes and regulate important organellar and cellular tasks. A recent study revealed the presence of four new proteins, the MIM complex (composed by Mim1 and Mim2), Ayr1, OMC7, and OMC8, that are able to form ion-conducting channels in the outer mitochondria membrane (OMM). These findings strongly indicate that the OMM is endowed with many solute-specific channels, in addition to porins and known channels mediating protein import into mitochondria. These solute-specific channels provide essential pathways for the controlled transport of ions and metabolites and may thus add a further layer of specificity to the regulation of mitochondrial function at the organelle-cytosol and/or inter-organellar interface. Future studies will be required to fully understand the way(s) of regulation of these new channels and to integrate them into signaling pathways within the cells.",
         "in vitro expression | ion channels | mitochondria | outer membrane | planar lipid bilayer | Œ≤-barrel proteins",
         "12",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "35",
         "35",
         "2-s2.0-85048702395",
         "10.1002/bies.201700245",
         "NULL",
         "29750374.0",
         "Endothelial Metabolic Control of Lymphangiogenesis",
         "ar",
         "Article",
         "Yu P.",
         "60069758;60017994",
         "Changhai Hospital;Yale School of Medicine",
         "Shanghai;New Haven",
         "China;United States",
         "4.0",
         "Yu, Pengchun;Wu, Guosheng;Lee, Heon Woo;Simons, Michael",
         "35799246300;56342554600;13002676700;7201538867",
         "60017994;60017994-60069758;60017994;60017994",
         "2018-06-01",
         "June 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "6",
         "1700245",
         "NULL",
         "Lymphangiogenesis is an important developmental process that is critical to regulation of fluid homeostasis, immune surveillance and response as well as pathogenesis of a number of diseases, among them cancer, inflammation, and heart failure. Specification, formation, and maturation of lymphatic blood vessels involves an interplay between a series of events orchestrated by various transcription factors that determine expression of key genes involved in lymphangiogenesis. These are traditionally thought to be under control of several key growth factors including vascular growth factor-C (VEGF-C) and fibroblast growth factors (FGFs). Recent insights into VEGF and FGF signaling point to their role in control of endothelial metabolic processes such as glycolysis and fatty acid oxidation that, in turn, play a major role in regulation of lymphangiogenesis. These advances have significantly increased our understanding of lymphatic biology and opened new therapeutic vistas. Here we review our current understanding of metabolic controls in the lymphatic vasculature.",
         "endothelial cells | fatty acid Œ≤-oxidation | glycolysis | growth factor signaling | lymphangiogenesis | lymphatics | metabolism",
         "12",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "HL053793",
         "National Institutes of Health"
        ],
        [
         "39",
         "39",
         "2-s2.0-85107490355",
         "10.1007/s13752-017-0268-4",
         "NULL",
         "NULL",
         "Reconsidering Morphology Through an Experimental Case Study",
         "ar",
         "Article",
         "Albertazzi L.",
         "60015986;60000481",
         "Universit√† di Trento;Universit√† degli Studi di Padova",
         "Trento;Padua",
         "Italy;Italy",
         "6.0",
         "Albertazzi, Liliana;Canal, Luisa;Chist√®, Paolo;De Rosa, Mara;Micciolo, Rocco;Minelli, Alessandro",
         "56365267800;55900401400;57203202727;57324928300;7005868993;7004539407",
         "60015986;60015986;60015986;60015986;60015986;60000481",
         "2017-09-01",
         "September 2017",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "12",
         "3",
         "NULL",
         "131-141",
         "This study analyzes shells of marine gastropods of a zoological museum and the Latin epithets expressing perceptual and connotative attributes that they have received in the standard, Linnaean nomenclature. Making use of the Osgood semantic differential, we presented the subjects with digital 3-D reproductions of the shell specimens to be subjectively evaluated according to 17 pairs of attributes. The results show that, overall, the subjective evaluations given by the subjects are consistent, which suggests that an intersubjective characterization of the shells as wholes according to their qualitative and morphological traits is possible and scientifically explicable. This may apply to other biological items as well. The results also show that the epithets given to the shells by taxonomists do not always reflect the perceptual evaluation of the general population, being a product of the well-known ambiguity in taxonomic categorization and naming.",
         "Linnaean nomenclature | Morphology | Shape invariants | Taxonomic categorization",
         "2",
         "0",
         "NULL",
         "NULL",
         "UNIBO",
         "undefined",
         "Universit√† di Bologna"
        ],
        [
         "40",
         "40",
         "2-s2.0-84995616598",
         "10.1007/s10441-016-9301-4",
         "NULL",
         "27853896.0",
         "Discrete Mesh Approach in Morphogenesis Modelling: the Example of Gastrulation",
         "ar",
         "Article",
         "Demongeot J.",
         "60108275;60104653",
         "Techniques for Biomedical Engineering and Complexity Management Informatics, Mathematics and Applications Grenoble (TIMC-IMAG);Universit√© Grenoble Alpes",
         "La Tronche;Saint Martin d'Heres",
         "France;France",
         "3.0",
         "Demongeot, J.;Lontos, A.;Promayon, E.",
         "7005372567;38561637100;6507228109",
         "60104653;60108275;60108275",
         "2016-12-01",
         "1 December 2016",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "64",
         "4",
         "NULL",
         "427-446",
         "Morphogenesis is a general concept in biology including all the processes which generate tissue shapes and cellular organizations in a living organism. Many hybrid formalizations (i.e., with both discrete and continuous parts) have been proposed for modelling morphogenesis in embryonic or adult animals, like gastrulation. We propose first to study the ventral furrow invagination as the initial step of gastrulation, early stage of embryogenesis. We focus on the study of the connection between the apical constriction of the ventral cells and the initiation of the invagination. For that, we have created a 3D biomechanical model of the embryo of the Drosophila melanogaster based on the finite element method. Each cell is modelled by an elastic hexahedron contour and is firmly attached to its neighbouring cells. A uniform initial distribution of elastic and contractile forces is applied to cells along the model. Numerical simulations show that invagination starts at ventral curved extremities of the embryo and then propagates to the ventral medial layer. Then, this observation already made in some experiments can be attributed uniquely to the specific shape of the embryo and we provide mechanical evidence to support it. Results of the simulations of the ‚Äúpill-shaped‚Äù geometry of the Drosophila melanogaster embryo are compared with those of a spherical geometry corresponding to the Xenopus l√¶vis embryo. Eventually, we propose to study the influence of cell proliferation on the end of the process of invagination represented by the closure of the ventral furrow.",
         "Discrete mesh modelling | Gastrulation | Morphogenesis | Ventral furrow",
         "0",
         "0",
         "NULL",
         "NULL",
         "EC",
         "undefined",
         "European Commission"
        ],
        [
         "42",
         "42",
         "2-s2.0-0023504894",
         "10.1007/BF00052064",
         "NULL",
         "3128032.0",
         "Mathematical model and simulation of retina and tectum opticum of lower vertebrates",
         "ar",
         "Article",
         "an der Heiden U.",
         "60022113;60008293",
         "Universit√§t Witten/Herdecke;Universit√§t Bremen",
         "Witten;Bremen",
         "Germany;Germany",
         "2.0",
         "an der Heiden, U.;Roth, G.",
         "6701417263;7202210916",
         "60022113;60008293",
         "1987-09-01",
         "September 1987",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "36",
         "3",
         "NULL",
         "179-212",
         "The processing of information within the retino-tectal visual system of amphibians is decomposed into five major operational stages, three of them taking place in the retina and two in the optic tectum. The stages in the retina involve (i) a spatially local high-pass filtering in connection to the perception of moving objects, (ii) separation of the receptor activity into ON- and OFF-channels regarding the distinction of objects on both light and dark backgrounds, (iii) spatial integration via near excitation and far-reaching inhibition. Variation of the spatial range of excitation and inhibition allows to account for typical activities observed in a variety of classes of retina ganglion cells. Mathematical description of the operations in the tectum opticum include (i) spatial summation of retinal output (mainly of class-2 and class-3 retina ganglion cells), and (ii) direct or indirect lateral inhibition between tectal cells. In the computer simulation, first the output of the mathematical retina model is computed which, then, is used as the input to the tectum model. The full spatio-temporal dynamics is taken into account. The simulations show that different combinations of strength of lateral inhibition on the one side and the response properties of the retina ganglion cells on the other side determine the response properties of tectal cell types involved in object recognition. ¬© 1987 Martinus Nijhoff/Dr W. Junk Publishers.",
         "NULL",
         "15",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "43",
         "43",
         "2-s2.0-85044358499",
         "10.1002/bies.201800029",
         "NULL",
         "29575018.0",
         "The Diversification of Early Emerging Metazoans: A Window into the Evolution of Animal Multicellularity",
         "ar",
         "Article",
         "Revilla-i-Domingo R.",
         "60025988",
         "Universit√§t Wien",
         "Vienna",
         "Austria",
         "2.0",
         "Revilla-i-Domingo, Roger;Simakov, Oleg",
         "6508022715;12783132000",
         "60025988;60025988",
         "2018-05-01",
         "May 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "5",
         "1800029",
         "NULL",
         "The biannual international workshop entitled ‚ÄúThe diversification of early emerging metazoans: A window into animal evolution?‚Äù took place at the Evangelische Akademie Tutzing, Germany, 11‚Äì14. September 2017. It was organized by Thomas Bosch (Kiel), Thomas Holstein (Heidelberg), and Ulrich Technau (Vienna), and it was sponsored by the Deutsche Forschungsgemeinschaft (DFG). The meeting gathered over 140 researchers to discuss the contribution of non-bilaterian metazoan models (Porifera, Ctenophora, Placozoa, and Cnidaria) to our understanding of: a. The evolution of metazoan developmental processes; b. Fundamental molecular mechanisms underlying metazoan features; and c. The complex interactions that animals establish with their environment.",
         "NULL",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "46",
         "46",
         "2-s2.0-85043275933",
         "10.1002/bies.201700241",
         "NULL",
         "29508910.0",
         "The Rise of the Cartwheel: Seeding the Centriole Organelle",
         "ar",
         "Article",
         "Guichard P.",
         "60028186;60004718",
         "√âcole Polytechnique F√©d√©rale de Lausanne;Universit√© de Gen√®ve",
         "Lausanne;Geneva",
         "Switzerland;Switzerland",
         "3.0",
         "Guichard, Paul;Hamel, Virginie;G√∂nczy, Pierre",
         "25936173500;8971305600;35446142600",
         "60004718;60004718;60028186",
         "2018-04-01",
         "April 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "4",
         "1700241",
         "NULL",
         "The cartwheel is a striking structure critical for building the centriole, a microtubule-based organelle fundamental for organizing centrosomes, cilia, and flagella. Over the last 50 years, the cartwheel has been described in many systems using electron microscopy, but the molecular nature of its constituent building blocks and their assembly mechanisms have long remained mysterious. Here, we review discoveries that led to the current understanding of cartwheel structure, assembly, and function. We focus on the key role of SAS-6 protein self-organization, both for building the signature ring-like structure with hub and spokes, as well as for their vertical stacking. The resemblance of assembly intermediates in vitro and in vivo leads us to propose a novel registration step in cartwheel biogenesis, whereby stacked SAS-6-containing rings are put in register through interactions with peripheral elements anchored to microtubules. We conclude by evoking some avenues for further uncovering cartwheel and centriole assembly mechanisms.",
         "basal bodies | cartwheel | centrioles | cilia | SAS-6",
         "37",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "ERC",
         "340227",
         "European Research Council"
        ],
        [
         "47",
         "47",
         "2-s2.0-85038865269",
         "10.1002/bies.201700018",
         "NULL",
         "29266341.0",
         "Microbiome-Germline Interactions and Their Transgenerational Implications",
         "ar",
         "Article",
         "Elgart M.",
         "119075263",
         "Biomolecular Sciences",
         "Rehovot",
         "Israel",
         "2.0",
         "Elgart, Michael;Soen, Yoav",
         "55964995000;6602347180",
         "119075263;119075263",
         "2018-04-01",
         "April 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "4",
         "1700018",
         "NULL",
         "It is becoming increasingly clear that most, if not all, animals and plants are associated with a diverse array of resident gut microbiota. This symbiosis is regulated by host-microbiome interactions which influence the development, homeostasis, adaptation and evolution of the host. Recent evidence indicated that these interactions can also affect the host germline and have a potential of supporting transgenerational effects, including inheritance of acquired characteristics. Taken together, the influence of gut bacteria on the host soma and germline could potentially give rise to emergent phenotypes, which may be partially inherited by three distinguishable modes of transgenerational influence of gut bacteria: 1) ‚Äúsoma-to-soma‚Äù 2) ‚Äúsoma-to-germline‚Äù and 3) ‚Äúsoma-germline-soma‚Äù. Here, we discuss these possibilities in light of evidence supporting bacterial-mediated modes of transgenerational inheritance.",
         "bacteria | germline | holobiont | influence | inheritance | soma | transgenerational",
         "9",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "48",
         "48",
         "2-s2.0-85041576154",
         "10.1002/bies.201700191",
         "NULL",
         "29405334.0",
         "Mitonuclear Mate Choice: A Missing Component of Sexual Selection Theory?",
         "ar",
         "Article",
         "Hill G.",
         "60011754",
         "Auburn University",
         "Auburn",
         "United States",
         "1.0",
         "Hill, Geoffrey E.",
         "35500077500",
         "60011754",
         "2018-03-01",
         "March 2018",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "40",
         "3",
         "1700191",
         "NULL",
         "The fitness of a eukaryote hinges on the coordinated function of the products of its nuclear and mitochondrial genomes in achieving oxidative phosphorylation (OXPHOS). I propose that sexual selection plays a key role in the maintenance of mitonuclear coadaptation across generations because it enables pre-zygotic sorting for coadapted mitonuclear genotypes. At each new generation, sexual reproduction creates new combinations of nuclear and mitochondrial genes, and the potential arises for mitonuclear incompatibilities and reduced fitness. In reviewing the literature, I hypothesize that individuals engaged in mate choice select partners with correct species-typical mitochondrial and nuclear genotypes as well as individuals with highly functional cellular respiration. The implication is that mate choice for compatible nuclear and mitochondrial genes can play a significant role in generating the patterns of ornamentation and preferences observed in animals. A number of testable predictions emerge from this mitonuclear compatibility hypothesis of sexual selection.",
         "indicator model | mitochondria | OXPHOS | reinforcement | sexual selection",
         "24",
         "0",
         "NULL",
         "NULL",
         "AU",
         "undefined",
         "Auburn University"
        ],
        [
         "54",
         "54",
         "2-s2.0-85007335058",
         "10.1002/bies.201600234",
         "NULL",
         "28009049.0",
         "Zebrafish adult pigment stem cells are multipotent and form pigment cells by a progressive fate restriction process: Clonal analysis identifies shared origin of all pigment cell types",
         "ar",
         "Article",
         "Kelsh R.N.",
         "60280747;60171512",
         "University of Bath, Department of Life Sciences;University of Bath, Department of Mathematical Sciences",
         "Bath;Bath",
         "United Kingdom;United Kingdom",
         "4.0",
         "Kelsh, Robert N.;Sosa, Karen C.;Owen, Jennifer P.;Yates, Christian A.",
         "7004607471;57192688132;57192680270;26532007600",
         "60280747;60280747;60171512;60171512",
         "2017-03-01",
         "1 March 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "3",
         "1600234",
         "NULL",
         "Skin pigment pattern formation is a paradigmatic example of pattern formation. In zebrafish, the adult body stripes are generated by coordinated rearrangement of three distinct pigment cell-types, black melanocytes, shiny iridophores and yellow xanthophores. A stem cell origin of melanocytes and iridophores has been proposed although the potency of those stem cells has remained unclear. Xanthophores, however, seemed to originate predominantly from proliferation of embryonic xanthophores. Now, data from Singh et al. shows that all three cell-types derive from shared stem cells, and that these cells generate peripheral neural cell-types too. Furthermore, clonal compositions are best explained by a progressive fate restriction model generating the individual cell-types. The numbers of adult pigment stem cells associated with the dorsal root ganglia remain low, but progenitor numbers increase significantly during larval development up to metamorphosis, likely via production of partially restricted progenitors on the spinal nerves.",
         "iridophore | melanocyte | neural crest | pigment pattern formation | stem cell | zebrafish",
         "14",
         "1",
         "repositoryam",
         "Green",
         "BBSRC",
         "BB/ M009122/1",
         "Biotechnology and Biological Sciences Research Council"
        ],
        [
         "55",
         "55",
         "2-s2.0-85008481774",
         "10.1002/bies.201600212",
         "NULL",
         "28026040.0",
         "Forces positioning the mitotic spindle: Theories, and now experiments",
         "ar",
         "Article",
         "Wu H.Y.",
         "60105360;60077572;60009982;60006303;60003261",
         "Simons Foundation;Harvard John A. Paulson School of Engineering and Applied Sciences;Harvard University;Harvard Faculty of Arts and Sciences;Courant Institute of Mathematical Sciences",
         "New York;Cambridge;Cambridge;Cambridge;New York",
         "United States;United States;United States;United States;United States",
         "4.0",
         "Wu, Hai Yin;Nazockdast, Ehssan;Shelley, Michael J.;Needleman, Daniel J.",
         "8641868200;24475132400;7006987491;6701620355",
         "60009982-60006303;60105360;60105360-60003261;60006303-60077572-60009982",
         "2017-02-01",
         "1 February 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "2",
         "1600212",
         "NULL",
         "The position of the spindle determines the position of the cleavage plane, and is thus crucial for cell division. Although spindle positioning has been extensively studied, the underlying forces ultimately responsible for moving the spindle remain poorly understood. A recent pioneering study by Garzon-Coral et al. uses magnetic tweezers to perform the first direct measurements of the forces involved in positioning the mitotic spindle. Combining this with molecular perturbations and geometrical effects, they use their data to argue that the forces that keep the spindle in its proper position for cell division arise from astral microtubules growing and pushing against the cell's cortex. Here, we review these ground-breaking experiments, the various biomechanical models for spindle positioning that they seek to differentiate, and discuss new questions raised by these measurements.",
         "force measurement | magnetic tweezers | microtubule | pronuclear migration and rotation | spindle | spindle positioning",
         "31",
         "0",
         "NULL",
         "NULL",
         "NIGMS",
         "R01GM104976",
         "National Institute of General Medical Sciences"
        ],
        [
         "57",
         "57",
         "2-s2.0-85008190908",
         "10.1002/bies.201600174",
         "NULL",
         "28054713.0",
         "Alternating terminal electron-acceptors at the basis of symbiogenesis: How oxygen ignited eukaryotic evolution",
         "ar",
         "Article",
         "Speijer D.",
         "60002483",
         "Universiteit van Amsterdam",
         "Amsterdam",
         "Netherlands",
         "1.0",
         "Speijer, Dave",
         "6602513652",
         "60002483",
         "2017-02-01",
         "1 February 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "2",
         "1600174",
         "NULL",
         "What kind of symbiosis between archaeon and bacterium gave rise to their eventual merger at the origin of the eukaryotes? I hypothesize that conditions favouring bacterial uptake were based on exchange of intermediate carbohydrate metabolites required by recurring changes in availability and use of the two different terminal electron chain acceptors, the bacterial one being oxygen. Oxygen won, and definitive loss of the archaeal membrane potential allowed permanent establishment of the bacterial partner as the proto-mitochondrion, further metabolic integration and highly efficient ATP production. This represents initial symbiogenesis, when crucial eukaryotic traits arose in response to the archaeon-bacterium merger. The attendant generation of internal reactive oxygen species (ROS) gave rise to a myriad of further eukaryotic adaptations, such as extreme mitochondrial genome reduction, nuclei, peroxisomes and meiotic sex. Eukaryotic origins could have started with shuffling intermediate metabolites as is still essential today.",
         "eukaryotic evolution | FADH /NADH ratio 2 | genome reduction | LECA | mitochondria | oxygen radicals | symbiogenesis",
         "30",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "58",
         "58",
         "2-s2.0-85007346825",
         "10.1002/bies.201600171",
         "NULL",
         "28026026.0",
         "MAMs are attractive targets for bacterial repurposing of the host cell: MAM-functions might be key for undermining an infected cell",
         "ar",
         "Article",
         "Escoll P.",
         "60008134;60001772",
         "CNRS Centre National de la Recherche Scientifique;Institut Pasteur, Paris",
         "Paris;Paris",
         "France;France",
         "3.0",
         "Escoll, Pedro;Rolando, Monica;Buchrieser, Carmen",
         "6603481451;56353351200;7003363257",
         "60001772-60008134;60001772-60008134;60001772-60008134",
         "2017-02-01",
         "1 February 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "2",
         "1600171",
         "NULL",
         "Pathogenic bacteria frequently target the endoplasmic reticulum (ER) and mitochondria in order to exploit host functions. ER-mitochondria inter-organelle communication is topologically sub-compartmentalized at mitochondria-associated ER membranes (MAMs). MAMs are specific membranous microdomains with unique regulatory functions such as lipid synthesis and trafficking, calcium homeostasis, mitochondrial morphology, inflammasome activation, autophagosome formation, and apoptosis. These important cellular processes are all modulated by pathogens to subvert host functions and promote infection, thus it is tempting to assume that pathogenic bacteria target MAMs to subvert these different pathways in their hosts. First lines of evidence that support this hypothesis come from Legionella pneumophila. This intracellular bacterium secretes an effector that exhibits sphingosine-1 phosphate lyase activity (LpSpl) that seems to target MAMs to modulate the autophagy response to infection. Here we thus propose the concept that MAMs could be targeted by pathogenic bacteria to undermine key host cellular processes.",
         "autophagy | Legionella pneumophila | mitochondria | mitochondria-associated ER membranes",
         "15",
         "0",
         "NULL",
         "NULL",
         "FRM",
         "ANR-10-LABX-62-IBEID",
         "Fondation pour la Recherche M√©dicale"
        ],
        [
         "59",
         "59",
         "2-s2.0-85006333268",
         "10.1002/bies.201600123",
         "NULL",
         "27930828.0",
         "Integrin and cadherin clusters: A robust way to organize adhesions for cell mechanics",
         "ar",
         "Article",
         "Changede R.",
         "60106367",
         "Mechanobiology Institute, National University of Singapore",
         "Singapore City",
         "Singapore",
         "2.0",
         "Changede, Rishita;Sheetz, Michael",
         "15838906800;7103347148",
         "60106367;60106367",
         "2017-01-01",
         "1 January 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "1",
         "NULL",
         "1-12",
         "Recent studies at the nanometer scale have revealed that relatively uniform clusters of adhesion proteins (50‚Äì100 nm) constitute the modular units of cell adhesion sites in both cell-matrix and cell-cell adhesions. Super resolution microscopy and membrane protein diffusion studies both suggest that even large focal adhesions are formed of 100 nm clusters that are loosely aggregated. Clusters of 20‚Äì50 adhesion molecules (integrins or cadherins) can support large forces through avidity binding interactions but can also be disassembled or endocytosed rapidly. Assembly of the clusters of integrins is force-independent and involves gathering integrins at ligand binding sites where they are stabilized by cytoplasmic adhesion proteins that crosslink the integrin cytoplasmic tails plus connect the clusters to the cell cytoskeleton. Cooperative-signaling events can occur in a single cluster without cascading to other clusters. Thus, the clusters appear to be very important elements in many cellular processes and can be considered as a critical functional module.",
         "adhesions | cadherin | cell forces | integrin | membrane signaling",
         "101",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "60",
         "60",
         "2-s2.0-85006094535",
         "10.1002/bies.201600186",
         "NULL",
         "27896818.0",
         "Coupling immunity and programmed cell suicide in prokaryotes: Life-or-death choices",
         "ar",
         "Article",
         "Koonin E.V.",
         "60100145;60022195;60001536;60001001;106627560",
         "McGovern Institute;Massachusetts Institute of Technology;National Library of Medicine (NLM);Broad Institute;Departments of Brain and Cognitive Science and Biological Engineering",
         "Cambridge;Cambridge;Bethesda;Cambridge;Cambridge",
         "United States;United States;United States;United States;United States",
         "2.0",
         "Koonin, Eugene V.;Zhang, Feng",
         "36039332000;58377871700",
         "60001536;60001001-60022195-60100145-106627560",
         "2017-01-01",
         "1 January 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "1",
         "NULL",
         "1-9",
         "Host-pathogen arms race is a universal, central aspect of the evolution of life. Most organisms evolved several distinct yet interacting strategies of anti-pathogen defense including resistance to parasite invasion, innate and adaptive immunity, and programmed cell death (PCD). The PCD is the means of last resort, a suicidal response to infection that is activated when resistance and immunity fail. An infected cell faces a decision between active defense and altruistic suicide or dormancy induction, depending on whether immunity is ‚Äúdeemed‚Äù capable of preventing parasite reproduction and consequent infection of other cells. In bacteria and archaea, immunity genes typically colocalize with PCD modules, such as toxins-antitoxins, suggestive of immunity-PCD coupling, likely mediated by shared proteins that sense damage and ‚Äúpredict‚Äù the outcome of infections. In type VI CRISPR-Cas systems, the same enzyme that inactivates the target RNA might execute cell suicide, in a case of ultimate integration of immunity and PCD.",
         "genotoxic stress sensing | immunity | programmed cell death | virus-host coevolution",
         "67",
         "1",
         "repositoryvor",
         "Green",
         "NIMH",
         "DP1MH100706",
         "National Institute of Mental Health"
        ],
        [
         "61",
         "61",
         "2-s2.0-84959519971",
         "10.1002/bies.201500199",
         "NULL",
         "26941227.0",
         "Coordination of cell decisions and promotion of phenotypic diversity in B. subtilis via pulsed behavior of the phosphorelay",
         "ar",
         "Article",
         "Schultz D.",
         "60002746",
         "Harvard Medical School",
         "Boston",
         "United States",
         "1.0",
         "Schultz, Daniel",
         "7202619781",
         "60002746",
         "2016-05-01",
         "1 May 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "5",
         "NULL",
         "440-445",
         "The phosphorelay of Bacillus subtilis, a kinase cascade that activates master regulator Spo0A‚àºP in response to starvation signals, is the core of a large network controlling the cell's decision to differentiate into sporulation and other phenotypes. This article reviews recent advances in understanding the origins and purposes of the complex dynamical behavior of the phosphorelay, which pulses with peaks of activity coordinated with the cell cycle. The transient imbalance in the expression of two critical genes caused by their strategic placement at opposing ends of the chromosome proved to be the key for this pulsed behavior. Feedback control loops in the phosphorelay use these pulses to implement a timer mechanism, which creates several windows of opportunity for phenotypic transitions over multiple generations. This strategy allows the cell to coordinate multiple differentiation programs in a decision process that fosters phenotypic diversity and adapts to current conditions.",
         "B. subtilis | Cell decision | Competence | Dynamics | Phenotypic diversity | Phosphorelay | Sporulation",
         "10",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "62",
         "62",
         "2-s2.0-84965188501",
         "10.1007/s40656-015-0089-3",
         "NULL",
         "26452776.0",
         "Managing variation in the investigation of organismal development: problems and opportunities",
         "ar",
         "Article",
         "Lowe J.",
         "60026479",
         "University of Exeter",
         "Exeter",
         "United Kingdom",
         "1.0",
         "Lowe, James W.E.",
         "57204157613",
         "60026479",
         "2015-12-01",
         "1 December 2015",
         "History and philosophy of the life sciences",
         "03919714",
         "15166",
         "NULL",
         "Journal",
         "37",
         "4",
         "NULL",
         "449-473",
         "This paper aims to clarify the consequences of new scientific and philosophical approaches for the practical-theoretical framework of modern developmental biology. I highlight normal development, and the instructive-permissive distinction, as key parts of this framework which shape how variation is conceptualised and managed. Furthermore, I establish the different dimensions of biological variation: the units, temporality and mode of variation. Using the analytical frame established by this, I interpret a selection of examples as challenges to the instructive-permissive distinction. These examples include the phenomena of developmental plasticity and transdifferentiation, the role of the microbiome in development, and new methodological approaches to standardisation and the assessment of causes. Furthermore, I argue that investigations into organismal development should investigate the effects of a wider range of kinds of variation including variation in the units, modes and temporalities of development. I close by examining various possible opportunities for producing and using normal development free of the assumptions of the instructive-permissive distinction. These opportunities are afforded by recent developments, which include new ways of producing standards incorporating more natural variation and being based on function rather than structure, and the ability to produce, store, and process large quantities of data.",
         "Developmental biology | Instructive-permissive distinction | Normal development | Standards | Variation",
         "4",
         "0",
         "repositoryam",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "64",
         "64",
         "2-s2.0-85002213664",
         "10.1007/s10739-016-9460-9",
         "NULL",
         "27921231.0",
         "The Cell and Protoplasm as Container, Object, and Substance, 1835‚Äì1861",
         "ar",
         "Article",
         "Liu D.",
         "60000745",
         "University of Illinois Urbana-Champaign",
         "Urbana",
         "United States",
         "1.0",
         "Liu, Daniel",
         "57192235608",
         "60000745",
         "2017-11-01",
         "1 November 2017",
         "Journal of the History of Biology",
         "00225010",
         "29662",
         "15730387",
         "Journal",
         "50",
         "4",
         "NULL",
         "889-925",
         "This article revisits the development of the protoplasm concept as it originally arose from critiques of the cell theory, and examines how the term ‚Äúprotoplasm‚Äù transformed from a botanical term of art in the 1840s to the so-called ‚Äúliving substance‚Äù and ‚Äúthe physical basis of life‚Äù two decades later. I show that there were two major shifts in biological materialism that needed to occur before protoplasm theory could be elevated to have equal status with cell theory in the nineteenth century. First, I argue that biologists had to accept that life could inhere in matter alone, regardless of form. Second, I argue that in the 1840s, ideas of what formless, biological matter was capable of dramatically changed: going from a ‚Äúcoagulation paradigm‚Äù (Pickstone, 1973) that had existed since Theophrastus, to a more robust conception of matter that was itself capable of movement and self-maintenance. In addition to revisiting Schleiden and Schwann‚Äôs original writings on cell theory, this article looks especially closely at Hugo von Mohl‚Äôs definition of the protoplasm concept in 1846, how it differed from his primordial utricle theory of cell structure two years earlier. This article draws on Lakoff and Johnson‚Äôs theory of ‚Äúontological metaphors‚Äù to show that the cell, primordial utricle, and protoplasm can be understood as material container, object, and substance, and that these overlapping distinctions help explain the chaotic and confusing early history of cell theory.",
         "Cambium | Cell | Coagulation | Matter | Primordial utricle | Protoplasm",
         "12",
         "0",
         "NULL",
         "NULL",
         "CHF",
         "undefined",
         "Chemical Heritage Foundation"
        ],
        [
         "66",
         "66",
         "2-s2.0-85027889285",
         "10.1002/bies.201700073",
         "NULL",
         "28792069.0",
         "Bloom syndrome helicase in meiosis: Pro-crossover functions of an anti-crossover protein",
         "ar",
         "Article",
         "Hatkevich T.",
         "60025111",
         "The University of North Carolina at Chapel Hill",
         "Chapel Hill",
         "United States",
         "2.0",
         "Hatkevich, Talia;Sekelsky, Jeff",
         "56197056300;6603810144",
         "60025111;60025111",
         "2017-09-01",
         "September 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "9",
         "1700073",
         "NULL",
         "The functions of the Bloom syndrome helicase (BLM) and its orthologs are well characterized in mitotic DNA damage repair, but their roles within the context of meiotic recombination are less clear. In meiotic recombination, multiple repair pathways are used to repair meiotic DSBs, and current studies suggest that BLM may regulate the use of these pathways. Based on literature from Saccharomyces cerevisiae, Arabidopsis thaliana, Mus musculus, Drosophila melanogaster, and Caenorhabditis elegans, we present a unified model for a critical meiotic role of BLM and its orthologs. In this model, BLM and its orthologs utilize helicase activity to regulate the use of various pathways in meiotic recombination by continuously disassembling recombination intermediates. This unwinding activity provides the meiotic program with a steady pool of early recombination substrates, increasing the probability for a DSB to be processed by the appropriate pathway. As a result of BLM activity, crossovers are properly placed throughout the genome, promoting proper chromosomal disjunction at the end of meiosis. This unified model can be used to further refine the complex role of BLM and its orthologs in meiotic recombination.",
         "bloom syndrome helicase | crossover patterning | double-strand break repair | meiosis | meiotic recombination | model organism genetics | resolution",
         "16",
         "0",
         "repositoryam",
         "Green",
         "NIGMS",
         "1R35GM118127",
         "National Institutes of Health"
        ],
        [
         "68",
         "68",
         "2-s2.0-85027098648",
         "10.1002/bies.201700050",
         "NULL",
         "28782202.0",
         "Evolution of peroxisomes illustrates symbiogenesis",
         "ar",
         "Article",
         "Speijer D.",
         "60002483",
         "Universiteit van Amsterdam",
         "Amsterdam",
         "Netherlands",
         "1.0",
         "Speijer, Dave",
         "6602513652",
         "60002483",
         "2017-09-01",
         "September 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "9",
         "1700050",
         "NULL",
         "Recently, the group of McBride reported a stunning observation regarding peroxisome biogenesis: newly born peroxisomes are hybrids of mitochondrial and ER-derived pre-peroxisomes. What was stunning? Studies performed with the yeast Saccharomyces cerevisiae had convincingly shown that peroxisomes are ER-derived, without indications for mitochondrial involvement. However, the recent finding using fibroblasts dovetails nicely with a mechanism inferred to be driving the eukaryotic invention of peroxisomes: reduction of mitochondrial reactive oxygen species (ROS) generation associated with fatty acid (FA) oxidation. This not only explains the mitochondrial involvement, but also its apparent absence in yeast. The latest results allow a reconstruction of the evolution of the yeast's highly derived metabolism and its limitations as a model organism in this instance. As I review here, peroxisomes are eukaryotic inventions reflecting mutual host endosymbiont adaptations: this is predicted by symbiogenetic theory, which states that the defining eukaryotic characteristics evolved as a result of mutual adaptations of two merging prokaryotes.",
         "beta-oxidation | FADH /NADH ratio 2 | peroxisome evolution | reverse electron transport (RET) | symbiogenesis",
         "26",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "69",
         "69",
         "2-s2.0-85026430693",
         "10.1002/bies.201700058",
         "NULL",
         "28752618.0",
         "Cell size control ‚Äì a mechanism for maintaining fitness and function",
         "ar",
         "Article",
         "Miettinen T.",
         "60022195;60022148;60021981;60017161;60008877",
         "Massachusetts Institute of Technology;University College London;A-Star, Institute of Molecular and Cell Biology;National University of Singapore;University of Dundee",
         "Cambridge;London;Singapore City;Singapore City;Dundee",
         "United States;United Kingdom;Singapore;Singapore;United Kingdom",
         "4.0",
         "Miettinen, Teemu P.;Caldez, Matias J.;Kaldis, Philipp;Bj√∂rklund, Mikael",
         "56052505000;36631417200;6701864610;7006565165",
         "60022148-60022195;60021981-60017161;60021981-60017161;60008877",
         "2017-09-01",
         "September 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "9",
         "1700058",
         "NULL",
         "The maintenance of cell size homeostasis has been studied for years in different cellular systems. With the focus on ‚Äòwhat regulates cell size‚Äô, the question ‚Äòwhy cell size needs to be maintained‚Äô has been largely overlooked. Recent evidence indicates that animal cells exhibit nonlinear cell size dependent growth rates and mitochondrial metabolism, which are maximal in intermediate sized cells within each cell population. Increases in intracellular distances and changes in the relative cell surface area impose biophysical limitations on cells, which can explain why growth and metabolic rates are maximal in a specific cell size range. Consistently, aberrant increases in cell size, for example through polyploidy, are typically disadvantageous to cellular metabolism, fitness and functionality. Accordingly, cellular hypertrophy can potentially predispose to or worsen metabolic diseases. We propose that cell size control may have emerged as a guardian of cellular fitness and metabolic activity.",
         "cell size control | fitness | metabolism | mevalonate pathway | mitochondria | polyploidy | statin",
         "52",
         "0",
         "repositoryam",
         "Green",
         "WT",
         "110275/Z/15/Z",
         "Wellcome Trust"
        ],
        [
         "70",
         "70",
         "2-s2.0-85023166990",
         "10.1007/s11097-017-9521-3",
         "NULL",
         "NULL",
         "Symmetry-breaking dynamics in development",
         "ar",
         "Article",
         "Moss Brender N.",
         "60033154",
         "Concordia University",
         "Montreal",
         "Canada",
         "1.0",
         "Moss Brender, Noah",
         "56861025000",
         "60033154",
         "2017-09-01",
         "1 September 2017",
         "Phenomenology and the Cognitive Sciences",
         "15687759",
         "144733",
         "15728676",
         "Journal",
         "16",
         "4",
         "NULL",
         "585-596",
         "Recognition of the plasticity of development ‚Äî from gene expression to neuroplasticity ‚Äî is increasingly undermining the traditional distinction between structure and function, or anatomy and behavior. At the same time, dynamic systems theory ‚Äî a set of tools and concepts drawn from the physical sciences ‚Äî has emerged as a way of describing what Maurice Merleau-Ponty calls the ‚Äúdynamic anatomy‚Äù of the living organism. This article surveys and synthesizes dynamic systems models of development from biology, neuroscience, and psychology in order to propose an integrated account of growth, learning, and behavior. Key to this account is the concept of self-differentiation or symmetry-breaking. I argue that development can be understood as a cascade of symmetry-breaking events brought about by the ongoing interactions of multiple, nested, nonlinear dynamic systems whose self-organizing behaviors gradually alter their own anatomical conditions. I begin by introducing the concept of symmetry-breaking as a way of understanding anatomical development. I then extend this approach to motor development by arguing that the organism‚Äôs behavior grows along with its body, like a new organ. Finally, I argue that the organism‚Äôs behavior and its world grow together dialectically, each driving the other to become more complex and asymmetrical through its own increasing asymmetry. Thus development turns out to be a form of cognition or sense-making, and cognition a form of development.",
         "Dynamic systems | Embodied cognition | Enactivism | Gene expression | Merleau-Ponty | Motor development",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "79",
         "79",
         "2-s2.0-85042038037",
         "10.1007/s13752-016-0253-3",
         "NULL",
         "NULL",
         "How Did the Eukaryotes Evolve?",
         "ar",
         "Article",
         "Barbieri M.",
         "60024690",
         "University of Ferrara",
         "Ferrara",
         "Italy",
         "1.0",
         "Barbieri, Marcello",
         "7102702241",
         "60024690",
         "2017-03-01",
         "March 2017",
         "Biological Theory",
         "15555542",
         "21101049622",
         "15555550",
         "Journal",
         "12",
         "1",
         "NULL",
         "13-26",
         "The fossil record shows that the stromatolites built by cyanobacteria 2 and 3 billion years ago are virtually identical to those built by their modern descendants, which is just a part of much evidence revealing that bacteria have barely changed in billions of years. They appeared very early in the history of life and have conserved their complexity (in terms of size, shape, and number of components) ever since. The eukaryotes, however, did the opposite. They repeatedly increased the complexity of their cells and eventually broke the cellular barrier and gave origin to all living creatures that we see around us. This gives us one of the major problems in evolution: why have the prokaryotes maintained the same complexity throughout the history of life while the eukaryotes have become increasingly more complex? Here it is shown that a solution does exist, but it is based on experimental data that so far have largely been ignored. It is based on the discovery that, in addition to the genetic code, many other organic codes exist in living systems. The potential to generate organic codes was already present in the common ancestor but was not transmitted indefinitely to all its descendants. After the genetic code and the signal transduction codes that gave origin to the first cells, the prokaryotes evolved no other organic code, whereas the ancestors of the eukaryotes continued to explore the coding space and gave origin to splicing codes, histone code, cytoskeleton codes, tubulin code, compartment codes, and sequence codes. This experimental fact suggests that the prokaryotes did not increase their complexity because they did not evolve new organic codes, whereas the eukaryotes became increasingly more complex because they maintained the potential to bring new codes into existence.",
         "Common ancestor | Complexity | Eukaryotes | Organic codes | Primary kingdoms | Prokaryotes | Sequence codes",
         "6",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "80",
         "80",
         "2-s2.0-85012030423",
         "10.1002/bies.201600231",
         "NULL",
         "28176337.0",
         "How fish color their skin: A paradigm for development and evolution of adult patterns: Multipotency, plasticity, and cell competition regulate proliferation and spreading of pigment cells in Zebrafish coloration",
         "ar",
         "Article",
         "N√ºsslein-Volhard C.",
         "60020329",
         "Max Planck Institute for Developmental Biology",
         "Tubingen",
         "Germany",
         "2.0",
         "N√ºsslein-Volhard, Christiane;Singh, Ajeet Pratap",
         "7005403321;57191258469",
         "60020329;60020329",
         "2017-03-01",
         "1 March 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "3",
         "1600231",
         "NULL",
         "Pigment cells in zebrafish ‚àí melanophores, iridophores, and xanthophores ‚àí originate from neural crest-derived stem cells associated with the dorsal root ganglia of the peripheral nervous system. Clonal analysis indicates that these progenitors remain multipotent and plastic beyond embryogenesis well into metamorphosis, when the adult color pattern develops. Pigment cells share a lineage with neuronal cells of the peripheral nervous system; progenitors propagate along the spinal nerves. The proliferation of pigment cells is regulated by competitive interactions among cells of the same type. An even spacing involves collective migration and contact inhibition of locomotion of the three cell types distributed in superimposed monolayers in the skin. This mode of coloring the skin is probably common to fish, whereas different patterns emerge by species specific cell interactions among the different pigment cell types. These interactions are mediated by channels involved in direct cell contact between the pigment cells, as well as unknown cues provided by the tissue environment.",
         "cell competition | coloration | multipotent stem cells | neural crest | pattern formation | peripheral nervous system | zebrafish",
         "79",
         "1",
         "publisherhybridgold",
         "Hybrid Gold",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "81",
         "81",
         "2-s2.0-85011103485",
         "10.1002/bies.201600233",
         "NULL",
         "28133765.0",
         "Boundaries of loop domains (insulators): Determinants of chromosome form and function in multicellular eukaryotes",
         "ar",
         "Article",
         "Chetverina D.",
         "60069617;60032420;60003269",
         "Institute of Gene Biology, Russian Academy of Sciences;Sidney Kimmel Medical College;Princeton University",
         "Moscow;Philadelphia;Princeton",
         "Russian Federation;United States;United States",
         "6.0",
         "Chetverina, Darya;Fujioka, Miki;Erokhin, Maksim;Georgiev, Pavel;Jaynes, James B.;Schedl, Paul",
         "23567857200;7103085654;35279621200;7102009868;7005923784;7005983204",
         "60069617;60032420;60069617;60069617;60032420;60003269-60069617",
         "2017-03-01",
         "1 March 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "3",
         "1600233",
         "NULL",
         "Chromosomes in multicellular animals are subdivided into a series of looped domains. In addition to being the underlying principle for organizing the chromatin fiber, looping is critical for processes ranging from gene regulation to recombination and repair. The subdivision of chromosomes into looped domains depends upon a special class of architectural elements called boundaries or insulators. These elements are distributed throughout the genome and are ubiquitous building blocks of chromosomes. In this review, we focus on features of boundaries that are critical in determining the topology of the looped domains and their genetic properties. We highlight the properties of fly boundaries that are likely to have an important bearing on the organization of looped domains in vertebrates, and discuss the functional consequences of the observed similarities and differences.",
         "chromosome architecture | gene regulation | insulators | loop topology | looped domain | TADs",
         "40",
         "0",
         "repositoryam",
         "Green",
         "NIGMS",
         "R01GM117458",
         "National Institute of General Medical Sciences"
        ],
        [
         "82",
         "82",
         "2-s2.0-84996436014",
         "10.1002/bies.201600154",
         "NULL",
         "27774671.0",
         "Centriole positioning in epithelial cells and its intimate relationship with planar cell polarity",
         "ar",
         "Article",
         "Carvajal-Gonzalez J.",
         "60017838;60012981",
         "Universidad de Extremadura;Icahn School of Medicine at Mount Sinai",
         "Badajoz;New York",
         "Spain;United States",
         "3.0",
         "Carvajal-Gonzalez, Jose Maria;Mulero-Navarro, Sonia;Mlodzik, Marek",
         "23968865800;6505778626;7006827593",
         "60017838;60017838;60012981",
         "2016-12-01",
         "1 December 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "12",
         "NULL",
         "1234-1245",
         "Planar cell polarity (PCP)-signaling and associated tissue polarization are evolutionarily conserved. A well documented feature of PCP-signaling in vertebrates is its link to centriole/cilia positioning, although the relationship of PCP and ciliogenesis is still debated. A recent report in Drosophila established that Frizzled (Fz)-PCP core signaling has an instructive input to polarized centriole positioning in non-ciliated Drosophila wing epithelia as a PCP read-out. Here, we review the impact of this observation in the context of recent descriptions of the relationship(s) of core Fz-PCP signaling and cilia/centriole positioning in epithelial and non-epithelial cells. All existing data are consistent with a model where Fz-PCP signaling functions upstream of centriole/cilia positioning, independent of ciliogenesis. The combined data sets indicate that the Fz-Dsh PCP complex is instructive for centriole/ciliary positioning via an actin-based mechanism. Thereby, centriole/cilia/centrosome positioning can be considered an evolutionarily conserved readout and common downstream effect of PCP-signaling from flies to mammals.",
         "actin cytoskeleton | centriole/MTOC localization | cilia | epithelial polarity | Frizzled-Vangl signaling | microtubules | planar cell polarity",
         "29",
         "0",
         "repositoryam",
         "Green",
         "NEI",
         "R01EY013256",
         "National Eye Institute"
        ],
        [
         "84",
         "84",
         "2-s2.0-84983394226",
         "10.1002/bies.201600100",
         "NULL",
         "27438915.0",
         "Exaptive origins of regulated mRNA decay in eukaryotes",
         "ar",
         "Article",
         "Hamid F.",
         "60095091;60011520",
         "School of Biological Sciences;King's College London",
         "Singapore City;London",
         "Singapore;United Kingdom",
         "2.0",
         "Hamid, Fursham M.;Makeyev, Eugene V.",
         "56318608700;6602198520",
         "60095091;60095091-60011520",
         "2016-09-01",
         "1 September 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "9",
         "NULL",
         "830-838",
         "Eukaryotic gene expression is extensively controlled at the level of mRNA stability and the mechanisms underlying this regulation are markedly different from their archaeal and bacterial counterparts. We propose that two such mechanisms, nonsense-mediated decay (NMD) and motif-specific transcript destabilization by CCCH-type zinc finger RNA-binding proteins, originated as a part of cellular defense against RNA pathogens. These branches of the mRNA turnover pathway might have been used by primeval eukaryotes alongside RNA interference to distinguish their own messages from those of RNA viruses and retrotransposable elements. We further hypothesize that the subsequent advent of ‚Äúprofessional‚Äù innate and adaptive immunity systems allowed NMD and the motif-triggered mechanisms to be efficiently repurposed for regulation of endogenous cellular transcripts. This scenario explains the rapid emergence of archetypical mRNA destabilization pathways in eukaryotes and argues that other aspects of post-transcriptional gene regulation in this lineage might have been derived through a similar exaptation route.",
         "antiviral defense | exaptation | mRNA decay | nonsense-mediated decay | regnase | roquin | tristetraprolin",
         "12",
         "1",
         "repositoryvor",
         "Green",
         "BBSRC",
         "BB/M001199/1",
         "Biotechnology and Biological Sciences Research Council"
        ],
        [
         "89",
         "89",
         "2-s2.0-84982276360",
         "10.1007/s10441-016-9286-z",
         "NULL",
         "27535677.0",
         "SMT and TOFT Integrable After All: A Reply to Bizzarri and Cucina",
         "ar",
         "Article",
         "Bedessem B.",
         "60104653",
         "Universit√© Grenoble Alpes",
         "Saint Martin d'Heres",
         "France",
         "2.0",
         "Bedessem, Baptiste;Ruphy, Stphanie",
         "56027030900;14421900800",
         "60104653;60104653",
         "2017-03-01",
         "1 March 2017",
         "Acta Biotheoretica",
         "00015342",
         "12456",
         "15728358",
         "Journal",
         "65",
         "1",
         "NULL",
         "81-85",
         "In a previous paper recently published in this journal, we argue that the two main theories of carcinogenesis (SMT and TOFT) should be considered as compatible, at the metaphysical, epistemological and biological levels. In a reply to our contribution, Bizzarri and Cucina claim we are wrong since SMT and TOFT are opposite and incompatible paradigms. Here, we show that their arguments are not satisfactory. Indeed, the authors go through the same mistakes that we already addressed. In particular, they confuse reductionism, as¬†an ontological frame, and genetic determinism, as a causal pathway. Beside, they make an inadequate use of the Kuhnian notion of paradigm shift. Finally, we confirm our previous conclusion: there is no strong argument to totally abandon the somatic mutation theory. It describes a partial causal pathway, compatible with the one proposed by TOFT.",
         "Cancer | Reductionism | TOFT",
         "5",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "90",
         "90",
         "2-s2.0-85007591523",
         "10.1002/bies.201600177",
         "NULL",
         "28044359.0",
         "Mitochondria in complex psychiatric disorders: Lessons from mouse models of 22q11.2 deletion syndrome: Hemizygous deletion of several mitochondrial genes in the 22q11.2 genomic region can lead to symptoms associated with neuropsychiatric disease",
         "ar",
         "Article",
         "Devaraju P.",
         "60019905",
         "St. Jude Children's Research Hospital",
         "Memphis",
         "United States",
         "2.0",
         "Devaraju, Prakash;Zakharenko, Stanislav S.",
         "57191374152;6701402982",
         "60019905;60019905",
         "2017-02-01",
         "1 February 2017",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "39",
         "2",
         "1600177",
         "NULL",
         "Mitochondrial ATP synthesis, calcium buffering, and trafficking affect neuronal function and survival. Several genes implicated in mitochondrial functions map within the genomic region associated with 22q11.2 deletion syndrome (22q11DS), which is a key genetic cause of neuropsychiatric diseases. Although neuropsychiatric diseases impose a serious health and economic burden, their etiology and pathogenesis remain largely unknown because of the dearth of valid animal models and the challenges in investigating the pathophysiology in neuronal circuits. Mouse models of 22q11DS are becoming valid tools for studying human psychiatric diseases, because they have hemizygous deletions of the genes that are deleted in patients and exhibit neuronal and behavioral abnormalities consistent with neuropsychiatric disease. The deletion of some 22q11DS genes implicated in mitochondrial function leads to abnormal neuronal and synaptic function. Herein, we summarize recent findings on mitochondrial dysfunction in 22q11DS and extend those findings to the larger context of schizophrenia and other neuropsychiatric diseases.",
         "22q11.2 deletion syndrome | ATP | calcium | mitochondria | schizophrenia | synapse | synaptic plasticity",
         "29",
         "0",
         "repositoryam",
         "Green",
         "NIH",
         "R01 MH095810",
         "National Institutes of Health"
        ],
        [
         "93",
         "93",
         "2-s2.0-84976580345",
         "10.1002/bies.201500180",
         "NULL",
         "27248008.0",
         "Antibiotic resistance and virulence: Understanding the link and its consequences for prophylaxis and therapy",
         "ar",
         "Article",
         "Guillard T.",
         "60123796;60123665;60019394;60016782;60003479",
         "Universit√© Paris Cit√©;Infection, Anti-Microbien, Mod√©lisation, √âvolution;Universit√© de Reims Champagne-Ardenne;Brigham and Women's Hospital;Hopital Louis-Mourier",
         "Paris;Paris;Reims;Boston;Colombes",
         "France;France;France;United States;France",
         "5.0",
         "Guillard, Thomas;Pons, St√©phanie;Roux, Damien;Pier, Gerald B.;Skurnik, David",
         "26431954200;57214717089;16064188600;7006307834;14819754300",
         "60016782-60019394;60016782;60016782-60123665-60123796-60003479;60016782;60016782",
         "2016-07-01",
         "1 July 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "7",
         "NULL",
         "682-693",
         "‚ÄúAntibiotic resistance is usually associated with a fitness cost‚Äù is frequently accepted as common knowledge in the field of infectious diseases. However, with the advances in high-throughput DNA sequencing that allows for a comprehensive analysis of bacterial pathogenesis at the genome scale, including antibiotic resistance genes, it appears that this paradigm might not be as solid as previously thought. Recent studies indicate that antibiotic resistance is able to enhance bacterial fitness in vivo with a concomitant increase in virulence during infections. As a consequence, strategies to minimize antibiotic resistance turn out to be not as simple as initially believed. Indeed, decreased antibiotic use may not be sufficient to let susceptible strains outcompete the resistant ones. Here, we put in perspective these findings and review alternative approaches, such as preventive and therapeutic anti-bacterial immunotherapies that have the potential to by-pass the classic antibiotics.",
         "antibiotic resistance | fitness | high-throughput sequencing | transposon sequencing | virulence",
         "37",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "95",
         "95",
         "2-s2.0-84976533555",
         "10.1002/bies.201600037",
         "NULL",
         "27173018.0",
         "YAP and TAZ in epithelial stem cells: A sensor for cell polarity, mechanical forces and tissue damage",
         "ar",
         "Article",
         "Elbediwy A.",
         "60107285",
         "The Francis Crick Institute",
         "London",
         "United Kingdom",
         "3.0",
         "Elbediwy, Ahmed;Vincent-Mistiaen, Zo√© I.;Thompson, Barry J.",
         "36805431300;56256376000;57208450526",
         "60107285;60107285;60107285",
         "2016-07-01",
         "1 July 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "7",
         "NULL",
         "644-653",
         "The YAP/TAZ family of transcriptional co-activators drives cell proliferation in epithelial tissues and cancers. Yet, how YAP and TAZ are physiologically regulated remains unclear. Here we review recent reports that YAP and TAZ act primarily as sensors of epithelial cell polarity, being inhibited when cells differentiate an apical membrane domain, and being activated when cells contact the extracellular matrix via their basal membrane domain. Apical signalling occurs via the canonical Crumbs/CRB-Hippo/MST-Warts/LATS kinase cascade to phosphorylate and inhibit YAP/TAZ. Basal signalling occurs via Integrins and Src family kinases to phosphorylate and activate YAP/TAZ. Thus, YAP/TAZ is localised to the nucleus in basal stem/progenitor cells and cytoplasm in differentiated squamous cells or columnar cells. In addition, other signals such as mechanical forces, tissue damage and possibly receptor tyrosine kinases (RTKs) can influence MST-LATS or Src family kinase activity to modulate YAP/TAZ activity.",
         "epithelial polarity | Hippo pathway | mechanosensing | mechanotransduction | TAZ | wound healing | YAP",
         "81",
         "1",
         "repositoryam",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "96",
         "96",
         "2-s2.0-84963795546",
         "10.1002/bies.201600003",
         "NULL",
         "27062292.0",
         "Intracellular evolution of mitochondrial DNA (mtDNA) and the tragedy of the cytoplasmic commons",
         "ar",
         "Article",
         "Haig D.",
         "60009982",
         "Harvard University",
         "Cambridge",
         "United States",
         "1.0",
         "Haig, David",
         "7005499789",
         "60009982",
         "2016-06-01",
         "1 June 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "6",
         "NULL",
         "549-555",
         "Mitochondria exist in large numbers per cell. Therefore, the strength of natural selection on individual mtDNAs for their contribution to cellular fitness is weak whereas the strength of selection in favor of mtDNAs that increase their own replication without regard for cellular functions is strong. This problem has been solved for most mitochondrial genes by their transfer to the nucleus but a few critical genes remain encoded by mtDNA. Organisms manage the evolution of mtDNA to prevent mutational decay of essential services mitochondria provide to their hosts. Bottlenecks of mitochondrial numbers in female germlines increase the homogeneity of mtDNAs within cells and allow intraorganismal selection to eliminate cells with low quality mitochondria. Mechanisms of intracellular \"quality control\" allow direct selection on the competence of individual mtDNAs. These processes maintain the integrity of mtDNAs within the germline but are inadequate to indefinitely maintain mitochondrial function in somatic cells. Mitochondria are domesticated prokaryotes that reside in large herds within cells.",
         "Aging | Apoptosis | Bottleneck | Mitochondria | Mitophagy | MtDNA | Public goods",
         "24",
         "0",
         "repositoryam",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "97",
         "97",
         "2-s2.0-84959474703",
         "10.1007/s10539-015-9497-8",
         "NULL",
         "NULL",
         "Biological regulation: controlling the system from within",
         "ar",
         "Article",
         "Bich L.",
         "60105873;60027856;60000179",
         "IFICC - Instituto de Filosof√≠a y Ciencias de la Complejidad;Universidad del Pais Vasco;√âcole Normale Sup√©rieure",
         "Santiago;Leioa;Paris",
         "Chile;Spain;France",
         "4.0",
         "Bich, Leonardo;Mossio, Matteo;Ruiz-Mirazo, Kepa;Moreno, Alvaro",
         "22978521800;23100694100;6602372127;35580695300",
         "60027856-60105873;60000179;60027856;60027856",
         "2016-03-01",
         "1 March 2016",
         "Biology and Philosophy",
         "01693867",
         "13699",
         "15728404",
         "Journal",
         "31",
         "2",
         "NULL",
         "237-265",
         "Biological regulation is what allows an organism to handle the effects of a perturbation, modulating its own constitutive dynamics in response to particular changes in internal and external conditions. With the central focus of analysis on the case of minimal living systems, we argue that regulation consists in a specific form of second-order control, exerted over the core (constitutive) regime of production and maintenance of the components that actually put together the organism. The main argument is that regulation requires a distinctive architecture of functional relationships, and specifically the action of a dedicated subsystem whose activity is dynamically decoupled from that of the constitutive regime. We distinguish between two major ways in which control mechanisms contribute to the maintenance of a biological organisation in response to internal and external perturbations: dynamic stability and regulation. Based on this distinction an explicit definition and a set of organisational requirements for regulation are provided, and thoroughly illustrated through the examples of bacterial chemotaxis and the lac-operon. The analysis enables us to mark out the differences between regulation and closely related concepts such as feedback, robustness and homeostasis.",
         "Control | Dynamic stability | Dynamical decoupling | Feedback | Organisation | Regulation",
         "88",
         "0",
         "NULL",
         "NULL",
         "FONDECYT",
         "FFI2011-25665",
         "Fondo Nacional de Desarrollo Cient√≠fico y Tecnol√≥gico"
        ],
        [
         "98",
         "98",
         "2-s2.0-84961385886",
         "10.1002/bies.201500124",
         "NULL",
         "26625861.0",
         "Variation is function: Are single cell differences functionally important?: Testing the hypothesis that single cell variation is required for aggregate function",
         "ar",
         "Article",
         "Dueck H.",
         "60102562;60006297;60003711",
         "School of Engineering and Applied Science;University of Pennsylvania;University of Pennsylvania Perelman School of Medicine",
         "Philadelphia;Philadelphia;Philadelphia",
         "United States;United States;United States",
         "3.0",
         "Dueck, Hannah;Eberwine, James;Kim, Junhyong",
         "55452573000;7005034882;14034265200",
         "60006297;60006297-60003711;60006297-60003711-60102562",
         "2016-02-01",
         "1 February 2016",
         "BioEssays",
         "02659247",
         "13174",
         "15211878",
         "Journal",
         "38",
         "2",
         "NULL",
         "172-180",
         "There is a growing appreciation of the extent of transcriptome variation across individual cells of the same cell type. While expression variation may be a byproduct of, for example, dynamic or homeostatic processes, here we consider whether single-cell molecular variation per se might be crucial for population-level function. Under this hypothesis, molecular variation indicates a diversity of hidden functional capacities within an ensemble of \"identical\" cells, and this functional diversity facilitates collective behavior that would be inaccessible to a homogenous population. In reviewing this topic, we explore possible functions that might be carried by a heterogeneous ensemble of cells; however, this question has proven difficult to test, both because methods to manipulate molecular variation are limited and because it is complicated to define, and measure, population-level function. We consider several possible methods to further pursue the hypothesis that \"variation is function\" through the use of comparative analysis and novel experimental techniques.",
         "Bet-hedging | Evolution of variation | Fractional response | Functional variation | Single cell transcriptome | Single cell variation",
         "53",
         "1",
         "repositoryvor",
         "Green",
         "NIMH",
         "U01MH098953",
         "National Institute of Mental Health"
        ]
       ],
       "shape": {
        "columns": 37,
        "rows": 6272
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>eid</th>\n",
       "      <th>doi</th>\n",
       "      <th>pii</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>title</th>\n",
       "      <th>subtype</th>\n",
       "      <th>subtypeDescription</th>\n",
       "      <th>creator</th>\n",
       "      <th>afid</th>\n",
       "      <th>...</th>\n",
       "      <th>pageRange</th>\n",
       "      <th>description</th>\n",
       "      <th>authkeywords</th>\n",
       "      <th>citedby_count</th>\n",
       "      <th>openaccess</th>\n",
       "      <th>freetoread</th>\n",
       "      <th>freetoreadLabel</th>\n",
       "      <th>fund_acr</th>\n",
       "      <th>fund_no</th>\n",
       "      <th>fund_sponsor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>2-s2.0-85209763600</td>\n",
       "      <td>10.1007/s11229-024-04815-5</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Driftability and niche construction</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>F√°bregas-Tejeda A.</td>\n",
       "      <td>60025063</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Niche construction is the process of organisms...</td>\n",
       "      <td>Drift | Driftability | Evolutionary causation ...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherhybridgold</td>\n",
       "      <td>Hybrid Gold</td>\n",
       "      <td>FWO</td>\n",
       "      <td>G070122N</td>\n",
       "      <td>Fonds Wetenschappelijk Onderzoek</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2-s2.0-85202899987</td>\n",
       "      <td>10.1007/s10539-024-09957-x</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Explanatory gaps in evolutionary theory</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Aaby B.H.</td>\n",
       "      <td>60025063;60010348</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Proponents of the extended evolutionary synthe...</td>\n",
       "      <td>Evolutionary theory | Explanatory gap | Natura...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherhybridgold</td>\n",
       "      <td>Hybrid Gold</td>\n",
       "      <td>NULL</td>\n",
       "      <td>3H210777</td>\n",
       "      <td>KU Leuven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2-s2.0-85175425123</td>\n",
       "      <td>10.1016/j.shpsa.2023.10.006</td>\n",
       "      <td>S0039368123001449</td>\n",
       "      <td>37907020.0</td>\n",
       "      <td>A Wolf in Sheep's Clothing: Idealisations and ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Serpico D.</td>\n",
       "      <td>60021361;60015986</td>\n",
       "      <td>...</td>\n",
       "      <td>72-83</td>\n",
       "      <td>Research in pharmacogenomics and precision med...</td>\n",
       "      <td>Genetic prediction | GWAS | Idealisations | Po...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>ERC</td>\n",
       "      <td>805498</td>\n",
       "      <td>European Research Council</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>2-s2.0-85152302150</td>\n",
       "      <td>10.1016/j.shpsa.2023.03.002</td>\n",
       "      <td>S0039368123000705</td>\n",
       "      <td>37068423.0</td>\n",
       "      <td>Joint representation: Modeling a phenomenon wi...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Yoshida Y.</td>\n",
       "      <td>60029445</td>\n",
       "      <td>...</td>\n",
       "      <td>67-76</td>\n",
       "      <td>Biologists often study particular biological s...</td>\n",
       "      <td>Collective cell migration | Generalization | M...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>UMN</td>\n",
       "      <td>undefined</td>\n",
       "      <td>University of Minnesota</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>2-s2.0-85148606416</td>\n",
       "      <td>10.3390/philosophies8010008</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Turing‚Äôs Biological Philosophy: Morphogenesis,...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Greif H.</td>\n",
       "      <td>60003675</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Alan M. Turing‚Äôs last published work and some ...</td>\n",
       "      <td>Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherfullgold</td>\n",
       "      <td>Gold</td>\n",
       "      <td>NCN</td>\n",
       "      <td>2020/37/B/HS1/01809</td>\n",
       "      <td>Narodowe Centrum Nauki</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3159</th>\n",
       "      <td>3159</td>\n",
       "      <td>2-s2.0-84898444008</td>\n",
       "      <td>10.1080/21550085.2014.885490</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>A Constructivist Approach Toward a General Def...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Meinard Y.</td>\n",
       "      <td>60068952;60012614;114099298</td>\n",
       "      <td>...</td>\n",
       "      <td>88-104</td>\n",
       "      <td>Biodiversity sciences witness a double dynamic...</td>\n",
       "      <td>economics | ecosystem management | philosophy ...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>repositoryam</td>\n",
       "      <td>Green</td>\n",
       "      <td>FRJ</td>\n",
       "      <td>undefined</td>\n",
       "      <td>Fondation pour la recherche juridique</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3160</th>\n",
       "      <td>3160</td>\n",
       "      <td>2-s2.0-27844440845</td>\n",
       "      <td>10.3197/096327105774462683</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The commons, game theory and aspects of human ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Dodds W.K.</td>\n",
       "      <td>60000689</td>\n",
       "      <td>...</td>\n",
       "      <td>411-425</td>\n",
       "      <td>Fundamental aspects of human use of the enviro...</td>\n",
       "      <td>Commons | Game theory | Global environment | H...</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3161</th>\n",
       "      <td>3161</td>\n",
       "      <td>2-s2.0-0036297383</td>\n",
       "      <td>10.3197/096327102129341073</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The role of customary institutions in the cons...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Virtanen P.</td>\n",
       "      <td>60011170</td>\n",
       "      <td>...</td>\n",
       "      <td>227-241</td>\n",
       "      <td>Recently the role of customary local instituti...</td>\n",
       "      <td>Africa | Biodiversity | Conservation areas | L...</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3162</th>\n",
       "      <td>3162</td>\n",
       "      <td>2-s2.0-0034962075</td>\n",
       "      <td>10.3197/096327101129340804</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Exotic species, naturalisation, and biological...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Hettinger N.</td>\n",
       "      <td>60013340</td>\n",
       "      <td>...</td>\n",
       "      <td>193-224</td>\n",
       "      <td>Contrary to frequent characterisations, exotic...</td>\n",
       "      <td>Exotics | Native | Nativism | Naturalisation |...</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3163</th>\n",
       "      <td>3163</td>\n",
       "      <td>2-s2.0-0343729540</td>\n",
       "      <td>10.3197/096327100129341976</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Does the convention on biodiversity safeguard ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>M√ºller F.G.</td>\n",
       "      <td>60033154</td>\n",
       "      <td>...</td>\n",
       "      <td>55-80</td>\n",
       "      <td>This paper attempts to assess and evaluate som...</td>\n",
       "      <td>Biological diversity | Commercialisation | Eco...</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6272 rows √ó 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0                 eid                           doi  \\\n",
       "0              0  2-s2.0-85209763600    10.1007/s11229-024-04815-5   \n",
       "1              1  2-s2.0-85202899987    10.1007/s10539-024-09957-x   \n",
       "2              2  2-s2.0-85175425123   10.1016/j.shpsa.2023.10.006   \n",
       "3              3  2-s2.0-85152302150   10.1016/j.shpsa.2023.03.002   \n",
       "4              4  2-s2.0-85148606416   10.3390/philosophies8010008   \n",
       "...          ...                 ...                           ...   \n",
       "3159        3159  2-s2.0-84898444008  10.1080/21550085.2014.885490   \n",
       "3160        3160  2-s2.0-27844440845    10.3197/096327105774462683   \n",
       "3161        3161   2-s2.0-0036297383    10.3197/096327102129341073   \n",
       "3162        3162   2-s2.0-0034962075    10.3197/096327101129340804   \n",
       "3163        3163   2-s2.0-0343729540    10.3197/096327100129341976   \n",
       "\n",
       "                    pii   pubmed_id  \\\n",
       "0                  NULL        NULL   \n",
       "1                  NULL        NULL   \n",
       "2     S0039368123001449  37907020.0   \n",
       "3     S0039368123000705  37068423.0   \n",
       "4                  NULL        NULL   \n",
       "...                 ...         ...   \n",
       "3159               NULL        NULL   \n",
       "3160               NULL        NULL   \n",
       "3161               NULL        NULL   \n",
       "3162               NULL        NULL   \n",
       "3163               NULL        NULL   \n",
       "\n",
       "                                                  title subtype  \\\n",
       "0                   Driftability and niche construction      ar   \n",
       "1               Explanatory gaps in evolutionary theory      ar   \n",
       "2     A Wolf in Sheep's Clothing: Idealisations and ...      ar   \n",
       "3     Joint representation: Modeling a phenomenon wi...      ar   \n",
       "4     Turing‚Äôs Biological Philosophy: Morphogenesis,...      ar   \n",
       "...                                                 ...     ...   \n",
       "3159  A Constructivist Approach Toward a General Def...      ar   \n",
       "3160  The commons, game theory and aspects of human ...      ar   \n",
       "3161  The role of customary institutions in the cons...      ar   \n",
       "3162  Exotic species, naturalisation, and biological...      ar   \n",
       "3163  Does the convention on biodiversity safeguard ...      ar   \n",
       "\n",
       "     subtypeDescription             creator                         afid  ...  \\\n",
       "0               Article  F√°bregas-Tejeda A.                     60025063  ...   \n",
       "1               Article           Aaby B.H.            60025063;60010348  ...   \n",
       "2               Article          Serpico D.            60021361;60015986  ...   \n",
       "3               Article          Yoshida Y.                     60029445  ...   \n",
       "4               Article            Greif H.                     60003675  ...   \n",
       "...                 ...                 ...                          ...  ...   \n",
       "3159            Article          Meinard Y.  60068952;60012614;114099298  ...   \n",
       "3160            Article          Dodds W.K.                     60000689  ...   \n",
       "3161            Article         Virtanen P.                     60011170  ...   \n",
       "3162            Article        Hettinger N.                     60013340  ...   \n",
       "3163            Article         M√ºller F.G.                     60033154  ...   \n",
       "\n",
       "     pageRange                                        description  \\\n",
       "0         NULL  Niche construction is the process of organisms...   \n",
       "1         NULL  Proponents of the extended evolutionary synthe...   \n",
       "2        72-83  Research in pharmacogenomics and precision med...   \n",
       "3        67-76  Biologists often study particular biological s...   \n",
       "4         NULL  Alan M. Turing‚Äôs last published work and some ...   \n",
       "...        ...                                                ...   \n",
       "3159    88-104  Biodiversity sciences witness a double dynamic...   \n",
       "3160   411-425  Fundamental aspects of human use of the enviro...   \n",
       "3161   227-241  Recently the role of customary local instituti...   \n",
       "3162   193-224  Contrary to frequent characterisations, exotic...   \n",
       "3163     55-80  This paper attempts to assess and evaluate som...   \n",
       "\n",
       "                                           authkeywords citedby_count  \\\n",
       "0     Drift | Driftability | Evolutionary causation ...             0   \n",
       "1     Evolutionary theory | Explanatory gap | Natura...             0   \n",
       "2     Genetic prediction | GWAS | Idealisations | Po...             0   \n",
       "3     Collective cell migration | Generalization | M...             0   \n",
       "4     Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...             1   \n",
       "...                                                 ...           ...   \n",
       "3159  economics | ecosystem management | philosophy ...             7   \n",
       "3160  Commons | Game theory | Global environment | H...            15   \n",
       "3161  Africa | Biodiversity | Conservation areas | L...            66   \n",
       "3162  Exotics | Native | Nativism | Naturalisation |...            55   \n",
       "3163  Biological diversity | Commercialisation | Eco...            12   \n",
       "\n",
       "     openaccess           freetoread freetoreadLabel fund_acr  \\\n",
       "0             1  publisherhybridgold     Hybrid Gold      FWO   \n",
       "1             1  publisherhybridgold     Hybrid Gold     NULL   \n",
       "2             1        repositoryvor           Green      ERC   \n",
       "3             0        repositoryvor           Green      UMN   \n",
       "4             1    publisherfullgold            Gold      NCN   \n",
       "...         ...                  ...             ...      ...   \n",
       "3159          0         repositoryam           Green      FRJ   \n",
       "3160          0                 NULL            NULL     NULL   \n",
       "3161          0                 NULL            NULL     NULL   \n",
       "3162          0                 NULL            NULL     NULL   \n",
       "3163          0                 NULL            NULL     NULL   \n",
       "\n",
       "                  fund_no                           fund_sponsor  \n",
       "0                G070122N       Fonds Wetenschappelijk Onderzoek  \n",
       "1                3H210777                              KU Leuven  \n",
       "2                  805498              European Research Council  \n",
       "3               undefined                University of Minnesota  \n",
       "4     2020/37/B/HS1/01809                 Narodowe Centrum Nauki  \n",
       "...                   ...                                    ...  \n",
       "3159            undefined  Fondation pour la recherche juridique  \n",
       "3160            undefined                                   NULL  \n",
       "3161            undefined                                   NULL  \n",
       "3162            undefined                                   NULL  \n",
       "3163            undefined                                   NULL  \n",
       "\n",
       "[6272 rows x 37 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cleaned  = df_cleaned[df_cleaned['subtypeDescription'] == 'Article']\n",
    "df_cleaned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "0c8abaf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Not in top 10', 'Biology and Philosophy', 'Acta Biotheoretica',\n",
       "       'Biological Theory', 'BioEssays',\n",
       "       'Studies in History and Philosophy of Science Part C :Studies in History and Philosophy of Biological and Biomedical Sciences',\n",
       "       'Journal of the History of Biology', 'Philosophy of Science',\n",
       "       'Behavioral and Brain Sciences',\n",
       "       'History and Philosophy of the Life Sciences', 'Biosemiotics'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cleaned.publicationName.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "e106820d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "Unnamed: 0",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "doi",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pii",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pubmed_id",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "title",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtype",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "subtypeDescription",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "creator",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "afid",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affilname",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_city",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "affiliation_country",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_count",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "author_names",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_ids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "author_afids",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "coverDate",
         "rawType": "datetime64[ns]",
         "type": "datetime"
        },
        {
         "name": "coverDisplayDate",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "publicationName",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "source_id",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "eIssn",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "aggregationType",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "volume",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "issueIdentifier",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "article_number",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pageRange",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "description",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "authkeywords",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "citedby_count",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "openaccess",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "freetoread",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "freetoreadLabel",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_acr",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_no",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "fund_sponsor",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "3a2bc506-9a96-4dbf-bdb1-c0798b4712dd",
       "rows": [
        [
         "263",
         "263",
         "2-s2.0-85013158199",
         "10.1007/s40656-017-0129-2",
         "NULL",
         "28205138.0",
         "Kant‚Äôs epigenesis: specificity and developmental constraints",
         "ar",
         "Article",
         "Demarest B.",
         "60002483",
         "Universiteit van Amsterdam",
         "Amsterdam",
         "Netherlands",
         "1.0",
         "Demarest, Boris",
         "56204837900",
         "60002483",
         "2017-03-01 00:00:00",
         "1 March 2017",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "39",
         "1",
         "3",
         "NULL",
         "In this paper, I argue that Kant adopted, throughout his career, a position that is much more akin to classical accounts of epigenesis, although he does reject the more radical forms of epigenesis proposed in his own time, and does make use of preformationist sounding terms. I argue that this is because Kant (1) thinks of what is pre-formed as a species, not an individual or a part of an individual; (2) has no qualm with the idea of a specific, teleological principle or force underlying generation, and conceives of germs and predispositions as specific constraints on such a principle or force. Neither of these conceptions of what is ‚Äúpreformed‚Äù, I argue, is in strict opposition to classical epigenesis. I further suggest that Kant‚Äôs lingering use of preformationist terminology is due to (1) his belief that this is required to account for the specificity of the specific generative force; (2) his resistance towards the unrestricted plasticity of the generative force in radical epigenesis, which violates species-fixism; and (3) his insistence on the internal, organic basis of developmental plasticity and variation within species. I conclude by suggesting that this terminological and interpretative peculiarity is partly due to a larger shift in the natural philosophical concerns surrounding the debate on epigenesis and preformation. Specifically, it is a sign that the original reasons for resisting epigenesis, namely its use of specific, teleological principles and its commitment to the natural production of biological structure, became less of a concern, whereas unrestricted plasticity and its undermining of fixism became a real issue, thereby also becoming the focal point of the debate.",
         "Epigenesis | Generative force | Kant | Preformation",
         "7",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "369",
         "369",
         "2-s2.0-85088262863",
         "10.1007/s40656-020-00329-8",
         "NULL",
         "32691291.0",
         "What‚Äôs all the fuss about? The inheritance of acquired traits is compatible with the Central Dogma",
         "ar",
         "Article",
         "Camacho M.P.",
         "60015457",
         "University of Kansas",
         "Lawrence",
         "United States",
         "1.0",
         "Camacho, M. Polo",
         "57218197797",
         "60015457",
         "2020-09-01 00:00:00",
         "1 September 2020",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "42",
         "3",
         "32",
         "NULL",
         "The Central Dogma of molecular biology, which holds that DNA makes protein and not the other way around, is as influential as it is controversial. Some believe the Dogma has outlived its usefulness, either because it fails to fully capture the ins-and-outs of protein synthesis (Griffiths and Stotz in Genetics and philosophy Cambridge introductions to philosophy and biology, Cambridge University Press, Cambridge, 2013; Stotz in Hist Philos Life Sci 28(4):533‚Äì548, 2006), because it turns on a confused notion of information (Sarkar in Molecular models of life, MIT Press, Cambridge, 2004), or because it problematically assumes the unidirectional flow of information from DNA to protein (Gottlieb, in: Oyama, Griffiths, Gray (eds), Cycles of contingency: developmental systems and evolution, MIT Press, Cambridge, 2001). This paper evaluates an underexplored defense of the Dogma, which relies on the assumption that the Dogma and the Inheritance of Acquired Traits, a principle which dates as far back as Jean Baptiste-Lamarck, are incompatible principles (Smith in The theory of evolution, Cambridge University Press, Cambridge, 1993; Judson in The eighth day of creation, Jonathan Cape, London, 1979; Dawkins in The extended phenotype, Oxford University Press, Oxford, 1970; Cobb in PLoS Biol 15(9):e2003243, 2017. https://doi.org/10.1371/journal.pbio.2003243; Wilkins in BioEssays 24(10):960‚Äì973, 2002. https://doi.org/10.1002/bies.10167; Graur The fallacious commingling of two unrelated hypotheses: ‚Äòthe central dogma‚Äô and ‚Äòdna makes rna makes protein‚Äô. Judge Starling., 2018. http://judgestarling.tumblr.com/post/177554581856/the-fallacious-commingling-of-two-unrelated). By appealing to empirical evidence in molecular science, I argue that this apparent incompatibility is indeed merely apparent. I conclude by briefly demonstrating how these considerations bear on the topic of conceptual pluralism in the philosophy of science (Stencel and Proszewska in Found Sci 23(4):603‚Äì620, 2018. https://doi.org/10.1007/s10699-017-9543-x; Lu and Bourrat in Br J Philos Sci 69(3):775‚Äì800, 2018. https://doi.org/10.1093/bjps/axx019).",
         "Difference-making | Epigenetics | Gene-centrism | Inheritance of acquired traits | Inheritance systems | Lamarkian inheritance | The Central Dogma",
         "4",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "390",
         "390",
         "2-s2.0-84995794096",
         "10.1007/s40656-016-0121-2",
         "NULL",
         "27854053.0",
         "Epigenetics: ambiguities and implications",
         "ar",
         "Article",
         "Stotz K.",
         "60019544;60010710",
         "Macquarie University;Faculty of Medicine and Health",
         "Sydney;Sydney",
         "Australia;Australia",
         "2.0",
         "Stotz, Karola;Griffiths, Paul",
         "6602314296;57203312735",
         "60019544;60010710",
         "2016-12-01 00:00:00",
         "1 December 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "4",
         "22",
         "NULL",
         "Everyone has heard of ‚Äòepigenetics‚Äô, but the term means different things to different researchers. Four important contemporary meanings are outlined in this paper. Epigenetics in its various senses has implications for development, heredity, and evolution, and also for medicine. Concerning development, it cements the vision of a reactive genome strongly coupled to its environment. Concerning heredity, both narrowly epigenetic and broader ‚Äòexogenetic‚Äô systems of inheritance play important roles in the construction of phenotypes. A thoroughly epigenetic model of development and evolution was Waddington‚Äôs aim when he introduced the term ‚Äòepigenetics‚Äô in the 1940s, but it has taken the modern development of molecular epigenetics to realize this aim. In the final sections of the paper we briefly outline some further implications of epigenetics for medicine and for the nature/nurture debate.",
         "Epigenesis | Epigenetic inheritance | Epigenetics | Exogenetic inheritance | Genetic accommodation | Genetic assimilation",
         "29",
         "0",
         "NULL",
         "NULL",
         "TWCF",
         "TWCF0063/AB37",
         "Templeton World Charity Foundation"
        ],
        [
         "622",
         "622",
         "2-s2.0-85180664555",
         "10.1007/s40656-023-00600-8",
         "NULL",
         "38153583.0",
         "Resilience and the shift of paradigm in ecology: a new name for an old concept or a different explanatory tool?",
         "ar",
         "Article",
         "Barbara L.",
         "60003668",
         "Universit√† degli Studi del Piemonte Orientale ‚ÄúAmedeo Avogadro‚Äù",
         "Vercelli",
         "Italy",
         "1.0",
         "Barbara, Lara",
         "58782307600",
         "60003668",
         "2024-03-01 00:00:00",
         "March 2024",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "46",
         "1",
         "2",
         "NULL",
         "In the shift from the balance of nature to the flux of nature paradigm, the concept of resilience has gained great traction in ecology. While it has been suggested that the concept of resilience does not imply a genuine departure from the balance of nature paradigm, I shall argue against this stance. To do so, I first show that the balance of nature paradigm and the related conception of a single-state equilibrium relies on what Eliot Sober has named the ‚ÄúNatural State Model (NSM)‚Äù, suggesting that the NSM has instead been dismissed in the flux of nature paradigm. I then focus on resilience as the main explanatory concept of the flux paradigm. After distinguishing between two main different understandings of ‚Äúresilience‚Äù, namely engineering resilience and ecological resilience, I argue that the former is close to the concept of balance or stability¬†and still part of the NSM, while the latter is not. Finally, I claim that ecological resilience is inconsistent with the NSM, concluding that this concept‚Äìbeing incompatible with the NSM‚Äìis not part of the balance of nature paradigm but rather a genuinely new explanatory tool.",
         "Balance of nature | Ecological resilience | Flux of nature | Natural state model",
         "1",
         "1",
         "publisherfree2read",
         "Bronze",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "682",
         "682",
         "2-s2.0-85063972115",
         "10.1007/s40656-019-0252-3",
         "NULL",
         "30937631.0",
         "Taxonomy and conservation science: interdependent and value-laden",
         "ar",
         "Article",
         "Conix S.",
         "60025063",
         "KU Leuven",
         "Leuven",
         "Belgium",
         "1.0",
         "Conix, Stijn",
         "57201803483",
         "60025063",
         "2019-06-01 00:00:00",
         "1 June 2019",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "41",
         "2",
         "15",
         "NULL",
         "The relation between conservation science and taxonomy is typically seen as a simple dependency of the former on the latter. This dependency is assumed to be strictly one-way to avoid normative concerns from conservation science inappropriately affecting the descriptive discipline of taxonomy. In this paper, I argue against this widely assumed standard view on the relation between these two disciplines by highlighting two important roles for conservation scientists in scientific decisions that are part of the internal stages of taxonomy. I show that these roles imply that the two disciplines should be interdependent and that value-judgments should play a substantial role in both.",
         "Conservation science | Species classification | Taxonomy | Values in science",
         "14",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "3H160214",
         "KU Leuven"
        ],
        [
         "686",
         "686",
         "2-s2.0-85058017987",
         "10.1007/s40656-018-0236-8",
         "NULL",
         "30523424.0",
         "At the intersection of medical geography and disease ecology: Mirko Grmek, Jacques May and the concept of pathocenosis",
         "ar",
         "Article",
         "Arrizabalaga J.",
         "60028181",
         "CSIC - Instituci√≥n Mila y Fontanals de Investigaci√≥n en Humanidades (IMF)",
         "Barcelona",
         "Spain",
         "1.0",
         "Arrizabalaga, Jon",
         "7006291897",
         "60028181",
         "2018-12-01 00:00:00",
         "1 December 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "4",
         "71",
         "NULL",
         "Environmental historians are not sufficiently aware of the extent to which mid twentieth-century thinkers turned to medical geography‚Äîoriginally a nineteenth-century area of study‚Äîin order to think through ideas of ecology, environment, and historical reasoning. This article outlines how the French‚ÄìCroatian Mirko D. Grmek (Krapina, 1924‚ÄìParis, 2000), a major thinker of his generation in the history of medicine, used those ideas in his studies of historical epidemiology. During the 1960s, Grmek attempted to provide, in the context of the Annales School‚Äôs research program under the leadership of Fernand Braudel, a new theoretical framework for a world history of disease. Its development was inspired by several sources, most notably the French‚ÄìAmerican Jacques M. May (Paris 1896‚ÄìTunisia, 1975), who was then pioneering an opening up of medical geography and movement towards the concept of disease ecology. The cornerstone of Grmek‚Äôs ‚Äúsynthetic approach‚Äù to the field was the notion of ‚Äúpathocenosis‚Äù. The diverse uses of this notion in the course of time‚Äîfrom his early agenda focused on a longue dur√©e history of diseases in Western Antiquity to his last, relating to the new epidemiological threat of (re)emerging infectious diseases, specifically HIV/aids‚Äîenables us firstly, to note how concepts of ecology sat uneasily alongside those of medical geography; secondly, to assess the reach and limits of his theoretical contribution to historical epidemiology; and thirdly, to understand better the uneven fortunes of his concept of pathocenosis at the end of the twentieth and beginning of the twenty-first centuries.",
         "Disease ecology | Fernand Braudel | Historical epidemiology | History of diseases | Jacques M. May | Medical geography | Mirko D. Grmek | Pathocenosis | School of the Annales",
         "10",
         "1",
         "publisherfree2read",
         "Bronze",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "691",
         "691",
         "2-s2.0-85046898458",
         "10.1007/s40656-018-0194-1",
         "NULL",
         "29761370.0",
         "Multispecies individuals",
         "ar",
         "Article",
         "Bourrat P.",
         "60025709;60019544;60010710",
         "The University of Sydney;Macquarie University;Faculty of Medicine and Health",
         "Sydney;Sydney;Sydney",
         "Australia;Australia;Australia",
         "2.0",
         "Bourrat, Pierrick;Griffiths, Paul E.",
         "36699240600;57203312735",
         "60019544-60025709-60010710;60010710",
         "2018-06-01 00:00:00",
         "1 June 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "2",
         "33",
         "NULL",
         "We assess the arguments for recognising functionally integrated multispecies consortia as genuine biological individuals, including cases of so-called ‚Äòholobionts‚Äô. We provide two examples in which the same core biochemical processes that sustain life are distributed across a consortium of individuals of different species. Although the same chemistry features in both examples, proponents of the holobiont as unit of evolution would recognize one of the two cases as a multispecies individual whilst they would consider the other as a compelling case of ecological dependence between separate individuals. Some widely used arguments in support of the ‚Äòholobiont‚Äô concept apply equally to both cases, suggesting that those arguments have misidentified what is at stake when seeking to identify a new level of biological individuality. One important aspect of biological individuality is evolutionary individuality. In line with other work on the evolution of individuality, we show that our cases can be distinguished by focusing on the fitness alignment between the partners of the consortia. We conclude that much of the evidence currently presented for the ubiquity and importance of multi-species individuals is simply not to the point, at least unless the issue of biological individuality is firmly divorced from the question of evolutionary individuality.",
         "Evolutionary transitions in individuality | Holobiont | Individuality | Symbiosis",
         "30",
         "0",
         "NULL",
         "NULL",
         "CEED",
         "DP0878650",
         "Centre of Excellence for Environmental Decisions, Australian Research Council"
        ],
        [
         "710",
         "710",
         "2-s2.0-84988353676",
         "10.1007/s40656-016-0113-2",
         "NULL",
         "27645228.0",
         "From ecological records to big data: the invention of global biodiversity",
         "ar",
         "Article",
         "Devictor V.",
         "60108488;60021260",
         "Universit√© de Montpellier;Universit√© Paris 1 Panth√©on-Sorbonne",
         "Montpellier;Paris",
         "France;France",
         "2.0",
         "Devictor, Vincent;Bensaude-Vincent, Bernadette",
         "15019082600;6603267674",
         "60021260-60108488;60021260",
         "2016-12-01 00:00:00",
         "1 December 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "4",
         "13",
         "NULL",
         "This paper is a critical assessment of the epistemological impact of the systematic quantification of nature with the accumulation of big datasets on the practice and orientation of ecological science. We examine the contents of big databases and argue that it is not just accumulated information; records are translated into digital data in a process that changes their meanings. In order to better understand what is at stake in the ‚Äòdatafication‚Äô process, we explore the context for the emergence and quantification of biodiversity in the 1980s, along with the concept of the global environment. In tracing the origin and development of the global biodiversity information facility (GBIF) we describe big data biodiversity projects as a techno-political construction dedicated to monitoring a new object: the global diversity. We argue that, biodiversity big data became a powerful driver behind the invention of the concept of the global environment, and a way to embed ecological science in the political agenda.",
         "Big data | Biodiversity | Ecology | Foucault | Politics",
         "50",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "713",
         "713",
         "2-s2.0-84940554967",
         "10.1007/s40656-015-0072-z",
         "NULL",
         "26141169.0",
         "‚ÄòBig science‚Äô in the field: experimenting with badgers and bovine TB, 1995‚Äì2015",
         "ar",
         "Article",
         "Cassidy A.",
         "60011520",
         "King's College London",
         "London",
         "United Kingdom",
         "1.0",
         "Cassidy, Angela",
         "55138529300",
         "60011520",
         "2015-10-01 00:00:00",
         "1 October 2015",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "37",
         "3",
         "NULL",
         "305-325",
         "Since wild badgers were first connected with outbreaks of bovine TB (bTB) in UK cattle herds in the early 1970s, the question of whether to cull them to control infections in cattle has been the subject of a protracted public and policy controversy. Following the recommendation of Prof. John Krebs that a ‚Äúscientifically based experimental trial‚Äù be carried out to test the effectiveness of badger culling, the Randomised Badger Culling Trial (RBCT) was commissioned by Government in 1998. One of the largest field experiments ever conducted in the UK, the RBCT sought to recreate the conditions of a randomised controlled trial (RCT) across approximately 3000¬†km<sup>2</sup> of the South West of England. Despite widespread expectations that the RBCT would provide the necessary evidence to resolve the controversy, its findings have instead been widely contested and reinterpreted, while arguments over badger culling have become increasingly polarised. This paper will investigate the complexities of field experimental knowledge by following the story of the RBCT from this initial proposal, through processes of research design, implementation, analysis, interpretation and reinterpretation of the findings by multiple actors. It asks what kind of experiment the RBCT actually was, and examines how it has contributed to the protracted controversy over whether to cull badgers in order to control bTB in cattle. Finally, it will explore the wider implications of this case for contemporary debates over the contribution that RCTs can make to formulating public policy.",
         "Agricultural research | Animal health | Ecology | Evidence based policy | Field science | RCTs | Science and policy",
         "12",
         "1",
         "repositoryvor",
         "Green",
         "WT",
         "RES-229-27-0007-A",
         "Wellcome Trust"
        ],
        [
         "1343",
         "1343",
         "2-s2.0-85050994742",
         "10.1007/s40656-018-0209-y",
         "NULL",
         "30066110.0",
         "The instability of field experiments: building an experimental research tradition on the rocky seashores (1950‚Äì1985)",
         "ar",
         "Article",
         "Grodwohl J.",
         "60031101;60013710",
         "University of Cambridge;Universidade Federal da Bahia",
         "Cambridge;Salvador",
         "United Kingdom;Brazil",
         "3.0",
         "Grodwohl, Jean Baptiste;Porto, Franco;El-Hani, Charbel N.",
         "55081908800;57203227374;6506237910",
         "60013710-60031101;60013710;60013710",
         "2018-09-01 00:00:00",
         "1 September 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "3",
         "45",
         "NULL",
         "In many experimental sciences, like particle physics or molecular biology, the proper place for establishing facts is the laboratory. In the sciences of population biology, however, the laboratory is often seen as a poor approximation of what occurs in nature. Results obtained in the field are usually more convincing. This raises special problems: it is much more difficult to obtain stable, repeatable results in the field, where environmental conditions vary out of the experimenter‚Äôs control, than in the laboratory. We examine here how this problem affected an influential experimental research tradition in community ecology, the study of the ecology of the rocky seashores. In the 1960s, a handful of North-American ecologists, most notably Joseph Connell, Robert Paine and Paul Dayton, made the rocky seashores a model study system for experimenting in the field. Their experiments were deceptively simple: they removed species living on the seashore and described the resulting effects on the local ecology. These experiments exerted a deep influence on community ecology. They provided evidence for speculative developments concerning the theory of interspecific competition, the factors responsible for species richness and the ecology of food webs. They also stimulated novel conceptual developments. In particular, Paine developed the predation hypothesis, which states that the presence of predators can favour species richness, before introducing the keystone species concept, according to which some species exert disproportionate effects on ecological systems. More broadly, these experiments gave support to a methodological trend in favour of field experimentation. Only controlled perturbations in the field, it seemed, provided a reliable method to get insights into the structure of ecological communities. However, as experiments were continued in time and repeated in different sites, divergent results appeared. We analyse here how intertidal researchers coped with the variability of environmental conditions and tried to stabilize their results. In the process, they reconsidered not only their early conclusions, but also the exclusive status given to field experiments. Expanding on this case study, we discuss some significant differences between laboratory and field experiments.",
         "Ecology | Field experiments | Food webs | Interspecific competition | Intertidal | Joseph Connell | Keystone species | Robert MacArthur | Robert Paine",
         "5",
         "0",
         "NULL",
         "NULL",
         "ONR",
         "undefined",
         "Office of Naval Research"
        ],
        [
         "1517",
         "1517",
         "2-s2.0-85188584139",
         "10.1007/s40656-024-00616-8",
         "NULL",
         "38530473.0",
         "The discovery of archaea: from observed anomaly to consequential restructuring of the phylogenetic tree",
         "ar",
         "Article",
         "Fry M.",
         "60022403",
         "Technion - Israel Institute of Technology",
         "Haifa",
         "Israel",
         "1.0",
         "Fry, Michael",
         "7102918054",
         "60022403",
         "2024-06-01 00:00:00",
         "June 2024",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "46",
         "2",
         "16",
         "NULL",
         "Observational and experimental discoveries of new factual entities such as objects, systems, or processes, are major contributors to some advances in the life sciences. Yet, whereas discovery of theories was extensively deliberated by philosophers of science, very little philosophical attention was paid to the discovery of factual entities. This paper examines historical and philosophical aspects of the experimental discovery by Carl Woese of archaea, prokaryotes that comprise one of the three principal domains of the phylogenetic tree. Borrowing Kuhn‚Äôs terminology, this discovery of a major biological entity was made during a ‚Äònormal science‚Äô project of building molecular taxonomy for prokaryotes. Unexpectedly, however, an observed anomaly instigated the discovery of archaea. Substantiation of the existence of the new archaeal entity and consequent reconstruction of the phylogenetic tree prompted replacement of a long-held model of a prokarya and eukarya bipartite tree of life by a new model of a tripartite tree comprising of bacteria, archaea, and eukarya. This paper explores the history and philosophical implications of the progression of Woese‚Äôs project from normal science to anomaly-instigated model-changing discovery. It is also shown that the consequential discoveries of RNA splicing and of ribozymes were similarly prompted by unexpected irregularities during normal science activities. It is thus submitted that some discoveries of factual biological entities are triggered by unforeseen observational or experimental anomalies.",
         "Archaea | Carl Woese | Microbial phylogenetics | Ribosomal RNA-based phylogenetics | Scientific discovery",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "NASA",
         "undefined",
         "National Aeronautics and Space Administration"
        ],
        [
         "1542",
         "1542",
         "2-s2.0-85181652890",
         "10.1007/s40656-023-00601-7",
         "NULL",
         "38190055.0",
         "On the nature of evolutionary explanations: a critical appraisal of Walter Bock‚Äôs approach with a new revised proposal",
         "ar",
         "Article",
         "de Santis M.D.",
         "60000036;101958458",
         "Universidade Federal do Rio de Janeiro;Museum Koenig Bonn",
         "Rio de Janeiro;Bonn",
         "Brazil;Germany",
         "1.0",
         "de Santis, Marcelo Domingos",
         "57204216893",
         "60000036-101958458",
         "2024-03-01 00:00:00",
         "March 2024",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "46",
         "1",
         "3",
         "NULL",
         "Walter Bock was committed to developing a framework for evolutionary biology. Bock repeatedly discussed how evolutionary explanations should be considered within the realm of Hempel‚Äôs deductive-nomological model of scientific explanations. Explanation in evolution would then consist of functional and evolutionary explanations, and within the latter, an explanation can be of nomological-deductive and historical narrative explanations. Thus, a complete evolutionary explanation should include, first, a deductive functional analysis, and then proceed through nomological and historical evolutionary explanations. However, I will argue that his views on the deductive proprieties of functional analysis and the deductive-nomological parts of evolution fail because of the nature of evolution, which contains a historical element that the logic of deduction and Hempel‚Äôs converting law model do not compass. Conversely, Bock‚Äôs historical approach gives a critical consideration of the historical narrative element of evolutionary explanation, which is fundamental to the methodology of the historical nature of evolutionary theory. Herein, I will expand and discuss a modern view of evolutionary explanations of traits that includes the currentacknowledgement of the differences between experimental and the historical sciences, including the token and type event dichotomy, that mutually illuminate each other in order to give us a well confirmed and coherent hypothesis for evolutionary explanations. Within this framework, I will argue that the duality of evolutionary explanations is related to two components of character evolution: origin, with its evolutionary pathways along with the history, and maintenance, the function (mainly a current function) for the character being selected.",
         "Deductive-nomological model | Evolutionary biology | Evolutionary explanation | Historical explanation | Narrative explanation",
         "1",
         "1",
         "repositoryvor",
         "Green",
         "CNPq",
         "150141/2022-0",
         "Conselho Nacional de Desenvolvimento Cient√≠fico e Tecnol√≥gico"
        ],
        [
         "1575",
         "1575",
         "2-s2.0-85165350433",
         "10.1007/s40656-023-00590-7",
         "NULL",
         "37477714.0",
         "Finding value-ladenness in evolutionary psychology: Examining Nelson‚Äôs arguments",
         "ar",
         "Article",
         "Amitani Y.",
         "60031404",
         "The University of Aizu",
         "Aizuwakamatsu",
         "Japan",
         "1.0",
         "Amitani, Yuichi",
         "56403129800",
         "60031404",
         "2023-09-01 00:00:00",
         "September 2023",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "45",
         "3",
         "36",
         "NULL",
         "Faced with the charge of value-ladenness in their theories, researchers in evolutionary psychology (EP) argue that their science is entirely free of values; their hypotheses only concern scientific facts, without any socio-cultural value judgments. Lynn Hankinson Nelson, a renowned feminist scholar of science, denies this. In her book and papers, Nelson finds that their hypotheses do contain evaluative components. One such example is the fear of snakes. While this fear was adaptive to the environment in the past, evolutionary psychologists argue that this trait is now ‚Äúmaladaptive‚Äù because city-dwellers would rarely encounter snakes in their environment. However, Nelson argues that labeling this trait ‚Äúmaladaptive‚Äù implies that this fear is irrational since this claim cannot be understood otherwise. This paper argues that this and other arguments made by Nelson for demonstrating the value-ladenness in EP‚Äôs hypotheses have serious flaws. For instance, we argue that investigating the psychological mechanisms behind the fear and their developmental and energy costs would allow for proper interpretation of evolutionary psychologists‚Äô claims for the maladaptive fear of snakes without any normative implication. We also maintain that some of her arguments fail to demonstrate their connection to the point at the center of the debates between EP and feminism. While Nelson may be right in stating that EP‚Äôs hypotheses have evaluative components, she does not prove their strong political or normative implications, which is central to the debate over EP.",
         "Evolutionary psychology | Feminist critique of science | Normativity | Sociocultural biases | Value-ladenness",
         "0",
         "0",
         "NULL",
         "NULL",
         "KAKEN",
         "21K00036",
         "Japan Society for the Promotion of Science"
        ],
        [
         "1607",
         "1607",
         "2-s2.0-85036630762",
         "10.1007/s40656-017-0176-8",
         "NULL",
         "29181597.0",
         "From lighthouse to hothouse: hospital hygiene, antibiotics and the evolution of infectious disease, 1950‚Äì1990",
         "ar",
         "Article",
         "Gradmann C.",
         "60010348",
         "Universitetet i Oslo",
         "Oslo",
         "Norway",
         "1.0",
         "Gradmann, Christoph",
         "6603365266",
         "60010348",
         "2018-03-01 00:00:00",
         "1 March 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "1",
         "8",
         "NULL",
         "Upon entering clinical medicine in the 1940s, antibiotic therapy seemed to complete a transformation of hospitals that originated in the late nineteenth century. Former death sinks had become harbingers of therapeutic progress. Yet this triumph was short-lived. The arrival of pathologies caused by resistant bacteria, and of nosocomial infections whose spread was helped by antibiotic therapies, seemed to be intimately related to modern anti-infective therapy. The place where such problems culminated were hospitals, which increasingly appeared as dangerous environments where attempts to combat infectious diseases had instead created hothouses of disease evolution. This paper will focus on one aspect of that history. It caused clinical medicine and hospital hygiene in particular to pay attention to a dimension of infectious disease it had previously paid little attention to thus far: The evolution of infectious disease‚Äîpreviously a matter of mostly theoretical interest‚Äîcame to be useful in explaining many phenomena observed. This did not turn hospital hygienists into geneticists, though it did give them an awareness that the evolution of infectious disease in a broad sense was something that did matter to them. The paper advances its argument by looking at three phases: The growing awareness of the hospital as a dangerous environment in the 1950s, comprehensive attempts at improving antibiotic therapy and hospital hygiene that followed from the 1960s and lastly the framing of such challenges as risk factors from the 1970s. In conclusion, I will argue that hospital hygiene, being inspired in particular by epidemiology and risk factor analysis, discussed its own specific version of disease emergence and therefore contributed to the 1980s debates around such topics. Being loosely connected to more specialized studies, it consisted of a re-interpretation of infectious disease centred around the temporality of such phenomena as they were encountered in day-to-day dealings of clinical wards.",
         "Antibiotics resistance | Disease evolution | Epidemiology | Hospital infections | Medical microbiology | Risk factors",
         "20",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1623",
         "1623",
         "2-s2.0-85141163937",
         "10.1007/s40656-022-00534-7",
         "NULL",
         "36326954.0",
         "Organic form and evolution: the morphological problem in twentieth-century italian biology",
         "ar",
         "Article",
         "Tamborini M.",
         "60011226",
         "Technische Universit√§t Darmstadt",
         "Darmstadt",
         "Germany",
         "1.0",
         "Tamborini, Marco",
         "56543381300",
         "60011226",
         "2022-12-01 00:00:00",
         "December 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "4",
         "54",
         "NULL",
         "This paper examines the efforts in evolution research to understand form‚Äôs structure that developed in Italy during the first half of the twentieth century. In particular, it analyzes how the organic approach in biology and the study of organic form merged in the morphological research agendas of Giuseppe Colosi (1892‚Äì1975) and Giuseppe Levi (1872‚Äì1965). These biologists sought to understand form‚Äôs inner composition and structure. First, I will briefly outline the morphological practices and frameworks used to study form changes and structures in the early twentieth century. Second, I will discuss what the Italian biologist Antonio Pensa (1874‚Äì1970) called the morphological problem. Third, I will examine Colosi‚Äôs response to the morphological problem. Fourth, I will analyze Levi‚Äôs morphological research program. As a result, this paper paves the way for a more nuanced and varied picture of the so-called ‚Äúorganicism movement‚Äù in the first half of the twentieth century by calling attention to morphology as practiced in Italian-speaking biology. In fact, alongside dialectical materialism and holistic biology, two of the main strands within organicism, the architectural approach to evolution as practiced in Italy and elsewhere had a profound impact on twentieth- and twenty-first-century organicism specifically and on evolutionary biology generally.",
         "NULL",
         "1",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "Technische Universit√§t Darmstadt"
        ],
        [
         "1638",
         "1638",
         "2-s2.0-85140180406",
         "10.1007/s40656-022-00531-w",
         "NULL",
         "36257998.0",
         "Phenotype-first hypotheses, spandrels and early metazoan evolution",
         "ar",
         "Article",
         "Rust J.",
         "60021742",
         "Stetson University",
         "De Land",
         "United States",
         "1.0",
         "Rust, Joshua",
         "35185427100",
         "60021742",
         "2022-12-01 00:00:00",
         "December 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "4",
         "48",
         "NULL",
         "Against the neo-Darwinian assumption that genetic factors are the principal source of variation upon which natural selection operates, a phenotype-first hypothesis strikes us as revolutionary because development would seem to constitute an independent source of variability. Richard Watson and his co-authors have argued that developmental memory constitutes one such variety of phenotypic variability. While this version of the phenotype-first hypothesis is especially well-suited for the late metazoan context, where animals have a sufficient history of selection from which to draw, appeals to developmental memory seem less plausible in the evolutionary context of the early metazoans. I provide an interpretation of Stuart Newman‚Äôs account of deep metazoan phylogenesis that suggests that spandrels are, in addition to developmental memory, an important reservoir of phenotypic variability. I conclude by arguing that Gerd M√ºller‚Äôs ‚Äúside-effect hypothesis‚Äù is an illuminating generalization of the proposed non-Watsonian version of the phenotype-first hypothesis.",
         "Early metazoan evolution | Novelty and Innovation | Plasticity-first hypothesis | Spandrels | Variability",
         "0",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1742",
         "1742",
         "2-s2.0-85128481682",
         "10.1007/s40656-022-00500-3",
         "NULL",
         "35441959.0",
         "Matthew‚Äôs (1915) climate and evolution, the ‚ÄúNew York School of Biogeography‚Äù, and the rise and fall of ‚ÄúHolarcticism‚Äù",
         "ar",
         "Article",
         "Morrone J.J.",
         "60032442",
         "Universidad Nacional Aut√≥noma de M√©xico",
         "Mexico",
         "Mexico",
         "1.0",
         "Morrone, Juan J.",
         "7003835311",
         "60032442",
         "2022-06-01 00:00:00",
         "June 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "2",
         "15",
         "NULL",
         "Climate and evolution (Matthew, 1915) represents an important contribution to evolutionary biogeography, that influenced several authors, notably Karl P. Schmidt, George S. Myers, George G. Simpson, Philip J. Darlington, Ernst Mayr, Thomas Barbour, John C. Poynton, Allen Keast, L√©on Croizat, Robin Craw, Michael Heads, and Osvaldo A. Reig. Authors belonging to the ‚ÄúNew York School of Zoogeography‚Äù ‚Äìa research community including Matthew, Schmidt, Myers and Simpson‚Äì accepted Matthew‚Äôs ‚ÄúHolarcticism‚Äù (north temperate centers of origin) and the permanence of ocean basins and continents, whereas others, especially panbiogeographers and cladistic biogeographers, were extremely critical and reacted against these ideas. ‚ÄúHolarcticism‚Äù has been falsified and rejected by dispersalists and the ‚ÄúNew York School of Zoogeography‚Äù disappeared in the 1970s. Matthew, however, continues being identified by panbiogeographers and cladistic biogeographers as a key representative of classic dispersalism, helping provide some cohesion to their research communities.",
         "Australism | Dispersalism | Tropicalism | ‚ÄúNew York School of Zoogeography‚Äù",
         "2",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1756",
         "1756",
         "2-s2.0-85131050189",
         "10.1007/s40656-022-00505-y",
         "NULL",
         "35648240.0",
         "The evolution of moral belief: support for the debunker‚Äôs causal premise",
         "ar",
         "Article",
         "Dale M.T.",
         "60013372",
         "The University of Texas at Austin",
         "Austin",
         "United States",
         "1.0",
         "Dale, Michael T.",
         "57204552155",
         "60013372",
         "2022-06-01 00:00:00",
         "June 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "2",
         "23",
         "NULL",
         "The causal premise of the evolutionary debunking argument contends that human moral beliefs are explained by the process of natural selection. While it is universally acknowledged that such a premise is fundamental to the debunker‚Äôs case, the vast majority of philosophers focus instead on the epistemic premise that natural selection does not track moral truth and the resulting skeptical conclusion(s). Recently, however, some have begun to concentrate on the causal premise. So far, the upshot of this small but growing literature has been that the causal premise is likely false due to the seemingly persuasive evidence that our moral beliefs are in fact not the result of natural selection. In this paper, I argue that this view is mistaken. Specifically, I advocate the Innate Biases Model, which contends that there is not only compelling evidence for an evolved cognitive capacity for acquiring norms but also for the existence of an evolutionarily instilled set of cognitive biases that make it either more or less likely that we adopt certain moral beliefs.",
         "Causal premise | Debunking | Ethics | Evolution | Innate biases model | Metaethics",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1757",
         "1757",
         "2-s2.0-85130744018",
         "10.1007/s40656-022-00493-z",
         "NULL",
         "35614290.0",
         "Values in evolutionary biology: a comparison between the contemporary debate on organic progress and Canguilhem‚Äôs biological philosophy",
         "ar",
         "Article",
         "De Cesare S.",
         "60004718",
         "Universit√© de Gen√®ve",
         "Geneva",
         "Switzerland",
         "1.0",
         "De Cesare, Silvia",
         "57205075108",
         "60004718",
         "2022-06-01 00:00:00",
         "June 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "2",
         "22",
         "NULL",
         "The aim of this paper is to make a comparison and build up a dialogue between two different philosophical approaches to values in evolutionary biology. First, I present the approach proposed by Alexander Rosenberg and Daniel McShea in their contribution to the contemporary debate on organic progress. i.e. the idea that there has been some kind of improvement concerning organisms over the history of life. Discussing organic progress raises the question of what ‚Äúbetter‚Äù exactly means. This requires an explicit clarification on what legitimately means to speak about ‚Äúgood‚Äù in evolutionary biology, thus to speak about values. Second, I move on to present an approach to values that has been proposed by Georges Canguilhem in the context of a different philosophical tradition (i.e. the ‚Äúcontinental‚Äù tradition). Canguilhem‚Äôs original theses are conceived in a Darwinian framework and clearly relate to the question of values in evolutionary biology. I shall then propose a comparison between these two heterogeneous perspectives on values by critically evaluating their common points and main differences. I will argue that both perspectives agree that the question of values in evolutionary biology takes on its full meaning with respect to the relationship between the organism and the environment. However, the framework for conceptualizing values in evolutionary biology provided by Rosenberg and McShea neglects a significant point highlighted by Canguilhem, i.e. the active role that the organism can play in evaluating the environment. In line with recent developments of biology (e.g. niche construction), this point can be easily integrated into Rosenberg and McShea‚Äôs framework. Finally, I will point out some main differences between the two perspectives relative to the specificity of Canguilhem‚Äôs biological philosophy.",
         "Fitness | Georges Canguilhem | Normativity | Organic progress | Philosophy of evolutionary biology | Values",
         "1",
         "1",
         "repositoryvor",
         "Green",
         "UNIGE",
         "undefined",
         "Universit√© de Gen√®ve"
        ],
        [
         "1772",
         "1772",
         "2-s2.0-85128076662",
         "10.1007/s40656-022-00485-z",
         "NULL",
         "35411477.0",
         "Natural selection according to Darwin: cause or effect?",
         "ar",
         "Article",
         "Bradley B.",
         "60001248",
         "Charles Sturt University",
         "Bathurst",
         "Australia",
         "1.0",
         "Bradley, Ben",
         "7201659223",
         "60001248",
         "2022-06-01 00:00:00",
         "June 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "2",
         "13",
         "NULL",
         "In the 1940s, the ‚Äòmodern synthesis‚Äô (MS) of Darwinism and genetics cast genetic mutation and recombination as the source of variability from which environmental events naturally select the fittest, such ‚Äònatural selection‚Äô constituting the cause of evolution. Recent biology increasingly challenges this view by casting genes as followers and awarding the leading role in the genesis of adaptations to the agency and plasticity of developing phenotypes‚Äîmaking natural selection a consequence of other causal processes. Both views of natural selection claim to capture the core of Darwin‚Äôs arguments in On the Origin of Species. Today, historians largely concur with the MS‚Äôs reading of Origin as a book aimed to prove natural selection the cause (vera causa) of adaptive change. This paper finds the evidence for that conclusion wanting. I undertake to examine the context and meaning of all Darwin‚Äôs known uses of the phrase vera causa, documenting in particular Darwin‚Äôs resistance to the pressure to prove natural selection a vera causa in letters written early in 1860. His resistance underlines the logical dependence of natural selection, an unobservable phenomenon, on the causal processes producing the observable events captured by the laws of inheritance, variation, and the struggle for existence, established in Chapters 1‚Äì3 of Origin.",
         "Agency | Darwinism | Development | Historiography | Phenotype | Vera causa",
         "5",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1776",
         "1776",
         "2-s2.0-85123424291",
         "10.1007/s40656-022-00482-2",
         "NULL",
         "35061130.0",
         "Evolution and ethics viewed from within two metaphors: machine and organism",
         "ar",
         "Article",
         "Ruse M.",
         "60002092",
         "Florida State University",
         "Tallahassee",
         "United States",
         "1.0",
         "Ruse, Michael",
         "7003479387",
         "60002092",
         "2022-03-01 00:00:00",
         "March 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "1",
         "1",
         "NULL",
         "How is moral thinking, ethics, related to evolutionary theorizing? There are two approaches, epitomized by Charles Darwin who works under the metaphor of the world as a machine, and by Herbert Spencer who works under the metaphor of the world as an organism. Although the author prefers the first approach, the aim of this paper is to give a disinterested account of both approaches.",
         "Charles Darwin | Ethics | Herbert Spencer | Mechanism | Naturalistic fallacy | Organicism | Progress",
         "2",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1788",
         "1788",
         "2-s2.0-85125611770",
         "10.1007/s40656-022-00483-1",
         "NULL",
         "35239015.0",
         "Biological constraints as norms in evolution",
         "ar",
         "Article",
         "Tahar M.",
         "60005899",
         "Universit√© Toulouse - Jean Jaur√®s",
         "Toulouse",
         "France",
         "1.0",
         "Tahar, Mathilde",
         "57474676100",
         "60005899",
         "2022-03-01 00:00:00",
         "March 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "1",
         "9",
         "NULL",
         "Biology seems to present local and transitory regularities rather than immutable laws. To account for these historically constituted regularities and to distinguish them from mathematical invariants, Mont√©vil and Mossio (Journal of Theoretical Biology 372:179‚Äì191, 2015) have proposed to speak of constraints. In this article we analyse the causal power of these constraints in the evolution of biodiversity, i.e., their positivity, but also the modality of their action on the directions taken by evolution. We argue that to fully account for the causal power of these constraints on evolution, they must be thought of in terms of normativity. In this way, we want to highlight two characteristics of the evolutionary constraints. The first, already emphasised as reported by Gould (The structure of evolutionary theory, Harvard University Press, 2002), is that these constraints are both produced by and producing biological evolution and that this circular causation creates true novelties. The second is that this specific causality, which generates unpredictability in evolution, stems not only from the historicity of biological constraints, but also from their internalisation through the practices of living beings.",
         "Biological agency | Biological constraints | Historicity | Normativity | Possibility space | Theory of evolution",
         "1",
         "1",
         "repositoryvor",
         "Green",
         "ICA",
         "undefined",
         "Institut Cl√©ment Ader, Universit√© de Toulouse"
        ],
        [
         "1794",
         "1794",
         "2-s2.0-85120699175",
         "10.1007/s40656-021-00476-6",
         "NULL",
         "34822012.0",
         "Medawar and Hamilton on the selective forces in the evolution of ageing",
         "ar",
         "Article",
         "Giaimo S.",
         "60000726",
         "Max-Planck-Institut f√ºr Evolutionsbiologie",
         "Plon",
         "Germany",
         "1.0",
         "Giaimo, Stefano",
         "55084930500",
         "60000726",
         "2021-12-01 00:00:00",
         "December 2021",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "43",
         "4",
         "124",
         "NULL",
         "Both Medawar and Hamilton contributed key ideas to the modern evolutionary theory of ageing. In particular, they both suggested that, in populations with overlapping generations, the force with which selection acts on traits declines with the age at which traits are expressed. This decline would eventually cause ageing to evolve. However, the biological literature diverges on the relationship between Medawar‚Äôs analysis of the force of selection and Hamilton‚Äôs. Some authors appear to believe that Hamilton perfected Medawar‚Äôs insightful, yet ultimately erroneous analysis of this force, while others see Hamilton‚Äôs analysis as a coherent development of, or the obvious complement to Medawar‚Äôs. Here, the relationship between the two analyses is revisited. Two things are argued for. First, most of Medawar‚Äôs alleged errors that Hamilton would had rectified seem not to be there. The origin of these perceived errors appears to be in a misinterpretation of Medawar‚Äôs writings. Second, the mathematics of Medawar and that of Hamilton show a significant overlap. However, different meanings are attached to the same mathematical expression. Medawar put forth an expression for the selective force on age-specific fitness. Hamilton proposed a full spectrum of selective forces each operating on age-specific fitness components, i.e. mortality and fertility. One of Hamilton‚Äôs expressions, possibly his most important, is of the same form as Medawar‚Äôs expression. But Hamilton‚Äôs selective forces on age-specific fitness components do not add up to yield Medawar‚Äôs selective force on age-specific fitness. It is concluded that Hamilton‚Äôs analysis should be considered neither as a correction to Medawar‚Äôs analysis nor as its obvious complement.",
         "Age | Reproductive value | Senescence | Theory",
         "2",
         "1",
         "repositoryvor",
         "Green",
         "MPG",
         "undefined",
         "Max-Planck-Gesellschaft"
        ],
        [
         "1796",
         "1796",
         "2-s2.0-85119518905",
         "10.1007/s40656-021-00471-x",
         "NULL",
         "34807335.0",
         "The red-beard evolutionary explanation of human sociality",
         "ar",
         "Article",
         "Koliofotis V.",
         "60022265",
         "Erasmus Universiteit Rotterdam",
         "Rotterdam",
         "Netherlands",
         "1.0",
         "Koliofotis, Vaios",
         "57220101412",
         "60022265",
         "2021-12-01 00:00:00",
         "December 2021",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "43",
         "4",
         "123",
         "NULL",
         "Recent evolutionary studies on cooperation devote specific attention to non-verbal expressions of emotions. In this paper, I examine Robert Frank‚Äôs popular attempt to explain emotions, non-verbal markers and social behaviours. Following this line of work, I focus on the green-beard explanation of social behaviours. In response to the criticisms raised against this controversial ultimate explanation, based on resources found in Frank‚Äôs work, I propose an alternative red-beard explanation of human sociality. The red-beard explanation explains the emergence and evolution of emotions, a proximate cause, rather than patterns of behaviour. In contrast to simple evolutionary models that invoke a green-beard mechanism, I demonstrate that the red-beard explanation can be evolutionary stable. Social emotions are a common cause of a social behaviour and a phenotypic marker and therefore cooperative behaviour cannot be suppressed without also changing the marker.",
         "Altruism | Cooperation | Emotion | Evolution | Green-beard | Signal",
         "0",
         "1",
         "repositoryvor",
         "Green",
         "TU/e",
         "undefined",
         "Technische Universiteit Eindhoven"
        ],
        [
         "1809",
         "1809",
         "2-s2.0-85118787330",
         "10.1007/s40656-021-00467-7",
         "NULL",
         "34755237.0",
         "Theistic evolution and evolutionary ethics: Henry Fairfield Osborn and Huxley‚Äôs legacy",
         "ar",
         "Article",
         "Ceccarelli D.",
         "60027509",
         "Universit√† degli Studi di Roma \"Tor Vergata\"",
         "Rome",
         "Italy",
         "1.0",
         "Ceccarelli, David",
         "57190867042",
         "60027509",
         "2021-12-01 00:00:00",
         "December 2021",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "43",
         "4",
         "114",
         "NULL",
         "Scholars have often considered evolutionary social theories a product of Positivist scientism and the naturalization of ethics. Yet the theistic foundations of many evolutionary theories proposed between the nineteenth and the twentieth centuries bolstered the belief that following natural laws was morally desirable, if not vital, to guaranteeing social and moral progress. In the early twentieth century, American paleontologist and leading evolutionist Henry Fairfield Osborn represented one of the most authoritative advocates of this interpretation of natural normativity. Particularly during the last years of his career, Osborn used theological arguments to reinforce his advocacy of evolutionary ethics and social control policies, which led him to challenge his ‚Äúold master‚Äù Thomas Huxley regarding the separation between evolution and moral conduct. This article examines the development of Osborn‚Äôs evolutionary ethics, with particular regard to its bearing on the American debate on euthenics. I argue that theistic topics played a major rhetorical role in the attempt to justify normative conclusions drawn from ostensible laws of evolution.",
         "Evolutionary ethics | Evolutionary social theories | Henry Fairfield Osborn | Theistic evolutionism | Thomas Huxley",
         "2",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "Yale University"
        ],
        [
         "1859",
         "1859",
         "2-s2.0-85063268708",
         "10.1007/s40656-019-0250-5",
         "NULL",
         "30895399.0",
         "Evolutionary causes as mechanisms: a critical analysis",
         "ar",
         "Article",
         "P√©rez-Gonz√°lez S.",
         "60002644",
         "Universitat de Val√®ncia",
         "Valencia",
         "Spain",
         "2.0",
         "P√©rez-Gonz√°lez, Sa√∫l;Luque, Victor J.",
         "57207937810;57189574934",
         "60002644;60002644",
         "2019-06-01 00:00:00",
         "1 June 2019",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "41",
         "2",
         "13",
         "NULL",
         "In this paper, we address the question whether a mechanistic approach can account for evolutionary causes. The last decade has seen a major attempt to account for natural selection as a mechanism. Nevertheless, we stress the relevance of broadening the debate by including the other evolutionary causes inside the mechanistic approach, in order to be a legitimate conceptual framework on the same footing as other approaches to evolutionary theory. We analyse the current debate on natural selection as a mechanism, and extend it to the rest of the evolutionary causes. We focus on three approaches that we call the stochastic view, the functional view, and the minimalist view. We argue that all of them are unable to account for evolutionary causes as mechanisms. It is concluded that the current mechanistic proposals cannot be accepted as a common framework for evolutionary causes. Finally, we outline some guidelines and requirements that any mechanistic proposal should meet in order to be applied to evolutionary theory.",
         "Evolutionary cause | Function | Mechanism | Natural selection | Stochastic",
         "2",
         "0",
         "NULL",
         "NULL",
         "MCIU",
         "FFI2017-89639-P",
         "Ministerio de Ciencia, Innovaci√≥n y Universidades"
        ],
        [
         "1873",
         "1873",
         "2-s2.0-85051763849",
         "10.1007/s40656-018-0196-z",
         "NULL",
         "29691669.0",
         "Thomas Hunt Morgan and the invisible gene: the right tool for the job",
         "ar",
         "Article",
         "Frezza G.",
         "105497271",
         "Unit of History of Medicine",
         "Rome",
         "Italy",
         "2.0",
         "Frezza, Giulia;Capocci, Mauro",
         "57194089802;57218315659",
         "105497271;105497271",
         "2018-06-01 00:00:00",
         "1 June 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "2",
         "31",
         "NULL",
         "The paper analyzes the early theory building process of Thomas Hunt Morgan (1866‚Äì1945) from the 1910s to the 1930s and the introduction of the invisible gene as a main explanatory unit of heredity. Morgan‚Äôs work marks the transition between two different styles of thought. In the early 1900s, he shifted from an embryological study of the development of the organism to a study of the mechanism of genetic inheritance and gene action. According to his contemporaries as well as to historiography, Morgan separated genetics from embryology, and the gene from the whole organism. Other scholars identified an underlying embryological focus in Morgan‚Äôs work throughout his career. Our paper aims to clarify the debate by concentrating on Morgan‚Äôs theory building‚Äîcharacterized by his confidence in the power of experimental methods, and carefully avoiding any ontological commitment towards the gene‚Äîand on the continuity of the questions to be addressed by both embryology and genetics.",
         "Embryology | Gene | Genetics | T. H. Morgan",
         "1",
         "0",
         "NULL",
         "NULL",
         "MIUR",
         "2010-RBFR10Q67A_002",
         "Ministero dell‚ÄôIstruzione, dell‚ÄôUniversit√† e della Ricerca"
        ],
        [
         "1894",
         "1894",
         "2-s2.0-85031723047",
         "10.1007/s40656-017-0160-3",
         "NULL",
         "29038982.0",
         "Hierarchy, determinism, and specificity in theories of development and evolution",
         "ar",
         "Article",
         "Deichmann U.",
         "60027161",
         "Ben-Gurion University of the Negev",
         "Beer-Sheva",
         "Israel",
         "1.0",
         "Deichmann, Ute",
         "57203390987",
         "60027161",
         "2017-12-01 00:00:00",
         "1 December 2017",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "39",
         "4",
         "33",
         "NULL",
         "The concepts of hierarchical organization, genetic determinism and biological specificity (for example of species, biologically relevant macromolecules, or genes) have played a crucial role in biology as a modern experimental science since its beginnings in the nineteenth century. The idea of genetic information (specificity) and genetic determination was at the basis of molecular biology that developed in the 1940s with macromolecules, viruses and prokaryotes as major objects of research often labelled ‚Äúreductionist‚Äù. However, the concepts have been marginalized or rejected in some of the research that in the late 1960s began to focus additionally on the molecularization of complex biological structures and functions using systems approaches. This paper challenges the view that ‚Äòmolecular reductionism‚Äô has been successfully replaced by holism and a focus on the collective behaviour of cellular entities. It argues instead that there are more fertile replacements for molecular ‚Äòreductionism‚Äô, in which genomics, embryology, biochemistry, and computer science intertwine and result in research that is as exact and causally predictive as earlier molecular biology.",
         "Big-data genomics | DST | Holism | Mechanistic systems biology | Molecular reductionism | Regulatory Genome",
         "9",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "National Human Genome Research Institute"
        ],
        [
         "1906",
         "1906",
         "2-s2.0-85000502026",
         "10.1007/s40656-016-0125-y",
         "NULL",
         "27910006.0",
         "William D. Hamilton‚Äôs Brazilian lectures and his unpublished model regarding Wynne-Edwards‚Äôs idea of natural selection. With a note on ‚Äòpluralism‚Äô and different philosophical approaches to evolution",
         "ar",
         "Article",
         "Coco E.",
         "60010146",
         "Universit√† degli Studi di Catania",
         "Catania",
         "Italy",
         "1.0",
         "Coco, Emanuele",
         "35108894200",
         "60010146",
         "2016-12-01 00:00:00",
         "1 December 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "4",
         "24",
         "NULL",
         "In 1975, the English evolutionist William Donald Hamilton (1936‚Äì2000) held in Brazil a series of lectures entitled ‚ÄúPopulation genetics and social behaviour‚Äù. The unpublished notes of these conferences‚Äîwritten by Hamilton and recently discovered at the British Library‚Äîoffer an opportunity to reflect on some of the author‚Äôs ideas about evolution. The year of the conference is particularly significant, as it took place shortly after the applications of the Price equation with which Hamilton was able to build a model that included several levels of selection. In this paper I mainly analyse the inaugural lecture in which Hamilton proposes a simple model to disprove the hypothesis supported by the British zoologist C. Vero Wynne-Edwards (1906‚Äì1997) regarding mechanisms to prevent ‚Äúover-exploitation of the food supply‚Äù in ‚Äúthe interests of the survival of the group‚Äù. The document presented here is of great historical interest. Not only because manuscript offers a model that‚Äîsince it was intended for teaching purposes‚Äîhad never before appeared in the published version, but also because of the general index of the lectures that accompanies it. The latter allows us to make some hypothetical considerations on the relationship and differences between kin-selection, group-selection and inclusive fitness that Hamilton wanted to present to the attentive, well-prepared audience of the foreign university that had invited him.",
         "Altruism | Philosophy | Pluralism | Resources | V.C. Wynne-Edwards | W.D. Hamilton",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "Leverhulme Trust"
        ],
        [
         "1907",
         "1907",
         "2-s2.0-84976291214",
         "10.1007/s40656-016-0102-5",
         "NULL",
         "27338570.0",
         "A causal dispositional account of fitness",
         "ar",
         "Article",
         "Trivi√±o V.",
         "60027856;60000130",
         "Universidad del Pais Vasco;Universidad de Murcia",
         "Leioa;Murcia",
         "Spain;Spain",
         "2.0",
         "Trivi√±o, Vanessa;Nu√±o de la Rosa, Laura",
         "57222621072;38460951700",
         "60000130;60027856",
         "2016-09-01 00:00:00",
         "1 September 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "3",
         "6",
         "NULL",
         "The notion of fitness is usually equated to reproductive success. However, this actualist approach presents some difficulties, mainly the explanatory circularity problem, which have lead philosophers of biology to offer alternative definitions in which fitness and reproductive success are distinguished. In this paper, we argue that none of these alternatives is satisfactory and, inspired by Mumford and Anjum‚Äôs dispositional theory of causation, we offer a definition of fitness as a causal dispositional property. We argue that, under this framework, the distinctiveness that biologists usually attribute to fitness‚Äînamely, the fact that fitness is something different from both the physical traits of an organism and the number of offspring it leaves‚Äîcan be explained, and the main problems associated with the concept of fitness can be solved. Firstly, we introduce Mumford and Anjum's dispositional theory of causation and present our definition of fitness as a causal disposition. We explain in detail each of the elements involved in our definition, namely: the relationship between fitness and the functional dispositions that compose it, the emergent character of fitness, and the context-sensitivity of fitness. Finally, we explain how fitness and realized fitness, as well as expected and realized fitness are distinguished in our approach to fitness as a causal disposition.",
         "Dispositions | Emergence | Expected fitness | Fitness | Realized fitness",
         "13",
         "0",
         "NULL",
         "NULL",
         "KLI",
         "undefined",
         "Fundaci√≥n S√©neca"
        ],
        [
         "1909",
         "1909",
         "2-s2.0-84923797329",
         "10.1007/s40656-014-0031-0",
         "NULL",
         "26013200.0",
         "Science, evolution and natural selection: In praise of Darwin at the Stazione Zoologica Anton Dohrn of Naples",
         "ar",
         "Article",
         "Ayala F.J.",
         "60007278",
         "University of California, Irvine",
         "Irvine",
         "United States",
         "1.0",
         "Ayala, Francisco J.",
         "57205383975",
         "60007278",
         "2015-01-01 00:00:00",
         "1 January 2015",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "NULL",
         "Journal",
         "36",
         "3",
         "NULL",
         "444-455",
         "Copernicus, Galileo, Newton and other physical scientists ushered in a conception of the universe as matter in motion governed by natural laws. Their discoveries brought about a fundamental revolution, namely a commitment to the postulate that the universe obeys immanent laws that can account for natural phenomena. The workings of the universe were brought into the realm of science: explanation through natural laws. Darwin completed the Copernican revolution by extending it to the living world. Darwin demonstrated the evolution of organisms. More important yet is that he discovered natural selection, the process that explains the 'design' of organisms. The adaptations and diversity of organisms, the origin of novel and complex species, even the origin of mankind, could now be explained by an orderly process of change governed by natural laws. The origin of species and the exquisite features of organisms had previously been explained as special creations of an omniscient God. Darwin brought them into the domain of science.",
         "Adaptation | Christiane groeben | Fossil record | Molecular evolution | Natural theology | Scientific revolution",
         "0",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "1910",
         "1910",
         "2-s2.0-84923817248",
         "10.1007/s40656-014-0005-2",
         "NULL",
         "NULL",
         "Mapping an expanding territory: Computer simulations in evolutionary biology",
         "ar",
         "Article",
         "Huneman P.",
         "60021260",
         "Universit√© Paris 1 Panth√©on-Sorbonne",
         "Paris",
         "France",
         "1.0",
         "Huneman, Philippe",
         "15048090500",
         "60021260",
         "2014-08-01 00:00:00",
         "1 August 2014",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "36",
         "1",
         "NULL",
         "60-89",
         "The pervasive use of computer simulations in the sciences brings novel epistemological issues discussed in the philosophy of science literature since about a decade. Evolutionary biology strongly relies on such simulations, and in relation to it there exists a research program (Artificial Life) that mainly studies simulations themselves. This paper addresses the specificity of computer simulations in evolutionary biology, in the context (described in Sect. 1) of a set of questions about their scope as explanations, the nature of validation processes and the relation between simulations and true experiments or mathematical models. After making distinctions, especially between a weak use where simulations test hypotheses about the world, and a strong use where they allow one to explore sets of evolutionary dynamics not necessarily extant in our world, I argue in Sect. 2 that (weak) simulations are likely to represent in virtue of the fact that they instantiate specific features of causal processes that may be isomorphic to features of some causal processes in the world, though the latter are always intertwined with a myriad of different processes and hence unlikely to be directly manipulated and studied. I therefore argue that these simulations are merely able to provide candidate explanations for real patterns. Section 3 ends up by placing strong and weak simulations in Levins' triangle, that conceives of simulations as devices trying to fulfil one or two among three incompatible epistemic values (precision, realism, genericity).",
         "A-Life | Computer simulations | Evolutionary biology | Explanations | Hypothesis testing | Models",
         "7",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "3471",
         "3471",
         "2-s2.0-85080056193",
         "10.1007/s40656-020-00305-2",
         "NULL",
         "32103386.0",
         "The stability of traits conception of the hologenome: An evolutionary account of holobiont individuality",
         "ar",
         "Article",
         "Su√°rez J.",
         "60026479;60001576",
         "University of Exeter;Universitat de Barcelona",
         "Exeter;Barcelona",
         "United Kingdom;Spain",
         "1.0",
         "Su√°rez, Javier",
         "57201743605",
         "60001576-60026479",
         "2020-03-01 00:00:00",
         "1 March 2020",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "42",
         "1",
         "11",
         "NULL",
         "Bourrat and Griffiths (Hist Philos Life Sci 40(2):33, 2018) have recently argued that most of the evidence presented by holobiont defenders to support the thesis that holobionts are evolutionary individuals is not to the point and is not even adequate to discriminate multispecies evolutionary individuals from other multispecies assemblages that would not be considered evolutionary individuals by most holobiont defenders. They further argue that an adequate criterion to distinguish the two categories is fitness alignment, presenting the notion of fitness boundedness as a criterion that allows divorcing true multispecies evolutionary individuals from other multispecies assemblages and provides an adequate criterion to single out genuine evolutionary multispecies assemblages. A consequence of their criterion is that holobionts, as conventionally defined by hologenome defenders, are not evolutionary individuals except in very rare cases, and for very specific host-symbiont associations. This paper is a critical response to Bourrat and Griffiths‚Äô arguments and a defence of the arguments presented by holobiont defenders. Drawing upon the case of the hologenomic basis of the evolution of sanguivory in vampire bats (Nat Ecol Evol 2:659‚Äì668, 2018), I argue that Bourrat and Griffiths overlook some aspects of the biological nature of the microbiome that justifies the thesis that holobionts are evolutionarily different to other multispecies assemblages. I argue that the hologenome theory of evolution should not define the hologenome as a collection of genomes, but as the sum of the host genome plus some traits of the microbiome which together constitute an evolutionary individual, a conception I refer to as the stability of traits conception of the hologenome. Based on that conception I argue that the evidence presented by holobiont defenders is to the point, and supports the thesis that holobionts are evolutionary individuals. In this sense, the paper offers an account of the holobiont that aims to foster a dialogue between hologenome advocates and hologenome critics.",
         "Biological individuality | Evolutionary individuality | Fitness alignment | Holobiont | Hologenome | Microbiota | Stability of traits | Symbiosis",
         "26",
         "0",
         "repositoryam",
         "Green",
         "MECD",
         "FFU16/02570",
         "University of Exeter"
        ],
        [
         "3475",
         "3475",
         "2-s2.0-84929335949",
         "10.1007/s40656-015-0067-9",
         "NULL",
         "26013643.0",
         "Levels of selection in Darwin's Origin of Species",
         "ar",
         "Article",
         "Chancellor G.",
         "NULL",
         "NULL",
         "NULL",
         "NULL",
         "1.0",
         "Chancellor, Gordon",
         "55263314800",
         "NULL",
         "2015-06-01 00:00:00",
         "1 June 2015",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "37",
         "2",
         "NULL",
         "131-157",
         "Abstract References in Darwin's Origin of Species to competition between units of selection at and above the level of individual organisms are enumerated. In many cases these references clearly speak of natural selection and do not support the view that Darwin thought selection only occurred at the level of the individual organism. Darwin did see organismal selection as the main process by which varieties were created but he also espoused what is here termed community and varietal selection. He saw no essential difference between varieties and species and the references show that he also believed that selection could operate at the species level.",
         "Darwin | Divergence | Evolution | Levels | Selection | Species",
         "2",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "3585",
         "3585",
         "2-s2.0-84924007092",
         "10.1007/s40656-014-0044-8",
         "NULL",
         "25515146.0",
         "Is there such a thing as \"group selection\" in the contextual analysis framework?",
         "ar",
         "Article",
         "Jeler C.",
         "60014596",
         "Universitatea Alexandru Ioan Cuza",
         "Iasi",
         "Romania",
         "1.0",
         "Jeler, Ciprian",
         "55247416100",
         "60014596",
         "2015-02-01 00:00:00",
         "1 February 2015",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "36",
         "4",
         "NULL",
         "484-502",
         "This paper argues that the contextual approach to natural selection does not offer an estimation of the contributions of individual and group selection to evolutionary change in multi-level selection scenarios, and that this is so because the term \"group selection\", as defined by the contextual approach, does not refer to a process taking place at the group level. In the contextual analysis framework, this term simply denotes an evolutionary change that takes place due to the fact that, overall, individual types do not share similar contexts or environments, and the only way to claim that such an evolutionary change is a result of selection is by admitting that \"group selection\" is in fact a kind of frequency-dependent selection, i.e. a selection process taking place at the individual level. Therefore, under the names \"individual selection\" and \"group selection\", the contextual approach actually isolates two aspects of the relation between individual types and their environment, and not two distinct levels of selection.",
         "Contextual analysis | Frequency-dependent selection | Group selection | Price approach",
         "9",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "3605",
         "3605",
         "2-s2.0-85149323116",
         "10.1007/s40656-023-00566-7",
         "NULL",
         "36862350.0",
         "Evolution within the body: the rise and fall of somatic Darwinism in the late nineteenth century",
         "ar",
         "Article",
         "Swiatczak B.",
         "60019118",
         "University of Science and Technology of China",
         "Hefei",
         "China",
         "1.0",
         "Swiatczak, Bartlomiej",
         "36505242700",
         "60019118",
         "2023-03-01 00:00:00",
         "March 2023",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "45",
         "1",
         "8",
         "NULL",
         "Originating in the work of Ernst Haeckel and Wilhelm Preyer, and advanced by a Prussian embryologist, Wilhelm Roux, the idea of struggle for existence between body parts helped to establish a framework, in which population cell dynamics rather than a predefined harmony guides adaptive changes in an organism. Intended to provide a causal-mechanical view of functional adjustments in body parts, this framework was also embraced later by early pioneers of immunology to address the question of vaccine effectiveness and pathogen resistance. As an extension of these early efforts, Elie Metchnikoff established an evolutionary vision of immunity, development, pathology, and senescence, in which phagocyte-driven selection and struggle promote adaptive changes in an organism. Despite its promising start, the idea of somatic evolution lost its appeal at the turn of the twentieth century giving way to a vision, in which an organism operates as a genetically uniform, harmonious entity.",
         "Clonal selection | Somatic Darwinism | Somatic evolution | Somatic selection | Wilhelm Roux",
         "0",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "3643",
         "3643",
         "2-s2.0-85102350902",
         "10.1007/s40656-021-00390-x",
         "NULL",
         "33694075.0",
         "Ageing and the goal of evolution",
         "ar",
         "Article",
         "Garson J.",
         "60032373",
         "The Graduate Center",
         "New York",
         "United States",
         "1.0",
         "Garson, Justin",
         "7006765291",
         "60032373",
         "2021-03-01 00:00:00",
         "March 2021",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "43",
         "1",
         "37",
         "NULL",
         "There is a certain metaphor that has enjoyed tremendous longevity in the evolution of ageing literature. According to this metaphor, nature has a certain goal or purpose, the perpetuation of the species, or, alternatively, the reproductive success of the individual. In relation to this goal, the individual organism has a function, job, or task, namely, to breed and, in some species, to raise its brood to maturity. On this picture, those who cannot, or can no longer, reproduce are somehow invisible to, or even dispensable to, the evolutionary process. Here, I argue that the metaphor should be discarded, not on the grounds that it is a metaphor, but on the grounds that this particular metaphor distorts our understanding of the evolution of ageing. One reason the metaphor is problematic is that it frames senescence and death as nature‚Äôs verdict on the value of older individuals. Instead, we should explore a different metaphor: the lengthy post-reproductive period in humans and some other animals is not an accident of culture, but designed by nature for the purpose of supporting and guiding younger generations. On this alternate picture, different stages of life have their own evolutionary rationales, their distinctive design features, their special mandates.",
         "Ageing | Biological function | Evolution of ageing | Goal-directedness | Teleology",
         "8",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "4109",
         "4109",
         "2-s2.0-85036509978",
         "10.1007/s40656-017-0173-y",
         "NULL",
         "29181739.0",
         "Stochasticity in cultural evolution: a revolution yet to¬†happen",
         "ar",
         "Article",
         "Billiard S.",
         "60175816;60104665",
         "Oxford Social Sciences Division;Universit√© de Lille",
         "Oxford;Lille",
         "United Kingdom;France",
         "2.0",
         "Billiard, Sylvain;Alvergne, Alexandra",
         "16314730300;15019013500",
         "60104665-60175816;60175816",
         "2018-03-01 00:00:00",
         "1 March 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "1",
         "9",
         "NULL",
         "Over the last 40¬†years¬†or so, there has been an explosion of cultural evolution research in anthropology and archaeology. In each discipline, cultural evolutionists investigate how interactions between individuals translate into group level patterns, with the aim of explaining the diachronic dynamics and diversity of cultural traits. However, while much attention has been given to deterministic processes (e.g. cultural transmission biases), we contend that current evolutionary accounts of cultural change are limited because they do not adopt a systematic stochastic approach (i.e. accounting for the role of chance). First, we show that, in contrast with the intense debates in ecology and population genetics, the importance of stochasticity in evolutionary processes has generated little discussion in the sciences of cultural evolution to date. Second, we speculate on the reasons, both ideological and methodological, why that should be so. Third, we highlight the inadequacy of genetically-inspired stochastic models in the context of cultural evolution modelling, and ask which fundamental stochastic processes might be more relevant to take up. We conclude that the field of cultural evolution would benefit from a stochastic revolution. For that to occur, stochastic models ought to be developed specifically for cultural data and not through a copy-pasting of neutral models from population genetics or ecology.",
         "Anthropology | Archaeology | Cultural evolution | Population ecology | Population genetics | Stochastic processes",
         "6",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "University of Oxford"
        ],
        [
         "4184",
         "4184",
         "2-s2.0-85036589355",
         "10.1007/s40656-017-0174-x",
         "NULL",
         "29168053.0",
         "The sociobiology of genes: the gene‚Äôs eye view as a unifying behavioural-ecological framework for biological evolution",
         "ar",
         "Article",
         "De Ti√®ge A.",
         "60033316",
         "Universiteit Gent",
         "Ghent",
         "Belgium",
         "4.0",
         "De Ti√®ge, Alexis;Van de Peer, Yves;Braeckman, Johan;Tanghe, Koen B.",
         "56012843400;7006594796;19233597900;55545970400",
         "60033316;60033316;60033316;60033316",
         "2018-03-01 00:00:00",
         "1 March 2018",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "40",
         "1",
         "6",
         "NULL",
         "Although classical evolutionary theory, i.e., population genetics and the Modern Synthesis, was already implicitly ‚Äògene-centred‚Äô, the organism was, in practice, still generally regarded as the individual unit of which a population is composed. The gene-centred approach to evolution only reached a logical conclusion with the advent of the gene-selectionist or gene‚Äôs eye view in the 1960s and 1970s. Whereas classical evolutionary theory can only work with (genotypically represented) fitness differences between individual organisms, gene-selectionism is capable of working with fitness differences among genes within the same organism and genome. Here, we explore the explanatory potential of ‚Äòintra-organismic‚Äô and ‚Äòintra-genomic‚Äô gene-selectionism, i.e., of a behavioural-ecological ‚Äògene‚Äôs eye view‚Äô on genetic, genomic and organismal evolution. First, we give a general outline of the framework and how it complements the‚Äîto some extent‚Äîstill ‚Äòorganism-centred‚Äô approach of classical evolutionary theory. Secondly, we give a more in-depth assessment of its explanatory potential for biological evolution, i.e., for Darwin‚Äôs ‚Äòcommon descent with modification‚Äô or, more specifically, for ‚Äòhistorical continuity or homology with modular evolutionary change‚Äô as it has been studied by evolutionary developmental biology (evo-devo) during the last few decades. In contrast with classical evolutionary theory, evo-devo focuses on ‚Äòwithin-organism‚Äô developmental processes. Given the capacity of gene-selectionism to adopt an intra-organismal gene‚Äôs eye view, we outline the relevance of the latter model for evo-devo. Overall, we aim for the conceptual integration between the gene‚Äôs eye view on the one hand, and more organism-centred evolutionary models (both classical evolutionary theory and evo-devo) on the other.",
         "Behavioural-ecological modelling of gene and genome evolution | Classical evolutionary theory | Evo-devo | Evolutionarily stable strategy (ESS) | Hierarchical organization | Intra-organismic and intra-genomic conflict | Organism‚Äôs versus gene‚Äôs eye view",
         "3",
         "0",
         "NULL",
         "NULL",
         "FNRS",
         "G001013N",
         "Fonds De La Recherche Scientifique - FNRS"
        ],
        [
         "4292",
         "4292",
         "2-s2.0-85113290982",
         "10.1007/s40656-021-00455-x",
         "NULL",
         "34427800.0",
         "Reevaluating the grandmother hypothesis",
         "ar",
         "Article",
         "Watkins A.",
         "60019674",
         "Boston University",
         "Boston",
         "United States",
         "1.0",
         "Watkins, Aja",
         "57212504118",
         "60019674",
         "2021-09-01 00:00:00",
         "September 2021",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "43",
         "3",
         "103",
         "NULL",
         "Menopause is an evolutionary mystery: how could living longer with no capacity to reproduce possibly be advantageous? Several explanations have been offered for why female humans, unlike our closest primate relatives, have such an extensive post-reproductive lifespan. Proponents of the so-called ‚Äúgrandmother hypothesis‚Äù suggest that older women are able to increase their fitness by helping to care for their grandchildren as allomothers. This paper first distinguishes the grandmother hypothesis from several other hypotheses that attempt to explain menopause, and then develops a formal model by which these hypotheses can be compared and tested by empirical researchers. The model is then modified and used to respond to a common objection to the grandmother hypothesis: that human fathers, rather than grandmothers, are better suited to be allomothers due to their physical strength and a high incentive to invest in their own children. However, fathers‚Äîunlike maternal grandmothers‚Äîcan never be sure that the children they are caring for are their own. Incorporating paternity uncertainty into the model demonstrates the conditions under which the grandmother hypothesis is more plausible than a hypothesis that focuses on the contributions of men.",
         "Grandmother hypothesis | Human evolution | Menopause | Paternity uncertainty",
         "3",
         "0",
         "NULL",
         "NULL",
         "NSF",
         "DGE- 1840990",
         "National Science Foundation"
        ],
        [
         "4390",
         "4390",
         "2-s2.0-85075378683",
         "10.1007/s40656-019-0294-6",
         "NULL",
         "31749015.0",
         "Drift as constitutive: conclusions from a formal reconstruction of population genetics",
         "ar",
         "Article",
         "Roff√© A.J.",
         "60001563",
         "Universidad de Buenos Aires",
         "Buenos Aires",
         "Argentina",
         "1.0",
         "Roff√©, Ariel Jonathan",
         "57194473784",
         "60001563",
         "2019-12-01 00:00:00",
         "1 December 2019",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "41",
         "4",
         "55",
         "NULL",
         "This article elaborates on McShea and Brandon‚Äôs idea that drift is unlike the rest of the evolutionary factors because it is constitutive rather than imposed on the evolutionary process. I show that the way they spelled out this idea renders it inadequate and is the reason why it received some (good) objections. I propose a different way in which their point could be understood, that rests on two general distinctions. The first is a distinction between the underlying mathematical apparatus used to formulate a theory and a concept proposed by that theory. With the aid of a formal reconstruction of a population genetic model, I show that drift belongs to the first category. That is, that drift is constitutive of population genetics in the same sense that multiplication is constitutive in classical mechanics, or that circle is constitutive in Ptolemaic astronomy. The second distinction is between eliminating a concept from a theory and setting its value to zero. I will show that even though drift can be set to zero just like the rest of the evolutionary factors (as others have noted in their criticism of McShea and Brandon), eliminating drift is much harder than eliminating those other factors, since it would require changing the entire mathematical apparatus of standard population genetic theory. I conclude by drawing some other implications from the proposed formal reconstruction.",
         "Analogy | Classical mechanics | Constitutivity | Drift | Formal reconstruction | Population genetics",
         "1",
         "0",
         "NULL",
         "NULL",
         "UBA",
         "UBACyT 20020170200106BA",
         "Universidad de Buenos Aires"
        ],
        [
         "4501",
         "4501",
         "2-s2.0-84978142039",
         "10.1007/s40656-016-0109-y",
         "NULL",
         "27385113.0",
         "Missing concepts in natural selection theory reconstructions",
         "ar",
         "Article",
         "Ginnobili S.",
         "60001563;117003953",
         "Universidad de Buenos Aires;Instituto de Estudios sobre la Ciencia y la Tecnolog√≠a",
         "Buenos Aires;Bernal",
         "Argentina;Argentina",
         "1.0",
         "Ginnobili, Santiago",
         "36607891300",
         "60001563-117003953",
         "2016-09-01 00:00:00",
         "1 September 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "3",
         "8",
         "NULL",
         "The concept of fitness has generated a lot of discussion in philosophy of biology. There is, however, relative agreement about the need to distinguish at least two uses of the term: ecological fitness on the one hand, and population genetics fitness on the other. The goal of this paper is to give an explication of the concept of ecological fitness by providing a reconstruction of the theory of natural selection in which this concept was framed, that is, based on the way the theory was put to use in Darwin‚Äôs main texts. I will contend that this reconstruction enables us to account for the current use of the theory of natural selection. The framework presupposed in the analysis will be that of metatheoretical structuralism. This framework will provide both a better understanding of the nature of ecological fitness and a more complete reconstruction of the theory. In particular, it will provide what I think is a better way of understanding how the concept of fitness is applied through heterogeneous cases. One of the major advantages of my way of thinking about natural selection theory is that it would not have the peculiar metatheoretical status that it has in other available views. I will argue that in order to achieve these goals it is necessary to make several concepts explicit, concepts that are frequently omitted in usual reconstructions.",
         "Darwin | Ecological fitness | Metatheoretical structuralism | Natural selection theory | Population genetics",
         "17",
         "0",
         "NULL",
         "NULL",
         "CONICET",
         "32/15 255",
         "Consejo Nacional de Investigaciones Cient√≠ficas y T√©cnicas"
        ],
        [
         "4599",
         "4599",
         "2-s2.0-85141094301",
         "10.1007/s40656-022-00539-2",
         "NULL",
         "36319715.0",
         "Technics and signs: anthropogenesis in Vygotsky, Leroi-Gourhan, and Stiegler",
         "ar",
         "Article",
         "Drain C.",
         "60010756",
         "Dartmouth College",
         "Hanover",
         "United States",
         "1.0",
         "Drain, Chris",
         "57191155124",
         "60010756",
         "2022-12-01 00:00:00",
         "December 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "4",
         "53",
         "NULL",
         "This paper reconstructs L.S. Vygotsky‚Äôs account of anthropogenesis with respect to the work of anthropologist Andr√© Leroi-Gourhan and late philosopher Bernard Stiegler, situating Vygotsky as a forerunner to recent theories that posit cultural scaffolding and niche construction as the main drivers of human cognitive evolution. One might think there is an immediate affinity between Vygotsky and the techno-centric accounts of Leroi-Gourhan and Stiegler. Following Leroi-Gourhan, Stiegler argues that ‚Äútechnics‚Äù is the main driver in the anthropogenic development of ‚Äúreflective consciousness.‚Äù Vygotsky likewise claims that ‚Äúpsychological tools‚Äù are responsible for the development of uniquely human forms of consciousness. However, closer inspection reveals deep disparities between Vygotsky and the French thinkers. In Stiegler‚Äôs philosophical redeployment of Leroi-Gourhan‚Äôs anthropology, ‚Äúreflective‚Äù cognition is the product of a prehistorical rupture in which some threshold of technical-cortical complexification is breached. For Vygotsky, on the other hand, the inverse scenario obtains. Technical development initially proceeds in tandem with the complexification of biologically based signaling behavior until the introduction of signs, which then radically restructure the cognitive apparatus. Due to inconsistencies regarding the equivalency of the technical and semiotic in Stiegler and Leroi-Gourhan, I advance a Vygotskian account where anthropogenesis is the result of semiotic rather than technical intervention. This aims to establish Vygotsky‚Äôs ‚ÄúCultural Historical‚Äù approach, and the Marxian-dialectical tradition from which he draws, as not only presaging recent naturalistic accounts of development, but offering a relevant theoretical program that may continue to inspire contemporary enculturated accounts of anthropogenesis.",
         "Andr√© Leroi-Gourhan | Anthropogenesis | Bernard Stiegler | Enculturated cognition | L.S. Vygotsky | Niche construction",
         "1",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "4700",
         "4700",
         "2-s2.0-85027417366",
         "10.1007/s40656-017-0152-3",
         "NULL",
         "28808914.0",
         "From Pleistocene to Holocene: the prehistory of southwest Asia in evolutionary context",
         "ar",
         "Article",
         "Watkins T.",
         "60110825;60027272",
         "Edinburgh Medical School;The University of Edinburgh",
         "Edinburgh;Edinburgh",
         "United Kingdom;United Kingdom",
         "1.0",
         "Watkins, Trevor",
         "35964345000",
         "60027272-60110825",
         "2017-09-01 00:00:00",
         "1 September 2017",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "39",
         "3",
         "22",
         "NULL",
         "In this paper I seek to show how cultural niche construction theory offers the potential to extend the human evolutionary story beyond the Pleistocene, through the Neolithic, towards the kind of very large-scale societies in which we live today. The study of the human past has been compartmentalised, each compartment using different analytical vocabularies, so that their accounts are written in mutually incompatible languages. In recent years social, cognitive and cultural evolutionary theories, building on a growing body of archaeological evidence, have made substantial sense of the social and cultural evolution of the genus Homo. However, specialists in this field of studies have found it difficult to extend their kind of analysis into the Holocene human world. Within southwest Asia the three or four millennia of the Neolithic period at the beginning of the Holocene represents a pivotal point, which saw the transformation of human society in the emergence of the first large-scale, permanent communities, the domestication of plants and animals, and the establishment of effective farming economies. Following the Neolithic, the pace of human social, economic and cultural evolution continued to increase. By 5000¬†years ago, in parts of southwest Asia and northeast Africa there were very large-scale urban societies, and the first large-scale states (kingdoms). An extension of cultural niche construction theory enables us to extend the evolutionary narrative of the Pleistocene into the Holocene, opening the way to developing a single, long-term, evolutionary account of human history.",
         "Cognitive-cultural co-evolution | Cultural niche construction theory | Epipalaeolithic | Neolithic | Southwest Asia",
         "15",
         "1",
         "repositoryvor",
         "Green",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "4707",
         "4707",
         "2-s2.0-84964047820",
         "10.1007/s40656-015-0090-x",
         "NULL",
         "26463099.0",
         "The Darwinian muddle on the division of labour: an attempt at clarification",
         "ar",
         "Article",
         "D‚ÄôHombres E.",
         "60198731;60031868",
         "Sciences, Philosophie, Histoire ‚Äì UMR 7219, Laboratoire SPHERE;Universit√© Catholique de Lyon",
         "Paris;Lyon",
         "France;France",
         "1.0",
         "D‚ÄôHombres, Emmanuel",
         "16678621800",
         "60198731-60031868",
         "2016-04-01 00:00:00",
         "1 April 2016",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "38",
         "1",
         "NULL",
         "1-22",
         "It is of philosophical and epistemological interest to examine how Darwin conceived the process of division of labour within Natural History. Darwin observed the advantages brought by division of labour to the human economy, and considered that the principle of divergence within nature, which is, according to him, one of the two ‚Äòkeystones‚Äô of his theory, gave comparable advantages. This led him to re-examine Milne-Edwards‚Äô view on the notion of division of physiological labour, and to introduce this with modifications into his naturalist writings. After a short review of the Darwinian historiography dealing with this issue, I first show the conceptual confusion into which Darwin plunges, when using a so-called economic argument to defend his thesis of the maximization of beings in a given territory due to division of labour. Following this I propose several hypotheses to explain these shifts, recurring in Darwin‚Äôs texts, from one conception and from one application to another, of the division of labour.",
         "Charles Darwin | Competition | Division of physiological labour | Henri Milne-Edwards | Principle of divergence | Productivity",
         "4",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "4807",
         "4807",
         "2-s2.0-85141590104",
         "10.1007/s40656-022-00546-3",
         "NULL",
         "36342559.0",
         "Humboldt, Darwin, and theory of evolution",
         "ar",
         "Article",
         "Stamenkoviƒá B.",
         "60068815",
         "University of Belgrade",
         "Belgrade",
         "Serbia",
         "1.0",
         "Stamenkoviƒá, Bogdana",
         "57217033782",
         "60068815",
         "2022-12-01 00:00:00",
         "December 2022",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "44",
         "4",
         "57",
         "NULL",
         "Numerous authors have examined the influence of other thinkers on Darwin‚Äôs formulation of some of the key concepts of the theory of evolution. Amongst those, Alexander von Humboldt often stands out ‚Äì a scholar who, following his intention to explain the interconnection of various parts of the natural system, seems to tackle the question of evolution but does not offer an explicit answer. In this article, I examine Humboldt‚Äôs thoughts on evolution and the origin of species and evaluate his contribution to Darwin‚Äôs theory of evolution. First, I analyse and explicate the fundamental assumptions and goals of Humboldt theory, and compare them to Darwin. Moving forward, I highlight the similarity of their methods, and argue that Humboldt and Darwin conduct similar investigation of fossil record and geographical distribution of species. Finally, I show that Humboldt acknowledges essential elements of Darwin‚Äôs theory of evolution: evidence given by fossil records, struggle for survival and relation between natural environment and living organisms. Humboldt, however, concludes we cannot know the evolution of species. I explain this stance, and contend that theories of Humboldt and Darwin turn out to be more similar than they seem, yet their different conclusions regarding the evolution of species stem from different initial assumptions underlying their respective frameworks.",
         "Darwin | Humboldt | Natural selection | Struggle for survival | Theory of evolution",
         "0",
         "0",
         "NULL",
         "NULL",
         "MPNTR",
         "undefined",
         "Ministarstvo Prosvete, Nauke i Tehnolo≈°kog Razvoja"
        ],
        [
         "4817",
         "4817",
         "2-s2.0-85029525457",
         "10.1007/s40656-017-0154-1",
         "NULL",
         "28900879.0",
         "Darwinism in metaethics: What if the universal acid cannot be contained?",
         "ar",
         "Article",
         "Severini E.",
         "60032350",
         "Sapienza Universit√† di Roma",
         "Rome",
         "Italy",
         "2.0",
         "Severini, Eleonora;Sterpetti, Fabio",
         "57188872185;56667730200",
         "60032350;60032350",
         "2017-09-01 00:00:00",
         "1 September 2017",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "39",
         "3",
         "27",
         "NULL",
         "The aim of this article is to explore the impact of Darwinism in metaethics and dispel some of the confusion surrounding it. While the prospects for a Darwinian metaethics appear to be improving, some underlying epistemological issues remain unclear. We will focus on the so-called Evolutionary Debunking Arguments (EDAs) which, when applied in metaethics, are defined as arguments that appeal to the evolutionary origins of moral beliefs so as to undermine their epistemic justification. The point is that an epistemic disanalogy can be identified in the debate on EDAs between moral beliefs and other kinds of beliefs, insofar as only the former are regarded as vulnerable to EDAs. First, we will analyze some significant debunking positions in metaethics in order to show that they do not provide adequate justification for such an epistemic disanalogy. Then, we will assess whether they can avoid the accusation of being epistemically incoherent by adopting the same evolutionary account for all kinds of beliefs. In other words, once it is argued that Darwinism has a corrosive impact on metaethics, what if its universal acid cannot be contained?",
         "Darwinism | Epistemic justification | Evolutionary debunking arguments | Metaethics | Moral realism | Scepticism",
         "2",
         "0",
         "NULL",
         "NULL",
         "NULL",
         "undefined",
         "NULL"
        ],
        [
         "4868",
         "4868",
         "2-s2.0-85208603422",
         "10.1007/s40656-024-00634-6",
         "NULL",
         "39503816.0",
         "The modern synthesis and ‚ÄúProgress‚Äù in evolution: a view from the journal literature",
         "ar",
         "Article",
         "Pence C.H.",
         "60000874",
         "Universit√© Catholique de Louvain",
         "Louvain-la-Neuve",
         "Belgium",
         "1.0",
         "Pence, Charles H.",
         "47061688800",
         "60000874",
         "2024-12-01 00:00:00",
         "December 2024",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "46",
         "4",
         "39",
         "NULL",
         "The concept of ‚Äúprogress‚Äù in evolutionary theory and its relationship to a putative notion of ‚ÄúProgress‚Äù in a global, normatively loaded sense of ‚Äúchange for the better‚Äù have been the subject of debate since Darwin admonished himself in a marginal note to avoid using the terms ‚Äòhigher‚Äô and ‚Äòlower.‚Äô While an increase in some kind of complexity in the natural world might seem self-evident, efforts to explicate this trend meet notorious philosophical difficulties. Numerous historians pin the Modern Synthesis as a pivotal moment in this history; Michael Ruse even provocatively hypothesizes that Ernst Mayr and other ‚Äúarchitects‚Äù of the Synthesis worked actively to eliminate Progress from evolutionary biology‚Äôs scientific purview. I evaluate these claims here with a textual analysis of the journals Evolution and Proceedings of the Royal Society B (a corpus of 27,762 documents), using a dynamic topic modeling approach to track the fate of the term ‚Äòprogress‚Äô across the Modern Synthesis. The claim that this term declines in importance for evolutionary theorizing over this period can, indeed, be supported; more tentative evidence is also provided that the discussion of ‚Äòprogress‚Äô is largely absent from the British context, emphasizing the role of American paleontology in the rise and fall of ‚Äòprogress‚Äô in 20th-century evolutionary biology.",
         "Digital humanities | Ernst Mayr | Evolution | Modern Synthesis | Progress | Textual analysis",
         "0",
         "0",
         "NULL",
         "NULL",
         "FNRS",
         "RECH-INNO-05",
         "Fonds De La Recherche Scientifique - FNRS"
        ],
        [
         "5117",
         "5117",
         "2-s2.0-84924011007",
         "10.1007/s40656-014-0051-9",
         "NULL",
         "26013314.0",
         "That 70s show: Regulation, evolution and development beyond molecular genetics",
         "ar",
         "Article",
         "Su√°rez-D√≠a E.",
         "112486446",
         "Cd. Univ",
         "Mexico City",
         "Mexico",
         "2.0",
         "Su√°rez-D√≠a, Edna;Garc√≠a-Deister, Vivette",
         "55914869800;37010864800",
         "112486446;112486446",
         "2015-02-01 00:00:00",
         "1 February 2015",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "36",
         "4",
         "NULL",
         "503-524",
         "This paper argues that the \"long 1970s\" (1969-1983) is an important though often overlooked period in the development of a rich landscape in the research of metabolism, development, and evolution. The period is marked by: shrinking public funding of basic science, shifting research agendas in molecular biology, the incorporation of new phenomena and experimental tools from previous biological research at the molecular level, and the development of recombinant DNA techniques. Research was reoriented towards eukaryotic cells and development, and in particular towards \"giant\" RNA processing and transcription. We will here focus on three different models of developmental regulation published in that period: the two models of eukaryotic genetic regulation at the transcriptional level that were developed by Georgii P. Georgiev on the one hand, and by Roy Britten and Eric Davidson on the other; and the model of genetic sufficiency and evolution of regulatory genes proposed by Emile Zuckerkandl. These three cases illustrate the range of exploratory hypotheses that characterised the challenging landscape of gene regulation in the 1970s, a period that in hindsight can be labelled as transitional, between the biology at the laboratory bench of the preceding period, and the biology of genetic engineering and intensive data-driven research that followed.",
         "Development | Hybridisation | Informofers | Regulation | Transcription",
         "6",
         "0",
         "NULL",
         "NULL",
         "CONACYT",
         "IN303111",
         "Consejo Nacional de Ciencia y Tecnolog√≠a"
        ],
        [
         "5145",
         "5145",
         "2-s2.0-85031753830",
         "10.1007/s40656-017-0164-z",
         "NULL",
         "29038942.0",
         "Fitting structure to function in gene regulatory networks",
         "ar",
         "Article",
         "Rothenberg E.V.",
         "60143895",
         "Division of Biology and Biological Engineering",
         "Pasadena",
         "United States",
         "1.0",
         "Rothenberg, Ellen V.",
         "7005964351",
         "60143895",
         "2017-12-01 00:00:00",
         "1 December 2017",
         "History and Philosophy of the Life Sciences",
         "03919714",
         "15166",
         "17426316",
         "Journal",
         "39",
         "4",
         "37",
         "NULL",
         "Cascades of transcriptional regulation are the common source of the forward drive in all developmental systems. Increases in complexity and specificity of gene expression at successive stages are based on the collaboration of varied combinations of transcription factors already expressed in the cells to turn on new genes, and the logical relationships between the transcription factors acting and becoming newly expressed from stage to stage are best visualized as gene regulatory networks. However, gene regulatory networks used in different developmental contexts underlie processes that actually operate through different sets of rules, which affect the kinetics, synchronicity, and logical properties of individual network nodes. Contrasting early embryonic development in flies and sea urchins with adult mammalian hematopoietic development from stem cells, major differences are seen in transcription factor dosage dependence, the silencing or damping impacts of repression, and the impact of cellular regulatory history on the parts of the genome that are accessible to transcription factor action in a given cell type. These different features not only affect the kinds of models that can illuminate developmental mechanisms in the respective biological systems, but also reflect the evolutionary needs of these biological systems to optimize different aspects of development.",
         "Boolean modeling | Chromatin accessibility | Developmental plasticity | Network topology | T lymphocyte development | Transcriptional regulation",
         "3",
         "0",
         "repositoryam",
         "Green",
         "NIAID",
         "R01AI095943",
         "National Institute of Allergy and Infectious Diseases"
        ]
       ],
       "shape": {
        "columns": 37,
        "rows": 240
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>eid</th>\n",
       "      <th>doi</th>\n",
       "      <th>pii</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>title</th>\n",
       "      <th>subtype</th>\n",
       "      <th>subtypeDescription</th>\n",
       "      <th>creator</th>\n",
       "      <th>afid</th>\n",
       "      <th>...</th>\n",
       "      <th>pageRange</th>\n",
       "      <th>description</th>\n",
       "      <th>authkeywords</th>\n",
       "      <th>citedby_count</th>\n",
       "      <th>openaccess</th>\n",
       "      <th>freetoread</th>\n",
       "      <th>freetoreadLabel</th>\n",
       "      <th>fund_acr</th>\n",
       "      <th>fund_no</th>\n",
       "      <th>fund_sponsor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>263</td>\n",
       "      <td>2-s2.0-85013158199</td>\n",
       "      <td>10.1007/s40656-017-0129-2</td>\n",
       "      <td>NULL</td>\n",
       "      <td>28205138.0</td>\n",
       "      <td>Kant‚Äôs epigenesis: specificity and development...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Demarest B.</td>\n",
       "      <td>60002483</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>In this paper, I argue that Kant adopted, thro...</td>\n",
       "      <td>Epigenesis | Generative force | Kant | Preform...</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>369</td>\n",
       "      <td>2-s2.0-85088262863</td>\n",
       "      <td>10.1007/s40656-020-00329-8</td>\n",
       "      <td>NULL</td>\n",
       "      <td>32691291.0</td>\n",
       "      <td>What‚Äôs all the fuss about? The inheritance of ...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Camacho M.P.</td>\n",
       "      <td>60015457</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The Central Dogma of molecular biology, which ...</td>\n",
       "      <td>Difference-making | Epigenetics | Gene-centris...</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390</th>\n",
       "      <td>390</td>\n",
       "      <td>2-s2.0-84995794096</td>\n",
       "      <td>10.1007/s40656-016-0121-2</td>\n",
       "      <td>NULL</td>\n",
       "      <td>27854053.0</td>\n",
       "      <td>Epigenetics: ambiguities and implications</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Stotz K.</td>\n",
       "      <td>60019544;60010710</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Everyone has heard of ‚Äòepigenetics‚Äô, but the t...</td>\n",
       "      <td>Epigenesis | Epigenetic inheritance | Epigenet...</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>TWCF</td>\n",
       "      <td>TWCF0063/AB37</td>\n",
       "      <td>Templeton World Charity Foundation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>622</th>\n",
       "      <td>622</td>\n",
       "      <td>2-s2.0-85180664555</td>\n",
       "      <td>10.1007/s40656-023-00600-8</td>\n",
       "      <td>NULL</td>\n",
       "      <td>38153583.0</td>\n",
       "      <td>Resilience and the shift of paradigm in ecolog...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Barbara L.</td>\n",
       "      <td>60003668</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>In the shift from the balance of nature to the...</td>\n",
       "      <td>Balance of nature | Ecological resilience | Fl...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherfree2read</td>\n",
       "      <td>Bronze</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>682</th>\n",
       "      <td>682</td>\n",
       "      <td>2-s2.0-85063972115</td>\n",
       "      <td>10.1007/s40656-019-0252-3</td>\n",
       "      <td>NULL</td>\n",
       "      <td>30937631.0</td>\n",
       "      <td>Taxonomy and conservation science: interdepend...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Conix S.</td>\n",
       "      <td>60025063</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The relation between conservation science and ...</td>\n",
       "      <td>Conservation science | Species classification ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>3H160214</td>\n",
       "      <td>KU Leuven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2707</th>\n",
       "      <td>2707</td>\n",
       "      <td>2-s2.0-85063972115</td>\n",
       "      <td>10.1007/s40656-019-0252-3</td>\n",
       "      <td>NULL</td>\n",
       "      <td>30937631.0</td>\n",
       "      <td>Taxonomy and conservation science: interdepend...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Conix S.</td>\n",
       "      <td>60025063</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>The relation between conservation science and ...</td>\n",
       "      <td>Conservation science | Species classification ...</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>3H160214</td>\n",
       "      <td>KU Leuven</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711</th>\n",
       "      <td>2711</td>\n",
       "      <td>2-s2.0-85058017987</td>\n",
       "      <td>10.1007/s40656-018-0236-8</td>\n",
       "      <td>NULL</td>\n",
       "      <td>30523424.0</td>\n",
       "      <td>At the intersection of medical geography and d...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Arrizabalaga J.</td>\n",
       "      <td>60028181</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>Environmental historians are not sufficiently ...</td>\n",
       "      <td>Disease ecology | Fernand Braudel | Historical...</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>publisherfree2read</td>\n",
       "      <td>Bronze</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2716</th>\n",
       "      <td>2716</td>\n",
       "      <td>2-s2.0-85046898458</td>\n",
       "      <td>10.1007/s40656-018-0194-1</td>\n",
       "      <td>NULL</td>\n",
       "      <td>29761370.0</td>\n",
       "      <td>Multispecies individuals</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Bourrat P.</td>\n",
       "      <td>60025709;60019544;60010710</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>We assess the arguments for recognising functi...</td>\n",
       "      <td>Evolutionary transitions in individuality | Ho...</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>CEED</td>\n",
       "      <td>DP0878650</td>\n",
       "      <td>Centre of Excellence for Environmental Decisio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2733</th>\n",
       "      <td>2733</td>\n",
       "      <td>2-s2.0-84988353676</td>\n",
       "      <td>10.1007/s40656-016-0113-2</td>\n",
       "      <td>NULL</td>\n",
       "      <td>27645228.0</td>\n",
       "      <td>From ecological records to big data: the inven...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Devictor V.</td>\n",
       "      <td>60108488;60021260</td>\n",
       "      <td>...</td>\n",
       "      <td>NULL</td>\n",
       "      <td>This paper is a critical assessment of the epi...</td>\n",
       "      <td>Big data | Biodiversity | Ecology | Foucault |...</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>NULL</td>\n",
       "      <td>undefined</td>\n",
       "      <td>NULL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2736</th>\n",
       "      <td>2736</td>\n",
       "      <td>2-s2.0-84940554967</td>\n",
       "      <td>10.1007/s40656-015-0072-z</td>\n",
       "      <td>NULL</td>\n",
       "      <td>26141169.0</td>\n",
       "      <td>‚ÄòBig science‚Äô in the field: experimenting with...</td>\n",
       "      <td>ar</td>\n",
       "      <td>Article</td>\n",
       "      <td>Cassidy A.</td>\n",
       "      <td>60011520</td>\n",
       "      <td>...</td>\n",
       "      <td>305-325</td>\n",
       "      <td>Since wild badgers were first connected with o...</td>\n",
       "      <td>Agricultural research | Animal health | Ecolog...</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>repositoryvor</td>\n",
       "      <td>Green</td>\n",
       "      <td>WT</td>\n",
       "      <td>RES-229-27-0007-A</td>\n",
       "      <td>Wellcome Trust</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>240 rows √ó 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0                 eid                         doi   pii  \\\n",
       "263          263  2-s2.0-85013158199   10.1007/s40656-017-0129-2  NULL   \n",
       "369          369  2-s2.0-85088262863  10.1007/s40656-020-00329-8  NULL   \n",
       "390          390  2-s2.0-84995794096   10.1007/s40656-016-0121-2  NULL   \n",
       "622          622  2-s2.0-85180664555  10.1007/s40656-023-00600-8  NULL   \n",
       "682          682  2-s2.0-85063972115   10.1007/s40656-019-0252-3  NULL   \n",
       "...          ...                 ...                         ...   ...   \n",
       "2707        2707  2-s2.0-85063972115   10.1007/s40656-019-0252-3  NULL   \n",
       "2711        2711  2-s2.0-85058017987   10.1007/s40656-018-0236-8  NULL   \n",
       "2716        2716  2-s2.0-85046898458   10.1007/s40656-018-0194-1  NULL   \n",
       "2733        2733  2-s2.0-84988353676   10.1007/s40656-016-0113-2  NULL   \n",
       "2736        2736  2-s2.0-84940554967   10.1007/s40656-015-0072-z  NULL   \n",
       "\n",
       "       pubmed_id                                              title subtype  \\\n",
       "263   28205138.0  Kant‚Äôs epigenesis: specificity and development...      ar   \n",
       "369   32691291.0  What‚Äôs all the fuss about? The inheritance of ...      ar   \n",
       "390   27854053.0          Epigenetics: ambiguities and implications      ar   \n",
       "622   38153583.0  Resilience and the shift of paradigm in ecolog...      ar   \n",
       "682   30937631.0  Taxonomy and conservation science: interdepend...      ar   \n",
       "...          ...                                                ...     ...   \n",
       "2707  30937631.0  Taxonomy and conservation science: interdepend...      ar   \n",
       "2711  30523424.0  At the intersection of medical geography and d...      ar   \n",
       "2716  29761370.0                           Multispecies individuals      ar   \n",
       "2733  27645228.0  From ecological records to big data: the inven...      ar   \n",
       "2736  26141169.0  ‚ÄòBig science‚Äô in the field: experimenting with...      ar   \n",
       "\n",
       "     subtypeDescription          creator                        afid  ...  \\\n",
       "263             Article      Demarest B.                    60002483  ...   \n",
       "369             Article     Camacho M.P.                    60015457  ...   \n",
       "390             Article         Stotz K.           60019544;60010710  ...   \n",
       "622             Article       Barbara L.                    60003668  ...   \n",
       "682             Article         Conix S.                    60025063  ...   \n",
       "...                 ...              ...                         ...  ...   \n",
       "2707            Article         Conix S.                    60025063  ...   \n",
       "2711            Article  Arrizabalaga J.                    60028181  ...   \n",
       "2716            Article       Bourrat P.  60025709;60019544;60010710  ...   \n",
       "2733            Article      Devictor V.           60108488;60021260  ...   \n",
       "2736            Article       Cassidy A.                    60011520  ...   \n",
       "\n",
       "     pageRange                                        description  \\\n",
       "263       NULL  In this paper, I argue that Kant adopted, thro...   \n",
       "369       NULL  The Central Dogma of molecular biology, which ...   \n",
       "390       NULL  Everyone has heard of ‚Äòepigenetics‚Äô, but the t...   \n",
       "622       NULL  In the shift from the balance of nature to the...   \n",
       "682       NULL  The relation between conservation science and ...   \n",
       "...        ...                                                ...   \n",
       "2707      NULL  The relation between conservation science and ...   \n",
       "2711      NULL  Environmental historians are not sufficiently ...   \n",
       "2716      NULL  We assess the arguments for recognising functi...   \n",
       "2733      NULL  This paper is a critical assessment of the epi...   \n",
       "2736   305-325  Since wild badgers were first connected with o...   \n",
       "\n",
       "                                           authkeywords citedby_count  \\\n",
       "263   Epigenesis | Generative force | Kant | Preform...             7   \n",
       "369   Difference-making | Epigenetics | Gene-centris...             4   \n",
       "390   Epigenesis | Epigenetic inheritance | Epigenet...            29   \n",
       "622   Balance of nature | Ecological resilience | Fl...             1   \n",
       "682   Conservation science | Species classification ...            14   \n",
       "...                                                 ...           ...   \n",
       "2707  Conservation science | Species classification ...            14   \n",
       "2711  Disease ecology | Fernand Braudel | Historical...            10   \n",
       "2716  Evolutionary transitions in individuality | Ho...            30   \n",
       "2733  Big data | Biodiversity | Ecology | Foucault |...            50   \n",
       "2736  Agricultural research | Animal health | Ecolog...            12   \n",
       "\n",
       "     openaccess          freetoread freetoreadLabel fund_acr  \\\n",
       "263           1       repositoryvor           Green     NULL   \n",
       "369           0                NULL            NULL     NULL   \n",
       "390           0                NULL            NULL     TWCF   \n",
       "622           1  publisherfree2read          Bronze     NULL   \n",
       "682           0                NULL            NULL     NULL   \n",
       "...         ...                 ...             ...      ...   \n",
       "2707          0                NULL            NULL     NULL   \n",
       "2711          1  publisherfree2read          Bronze     NULL   \n",
       "2716          0                NULL            NULL     CEED   \n",
       "2733          0                NULL            NULL     NULL   \n",
       "2736          1       repositoryvor           Green       WT   \n",
       "\n",
       "                fund_no                                       fund_sponsor  \n",
       "263           undefined                                               NULL  \n",
       "369           undefined                                               NULL  \n",
       "390       TWCF0063/AB37                 Templeton World Charity Foundation  \n",
       "622           undefined                                               NULL  \n",
       "682            3H160214                                          KU Leuven  \n",
       "...                 ...                                                ...  \n",
       "2707           3H160214                                          KU Leuven  \n",
       "2711          undefined                                               NULL  \n",
       "2716          DP0878650  Centre of Excellence for Environmental Decisio...  \n",
       "2733          undefined                                               NULL  \n",
       "2736  RES-229-27-0007-A                                     Wellcome Trust  \n",
       "\n",
       "[240 rows x 37 columns]"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_cleaned[df_cleaned['publicationName'] == \"Biology and Philosophy\"]\n",
    "#df_cleaned[df_cleaned['publicationName'] == \"Biological Theory\"]\n",
    "df_cleaned[df_cleaned['publicationName'] =='History and Philosophy of the Life Sciences']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c90f734d",
   "metadata": {},
   "outputs": [],
   "source": [
    "docs = df_cleaned.description.to_list()\n",
    "\n",
    "df_cleaned['coverDate'] = pd.to_datetime(df_cleaned['coverDate'])\n",
    "df_cleaned['year'] = df_cleaned['coverDate'].dt.year\n",
    "\n",
    "columns_to_combine = ['creator', 'year', 'title']\n",
    "df_cleaned['combined'] = df_cleaned[columns_to_combine].apply(lambda row: ', '.join(map(str, row)), axis=1)\n",
    "node_text = df_cleaned.combined.to_list()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "21abc828",
   "metadata": {},
   "source": [
    "## üåå BERTopic model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "22ce97ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "7b8e2dff3f364295b9c1ae96d6b93d8a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Batches:   0%|          | 0/196 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Pre-calculate embeddings\n",
    "from sentence_transformers import SentenceTransformer\n",
    "embedding_model = SentenceTransformer(\"all-MiniLM-L6-v2\")\n",
    "embeddings = embedding_model.encode(docs, show_progress_bar=True)\n",
    "\n",
    "# Pre-reduce embeddings for visualization purposes\n",
    "reduced_embeddings = UMAP(n_neighbors=10, n_components=2, min_dist=0.0, metric='cosine', random_state=42).fit_transform(embeddings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2841473b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define sub-models\n",
    "from hdbscan import HDBSCAN\n",
    "from umap import UMAP\n",
    "umap_model = UMAP(n_neighbors=10, n_components=2, min_dist=0.0, metric='cosine', random_state=42)\n",
    "hdbscan_model = HDBSCAN(min_cluster_size=20, metric='euclidean', cluster_selection_method='eom', prediction_data=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a399d883",
   "metadata": {},
   "source": [
    "## Visualisation with datamapplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1b01467c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-04-28 17:51:35,885 - BERTopic - Dimensionality - Fitting the dimensionality reduction algorithm\n",
      "2025-04-28 17:51:43,952 - BERTopic - Dimensionality - Completed ‚úì\n",
      "2025-04-28 17:51:43,953 - BERTopic - Cluster - Start clustering the reduced embeddings\n",
      "2025-04-28 17:51:44,074 - BERTopic - Cluster - Completed ‚úì\n",
      "2025-04-28 17:51:44,078 - BERTopic - Representation - Extracting topics from clusters using representation models.\n",
      "2025-04-28 17:51:44,746 - BERTopic - Representation - Completed ‚úì\n"
     ]
    }
   ],
   "source": [
    "from bertopic import BERTopic\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "\n",
    "\n",
    "# Create a CountVectorizer with the custom stopwords\n",
    "vectorizer_model = CountVectorizer(stop_words=\"english\")\n",
    "\n",
    "\n",
    "\n",
    "topic_model = BERTopic(\n",
    "  # Sub-models\n",
    "  embedding_model=embedding_model,\n",
    "  umap_model=umap_model,\n",
    "  hdbscan_model=hdbscan_model,\n",
    "  #representation_model=representation_model,\n",
    "  vectorizer_model=vectorizer_model,\n",
    "  \n",
    "  # Hyperparameters\n",
    "  top_n_words=10,\n",
    "  verbose=True\n",
    ")\n",
    "\n",
    "# Train model\n",
    "topics, probs = topic_model.fit_transform(docs, embeddings)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "489a7852",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of outliers: 2294\n"
     ]
    }
   ],
   "source": [
    "topic_info = topic_model.get_topic_info()\n",
    "outlier_count = topic_info[topic_info['Topic'] == -1]['Count'].values[0]\n",
    "print(f\"Number of outliers: {outlier_count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "bf77824d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.plotly.v1+json": {
       "config": {
        "plotlyServerURL": "https://plot.ly"
       },
       "data": [
        {
         "coloraxis": "coloraxis",
         "hovertemplate": "x: %{x}<br>y: %{y}<br>Similarity Score: %{z}<extra></extra>",
         "name": "0",
         "type": "heatmap",
         "x": [
          "0_dna_transcription_rna",
          "1_mendel_vries_century",
          "2_cultural_culture_transmis...",
          "3_genetic_genetics_ethical",
          "4_systems_energy_complexity",
          "5_ecology_ecological_ecolog...",
          "6_darwin_natural_selection",
          "7_host_bacteria_eukaryotic",
          "8_language_linguistic_commu...",
          "9_information_gene_genome",
          "10_explanations_ultimate_pr...",
          "11_species_taxa_taxon",
          "12_political_nature_ecologi...",
          "13_biology_systems_engineer...",
          "14_god_theology_theological",
          "15_drift_selection_causal",
          "16_game_cooperation_punishm...",
          "17_fitness_propensity_selec...",
          "18_semiotic_umwelt_signs",
          "19_cell_sleep_cells",
          "20_consciousness_ual_cognit...",
          "21_moral_norms_morality",
          "22_niche_construction_organ...",
          "23_heritability_intelligenc...",
          "24_sex_female_chromosome",
          "25_design_intelligent_id",
          "26_welfare_suffering_ecosys...",
          "27_microbiome_microbial_gut",
          "28_medicine_disease_medical",
          "29_altruism_altruistic_group",
          "30_der_la_morphological",
          "31_cancer_tumor_cells",
          "32_biodiversity_conservatio...",
          "33_mental_schizophrenia_dis...",
          "34_ad_proteins_golgi",
          "35_food_agricultural_biodiv...",
          "36_prey_predator_decline",
          "37_aesthetic_art_aesthetics",
          "38_mitochondrial_mitochondr...",
          "39_model_hiv_disease",
          "40_function_functions_funct...",
          "41_race_racial_eugenics",
          "42_ethical_ethics_students",
          "43_agency_intentionality_pe...",
          "44_metazoans_cnidarians_dev...",
          "45_synthesis_popper_evoluti...",
          "46_invasion_native_species",
          "47_kin_women_children",
          "48_cas_crispr_antibiotic",
          "49_causal_mechanism_pluralism",
          "50_crop_farmers_organic",
          "51_epistemology_knowledge_e...",
          "52_selection_natural_view",
          "53_restoration_biodiversity...",
          "54_debunking_moral_beliefs",
          "55_dst_developmental_replic...",
          "56_biodiversity_conservatio...",
          "57_editing_gene_therapy",
          "58_selection_operant_replic...",
          "59_adaptationism_adaptation...",
          "60_code_amino_codes",
          "61_science_history_whewell",
          "62_growth_mathematical_plant",
          "63_handedness_lateralizatio...",
          "64_networks_regulatory_netw...",
          "65_psycinfo_apa_psychological",
          "66_brain_cognitive_dorsal",
          "67_integration_scientific_k...",
          "68_religion_religious_belief",
          "69_digital_guattari_ai",
          "70_regeneration_morphogenet...",
          "71_competition_coexistence_...",
          "72_teleology_teleological_p...",
          "73_mating_sexual_sex",
          "74_tes_p53_retrocopies",
          "75_sociobiology_wilson_mant..."
         ],
         "xaxis": "x",
         "y": [
          "0_dna_transcription_rna",
          "1_mendel_vries_century",
          "2_cultural_culture_transmis...",
          "3_genetic_genetics_ethical",
          "4_systems_energy_complexity",
          "5_ecology_ecological_ecolog...",
          "6_darwin_natural_selection",
          "7_host_bacteria_eukaryotic",
          "8_language_linguistic_commu...",
          "9_information_gene_genome",
          "10_explanations_ultimate_pr...",
          "11_species_taxa_taxon",
          "12_political_nature_ecologi...",
          "13_biology_systems_engineer...",
          "14_god_theology_theological",
          "15_drift_selection_causal",
          "16_game_cooperation_punishm...",
          "17_fitness_propensity_selec...",
          "18_semiotic_umwelt_signs",
          "19_cell_sleep_cells",
          "20_consciousness_ual_cognit...",
          "21_moral_norms_morality",
          "22_niche_construction_organ...",
          "23_heritability_intelligenc...",
          "24_sex_female_chromosome",
          "25_design_intelligent_id",
          "26_welfare_suffering_ecosys...",
          "27_microbiome_microbial_gut",
          "28_medicine_disease_medical",
          "29_altruism_altruistic_group",
          "30_der_la_morphological",
          "31_cancer_tumor_cells",
          "32_biodiversity_conservatio...",
          "33_mental_schizophrenia_dis...",
          "34_ad_proteins_golgi",
          "35_food_agricultural_biodiv...",
          "36_prey_predator_decline",
          "37_aesthetic_art_aesthetics",
          "38_mitochondrial_mitochondr...",
          "39_model_hiv_disease",
          "40_function_functions_funct...",
          "41_race_racial_eugenics",
          "42_ethical_ethics_students",
          "43_agency_intentionality_pe...",
          "44_metazoans_cnidarians_dev...",
          "45_synthesis_popper_evoluti...",
          "46_invasion_native_species",
          "47_kin_women_children",
          "48_cas_crispr_antibiotic",
          "49_causal_mechanism_pluralism",
          "50_crop_farmers_organic",
          "51_epistemology_knowledge_e...",
          "52_selection_natural_view",
          "53_restoration_biodiversity...",
          "54_debunking_moral_beliefs",
          "55_dst_developmental_replic...",
          "56_biodiversity_conservatio...",
          "57_editing_gene_therapy",
          "58_selection_operant_replic...",
          "59_adaptationism_adaptation...",
          "60_code_amino_codes",
          "61_science_history_whewell",
          "62_growth_mathematical_plant",
          "63_handedness_lateralizatio...",
          "64_networks_regulatory_netw...",
          "65_psycinfo_apa_psychological",
          "66_brain_cognitive_dorsal",
          "67_integration_scientific_k...",
          "68_religion_religious_belief",
          "69_digital_guattari_ai",
          "70_regeneration_morphogenet...",
          "71_competition_coexistence_...",
          "72_teleology_teleological_p...",
          "73_mating_sexual_sex",
          "74_tes_p53_retrocopies",
          "75_sociobiology_wilson_mant..."
         ],
         "yaxis": "y",
         "z": [
          [
           0.9999999,
           0.2685101,
           0.22701338,
           0.29576847,
           0.26295295,
           0.11780151,
           0.17696066,
           0.46817005,
           0.19236796,
           0.5879825,
           0.323318,
           0.20027825,
           0.045533508,
           0.3661466,
           0.1800473,
           0.2601187,
           0.11935642,
           0.24953601,
           0.2508548,
           0.5690415,
           0.264264,
           0.1429251,
           0.23867185,
           0.28887868,
           0.52372575,
           0.10383075,
           0.045971036,
           0.31766886,
           0.110171095,
           0.14542629,
           0.49683112,
           0.6596825,
           0.04398114,
           0.31635642,
           0.7167218,
           0.025960436,
           0.01182943,
           0.051570304,
           0.6570456,
           0.03301058,
           0.25393137,
           0.122526765,
           -0.039714284,
           0.1740633,
           0.43265122,
           0.23839131,
           0.100331455,
           0.15686393,
           0.629904,
           0.34782946,
           0.099794686,
           0.22482464,
           0.24732104,
           -0.028879616,
           0.19150223,
           0.3148466,
           0.09521504,
           0.40465862,
           0.22151735,
           0.32058775,
           0.5725105,
           0.1124645,
           0.16856498,
           0.3388145,
           0.54143524,
           0.10730664,
           0.3417073,
           0.39038604,
           0.17601353,
           0.015973233,
           0.41458988,
           0.10086088,
           0.21978997,
           0.20896237,
           0.80309767,
           0.19172727
          ],
          [
           0.2685101,
           0.9999996,
           0.5770917,
           0.60431993,
           0.6039861,
           0.67165565,
           0.845045,
           0.57479584,
           0.4588259,
           0.69426346,
           0.68707216,
           0.6440343,
           0.5365589,
           0.73045397,
           0.6050845,
           0.663652,
           0.38577652,
           0.6194589,
           0.67059785,
           0.35872313,
           0.5051764,
           0.4610113,
           0.5519347,
           0.6253716,
           0.5501931,
           0.60836315,
           0.43149155,
           0.44269583,
           0.6793354,
           0.45205218,
           0.6354992,
           0.3816826,
           0.46684337,
           0.43793917,
           0.17965631,
           0.29760963,
           0.2293008,
           0.5208649,
           0.32251978,
           0.13577178,
           0.64716154,
           0.78244257,
           0.21099447,
           0.5797262,
           0.49060893,
           0.7642733,
           0.42163575,
           0.45614567,
           0.36281985,
           0.71993446,
           0.3848536,
           0.6978781,
           0.62491965,
           0.3363461,
           0.6188692,
           0.6473645,
           0.51476365,
           0.53898585,
           0.51634777,
           0.5971838,
           0.518215,
           0.82236946,
           0.42835462,
           0.4771763,
           0.43102294,
           0.3375904,
           0.5349815,
           0.7597969,
           0.3975039,
           0.4476083,
           0.48034602,
           0.39876753,
           0.6765905,
           0.47350827,
           0.34191045,
           0.74534744
          ],
          [
           0.22701338,
           0.5770917,
           1,
           0.47356224,
           0.61825264,
           0.57171,
           0.6382091,
           0.48703873,
           0.6973381,
           0.5732001,
           0.68532836,
           0.5751797,
           0.51613283,
           0.561466,
           0.6013083,
           0.6331134,
           0.58074945,
           0.59916973,
           0.6438215,
           0.20181575,
           0.6346712,
           0.7072115,
           0.63336706,
           0.6378759,
           0.4464618,
           0.57866615,
           0.472066,
           0.3814762,
           0.5182632,
           0.64444077,
           0.5594946,
           0.3367659,
           0.48488057,
           0.507269,
           0.08240308,
           0.3484208,
           0.23291108,
           0.71151245,
           0.29164693,
           0.11998485,
           0.6086157,
           0.6320803,
           0.3051798,
           0.6549555,
           0.44709882,
           0.7243743,
           0.46219036,
           0.6072897,
           0.25876004,
           0.6446583,
           0.26411992,
           0.7438786,
           0.64430845,
           0.39128238,
           0.6431493,
           0.64217156,
           0.44447201,
           0.40657943,
           0.6110548,
           0.69370204,
           0.35976762,
           0.59731805,
           0.30762172,
           0.5375301,
           0.4085385,
           0.5062263,
           0.69849753,
           0.5374627,
           0.7208799,
           0.54998446,
           0.3177487,
           0.41845256,
           0.57373345,
           0.5326069,
           0.3364722,
           0.675098
          ],
          [
           0.29576847,
           0.60431993,
           0.47356224,
           1.0000002,
           0.2938822,
           0.39870578,
           0.4649366,
           0.34174132,
           0.26724786,
           0.6583301,
           0.47818518,
           0.41500238,
           0.4402387,
           0.48265365,
           0.48488432,
           0.4480081,
           0.3201676,
           0.43184647,
           0.4545643,
           0.11277555,
           0.3008553,
           0.47863793,
           0.32917017,
           0.61399543,
           0.40209943,
           0.46367598,
           0.48451063,
           0.38154972,
           0.6369028,
           0.4442507,
           0.38046908,
           0.40709442,
           0.5139535,
           0.52596205,
           0.063744634,
           0.3722489,
           -0.03288702,
           0.38172644,
           0.27763116,
           0.09175721,
           0.45009127,
           0.6800288,
           0.48545787,
           0.45012075,
           0.18760656,
           0.45302218,
           0.37536418,
           0.450158,
           0.33600155,
           0.50777996,
           0.4089388,
           0.44650662,
           0.5105183,
           0.35824555,
           0.5403272,
           0.44440007,
           0.44801867,
           0.84397006,
           0.38898024,
           0.39114416,
           0.3453254,
           0.53021574,
           0.084305674,
           0.34793484,
           0.26599735,
           0.2680086,
           0.3463261,
           0.58029544,
           0.38839763,
           0.41029453,
           0.14028722,
           0.11036882,
           0.45366684,
           0.41248342,
           0.41339225,
           0.6801667
          ],
          [
           0.26295295,
           0.6039861,
           0.61825264,
           0.2938822,
           1,
           0.69730985,
           0.6319026,
           0.60897833,
           0.56403655,
           0.6456769,
           0.7845791,
           0.5727831,
           0.4649367,
           0.7982445,
           0.56654775,
           0.71191216,
           0.61858785,
           0.7197523,
           0.6863833,
           0.52432096,
           0.70300925,
           0.52228993,
           0.68426293,
           0.53155124,
           0.42279673,
           0.59047234,
           0.45299366,
           0.3935427,
           0.4510625,
           0.54136574,
           0.6357767,
           0.43295753,
           0.36331218,
           0.39679104,
           0.22999819,
           0.248216,
           0.48402262,
           0.48761654,
           0.31264603,
           0.32795003,
           0.6764516,
           0.44046968,
           0.15684663,
           0.6960261,
           0.56180096,
           0.7825592,
           0.35786608,
           0.4304492,
           0.3320602,
           0.77774507,
           0.24094716,
           0.7522172,
           0.63394725,
           0.2988979,
           0.5899894,
           0.7493782,
           0.46069404,
           0.27099377,
           0.6807524,
           0.6959624,
           0.5722574,
           0.5556972,
           0.61132514,
           0.47451368,
           0.6770265,
           0.40536982,
           0.6272852,
           0.6375905,
           0.48402858,
           0.49741793,
           0.53682894,
           0.62247944,
           0.70528305,
           0.47125036,
           0.2727454,
           0.5667244
          ],
          [
           0.11780151,
           0.67165565,
           0.57171,
           0.39870578,
           0.69730985,
           1.0000002,
           0.71744907,
           0.5391501,
           0.44527215,
           0.53110564,
           0.7347893,
           0.71625066,
           0.782031,
           0.6657643,
           0.5592102,
           0.67098653,
           0.50711995,
           0.63623476,
           0.7614052,
           0.28133273,
           0.54304194,
           0.50550574,
           0.77887666,
           0.56134355,
           0.3662238,
           0.5406207,
           0.7764287,
           0.48361975,
           0.5517028,
           0.51738185,
           0.48549914,
           0.30400252,
           0.74962234,
           0.29940736,
           0.07052102,
           0.5628555,
           0.53348017,
           0.576864,
           0.17257017,
           0.28129345,
           0.6742226,
           0.5806762,
           0.32623255,
           0.6589807,
           0.4773838,
           0.7073237,
           0.7265222,
           0.47562587,
           0.20150493,
           0.7462462,
           0.5275159,
           0.6972559,
           0.6404181,
           0.7021847,
           0.63561165,
           0.624308,
           0.7515613,
           0.30132908,
           0.59382224,
           0.6154038,
           0.35371113,
           0.67384815,
           0.57644314,
           0.34262055,
           0.45396864,
           0.33510888,
           0.44793463,
           0.67394936,
           0.43426305,
           0.5437615,
           0.44027832,
           0.72730744,
           0.675863,
           0.43486795,
           0.1630773,
           0.69953
          ],
          [
           0.17696066,
           0.845045,
           0.6382091,
           0.4649366,
           0.6319026,
           0.71744907,
           1.0000001,
           0.56144404,
           0.5269357,
           0.5881506,
           0.7246599,
           0.71923965,
           0.59223425,
           0.6361711,
           0.7066362,
           0.71248794,
           0.45505622,
           0.6402838,
           0.6659212,
           0.23472802,
           0.56380624,
           0.54301286,
           0.6283139,
           0.5346448,
           0.4763285,
           0.69353664,
           0.50221294,
           0.42847916,
           0.61407745,
           0.50899,
           0.5918284,
           0.32751936,
           0.56088495,
           0.4127518,
           0.043535084,
           0.30432698,
           0.3113984,
           0.58637154,
           0.23766881,
           0.13248432,
           0.64758,
           0.7242933,
           0.19332582,
           0.6044078,
           0.5239885,
           0.85124874,
           0.53494596,
           0.45659918,
           0.28702196,
           0.6796168,
           0.3163221,
           0.760568,
           0.70102787,
           0.39594126,
           0.71570325,
           0.62292796,
           0.5842577,
           0.42541456,
           0.55829334,
           0.68133867,
           0.40436217,
           0.77748185,
           0.3848377,
           0.46206087,
           0.3717106,
           0.36559486,
           0.5969498,
           0.652513,
           0.48233593,
           0.46658343,
           0.40186167,
           0.50203145,
           0.6657206,
           0.49402738,
           0.3141365,
           0.7436501
          ],
          [
           0.46817005,
           0.57479584,
           0.48703873,
           0.34174132,
           0.60897833,
           0.5391501,
           0.56144404,
           1.0000001,
           0.37717244,
           0.6151585,
           0.6490159,
           0.66423327,
           0.35510144,
           0.6286075,
           0.4989875,
           0.565574,
           0.3815995,
           0.5700244,
           0.5472866,
           0.49892804,
           0.47608545,
           0.4341516,
           0.6325159,
           0.44312817,
           0.60254353,
           0.37896192,
           0.39457053,
           0.7355951,
           0.4262957,
           0.432329,
           0.6421176,
           0.48519802,
           0.4117242,
           0.37061387,
           0.4161122,
           0.24786958,
           0.31227782,
           0.33322218,
           0.5495025,
           0.25848404,
           0.5590937,
           0.41400948,
           0.097806014,
           0.44656944,
           0.6668359,
           0.63382316,
           0.42025366,
           0.35875276,
           0.7177167,
           0.6238266,
           0.33779222,
           0.5506132,
           0.553038,
           0.25153688,
           0.5225654,
           0.59395397,
           0.49436513,
           0.3499304,
           0.5034174,
           0.64730453,
           0.5972512,
           0.40107954,
           0.50912786,
           0.43271834,
           0.5126723,
           0.20375454,
           0.47938967,
           0.5652318,
           0.34932783,
           0.24056144,
           0.56044805,
           0.504418,
           0.53998995,
           0.43761986,
           0.48031706,
           0.56807953
          ],
          [
           0.19236796,
           0.4588259,
           0.6973381,
           0.26724786,
           0.56403655,
           0.44527215,
           0.5269357,
           0.37717244,
           1,
           0.5456356,
           0.5877118,
           0.49943873,
           0.37424618,
           0.5017706,
           0.46225336,
           0.48215076,
           0.49101713,
           0.44990355,
           0.64559054,
           0.15317827,
           0.673951,
           0.579686,
           0.5276861,
           0.48112923,
           0.3371731,
           0.45766783,
           0.28155944,
           0.23417065,
           0.39318722,
           0.48160115,
           0.57862395,
           0.24299252,
           0.29921573,
           0.40191588,
           0.08659868,
           0.17875074,
           0.1764372,
           0.5344621,
           0.16943464,
           0.058416076,
           0.55388117,
           0.44128454,
           0.11698954,
           0.65839124,
           0.46817455,
           0.607015,
           0.2781164,
           0.3991673,
           0.20305581,
           0.5458218,
           0.10408076,
           0.65130657,
           0.4995614,
           0.22873831,
           0.539604,
           0.55069333,
           0.28744438,
           0.2373264,
           0.49824113,
           0.54822373,
           0.40233302,
           0.45745003,
           0.23688217,
           0.53188956,
           0.35816103,
           0.5299243,
           0.68272483,
           0.4456621,
           0.5299238,
           0.46890217,
           0.3148841,
           0.34169728,
           0.5310446,
           0.40281606,
           0.26111823,
           0.4978754
          ],
          [
           0.5879825,
           0.69426346,
           0.5732001,
           0.6583301,
           0.6456769,
           0.53110564,
           0.5881506,
           0.6151585,
           0.5456356,
           0.99999994,
           0.71918297,
           0.6256726,
           0.35740453,
           0.74739337,
           0.5022926,
           0.6676202,
           0.3970006,
           0.61636436,
           0.62786317,
           0.45849514,
           0.574262,
           0.42621535,
           0.55410254,
           0.72510076,
           0.6010779,
           0.504663,
           0.35598654,
           0.48092055,
           0.49291956,
           0.43637627,
           0.7106916,
           0.5642384,
           0.35223,
           0.56595397,
           0.32997167,
           0.22185077,
           0.16293192,
           0.39445525,
           0.4871124,
           0.12007581,
           0.6836642,
           0.56625414,
           0.16069436,
           0.61450005,
           0.49978954,
           0.66830444,
           0.3035974,
           0.39406526,
           0.55868495,
           0.75246394,
           0.3156681,
           0.667561,
           0.6637057,
           0.2163308,
           0.570647,
           0.6996659,
           0.43988317,
           0.6035587,
           0.605339,
           0.6413144,
           0.726918,
           0.53681195,
           0.35096908,
           0.5188186,
           0.6631793,
           0.3928072,
           0.57916665,
           0.7533764,
           0.39515662,
           0.36389625,
           0.4617615,
           0.3266872,
           0.6353307,
           0.46320087,
           0.6177729,
           0.6034175
          ],
          [
           0.323318,
           0.68707216,
           0.68532836,
           0.47818518,
           0.7845791,
           0.7347893,
           0.7246599,
           0.6490159,
           0.5877118,
           0.71918297,
           1,
           0.670044,
           0.48694342,
           0.7307558,
           0.61796856,
           0.8740815,
           0.61997545,
           0.808287,
           0.6939803,
           0.3984379,
           0.6657063,
           0.59382856,
           0.70187366,
           0.7264424,
           0.5306895,
           0.63494813,
           0.54133254,
           0.51245505,
           0.58311474,
           0.6375329,
           0.66390204,
           0.4889573,
           0.49694854,
           0.53703266,
           0.20636967,
           0.29471713,
           0.39975125,
           0.5157662,
           0.37469065,
           0.28070238,
           0.770931,
           0.5703069,
           0.2220411,
           0.7373692,
           0.57357705,
           0.82819325,
           0.5322288,
           0.54043573,
           0.42906955,
           0.893926,
           0.3522287,
           0.82663685,
           0.8168747,
           0.34569287,
           0.7875968,
           0.75599223,
           0.5366349,
           0.45802158,
           0.7437234,
           0.8191596,
           0.49736315,
           0.5778008,
           0.48424906,
           0.5660745,
           0.5719954,
           0.5258739,
           0.64162666,
           0.6869068,
           0.54255664,
           0.37515286,
           0.4839296,
           0.5758313,
           0.70793015,
           0.58102703,
           0.3957225,
           0.67045397
          ],
          [
           0.20027825,
           0.6440343,
           0.5751797,
           0.41500238,
           0.5727831,
           0.71625066,
           0.71923965,
           0.66423327,
           0.49943873,
           0.6256726,
           0.670044,
           0.99999994,
           0.49167132,
           0.5935638,
           0.5236243,
           0.60660183,
           0.36472473,
           0.54647815,
           0.6443757,
           0.219022,
           0.5211369,
           0.45877326,
           0.7152809,
           0.5335366,
           0.53376245,
           0.45693105,
           0.51310754,
           0.4922433,
           0.46811038,
           0.4135701,
           0.66486764,
           0.35080862,
           0.60324174,
           0.36411208,
           0.09620592,
           0.3348417,
           0.3455925,
           0.48538888,
           0.30268133,
           0.13978042,
           0.638253,
           0.6016524,
           0.14792338,
           0.5356242,
           0.62001574,
           0.7113774,
           0.61667943,
           0.41594526,
           0.37949806,
           0.6624867,
           0.37625685,
           0.6529417,
           0.62498295,
           0.46318576,
           0.60449666,
           0.5852476,
           0.7007328,
           0.30819505,
           0.53672975,
           0.6316362,
           0.44935286,
           0.5701931,
           0.36660254,
           0.41964728,
           0.39466536,
           0.32696578,
           0.5325008,
           0.63247585,
           0.4042943,
           0.36185256,
           0.46613458,
           0.5672029,
           0.60164595,
           0.47492856,
           0.30311173,
           0.6134572
          ],
          [
           0.045533508,
           0.5365589,
           0.51613283,
           0.4402387,
           0.4649367,
           0.782031,
           0.59223425,
           0.35510144,
           0.37424618,
           0.35740453,
           0.48694342,
           0.49167132,
           0.99999994,
           0.4507719,
           0.6500286,
           0.416849,
           0.3981716,
           0.40419558,
           0.7164495,
           0.09045708,
           0.4596771,
           0.5307186,
           0.6213599,
           0.39353955,
           0.21990871,
           0.5630742,
           0.822883,
           0.3479465,
           0.55447894,
           0.4707657,
           0.3078357,
           0.18996637,
           0.79868937,
           0.23093322,
           0.0011429563,
           0.61729693,
           0.2567033,
           0.658365,
           0.07386372,
           0.04752286,
           0.50480056,
           0.61106485,
           0.4462842,
           0.6225983,
           0.28560936,
           0.54995865,
           0.63552845,
           0.38890845,
           0.049319923,
           0.5448414,
           0.44702974,
           0.5410049,
           0.45095977,
           0.78187954,
           0.59760535,
           0.42505634,
           0.6675934,
           0.33116388,
           0.3424797,
           0.4175244,
           0.20369807,
           0.7118881,
           0.31113243,
           0.22951262,
           0.17410634,
           0.19280143,
           0.37357092,
           0.46445298,
           0.4719981,
           0.7006599,
           0.22403322,
           0.45718265,
           0.6342977,
           0.2674052,
           0.1011242,
           0.7263181
          ],
          [
           0.3661466,
           0.73045397,
           0.561466,
           0.48265365,
           0.7982445,
           0.6657643,
           0.6361711,
           0.6286075,
           0.5017706,
           0.74739337,
           0.7307558,
           0.5935638,
           0.4507719,
           1.0000001,
           0.5191119,
           0.6083807,
           0.43085426,
           0.6350051,
           0.69014275,
           0.5307427,
           0.6260784,
           0.39672327,
           0.6117374,
           0.5484868,
           0.43384975,
           0.6422883,
           0.45270583,
           0.48973292,
           0.5414591,
           0.42507,
           0.634946,
           0.51667964,
           0.42132896,
           0.36546895,
           0.33348155,
           0.26021343,
           0.28233248,
           0.47896883,
           0.3538168,
           0.21280618,
           0.6870829,
           0.500029,
           0.17810848,
           0.6899468,
           0.50387895,
           0.75807106,
           0.38595065,
           0.2949734,
           0.49243078,
           0.8017052,
           0.34377092,
           0.7302184,
           0.57074094,
           0.28523958,
           0.587404,
           0.6809652,
           0.51008195,
           0.49947256,
           0.57188296,
           0.6278178,
           0.6448718,
           0.65362287,
           0.47256804,
           0.41223884,
           0.6524055,
           0.35343757,
           0.52536356,
           0.809203,
           0.3728461,
           0.5033655,
           0.5368296,
           0.43256947,
           0.6955608,
           0.376272,
           0.37213257,
           0.62835395
          ],
          [
           0.1800473,
           0.6050845,
           0.6013083,
           0.48488432,
           0.56654775,
           0.5592102,
           0.7066362,
           0.4989875,
           0.46225336,
           0.5022926,
           0.61796856,
           0.5236243,
           0.6500286,
           0.5191119,
           1,
           0.56528264,
           0.43972945,
           0.550593,
           0.62906957,
           0.17422931,
           0.53280944,
           0.62876457,
           0.5421504,
           0.46778232,
           0.37464306,
           0.77443564,
           0.5427282,
           0.34980434,
           0.57542837,
           0.52220964,
           0.4947403,
           0.30986387,
           0.5420323,
           0.45151782,
           0.06106789,
           0.3047034,
           0.17951426,
           0.5986618,
           0.22439095,
           0.08406296,
           0.5730623,
           0.5939942,
           0.32305533,
           0.6080717,
           0.40579033,
           0.726402,
           0.41871062,
           0.395707,
           0.2745856,
           0.60538137,
           0.25058568,
           0.70775867,
           0.58939826,
           0.38784117,
           0.7564469,
           0.54221743,
           0.4840384,
           0.4512871,
           0.4677229,
           0.59450465,
           0.3688674,
           0.6670327,
           0.2673148,
           0.40747613,
           0.28111014,
           0.27652937,
           0.54941124,
           0.48712263,
           0.75399184,
           0.519655,
           0.32012728,
           0.3582857,
           0.69433796,
           0.41245887,
           0.27874053,
           0.7269757
          ],
          [
           0.2601187,
           0.663652,
           0.6331134,
           0.4480081,
           0.71191216,
           0.67098653,
           0.71248794,
           0.565574,
           0.48215076,
           0.6676202,
           0.8740815,
           0.60660183,
           0.416849,
           0.6083807,
           0.56528264,
           1,
           0.57189584,
           0.8354319,
           0.55176544,
           0.3567073,
           0.53530157,
           0.49760714,
           0.6088575,
           0.72188944,
           0.527823,
           0.5883587,
           0.44407195,
           0.4381666,
           0.4878962,
           0.5761593,
           0.5651019,
           0.45937356,
           0.4370888,
           0.49880898,
           0.12696216,
           0.23393746,
           0.44722039,
           0.44001514,
           0.3527137,
           0.32847518,
           0.6541111,
           0.5943558,
           0.12634352,
           0.5932962,
           0.48035365,
           0.7874513,
           0.49790865,
           0.5343306,
           0.36070365,
           0.7734679,
           0.33379474,
           0.7411648,
           0.8700013,
           0.2919079,
           0.6779439,
           0.6905884,
           0.49167436,
           0.4295028,
           0.7793511,
           0.81610155,
           0.4089477,
           0.5044414,
           0.51230335,
           0.50637746,
           0.54434264,
           0.47354093,
           0.5695826,
           0.5705027,
           0.46217197,
           0.2909247,
           0.40427572,
           0.6000842,
           0.61848766,
           0.6081903,
           0.37131685,
           0.6027076
          ],
          [
           0.11935642,
           0.38577652,
           0.58074945,
           0.3201676,
           0.61858785,
           0.50711995,
           0.45505622,
           0.3815995,
           0.49101713,
           0.3970006,
           0.61997545,
           0.36472473,
           0.3981716,
           0.43085426,
           0.43972945,
           0.57189584,
           1,
           0.6413086,
           0.4933614,
           0.20720491,
           0.52461493,
           0.758342,
           0.4889772,
           0.4842432,
           0.33398068,
           0.42525283,
           0.48332602,
           0.21561968,
           0.331896,
           0.7398397,
           0.364326,
           0.27100968,
           0.3766077,
           0.3962857,
           0.046109937,
           0.26075208,
           0.43928856,
           0.44487345,
           0.17902802,
           0.26880696,
           0.50417566,
           0.37955308,
           0.34903407,
           0.580469,
           0.3519144,
           0.54759395,
           0.3461689,
           0.57962877,
           0.16813633,
           0.52318853,
           0.19659819,
           0.53619474,
           0.5513275,
           0.3010569,
           0.5595716,
           0.53825796,
           0.3313259,
           0.25070474,
           0.6166614,
           0.5623583,
           0.28734154,
           0.34232524,
           0.3631689,
           0.44953462,
           0.39394248,
           0.4473778,
           0.46671748,
           0.33985674,
           0.5127716,
           0.4030963,
           0.24518456,
           0.59126925,
           0.45143864,
           0.62714887,
           0.16148537,
           0.51665664
          ],
          [
           0.24953601,
           0.6194589,
           0.59916973,
           0.43184647,
           0.7197523,
           0.63623476,
           0.6402838,
           0.5700244,
           0.44990355,
           0.61636436,
           0.808287,
           0.54647815,
           0.40419558,
           0.6350051,
           0.550593,
           0.8354319,
           0.6413086,
           1,
           0.5541668,
           0.36815503,
           0.53710574,
           0.5525186,
           0.6454561,
           0.7206676,
           0.53514504,
           0.5335057,
           0.4880643,
           0.37322456,
           0.4536992,
           0.65584266,
           0.56586516,
           0.45018876,
           0.4455276,
           0.5249975,
           0.14321338,
           0.2283988,
           0.4560598,
           0.49136135,
           0.37453872,
           0.30551428,
           0.67587173,
           0.5319165,
           0.17577575,
           0.59186983,
           0.5052607,
           0.7652271,
           0.4401756,
           0.61561704,
           0.3863702,
           0.71106637,
           0.3227077,
           0.7108095,
           0.80044675,
           0.29583633,
           0.63166046,
           0.72036016,
           0.47815686,
           0.38060567,
           0.7415786,
           0.8413718,
           0.4473945,
           0.47015524,
           0.5417839,
           0.5065243,
           0.54173875,
           0.47774968,
           0.5623563,
           0.5271426,
           0.45424116,
           0.3167013,
           0.42998022,
           0.60768265,
           0.5889057,
           0.67751455,
           0.315341,
           0.5911232
          ],
          [
           0.2508548,
           0.67059785,
           0.6438215,
           0.4545643,
           0.6863833,
           0.7614052,
           0.6659212,
           0.5472866,
           0.64559054,
           0.62786317,
           0.6939803,
           0.6443757,
           0.7164495,
           0.69014275,
           0.62906957,
           0.55176544,
           0.4933614,
           0.5541668,
           1,
           0.35438186,
           0.7271825,
           0.60101223,
           0.73102504,
           0.57454354,
           0.45356894,
           0.5693738,
           0.6469999,
           0.43130356,
           0.64043844,
           0.56573975,
           0.6156575,
           0.34956446,
           0.5945514,
           0.440278,
           0.2094173,
           0.38482445,
           0.3670066,
           0.68667793,
           0.21993323,
           0.16461924,
           0.7142015,
           0.6079328,
           0.3112672,
           0.7956196,
           0.5685498,
           0.6899712,
           0.5074549,
           0.4729779,
           0.26275063,
           0.75898194,
           0.32162112,
           0.7234951,
           0.5746496,
           0.51008487,
           0.6440805,
           0.6464849,
           0.5666832,
           0.36900148,
           0.5522243,
           0.57557464,
           0.45869488,
           0.7007773,
           0.42511034,
           0.49223727,
           0.4597832,
           0.41974515,
           0.58297575,
           0.63959795,
           0.54133344,
           0.65659356,
           0.48023134,
           0.505375,
           0.75789344,
           0.45073217,
           0.29116738,
           0.7233943
          ],
          [
           0.5690415,
           0.35872313,
           0.20181575,
           0.11277555,
           0.52432096,
           0.28133273,
           0.23472802,
           0.49892804,
           0.15317827,
           0.45849514,
           0.3984379,
           0.219022,
           0.09045708,
           0.5307427,
           0.17422931,
           0.3567073,
           0.20720491,
           0.36815503,
           0.35438186,
           1.0000002,
           0.36854842,
           0.12783176,
           0.306718,
           0.23685282,
           0.40471286,
           0.11727527,
           0.10836005,
           0.31489933,
           0.1876654,
           0.17440794,
           0.45842135,
           0.49975178,
           0.040494725,
           0.18565819,
           0.6530219,
           0.004894115,
           0.38026428,
           0.0724799,
           0.5432359,
           0.3411456,
           0.35749376,
           0.13914026,
           -0.06303092,
           0.25791895,
           0.49203748,
           0.31917003,
           0.064265415,
           0.1612271,
           0.47957587,
           0.46124858,
           0.10540104,
           0.2636106,
           0.23562354,
           -0.023876823,
           0.16378051,
           0.40239328,
           0.13958982,
           0.19819245,
           0.2966645,
           0.31884557,
           0.55278254,
           0.17168361,
           0.5764308,
           0.27699414,
           0.6162199,
           0.096346945,
           0.3164358,
           0.39883292,
           0.1394852,
           0.07455467,
           0.61632407,
           0.3552805,
           0.3393423,
           0.20465314,
           0.4019571,
           0.2171666
          ],
          [
           0.264264,
           0.5051764,
           0.6346712,
           0.3008553,
           0.70300925,
           0.54304194,
           0.56380624,
           0.47608545,
           0.673951,
           0.574262,
           0.6657063,
           0.5211369,
           0.4596771,
           0.6260784,
           0.53280944,
           0.53530157,
           0.52461493,
           0.53710574,
           0.7271825,
           0.36854842,
           1.0000001,
           0.60646987,
           0.64197296,
           0.5235359,
           0.3939975,
           0.5326364,
           0.4532541,
           0.34805733,
           0.5252805,
           0.55444866,
           0.5704098,
           0.32362956,
           0.3710997,
           0.52988493,
           0.2525279,
           0.15293953,
           0.28261185,
           0.60180926,
           0.26556757,
           0.06397529,
           0.653974,
           0.42109957,
           0.1762813,
           0.8503798,
           0.5601213,
           0.64688957,
           0.32669836,
           0.40450618,
           0.2520395,
           0.67373633,
           0.12984237,
           0.7547256,
           0.57522833,
           0.25904292,
           0.59939486,
           0.58818996,
           0.38644776,
           0.27259815,
           0.63447964,
           0.6232996,
           0.44232792,
           0.542879,
           0.3033529,
           0.60554993,
           0.48631442,
           0.6432147,
           0.79655915,
           0.5249596,
           0.5760839,
           0.5051898,
           0.42634517,
           0.38671607,
           0.6908771,
           0.420116,
           0.2978469,
           0.5545346
          ],
          [
           0.1429251,
           0.4610113,
           0.7072115,
           0.47863793,
           0.52228993,
           0.50550574,
           0.54301286,
           0.4341516,
           0.579686,
           0.42621535,
           0.59382856,
           0.45877326,
           0.5307186,
           0.39672327,
           0.62876457,
           0.49760714,
           0.758342,
           0.5525186,
           0.60101223,
           0.12783176,
           0.60646987,
           1,
           0.54731894,
           0.55933785,
           0.3761657,
           0.48544335,
           0.60885084,
           0.31113678,
           0.48416644,
           0.786759,
           0.43484128,
           0.23010972,
           0.5258504,
           0.53180057,
           0.043600366,
           0.30747324,
           0.26057282,
           0.58787477,
           0.20581737,
           0.13031389,
           0.5381422,
           0.5298488,
           0.5012027,
           0.6553906,
           0.42601424,
           0.55773616,
           0.41466025,
           0.6508802,
           0.20196125,
           0.5342438,
           0.2533151,
           0.5953151,
           0.55738914,
           0.3892682,
           0.7036215,
           0.5447324,
           0.377528,
           0.40512836,
           0.54326713,
           0.55797905,
           0.3020404,
           0.4776913,
           0.2527669,
           0.5463517,
           0.30065787,
           0.4908169,
           0.62793934,
           0.37024182,
           0.7032993,
           0.5052161,
           0.23767167,
           0.4009906,
           0.533132,
           0.5835444,
           0.21523044,
           0.7098983
          ],
          [
           0.23867185,
           0.5519347,
           0.63336706,
           0.32917017,
           0.68426293,
           0.77887666,
           0.6283139,
           0.6325159,
           0.5276861,
           0.55410254,
           0.70187366,
           0.7152809,
           0.6213599,
           0.6117374,
           0.5421504,
           0.6088575,
           0.4889772,
           0.6454561,
           0.73102504,
           0.306718,
           0.64197296,
           0.54731894,
           1.0000002,
           0.57674766,
           0.486378,
           0.48487347,
           0.64536905,
           0.47144097,
           0.45128447,
           0.561307,
           0.657694,
           0.38944274,
           0.62151414,
           0.4159702,
           0.1660193,
           0.44281211,
           0.468726,
           0.6125717,
           0.26012647,
           0.16028234,
           0.6743808,
           0.4730587,
           0.23054567,
           0.6574838,
           0.65371704,
           0.72662574,
           0.6139828,
           0.5158296,
           0.29990464,
           0.68058634,
           0.4236071,
           0.7144847,
           0.66454506,
           0.5715568,
           0.588038,
           0.706298,
           0.6601646,
           0.2883716,
           0.6278991,
           0.7265377,
           0.42664117,
           0.5324427,
           0.529495,
           0.48219338,
           0.45818275,
           0.41247374,
           0.61727035,
           0.54684556,
           0.48598155,
           0.4684855,
           0.5646697,
           0.69857806,
           0.6194102,
           0.5051565,
           0.3454734,
           0.61852103
          ],
          [
           0.28887868,
           0.6253716,
           0.6378759,
           0.61399543,
           0.53155124,
           0.56134355,
           0.5346448,
           0.44312817,
           0.48112923,
           0.72510076,
           0.7264424,
           0.5335366,
           0.39353955,
           0.5484868,
           0.46778232,
           0.72188944,
           0.4842432,
           0.7206676,
           0.57454354,
           0.23685282,
           0.5235359,
           0.55933785,
           0.57674766,
           0.9999999,
           0.5241675,
           0.4999249,
           0.44085726,
           0.40839875,
           0.4970496,
           0.6116413,
           0.53189063,
           0.39592463,
           0.41369483,
           0.7088943,
           0.08337465,
           0.24813472,
           0.26204938,
           0.4995976,
           0.31727844,
           0.15662631,
           0.6278,
           0.62104136,
           0.24552606,
           0.6237458,
           0.37112355,
           0.602158,
           0.39554954,
           0.59830153,
           0.3422297,
           0.6687726,
           0.37604538,
           0.6530897,
           0.74873626,
           0.26056665,
           0.59992,
           0.6741071,
           0.4023514,
           0.5163574,
           0.6587463,
           0.6887499,
           0.41334468,
           0.48376548,
           0.34003013,
           0.5839486,
           0.48998183,
           0.58723915,
           0.5828959,
           0.5642381,
           0.4970188,
           0.33940497,
           0.29846108,
           0.39100075,
           0.5366581,
           0.5960351,
           0.36881757,
           0.5991821
          ],
          [
           0.52372575,
           0.5501931,
           0.4464618,
           0.40209943,
           0.42279673,
           0.3662238,
           0.4763285,
           0.60254353,
           0.3371731,
           0.6010779,
           0.5306895,
           0.53376245,
           0.21990871,
           0.43384975,
           0.37464306,
           0.527823,
           0.33398068,
           0.53514504,
           0.45356894,
           0.40471286,
           0.3939975,
           0.3761657,
           0.486378,
           0.5241675,
           1,
           0.2497467,
           0.26523304,
           0.32051313,
           0.23884672,
           0.36072668,
           0.6473931,
           0.40959215,
           0.28162953,
           0.41833353,
           0.32436585,
           0.11023255,
           0.2798879,
           0.30419624,
           0.5412766,
           0.15400402,
           0.4874248,
           0.41902727,
           0.025198543,
           0.33176684,
           0.63021326,
           0.50965524,
           0.2771292,
           0.54303706,
           0.42381144,
           0.5199639,
           0.28821677,
           0.44368213,
           0.54064184,
           0.14537472,
           0.36618948,
           0.5701467,
           0.34172252,
           0.40418422,
           0.46839285,
           0.579103,
           0.44800964,
           0.29631013,
           0.35082662,
           0.5647901,
           0.46370953,
           0.25894743,
           0.47462553,
           0.38829917,
           0.2864183,
           0.1386054,
           0.49716237,
           0.41202563,
           0.41629732,
           0.7688309,
           0.55611837,
           0.4367326
          ],
          [
           0.10383075,
           0.60836315,
           0.57866615,
           0.46367598,
           0.59047234,
           0.5406207,
           0.69353664,
           0.37896192,
           0.45766783,
           0.504663,
           0.63494813,
           0.45693105,
           0.5630742,
           0.6422883,
           0.77443564,
           0.5883587,
           0.42525283,
           0.5335057,
           0.5693738,
           0.11727527,
           0.5326364,
           0.48544335,
           0.48487347,
           0.4999249,
           0.2497467,
           0.9999999,
           0.44861758,
           0.31346098,
           0.50428057,
           0.4512436,
           0.4028868,
           0.27273145,
           0.46674183,
           0.3805704,
           0.028100204,
           0.24621788,
           0.10264244,
           0.6132419,
           0.13999438,
           -0.0007876803,
           0.542238,
           0.545994,
           0.29293904,
           0.64418435,
           0.29375374,
           0.7483556,
           0.3782785,
           0.25814673,
           0.21852812,
           0.61340445,
           0.2531575,
           0.75314355,
           0.61191225,
           0.30341637,
           0.74284166,
           0.5229823,
           0.4412396,
           0.4787023,
           0.4777716,
           0.5777291,
           0.3622744,
           0.68964267,
           0.17047313,
           0.3482976,
           0.28609288,
           0.3343535,
           0.52002776,
           0.54839265,
           0.5803045,
           0.5398401,
           0.2094554,
           0.26513702,
           0.6665116,
           0.29369807,
           0.22425818,
           0.6369157
          ],
          [
           0.045971036,
           0.43149155,
           0.472066,
           0.48451063,
           0.45299366,
           0.7764287,
           0.50221294,
           0.39457053,
           0.28155944,
           0.35598654,
           0.54133254,
           0.51310754,
           0.822883,
           0.45270583,
           0.5427282,
           0.44407195,
           0.48332602,
           0.4880643,
           0.6469999,
           0.10836005,
           0.4532541,
           0.60885084,
           0.64536905,
           0.44085726,
           0.26523304,
           0.44861758,
           0.9999999,
           0.36328706,
           0.501544,
           0.5544702,
           0.27814078,
           0.20895205,
           0.88651407,
           0.2664717,
           -0.025827818,
           0.64744586,
           0.3836156,
           0.5752386,
           0.118280545,
           0.1764237,
           0.52788585,
           0.48145902,
           0.57674456,
           0.6080508,
           0.326568,
           0.4960928,
           0.72670996,
           0.47717714,
           0.119027875,
           0.55715066,
           0.57631177,
           0.51430583,
           0.50406075,
           0.77358055,
           0.59821343,
           0.4439895,
           0.73458123,
           0.402382,
           0.44841444,
           0.47667903,
           0.20963028,
           0.51425815,
           0.39448303,
           0.25643492,
           0.20804788,
           0.26540795,
           0.34155652,
           0.42081052,
           0.42908114,
           0.5358182,
           0.25430998,
           0.52721155,
           0.5550761,
           0.37665433,
           0.12935565,
           0.68862677
          ],
          [
           0.31766886,
           0.44269583,
           0.3814762,
           0.38154972,
           0.3935427,
           0.48361975,
           0.42847916,
           0.7355951,
           0.23417065,
           0.48092055,
           0.51245505,
           0.4922433,
           0.3479465,
           0.48973292,
           0.34980434,
           0.4381666,
           0.21561968,
           0.37322456,
           0.43130356,
           0.31489933,
           0.34805733,
           0.31113678,
           0.47144097,
           0.40839875,
           0.32051313,
           0.31346098,
           0.36328706,
           1,
           0.5062075,
           0.2817528,
           0.31614235,
           0.37688243,
           0.39642406,
           0.33927742,
           0.2544548,
           0.3189332,
           0.20295945,
           0.23167923,
           0.32857186,
           0.25569975,
           0.44773465,
           0.38263837,
           0.14365548,
           0.38354138,
           0.3409118,
           0.41305053,
           0.3676529,
           0.28313205,
           0.5806009,
           0.523512,
           0.3669544,
           0.41960096,
           0.400133,
           0.24253805,
           0.4557429,
           0.3420932,
           0.4331143,
           0.35775518,
           0.3743738,
           0.42663565,
           0.39515632,
           0.36120504,
           0.34279984,
           0.20648077,
           0.38237008,
           0.18952382,
           0.27714372,
           0.5187668,
           0.25108495,
           0.23141006,
           0.26342165,
           0.30235976,
           0.39739168,
           0.24447039,
           0.3089768,
           0.5350443
          ],
          [
           0.110171095,
           0.6793354,
           0.5182632,
           0.6369028,
           0.4510625,
           0.5517028,
           0.61407745,
           0.4262957,
           0.39318722,
           0.49291956,
           0.58311474,
           0.46811038,
           0.55447894,
           0.5414591,
           0.57542837,
           0.4878962,
           0.331896,
           0.4536992,
           0.64043844,
           0.1876654,
           0.5252805,
           0.48416644,
           0.45128447,
           0.4970496,
           0.23884672,
           0.50428057,
           0.501544,
           0.5062075,
           1,
           0.43204665,
           0.38341475,
           0.33059716,
           0.48647654,
           0.5593283,
           0.06342038,
           0.2963872,
           0.13200597,
           0.52140284,
           0.13578609,
           0.29374626,
           0.5989969,
           0.6876081,
           0.30837685,
           0.6158883,
           0.29765543,
           0.5794332,
           0.38541108,
           0.36995694,
           0.28409982,
           0.63596785,
           0.29585633,
           0.5813428,
           0.48629838,
           0.36349922,
           0.58601725,
           0.46834612,
           0.4456405,
           0.5136957,
           0.44327432,
           0.46246493,
           0.3140738,
           0.6816212,
           0.21973759,
           0.34421486,
           0.24683464,
           0.3837988,
           0.43924493,
           0.5954058,
           0.46734327,
           0.5097311,
           0.22602636,
           0.21341144,
           0.6289892,
           0.30518645,
           0.21267605,
           0.721626
          ],
          [
           0.14542629,
           0.45205218,
           0.64444077,
           0.4442507,
           0.54136574,
           0.51738185,
           0.50899,
           0.432329,
           0.48160115,
           0.43637627,
           0.6375329,
           0.4135701,
           0.4707657,
           0.42507,
           0.52220964,
           0.5761593,
           0.7398397,
           0.65584266,
           0.56573975,
           0.17440794,
           0.55444866,
           0.786759,
           0.561307,
           0.6116413,
           0.36072668,
           0.4512436,
           0.5544702,
           0.2817528,
           0.43204665,
           0.9999999,
           0.3871858,
           0.2910762,
           0.48146397,
           0.5619151,
           0.06549145,
           0.30074438,
           0.28480452,
           0.5679414,
           0.21415313,
           0.12028812,
           0.55890155,
           0.4918172,
           0.41672873,
           0.62554693,
           0.38105968,
           0.56497395,
           0.41384697,
           0.6443709,
           0.20020324,
           0.581521,
           0.24840966,
           0.5682118,
           0.6562443,
           0.3440714,
           0.6048533,
           0.5450049,
           0.3776316,
           0.36828715,
           0.6301509,
           0.62761474,
           0.28577903,
           0.40393957,
           0.31937522,
           0.54171467,
           0.3426442,
           0.523544,
           0.5485527,
           0.3709631,
           0.58407956,
           0.39959392,
           0.25795785,
           0.4419979,
           0.48599926,
           0.59585315,
           0.22198631,
           0.6352123
          ],
          [
           0.49683112,
           0.6354992,
           0.5594946,
           0.38046908,
           0.6357767,
           0.48549914,
           0.5918284,
           0.6421176,
           0.57862395,
           0.7106916,
           0.66390204,
           0.66486764,
           0.3078357,
           0.634946,
           0.4947403,
           0.5651019,
           0.364326,
           0.56586516,
           0.6156575,
           0.45842135,
           0.5704098,
           0.43484128,
           0.657694,
           0.53189063,
           0.6473931,
           0.4028868,
           0.27814078,
           0.31614235,
           0.38341475,
           0.3871858,
           0.99999964,
           0.53025556,
           0.25176936,
           0.4534496,
           0.3867254,
           0.1561628,
           0.2537861,
           0.43239534,
           0.41381976,
           0.09453935,
           0.6324442,
           0.4861218,
           0.06325229,
           0.5049834,
           0.7974724,
           0.69272447,
           0.28889567,
           0.38744384,
           0.44788945,
           0.65117705,
           0.20119196,
           0.6392051,
           0.58023953,
           0.19991323,
           0.4943265,
           0.7857237,
           0.3637895,
           0.39616397,
           0.5083456,
           0.65633017,
           0.5694319,
           0.46894637,
           0.42579514,
           0.6321794,
           0.5777785,
           0.33201292,
           0.6974411,
           0.594294,
           0.382561,
           0.27241147,
           0.75757754,
           0.43906105,
           0.55652815,
           0.47372794,
           0.5309708,
           0.49547088
          ],
          [
           0.6596825,
           0.3816826,
           0.3367659,
           0.40709442,
           0.43295753,
           0.30400252,
           0.32751936,
           0.48519802,
           0.24299252,
           0.5642384,
           0.4889573,
           0.35080862,
           0.18996637,
           0.51667964,
           0.30986387,
           0.45937356,
           0.27100968,
           0.45018876,
           0.34956446,
           0.49975178,
           0.32362956,
           0.23010972,
           0.38944274,
           0.39592463,
           0.40959215,
           0.27273145,
           0.20895205,
           0.37688243,
           0.33059716,
           0.2910762,
           0.53025556,
           0.99999964,
           0.21996099,
           0.3144042,
           0.5074952,
           0.10782924,
           0.17010623,
           0.19125184,
           0.49020684,
           0.25703278,
           0.39779824,
           0.30208683,
           -0.003094431,
           0.3141868,
           0.42163557,
           0.4172746,
           0.27261072,
           0.24454221,
           0.549104,
           0.48936796,
           0.17565967,
           0.37773827,
           0.4413991,
           0.112612545,
           0.3509015,
           0.4299584,
           0.29723072,
           0.4705179,
           0.402969,
           0.46729022,
           0.43883404,
           0.26042646,
           0.2841651,
           0.3085335,
           0.51717234,
           0.15182239,
           0.34168327,
           0.49709445,
           0.25956082,
           0.14845365,
           0.44646567,
           0.2790044,
           0.3373701,
           0.2865951,
           0.69627297,
           0.31459355
          ],
          [
           0.04398114,
           0.46684337,
           0.48488057,
           0.5139535,
           0.36331218,
           0.74962234,
           0.56088495,
           0.4117242,
           0.29921573,
           0.35223,
           0.49694854,
           0.60324174,
           0.79868937,
           0.42132896,
           0.5420323,
           0.4370888,
           0.3766077,
           0.4455276,
           0.5945514,
           0.040494725,
           0.3710997,
           0.5258504,
           0.62151414,
           0.41369483,
           0.28162953,
           0.46674183,
           0.88651407,
           0.39642406,
           0.48647654,
           0.48146397,
           0.25176936,
           0.21996099,
           1.0000001,
           0.23171361,
           -0.03403554,
           0.6729171,
           0.35227162,
           0.5606792,
           0.13912043,
           0.1584574,
           0.4899786,
           0.5562118,
           0.45445126,
           0.5143877,
           0.2836018,
           0.51538336,
           0.7992276,
           0.43506938,
           0.17851514,
           0.50028014,
           0.5802207,
           0.48992625,
           0.50845075,
           0.82747746,
           0.58138204,
           0.37337583,
           0.88819706,
           0.43186593,
           0.4116754,
           0.46450832,
           0.18868077,
           0.55250585,
           0.3107288,
           0.20881715,
           0.1515711,
           0.22662318,
           0.31335306,
           0.44435823,
           0.41329226,
           0.50056267,
           0.20200738,
           0.5316307,
           0.5153706,
           0.37068775,
           0.14468488,
           0.6820221
          ],
          [
           0.31635642,
           0.43793917,
           0.507269,
           0.52596205,
           0.39679104,
           0.29940736,
           0.4127518,
           0.37061387,
           0.40191588,
           0.56595397,
           0.53703266,
           0.36411208,
           0.23093322,
           0.36546895,
           0.45151782,
           0.49880898,
           0.3962857,
           0.5249975,
           0.440278,
           0.18565819,
           0.52988493,
           0.53180057,
           0.4159702,
           0.7088943,
           0.41833353,
           0.3805704,
           0.2664717,
           0.33927742,
           0.5593283,
           0.5619151,
           0.4534496,
           0.3144042,
           0.23171361,
           1,
           0.1553535,
           0.0516058,
           0.09705918,
           0.38364327,
           0.28116417,
           0.091758214,
           0.50169265,
           0.48024392,
           0.1624986,
           0.5151665,
           0.34678054,
           0.47574317,
           0.18648328,
           0.45723832,
           0.33623153,
           0.48453078,
           0.1615791,
           0.52205414,
           0.5941908,
           0.09415732,
           0.48500574,
           0.5662199,
           0.22729485,
           0.46063688,
           0.5211443,
           0.57833254,
           0.37719285,
           0.32893914,
           0.17232542,
           0.558689,
           0.39836705,
           0.5938384,
           0.632463,
           0.36298513,
           0.50919646,
           0.24783386,
           0.17008993,
           0.18819784,
           0.41693678,
           0.47004136,
           0.36082557,
           0.5339324
          ],
          [
           0.7167218,
           0.17965631,
           0.08240308,
           0.063744634,
           0.22999819,
           0.07052102,
           0.043535084,
           0.4161122,
           0.08659868,
           0.32997167,
           0.20636967,
           0.09620592,
           0.0011429563,
           0.33348155,
           0.06106789,
           0.12696216,
           0.046109937,
           0.14321338,
           0.2094173,
           0.6530219,
           0.2525279,
           0.043600366,
           0.1660193,
           0.08337465,
           0.32436585,
           0.028100204,
           -0.025827818,
           0.2544548,
           0.06342038,
           0.06549145,
           0.3867254,
           0.5074952,
           -0.03403554,
           0.1553535,
           0.99999964,
           -0.034979127,
           0.037699163,
           -0.011352966,
           0.56535053,
           0.039611407,
           0.17747624,
           0.0033892207,
           -0.1069455,
           0.1304555,
           0.43016368,
           0.11595979,
           0.020196851,
           0.039366845,
           0.5626541,
           0.28589833,
           0.007541804,
           0.11188494,
           0.090780795,
           -0.070819415,
           0.07582425,
           0.17686683,
           0.039447334,
           0.17991467,
           0.10423212,
           0.15095672,
           0.4103737,
           0.072330154,
           0.21799383,
           0.24461481,
           0.4075849,
           -0.008932435,
           0.26454222,
           0.2960385,
           0.0755681,
           0.032204498,
           0.43020368,
           0.10569686,
           0.16084579,
           0.07738396,
           0.49399137,
           0.08131992
          ],
          [
           0.025960436,
           0.29760963,
           0.3484208,
           0.3722489,
           0.248216,
           0.5628555,
           0.30432698,
           0.24786958,
           0.17875074,
           0.22185077,
           0.29471713,
           0.3348417,
           0.61729693,
           0.26021343,
           0.3047034,
           0.23393746,
           0.26075208,
           0.2283988,
           0.38482445,
           0.004894115,
           0.15293953,
           0.30747324,
           0.44281211,
           0.24813472,
           0.11023255,
           0.24621788,
           0.64744586,
           0.3189332,
           0.2963872,
           0.30074438,
           0.1561628,
           0.10782924,
           0.6729171,
           0.0516058,
           -0.034979127,
           1,
           0.22795624,
           0.32726043,
           0.05486635,
           0.095054135,
           0.29357836,
           0.40005672,
           0.3704427,
           0.30233008,
           0.09708142,
           0.28276375,
           0.57142806,
           0.37285724,
           0.058032792,
           0.29649398,
           0.75001496,
           0.26082414,
           0.27623188,
           0.7397666,
           0.3100155,
           0.2503528,
           0.58833146,
           0.28157628,
           0.22408253,
           0.2501675,
           0.104198925,
           0.34266058,
           0.31260985,
           0.0469635,
           0.085063934,
           0.03290043,
           0.11601049,
           0.3242889,
           0.2802437,
           0.38150427,
           0.16594335,
           0.41401857,
           0.27617097,
           0.16778988,
           0.04552248,
           0.4323024
          ],
          [
           0.01182943,
           0.2293008,
           0.23291108,
           -0.03288702,
           0.48402262,
           0.53348017,
           0.3113984,
           0.31227782,
           0.1764372,
           0.16293192,
           0.39975125,
           0.3455925,
           0.2567033,
           0.28233248,
           0.17951426,
           0.44722039,
           0.43928856,
           0.4560598,
           0.3670066,
           0.38026428,
           0.28261185,
           0.26057282,
           0.468726,
           0.26204938,
           0.2798879,
           0.10264244,
           0.3836156,
           0.20295945,
           0.13200597,
           0.28480452,
           0.2537861,
           0.17010623,
           0.35227162,
           0.09705918,
           0.037699163,
           0.22795624,
           0.9999999,
           0.17167333,
           0.102970675,
           0.6730403,
           0.30760744,
           0.20163006,
           -0.022364795,
           0.24095462,
           0.36992007,
           0.34062126,
           0.44376612,
           0.3999657,
           0.11546685,
           0.33255747,
           0.25851554,
           0.24543753,
           0.36048156,
           0.324498,
           0.18087739,
           0.371821,
           0.40950572,
           -0.03166621,
           0.43566954,
           0.37868312,
           0.1469574,
           0.13073856,
           0.6860709,
           0.17374198,
           0.37517864,
           0.14162722,
           0.19750269,
           0.15804774,
           0.14630535,
           0.1412581,
           0.3747959,
           0.8007505,
           0.25860083,
           0.4350878,
           0.051237836,
           0.21449298
          ],
          [
           0.051570304,
           0.5208649,
           0.71151245,
           0.38172644,
           0.48761654,
           0.576864,
           0.58637154,
           0.33322218,
           0.5344621,
           0.39445525,
           0.5157662,
           0.48538888,
           0.658365,
           0.47896883,
           0.5986618,
           0.44001514,
           0.44487345,
           0.49136135,
           0.68667793,
           0.0724799,
           0.60180926,
           0.58787477,
           0.6125717,
           0.4995976,
           0.30419624,
           0.6132419,
           0.5752386,
           0.23167923,
           0.52140284,
           0.5679414,
           0.43239534,
           0.19125184,
           0.5606792,
           0.38364327,
           -0.011352966,
           0.32726043,
           0.17167333,
           1,
           0.12073395,
           -0.026969304,
           0.56106865,
           0.5462963,
           0.36995336,
           0.67339087,
           0.35834122,
           0.6438228,
           0.42258352,
           0.45836467,
           0.061814167,
           0.52656895,
           0.24023825,
           0.64968795,
           0.543889,
           0.47440672,
           0.563599,
           0.48366004,
           0.46898338,
           0.31414258,
           0.47898054,
           0.5603634,
           0.25195962,
           0.6510253,
           0.21127361,
           0.45879287,
           0.21392637,
           0.45188755,
           0.55957794,
           0.4200647,
           0.5536435,
           0.64564013,
           0.2485086,
           0.35735524,
           0.5654141,
           0.42262644,
           0.15342729,
           0.62227553
          ],
          [
           0.6570456,
           0.32251978,
           0.29164693,
           0.27763116,
           0.31264603,
           0.17257017,
           0.23766881,
           0.5495025,
           0.16943464,
           0.4871124,
           0.37469065,
           0.30268133,
           0.07386372,
           0.3538168,
           0.22439095,
           0.3527137,
           0.17902802,
           0.37453872,
           0.21993323,
           0.5432359,
           0.26556757,
           0.20581737,
           0.26012647,
           0.31727844,
           0.5412766,
           0.13999438,
           0.118280545,
           0.32857186,
           0.13578609,
           0.21415313,
           0.41381976,
           0.49020684,
           0.13912043,
           0.28116417,
           0.56535053,
           0.05486635,
           0.102970675,
           0.12073395,
           1,
           0.051044147,
           0.30930316,
           0.17992815,
           -0.010989228,
           0.1587272,
           0.40085715,
           0.3130861,
           0.10974285,
           0.3270247,
           0.5662333,
           0.3351836,
           0.14187154,
           0.285948,
           0.31061965,
           0.04996557,
           0.23694853,
           0.32203376,
           0.19503596,
           0.30237874,
           0.2786774,
           0.39112717,
           0.46385145,
           0.16657388,
           0.28194606,
           0.33404547,
           0.3836077,
           0.123445936,
           0.35570854,
           0.3401605,
           0.21941045,
           0.01434503,
           0.3311509,
           0.18982041,
           0.28608668,
           0.3350593,
           0.57041156,
           0.2561099
          ],
          [
           0.03301058,
           0.13577178,
           0.11998485,
           0.09175721,
           0.32795003,
           0.28129345,
           0.13248432,
           0.25848404,
           0.058416076,
           0.12007581,
           0.28070238,
           0.13978042,
           0.04752286,
           0.21280618,
           0.08406296,
           0.32847518,
           0.26880696,
           0.30551428,
           0.16461924,
           0.3411456,
           0.06397529,
           0.13031389,
           0.16028234,
           0.15662631,
           0.15400402,
           -0.0007876803,
           0.1764237,
           0.25569975,
           0.29374626,
           0.12028812,
           0.09453935,
           0.25703278,
           0.1584574,
           0.091758214,
           0.039611407,
           0.095054135,
           0.6730403,
           -0.026969304,
           0.051044147,
           0.99999976,
           0.1747055,
           0.1287254,
           -0.02135377,
           0.07252748,
           0.10866882,
           0.17179473,
           0.21535581,
           0.24784338,
           0.27858213,
           0.24357674,
           0.22624427,
           0.08082774,
           0.17419791,
           0.119278565,
           0.06569591,
           0.23355687,
           0.22202656,
           0.124253765,
           0.27401483,
           0.18496881,
           0.14092119,
           -0.015668616,
           0.5115397,
           0.025446977,
           0.33685905,
           0.021209074,
           -0.0062870625,
           0.11955565,
           0.06538344,
           0.059029654,
           0.23391873,
           0.46886098,
           0.10798713,
           0.280141,
           0.06363537,
           0.13636301
          ],
          [
           0.25393137,
           0.64716154,
           0.6086157,
           0.45009127,
           0.6764516,
           0.6742226,
           0.64758,
           0.5590937,
           0.55388117,
           0.6836642,
           0.770931,
           0.638253,
           0.50480056,
           0.6870829,
           0.5730623,
           0.6541111,
           0.50417566,
           0.67587173,
           0.7142015,
           0.35749376,
           0.653974,
           0.5381422,
           0.6743808,
           0.6278,
           0.4874248,
           0.542238,
           0.52788585,
           0.44773465,
           0.5989969,
           0.55890155,
           0.6324442,
           0.39779824,
           0.4899786,
           0.50169265,
           0.17747624,
           0.29357836,
           0.30760744,
           0.56106865,
           0.30930316,
           0.1747055,
           0.9999999,
           0.537306,
           0.26309642,
           0.7332587,
           0.53693724,
           0.7259369,
           0.40848082,
           0.47059536,
           0.3542868,
           0.8040923,
           0.31468976,
           0.71107763,
           0.70701444,
           0.36463773,
           0.63840115,
           0.68817234,
           0.5224427,
           0.39005983,
           0.6861232,
           0.69627225,
           0.4944253,
           0.5871638,
           0.41713256,
           0.54887867,
           0.4887079,
           0.487532,
           0.60762846,
           0.6490009,
           0.4691131,
           0.42965505,
           0.48807877,
           0.46814686,
           0.7209524,
           0.5215195,
           0.33594376,
           0.6428611
          ],
          [
           0.122526765,
           0.78244257,
           0.6320803,
           0.6800288,
           0.44046968,
           0.5806762,
           0.7242933,
           0.41400948,
           0.44128454,
           0.56625414,
           0.5703069,
           0.6016524,
           0.61106485,
           0.500029,
           0.5939942,
           0.5943558,
           0.37955308,
           0.5319165,
           0.6079328,
           0.13914026,
           0.42109957,
           0.5298488,
           0.4730587,
           0.62104136,
           0.41902727,
           0.545994,
           0.48145902,
           0.38263837,
           0.6876081,
           0.4918172,
           0.4861218,
           0.30208683,
           0.5562118,
           0.48024392,
           0.0033892207,
           0.40005672,
           0.20163006,
           0.5462963,
           0.17992815,
           0.1287254,
           0.537306,
           1.0000001,
           0.26281223,
           0.5261445,
           0.33378285,
           0.6430838,
           0.5004221,
           0.4823181,
           0.22565621,
           0.5824726,
           0.35765505,
           0.57274175,
           0.6039097,
           0.44424054,
           0.59721863,
           0.5255633,
           0.54108274,
           0.53869784,
           0.46229506,
           0.5196457,
           0.30451334,
           0.7359301,
           0.24911143,
           0.41424695,
           0.26534703,
           0.3338573,
           0.48377943,
           0.5698155,
           0.46432707,
           0.48413214,
           0.23729932,
           0.36630607,
           0.553933,
           0.45760193,
           0.262381,
           0.7252711
          ],
          [
           -0.039714284,
           0.21099447,
           0.3051798,
           0.48545787,
           0.15684663,
           0.32623255,
           0.19332582,
           0.097806014,
           0.11698954,
           0.16069436,
           0.2220411,
           0.14792338,
           0.4462842,
           0.17810848,
           0.32305533,
           0.12634352,
           0.34903407,
           0.17577575,
           0.3112672,
           -0.06303092,
           0.1762813,
           0.5012027,
           0.23054567,
           0.24552606,
           0.025198543,
           0.29293904,
           0.57674456,
           0.14365548,
           0.30837685,
           0.41672873,
           0.06325229,
           -0.003094431,
           0.45445126,
           0.1624986,
           -0.1069455,
           0.3704427,
           -0.022364795,
           0.36995336,
           -0.010989228,
           -0.02135377,
           0.26309642,
           0.26281223,
           1.0000001,
           0.37511647,
           0.01546539,
           0.20641558,
           0.28720582,
           0.23470378,
           -0.012987137,
           0.26844943,
           0.29523113,
           0.25536305,
           0.19337237,
           0.4067212,
           0.4084524,
           0.20236146,
           0.26223353,
           0.36472285,
           0.19955185,
           0.1333093,
           0.039601266,
           0.3159029,
           0.043550976,
           0.118712634,
           0.007253386,
           0.19744653,
           0.12488465,
           0.24053839,
           0.33040723,
           0.43026158,
           0.039834034,
           0.06562365,
           0.26230973,
           0.14688833,
           -0.011307172,
           0.48638573
          ],
          [
           0.1740633,
           0.5797262,
           0.6549555,
           0.45012075,
           0.6960261,
           0.6589807,
           0.6044078,
           0.44656944,
           0.65839124,
           0.61450005,
           0.7373692,
           0.5356242,
           0.6225983,
           0.6899468,
           0.6080717,
           0.5932962,
           0.580469,
           0.59186983,
           0.7956196,
           0.25791895,
           0.8503798,
           0.6553906,
           0.6574838,
           0.6237458,
           0.33176684,
           0.64418435,
           0.6080508,
           0.38354138,
           0.6158883,
           0.62554693,
           0.5049834,
           0.3141868,
           0.5143877,
           0.5151665,
           0.1304555,
           0.30233008,
           0.24095462,
           0.67339087,
           0.1587272,
           0.07252748,
           0.7332587,
           0.5261445,
           0.37511647,
           0.9999999,
           0.41974455,
           0.6785259,
           0.42995018,
           0.42736596,
           0.21502388,
           0.7736366,
           0.27904248,
           0.7775191,
           0.627999,
           0.39916122,
           0.7432597,
           0.62361753,
           0.46080804,
           0.3907107,
           0.6288957,
           0.62759244,
           0.4109401,
           0.63797176,
           0.3144981,
           0.5340538,
           0.41831055,
           0.6370834,
           0.649766,
           0.5951759,
           0.57876456,
           0.64717853,
           0.3470375,
           0.38330257,
           0.73148173,
           0.42412496,
           0.23057483,
           0.68061197
          ],
          [
           0.43265122,
           0.49060893,
           0.44709882,
           0.18760656,
           0.56180096,
           0.4773838,
           0.5239885,
           0.6668359,
           0.46817455,
           0.49978954,
           0.57357705,
           0.62001574,
           0.28560936,
           0.50387895,
           0.40579033,
           0.48035365,
           0.3519144,
           0.5052607,
           0.5685498,
           0.49203748,
           0.5601213,
           0.42601424,
           0.65371704,
           0.37112355,
           0.63021326,
           0.29375374,
           0.326568,
           0.3409118,
           0.29765543,
           0.38105968,
           0.7974724,
           0.42163557,
           0.2836018,
           0.34678054,
           0.43016368,
           0.09708142,
           0.36992007,
           0.35834122,
           0.40085715,
           0.10866882,
           0.53693724,
           0.33378285,
           0.01546539,
           0.41974455,
           1,
           0.58654094,
           0.32694596,
           0.38306397,
           0.4120943,
           0.54883754,
           0.13496253,
           0.5237404,
           0.50002384,
           0.20847143,
           0.42522198,
           0.6161793,
           0.3485278,
           0.19606015,
           0.45293823,
           0.59507364,
           0.4887695,
           0.3392278,
           0.46405697,
           0.604273,
           0.4754004,
           0.2579334,
           0.6594084,
           0.43008906,
           0.32217494,
           0.19976611,
           0.7157248,
           0.5043088,
           0.52677447,
           0.45848352,
           0.45953757,
           0.42807245
          ],
          [
           0.23839131,
           0.7642733,
           0.7243743,
           0.45302218,
           0.7825592,
           0.7073237,
           0.85124874,
           0.63382316,
           0.607015,
           0.66830444,
           0.82819325,
           0.7113774,
           0.54995865,
           0.75807106,
           0.726402,
           0.7874513,
           0.54759395,
           0.7652271,
           0.6899712,
           0.31917003,
           0.64688957,
           0.55773616,
           0.72662574,
           0.602158,
           0.50965524,
           0.7483556,
           0.4960928,
           0.41305053,
           0.5794332,
           0.56497395,
           0.69272447,
           0.4172746,
           0.51538336,
           0.47574317,
           0.11595979,
           0.28276375,
           0.34062126,
           0.6438228,
           0.3130861,
           0.17179473,
           0.7259369,
           0.6430838,
           0.20641558,
           0.6785259,
           0.58654094,
           1.0000001,
           0.4931442,
           0.44770694,
           0.37698472,
           0.7756249,
           0.29597813,
           0.8741669,
           0.7709407,
           0.36971623,
           0.74790573,
           0.7643037,
           0.5863889,
           0.44794837,
           0.67023456,
           0.8327869,
           0.5311013,
           0.7081171,
           0.44398886,
           0.52327853,
           0.49144375,
           0.4302603,
           0.6725752,
           0.6893027,
           0.5219912,
           0.4951768,
           0.47904485,
           0.5320432,
           0.7365668,
           0.53126943,
           0.35267466,
           0.7069427
          ],
          [
           0.100331455,
           0.42163575,
           0.46219036,
           0.37536418,
           0.35786608,
           0.7265222,
           0.53494596,
           0.42025366,
           0.2781164,
           0.3035974,
           0.5322288,
           0.61667943,
           0.63552845,
           0.38595065,
           0.41871062,
           0.49790865,
           0.3461689,
           0.4401756,
           0.5074549,
           0.064265415,
           0.32669836,
           0.41466025,
           0.6139828,
           0.39554954,
           0.2771292,
           0.3782785,
           0.72670996,
           0.3676529,
           0.38541108,
           0.41384697,
           0.28889567,
           0.27261072,
           0.7992276,
           0.18648328,
           0.020196851,
           0.57142806,
           0.44376612,
           0.42258352,
           0.10974285,
           0.21535581,
           0.40848082,
           0.5004221,
           0.28720582,
           0.42995018,
           0.32694596,
           0.4931442,
           1.0000002,
           0.37651685,
           0.22864565,
           0.46988553,
           0.5854268,
           0.4404376,
           0.5208225,
           0.7084881,
           0.5130777,
           0.35864043,
           0.72232974,
           0.35950878,
           0.4327584,
           0.5017439,
           0.14777452,
           0.4195482,
           0.34840536,
           0.18521178,
           0.19409458,
           0.20825782,
           0.29449835,
           0.43657574,
           0.35829574,
           0.34997535,
           0.26561075,
           0.6156856,
           0.39356095,
           0.35165575,
           0.19513932,
           0.5207057
          ],
          [
           0.15686393,
           0.45614567,
           0.6072897,
           0.450158,
           0.4304492,
           0.47562587,
           0.45659918,
           0.35875276,
           0.3991673,
           0.39406526,
           0.54043573,
           0.41594526,
           0.38890845,
           0.2949734,
           0.395707,
           0.5343306,
           0.57962877,
           0.61561704,
           0.4729779,
           0.1612271,
           0.40450618,
           0.6508802,
           0.5158296,
           0.59830153,
           0.54303706,
           0.25814673,
           0.47717714,
           0.28313205,
           0.36995694,
           0.6443709,
           0.38744384,
           0.24454221,
           0.43506938,
           0.45723832,
           0.039366845,
           0.37285724,
           0.3999657,
           0.45836467,
           0.3270247,
           0.24784338,
           0.47059536,
           0.4823181,
           0.23470378,
           0.42736596,
           0.38306397,
           0.44770694,
           0.37651685,
           1.0000001,
           0.146902,
           0.47611427,
           0.347641,
           0.42527294,
           0.54589564,
           0.38133943,
           0.42268908,
           0.53457683,
           0.36933714,
           0.3447149,
           0.51808363,
           0.5335328,
           0.1976138,
           0.33989048,
           0.403162,
           0.46717143,
           0.27529496,
           0.37088764,
           0.47271922,
           0.30775356,
           0.51009727,
           0.31147265,
           0.2243652,
           0.47449192,
           0.38001648,
           0.75485027,
           0.24477811,
           0.52805674
          ],
          [
           0.629904,
           0.36281985,
           0.25876004,
           0.33600155,
           0.3320602,
           0.20150493,
           0.28702196,
           0.7177167,
           0.20305581,
           0.55868495,
           0.42906955,
           0.37949806,
           0.049319923,
           0.49243078,
           0.2745856,
           0.36070365,
           0.16813633,
           0.3863702,
           0.26275063,
           0.47957587,
           0.2520395,
           0.20196125,
           0.29990464,
           0.3422297,
           0.42381144,
           0.21852812,
           0.119027875,
           0.5806009,
           0.28409982,
           0.20020324,
           0.44788945,
           0.549104,
           0.17851514,
           0.33623153,
           0.5626541,
           0.058032792,
           0.11546685,
           0.061814167,
           0.5662333,
           0.27858213,
           0.3542868,
           0.22565621,
           -0.012987137,
           0.21502388,
           0.4120943,
           0.37698472,
           0.22864565,
           0.146902,
           1.0000001,
           0.3927483,
           0.2503361,
           0.29990608,
           0.33723077,
           -0.0035993569,
           0.31324184,
           0.34802508,
           0.25846213,
           0.4376952,
           0.29595983,
           0.428682,
           0.60493517,
           0.16272798,
           0.24271289,
           0.30311674,
           0.46478748,
           0.111997694,
           0.293495,
           0.433792,
           0.1868999,
           0.045531858,
           0.3765611,
           0.1984641,
           0.27703565,
           0.2392836,
           0.55078614,
           0.31513467
          ],
          [
           0.34782946,
           0.71993446,
           0.6446583,
           0.50777996,
           0.77774507,
           0.7462462,
           0.6796168,
           0.6238266,
           0.5458218,
           0.75246394,
           0.893926,
           0.6624867,
           0.5448414,
           0.8017052,
           0.60538137,
           0.7734679,
           0.52318853,
           0.71106637,
           0.75898194,
           0.46124858,
           0.67373633,
           0.5342438,
           0.68058634,
           0.6687726,
           0.5199639,
           0.61340445,
           0.55715066,
           0.523512,
           0.63596785,
           0.581521,
           0.65117705,
           0.48936796,
           0.50028014,
           0.48453078,
           0.28589833,
           0.29649398,
           0.33255747,
           0.52656895,
           0.3351836,
           0.24357674,
           0.8040923,
           0.5824726,
           0.26844943,
           0.7736366,
           0.54883754,
           0.7756249,
           0.46988553,
           0.47611427,
           0.3927483,
           1,
           0.34455398,
           0.77958125,
           0.7286129,
           0.36409527,
           0.7024551,
           0.7291031,
           0.54354143,
           0.45743626,
           0.7033248,
           0.7102967,
           0.52526665,
           0.6454602,
           0.45191193,
           0.5210256,
           0.6052033,
           0.44415662,
           0.58311814,
           0.7636051,
           0.4851138,
           0.47269398,
           0.50818187,
           0.5012045,
           0.7628654,
           0.516321,
           0.38805705,
           0.7054722
          ],
          [
           0.099794686,
           0.3848536,
           0.26411992,
           0.4089388,
           0.24094716,
           0.5275159,
           0.3163221,
           0.33779222,
           0.10408076,
           0.3156681,
           0.3522287,
           0.37625685,
           0.44702974,
           0.34377092,
           0.25058568,
           0.33379474,
           0.19659819,
           0.3227077,
           0.32162112,
           0.10540104,
           0.12984237,
           0.2533151,
           0.4236071,
           0.37604538,
           0.28821677,
           0.2531575,
           0.57631177,
           0.3669544,
           0.29585633,
           0.24840966,
           0.20119196,
           0.17565967,
           0.5802207,
           0.1615791,
           0.007541804,
           0.75001496,
           0.25851554,
           0.24023825,
           0.14187154,
           0.22624427,
           0.31468976,
           0.35765505,
           0.29523113,
           0.27904248,
           0.13496253,
           0.29597813,
           0.5854268,
           0.347641,
           0.2503361,
           0.34455398,
           0.9999999,
           0.28143692,
           0.369128,
           0.541304,
           0.30728847,
           0.3083718,
           0.4970417,
           0.41270608,
           0.2892018,
           0.33761743,
           0.22874227,
           0.27057603,
           0.36484748,
           0.09147529,
           0.198307,
           0.09882512,
           0.08072518,
           0.34410387,
           0.18507054,
           0.23220201,
           0.24780196,
           0.39609146,
           0.27557835,
           0.2579401,
           0.16114359,
           0.43935668
          ],
          [
           0.22482464,
           0.6978781,
           0.7438786,
           0.44650662,
           0.7522172,
           0.6972559,
           0.760568,
           0.5506132,
           0.65130657,
           0.667561,
           0.82663685,
           0.6529417,
           0.5410049,
           0.7302184,
           0.70775867,
           0.7411648,
           0.53619474,
           0.7108095,
           0.7234951,
           0.2636106,
           0.7547256,
           0.5953151,
           0.7144847,
           0.6530897,
           0.44368213,
           0.75314355,
           0.51430583,
           0.41960096,
           0.5813428,
           0.5682118,
           0.6392051,
           0.37773827,
           0.48992625,
           0.52205414,
           0.11188494,
           0.26082414,
           0.24543753,
           0.64968795,
           0.285948,
           0.08082774,
           0.71107763,
           0.57274175,
           0.25536305,
           0.7775191,
           0.5237404,
           0.8741669,
           0.4404376,
           0.42527294,
           0.29990608,
           0.77958125,
           0.28143692,
           0.9999998,
           0.74817723,
           0.34970355,
           0.8075279,
           0.7388724,
           0.52315074,
           0.41045046,
           0.673013,
           0.7808901,
           0.48132396,
           0.7063055,
           0.36443454,
           0.5566448,
           0.49141946,
           0.55737734,
           0.7187109,
           0.69139344,
           0.6140157,
           0.5066604,
           0.4227093,
           0.44437236,
           0.74396354,
           0.47131124,
           0.3240606,
           0.6792282
          ],
          [
           0.24732104,
           0.62491965,
           0.64430845,
           0.5105183,
           0.63394725,
           0.6404181,
           0.70102787,
           0.553038,
           0.4995614,
           0.6637057,
           0.8168747,
           0.62498295,
           0.45095977,
           0.57074094,
           0.58939826,
           0.8700013,
           0.5513275,
           0.80044675,
           0.5746496,
           0.23562354,
           0.57522833,
           0.55738914,
           0.66454506,
           0.74873626,
           0.54064184,
           0.61191225,
           0.50406075,
           0.400133,
           0.48629838,
           0.6562443,
           0.58023953,
           0.4413991,
           0.50845075,
           0.5941908,
           0.090780795,
           0.27623188,
           0.36048156,
           0.543889,
           0.31061965,
           0.17419791,
           0.70701444,
           0.6039097,
           0.19337237,
           0.627999,
           0.50002384,
           0.7709407,
           0.5208225,
           0.54589564,
           0.33723077,
           0.7286129,
           0.369128,
           0.74817723,
           0.99999976,
           0.3394648,
           0.6767551,
           0.7120262,
           0.5289586,
           0.47022,
           0.8571117,
           0.8541095,
           0.40121552,
           0.49022704,
           0.41417193,
           0.59604514,
           0.46808705,
           0.5453603,
           0.61301094,
           0.52686375,
           0.4691313,
           0.30371258,
           0.37167853,
           0.5372644,
           0.6193569,
           0.6409944,
           0.39548382,
           0.62588084
          ],
          [
           -0.028879616,
           0.3363461,
           0.39128238,
           0.35824555,
           0.2988979,
           0.7021847,
           0.39594126,
           0.25153688,
           0.22873831,
           0.2163308,
           0.34569287,
           0.46318576,
           0.78187954,
           0.28523958,
           0.38784117,
           0.2919079,
           0.3010569,
           0.29583633,
           0.51008487,
           -0.023876823,
           0.25904292,
           0.3892682,
           0.5715568,
           0.26056665,
           0.14537472,
           0.30341637,
           0.77358055,
           0.24253805,
           0.36349922,
           0.3440714,
           0.19991323,
           0.112612545,
           0.82747746,
           0.09415732,
           -0.070819415,
           0.7397666,
           0.324498,
           0.47440672,
           0.04996557,
           0.119278565,
           0.36463773,
           0.44424054,
           0.4067212,
           0.39916122,
           0.20847143,
           0.36971623,
           0.7084881,
           0.38133943,
           -0.0035993569,
           0.36409527,
           0.541304,
           0.34970355,
           0.3394648,
           0.9999999,
           0.40147653,
           0.32024866,
           0.7432883,
           0.23960097,
           0.28523645,
           0.33342886,
           0.08138111,
           0.47513562,
           0.3003286,
           0.07795922,
           0.07208508,
           0.12746122,
           0.20658144,
           0.34815693,
           0.32932308,
           0.46954596,
           0.19778197,
           0.4984269,
           0.3764382,
           0.21176751,
           0.039357092,
           0.5104016
          ],
          [
           0.19150223,
           0.6188692,
           0.6431493,
           0.5403272,
           0.5899894,
           0.63561165,
           0.71570325,
           0.5225654,
           0.539604,
           0.570647,
           0.7875968,
           0.60449666,
           0.59760535,
           0.587404,
           0.7564469,
           0.6779439,
           0.5595716,
           0.63166046,
           0.6440805,
           0.16378051,
           0.59939486,
           0.7036215,
           0.588038,
           0.59992,
           0.36618948,
           0.74284166,
           0.59821343,
           0.4557429,
           0.58601725,
           0.6048533,
           0.4943265,
           0.3509015,
           0.58138204,
           0.48500574,
           0.07582425,
           0.3100155,
           0.18087739,
           0.563599,
           0.23694853,
           0.06569591,
           0.63840115,
           0.59721863,
           0.4084524,
           0.7432597,
           0.42522198,
           0.74790573,
           0.5130777,
           0.42268908,
           0.31324184,
           0.7024551,
           0.30728847,
           0.8075279,
           0.6767551,
           0.40147653,
           0.99999976,
           0.5803075,
           0.5239241,
           0.49502787,
           0.5310403,
           0.65419745,
           0.3835792,
           0.653286,
           0.26292002,
           0.45343238,
           0.33409145,
           0.43573886,
           0.5697858,
           0.5815146,
           0.6383776,
           0.4830647,
           0.3051764,
           0.38464677,
           0.67791784,
           0.43766475,
           0.269589,
           0.7394287
          ],
          [
           0.3148466,
           0.6473645,
           0.64217156,
           0.44440007,
           0.7493782,
           0.624308,
           0.62292796,
           0.59395397,
           0.55069333,
           0.6996659,
           0.75599223,
           0.5852476,
           0.42505634,
           0.6809652,
           0.54221743,
           0.6905884,
           0.53825796,
           0.72036016,
           0.6464849,
           0.40239328,
           0.58818996,
           0.5447324,
           0.706298,
           0.6741071,
           0.5701467,
           0.5229823,
           0.4439895,
           0.3420932,
           0.46834612,
           0.5450049,
           0.7857237,
           0.4299584,
           0.37337583,
           0.5662199,
           0.17686683,
           0.2503528,
           0.371821,
           0.48366004,
           0.32203376,
           0.23355687,
           0.68817234,
           0.5255633,
           0.20236146,
           0.62361753,
           0.6161793,
           0.7643037,
           0.35864043,
           0.53457683,
           0.34802508,
           0.7291031,
           0.3083718,
           0.7388724,
           0.7120262,
           0.32024866,
           0.5803075,
           0.9999999,
           0.44788522,
           0.4010762,
           0.65368056,
           0.75035673,
           0.53724086,
           0.5088407,
           0.53802717,
           0.59413517,
           0.57941175,
           0.4342885,
           0.6637527,
           0.6199947,
           0.45874152,
           0.3632873,
           0.5765928,
           0.52168036,
           0.6156113,
           0.5525853,
           0.38564792,
           0.6119303
          ],
          [
           0.09521504,
           0.51476365,
           0.44447201,
           0.44801867,
           0.46069404,
           0.7515613,
           0.5842577,
           0.49436513,
           0.28744438,
           0.43988317,
           0.5366349,
           0.7007328,
           0.6675934,
           0.51008195,
           0.4840384,
           0.49167436,
           0.3313259,
           0.47815686,
           0.5666832,
           0.13958982,
           0.38644776,
           0.377528,
           0.6601646,
           0.4023514,
           0.34172252,
           0.4412396,
           0.73458123,
           0.4331143,
           0.4456405,
           0.3776316,
           0.3637895,
           0.29723072,
           0.88819706,
           0.22729485,
           0.039447334,
           0.58833146,
           0.40950572,
           0.46898338,
           0.19503596,
           0.22202656,
           0.5224427,
           0.54108274,
           0.26223353,
           0.46080804,
           0.3485278,
           0.5863889,
           0.72232974,
           0.36933714,
           0.25846213,
           0.54354143,
           0.4970417,
           0.52315074,
           0.5289586,
           0.7432883,
           0.5239241,
           0.44788522,
           1,
           0.37295842,
           0.43805683,
           0.51032746,
           0.28276694,
           0.5504991,
           0.3812046,
           0.20715283,
           0.2651375,
           0.2004317,
           0.3410722,
           0.52577734,
           0.3068862,
           0.43116468,
           0.31546462,
           0.6074764,
           0.5173289,
           0.3649522,
           0.1708761,
           0.5922406
          ],
          [
           0.40465862,
           0.53898585,
           0.40657943,
           0.84397006,
           0.27099377,
           0.30132908,
           0.42541456,
           0.3499304,
           0.2373264,
           0.6035587,
           0.45802158,
           0.30819505,
           0.33116388,
           0.49947256,
           0.4512871,
           0.4295028,
           0.25070474,
           0.38060567,
           0.36900148,
           0.19819245,
           0.27259815,
           0.40512836,
           0.2883716,
           0.5163574,
           0.40418422,
           0.4787023,
           0.402382,
           0.35775518,
           0.5136957,
           0.36828715,
           0.39616397,
           0.4705179,
           0.43186593,
           0.46063688,
           0.17991467,
           0.28157628,
           -0.03166621,
           0.31414258,
           0.30237874,
           0.124253765,
           0.39005983,
           0.53869784,
           0.36472285,
           0.3907107,
           0.19606015,
           0.44794837,
           0.35950878,
           0.3447149,
           0.4376952,
           0.45743626,
           0.41270608,
           0.41045046,
           0.47022,
           0.23960097,
           0.49502787,
           0.4010762,
           0.37295842,
           0.99999964,
           0.33968285,
           0.39463586,
           0.3608119,
           0.42783964,
           0.09955975,
           0.32642904,
           0.28469813,
           0.23606053,
           0.3428784,
           0.5031973,
           0.3383988,
           0.31951833,
           0.22477663,
           0.06504501,
           0.39383444,
           0.35796058,
           0.5109043,
           0.5875238
          ],
          [
           0.22151735,
           0.51634777,
           0.6110548,
           0.38898024,
           0.6807524,
           0.59382224,
           0.55829334,
           0.5034174,
           0.49824113,
           0.605339,
           0.7437234,
           0.53672975,
           0.3424797,
           0.57188296,
           0.4677229,
           0.7793511,
           0.6166614,
           0.7415786,
           0.5522243,
           0.2966645,
           0.63447964,
           0.54326713,
           0.6278991,
           0.6587463,
           0.46839285,
           0.4777716,
           0.44841444,
           0.3743738,
           0.44327432,
           0.6301509,
           0.5083456,
           0.402969,
           0.4116754,
           0.5211443,
           0.10423212,
           0.22408253,
           0.43566954,
           0.47898054,
           0.2786774,
           0.27401483,
           0.6861232,
           0.46229506,
           0.19955185,
           0.6288957,
           0.45293823,
           0.67023456,
           0.4327584,
           0.51808363,
           0.29595983,
           0.7033248,
           0.2892018,
           0.673013,
           0.8571117,
           0.28523645,
           0.5310403,
           0.65368056,
           0.43805683,
           0.33968285,
           1.0000001,
           0.7557454,
           0.3724576,
           0.39539975,
           0.4418686,
           0.5110794,
           0.5097699,
           0.5826159,
           0.56264615,
           0.5279654,
           0.47627026,
           0.29095864,
           0.36273855,
           0.5617432,
           0.58697325,
           0.62244475,
           0.34355134,
           0.5275327
          ],
          [
           0.32058775,
           0.5971838,
           0.69370204,
           0.39114416,
           0.6959624,
           0.6154038,
           0.68133867,
           0.64730453,
           0.54822373,
           0.6413144,
           0.8191596,
           0.6316362,
           0.4175244,
           0.6278178,
           0.59450465,
           0.81610155,
           0.5623583,
           0.8413718,
           0.57557464,
           0.31884557,
           0.6232996,
           0.55797905,
           0.7265377,
           0.6887499,
           0.579103,
           0.5777291,
           0.47667903,
           0.42663565,
           0.46246493,
           0.62761474,
           0.65633017,
           0.46729022,
           0.46450832,
           0.57833254,
           0.15095672,
           0.2501675,
           0.37868312,
           0.5603634,
           0.39112717,
           0.18496881,
           0.69627225,
           0.5196457,
           0.1333093,
           0.62759244,
           0.59507364,
           0.8327869,
           0.5017439,
           0.5335328,
           0.428682,
           0.7102967,
           0.33761743,
           0.7808901,
           0.8541095,
           0.33342886,
           0.65419745,
           0.75035673,
           0.51032746,
           0.39463586,
           0.7557454,
           0.9999999,
           0.4807404,
           0.47292113,
           0.47442383,
           0.6060597,
           0.50684917,
           0.58955216,
           0.7019075,
           0.5455693,
           0.51483953,
           0.3106541,
           0.47038066,
           0.5632713,
           0.59496135,
           0.6137836,
           0.4405852,
           0.58985126
          ],
          [
           0.5725105,
           0.518215,
           0.35976762,
           0.3453254,
           0.5722574,
           0.35371113,
           0.40436217,
           0.5972512,
           0.40233302,
           0.726918,
           0.49736315,
           0.44935286,
           0.20369807,
           0.6448718,
           0.3688674,
           0.4089477,
           0.28734154,
           0.4473945,
           0.45869488,
           0.55278254,
           0.44232792,
           0.3020404,
           0.42664117,
           0.41334468,
           0.44800964,
           0.3622744,
           0.20963028,
           0.39515632,
           0.3140738,
           0.28577903,
           0.5694319,
           0.43883404,
           0.18868077,
           0.37719285,
           0.4103737,
           0.104198925,
           0.1469574,
           0.25195962,
           0.46385145,
           0.14092119,
           0.4944253,
           0.30451334,
           0.039601266,
           0.4109401,
           0.4887695,
           0.5311013,
           0.14777452,
           0.1976138,
           0.60493517,
           0.52526665,
           0.22874227,
           0.48132396,
           0.40121552,
           0.08138111,
           0.3835792,
           0.53724086,
           0.28276694,
           0.3608119,
           0.3724576,
           0.4807404,
           1.0000002,
           0.35546118,
           0.36933553,
           0.36963642,
           0.57966125,
           0.19375247,
           0.45181316,
           0.5430747,
           0.25072536,
           0.21106684,
           0.44585013,
           0.26955622,
           0.49001008,
           0.27174127,
           0.47654274,
           0.40058595
          ],
          [
           0.1124645,
           0.82236946,
           0.59731805,
           0.53021574,
           0.5556972,
           0.67384815,
           0.77748185,
           0.40107954,
           0.45745003,
           0.53681195,
           0.5778008,
           0.5701931,
           0.7118881,
           0.65362287,
           0.6670327,
           0.5044414,
           0.34232524,
           0.47015524,
           0.7007773,
           0.17168361,
           0.542879,
           0.4776913,
           0.5324427,
           0.48376548,
           0.29631013,
           0.68964267,
           0.51425815,
           0.36120504,
           0.6816212,
           0.40393957,
           0.46894637,
           0.26042646,
           0.55250585,
           0.32893914,
           0.072330154,
           0.34266058,
           0.13073856,
           0.6510253,
           0.16657388,
           -0.015668616,
           0.5871638,
           0.7359301,
           0.3159029,
           0.63797176,
           0.3392278,
           0.7081171,
           0.4195482,
           0.33989048,
           0.16272798,
           0.6454602,
           0.27057603,
           0.7063055,
           0.49022704,
           0.47513562,
           0.653286,
           0.5088407,
           0.5504991,
           0.42783964,
           0.39539975,
           0.47292113,
           0.35546118,
           0.99999994,
           0.23230273,
           0.35029587,
           0.27738917,
           0.30952108,
           0.50502616,
           0.71541506,
           0.4873348,
           0.6318215,
           0.29332513,
           0.30120146,
           0.70962334,
           0.29191917,
           0.21219528,
           0.7531352
          ],
          [
           0.16856498,
           0.42835462,
           0.30762172,
           0.084305674,
           0.61132514,
           0.57644314,
           0.3848377,
           0.50912786,
           0.23688217,
           0.35096908,
           0.48424906,
           0.36660254,
           0.31113243,
           0.47256804,
           0.2673148,
           0.51230335,
           0.3631689,
           0.5417839,
           0.42511034,
           0.5764308,
           0.3033529,
           0.2527669,
           0.529495,
           0.34003013,
           0.35082662,
           0.17047313,
           0.39448303,
           0.34279984,
           0.21973759,
           0.31937522,
           0.42579514,
           0.2841651,
           0.3107288,
           0.17232542,
           0.21799383,
           0.31260985,
           0.6860709,
           0.21127361,
           0.28194606,
           0.5115397,
           0.41713256,
           0.24911143,
           0.043550976,
           0.3144981,
           0.46405697,
           0.44398886,
           0.34840536,
           0.403162,
           0.24271289,
           0.45191193,
           0.36484748,
           0.36443454,
           0.41417193,
           0.3003286,
           0.26292002,
           0.53802717,
           0.3812046,
           0.09955975,
           0.4418686,
           0.47442383,
           0.36933553,
           0.23230273,
           1.0000004,
           0.24636467,
           0.453777,
           0.18181708,
           0.3202879,
           0.35320264,
           0.18837552,
           0.1776154,
           0.6135709,
           0.68337446,
           0.35711074,
           0.36458445,
           0.110935465,
           0.3317477
          ],
          [
           0.3388145,
           0.4771763,
           0.5375301,
           0.34793484,
           0.47451368,
           0.34262055,
           0.46206087,
           0.43271834,
           0.53188956,
           0.5188186,
           0.5660745,
           0.41964728,
           0.22951262,
           0.41223884,
           0.40747613,
           0.50637746,
           0.44953462,
           0.5065243,
           0.49223727,
           0.27699414,
           0.60554993,
           0.5463517,
           0.48219338,
           0.5839486,
           0.5647901,
           0.3482976,
           0.25643492,
           0.20648077,
           0.34421486,
           0.54171467,
           0.6321794,
           0.3085335,
           0.20881715,
           0.558689,
           0.24461481,
           0.0469635,
           0.17374198,
           0.45879287,
           0.33404547,
           0.025446977,
           0.54887867,
           0.41424695,
           0.118712634,
           0.5340538,
           0.604273,
           0.52327853,
           0.18521178,
           0.46717143,
           0.30311674,
           0.5210256,
           0.09147529,
           0.5566448,
           0.59604514,
           0.07795922,
           0.45343238,
           0.59413517,
           0.20715283,
           0.32642904,
           0.5110794,
           0.6060597,
           0.36963642,
           0.35029587,
           0.24636467,
           1,
           0.41090626,
           0.6029116,
           0.72719014,
           0.3591796,
           0.43648812,
           0.22193575,
           0.4400743,
           0.300355,
           0.45200968,
           0.5972856,
           0.41009384,
           0.43259948
          ],
          [
           0.54143524,
           0.43102294,
           0.4085385,
           0.26599735,
           0.6770265,
           0.45396864,
           0.3717106,
           0.5126723,
           0.35816103,
           0.6631793,
           0.5719954,
           0.39466536,
           0.17410634,
           0.6524055,
           0.28111014,
           0.54434264,
           0.39394248,
           0.54173875,
           0.4597832,
           0.6162199,
           0.48631442,
           0.30065787,
           0.45818275,
           0.48998183,
           0.46370953,
           0.28609288,
           0.20804788,
           0.38237008,
           0.24683464,
           0.3426442,
           0.5777785,
           0.51717234,
           0.1515711,
           0.39836705,
           0.4075849,
           0.085063934,
           0.37517864,
           0.21392637,
           0.3836077,
           0.33685905,
           0.4887079,
           0.26534703,
           0.007253386,
           0.41831055,
           0.4754004,
           0.49144375,
           0.19409458,
           0.27529496,
           0.46478748,
           0.6052033,
           0.198307,
           0.49141946,
           0.46808705,
           0.07208508,
           0.33409145,
           0.57941175,
           0.2651375,
           0.28469813,
           0.5097699,
           0.50684917,
           0.57966125,
           0.27738917,
           0.453777,
           0.41090626,
           0.9999997,
           0.24742833,
           0.44525802,
           0.51473117,
           0.2465164,
           0.21106039,
           0.53408104,
           0.45781446,
           0.42995203,
           0.36092734,
           0.452132,
           0.33781207
          ],
          [
           0.10730664,
           0.3375904,
           0.5062263,
           0.2680086,
           0.40536982,
           0.33510888,
           0.36559486,
           0.20375454,
           0.5299243,
           0.3928072,
           0.5258739,
           0.32696578,
           0.19280143,
           0.35343757,
           0.27652937,
           0.47354093,
           0.4473778,
           0.47774968,
           0.41974515,
           0.096346945,
           0.6432147,
           0.4908169,
           0.41247374,
           0.58723915,
           0.25894743,
           0.3343535,
           0.26540795,
           0.18952382,
           0.3837988,
           0.523544,
           0.33201292,
           0.15182239,
           0.22662318,
           0.5938384,
           -0.008932435,
           0.03290043,
           0.14162722,
           0.45188755,
           0.123445936,
           0.021209074,
           0.487532,
           0.3338573,
           0.19744653,
           0.6370834,
           0.2579334,
           0.4302603,
           0.20825782,
           0.37088764,
           0.111997694,
           0.44415662,
           0.09882512,
           0.55737734,
           0.5453603,
           0.12746122,
           0.43573886,
           0.4342885,
           0.2004317,
           0.23606053,
           0.5826159,
           0.58955216,
           0.19375247,
           0.30952108,
           0.18181708,
           0.6029116,
           0.24742833,
           1,
           0.6002382,
           0.31244227,
           0.45124444,
           0.24442463,
           0.1875605,
           0.23823088,
           0.39373004,
           0.4180625,
           0.15911904,
           0.3748598
          ],
          [
           0.3417073,
           0.5349815,
           0.69849753,
           0.3463261,
           0.6272852,
           0.44793463,
           0.5969498,
           0.47938967,
           0.68272483,
           0.57916665,
           0.64162666,
           0.5325008,
           0.37357092,
           0.52536356,
           0.54941124,
           0.5695826,
           0.46671748,
           0.5623563,
           0.58297575,
           0.3164358,
           0.79655915,
           0.62793934,
           0.61727035,
           0.5828959,
           0.47462553,
           0.52002776,
           0.34155652,
           0.27714372,
           0.43924493,
           0.5485527,
           0.6974411,
           0.34168327,
           0.31335306,
           0.632463,
           0.26454222,
           0.11601049,
           0.19750269,
           0.55957794,
           0.35570854,
           -0.0062870625,
           0.60762846,
           0.48377943,
           0.12488465,
           0.649766,
           0.6594084,
           0.6725752,
           0.29449835,
           0.47271922,
           0.293495,
           0.58311814,
           0.08072518,
           0.7187109,
           0.61301094,
           0.20658144,
           0.5697858,
           0.6637527,
           0.3410722,
           0.3428784,
           0.56264615,
           0.7019075,
           0.45181316,
           0.50502616,
           0.3202879,
           0.72719014,
           0.44525802,
           0.6002382,
           1,
           0.4692121,
           0.5969676,
           0.39794922,
           0.4660904,
           0.36268508,
           0.59617615,
           0.46870667,
           0.39736164,
           0.5466872
          ],
          [
           0.39038604,
           0.7597969,
           0.5374627,
           0.58029544,
           0.6375905,
           0.67394936,
           0.652513,
           0.5652318,
           0.4456621,
           0.7533764,
           0.6869068,
           0.63247585,
           0.46445298,
           0.809203,
           0.48712263,
           0.5705027,
           0.33985674,
           0.5271426,
           0.63959795,
           0.39883292,
           0.5249596,
           0.37024182,
           0.54684556,
           0.5642381,
           0.38829917,
           0.54839265,
           0.42081052,
           0.5187668,
           0.5954058,
           0.3709631,
           0.594294,
           0.49709445,
           0.44435823,
           0.36298513,
           0.2960385,
           0.3242889,
           0.15804774,
           0.4200647,
           0.3401605,
           0.11955565,
           0.6490009,
           0.5698155,
           0.24053839,
           0.5951759,
           0.43008906,
           0.6893027,
           0.43657574,
           0.30775356,
           0.433792,
           0.7636051,
           0.34410387,
           0.69139344,
           0.52686375,
           0.34815693,
           0.5815146,
           0.6199947,
           0.52577734,
           0.5031973,
           0.5279654,
           0.5455693,
           0.5430747,
           0.71541506,
           0.35320264,
           0.3591796,
           0.51473117,
           0.31244227,
           0.4692121,
           1.0000002,
           0.39456505,
           0.44246635,
           0.43147328,
           0.34253561,
           0.6361113,
           0.32268196,
           0.40707183,
           0.63355494
          ],
          [
           0.17601353,
           0.3975039,
           0.7208799,
           0.38839763,
           0.48402858,
           0.43426305,
           0.48233593,
           0.34932783,
           0.5299238,
           0.39515662,
           0.54255664,
           0.4042943,
           0.4719981,
           0.3728461,
           0.75399184,
           0.46217197,
           0.5127716,
           0.45424116,
           0.54133344,
           0.1394852,
           0.5760839,
           0.7032993,
           0.48598155,
           0.4970188,
           0.2864183,
           0.5803045,
           0.42908114,
           0.25108495,
           0.46734327,
           0.58407956,
           0.382561,
           0.25956082,
           0.41329226,
           0.50919646,
           0.0755681,
           0.2802437,
           0.14630535,
           0.5536435,
           0.21941045,
           0.06538344,
           0.4691131,
           0.46432707,
           0.33040723,
           0.57876456,
           0.32217494,
           0.5219912,
           0.35829574,
           0.51009727,
           0.1868999,
           0.4851138,
           0.18507054,
           0.6140157,
           0.4691313,
           0.32932308,
           0.6383776,
           0.45874152,
           0.3068862,
           0.3383988,
           0.47627026,
           0.51483953,
           0.25072536,
           0.4873348,
           0.18837552,
           0.43648812,
           0.2465164,
           0.45124444,
           0.5969676,
           0.39456505,
           0.9999999,
           0.44010442,
           0.22289152,
           0.30488232,
           0.5283457,
           0.41175652,
           0.2270003,
           0.5821589
          ],
          [
           0.015973233,
           0.4476083,
           0.54998446,
           0.41029453,
           0.49741793,
           0.5437615,
           0.46658343,
           0.24056144,
           0.46890217,
           0.36389625,
           0.37515286,
           0.36185256,
           0.7006599,
           0.5033655,
           0.519655,
           0.2909247,
           0.4030963,
           0.3167013,
           0.65659356,
           0.07455467,
           0.5051898,
           0.5052161,
           0.4684855,
           0.33940497,
           0.1386054,
           0.5398401,
           0.5358182,
           0.23141006,
           0.5097311,
           0.39959392,
           0.27241147,
           0.14845365,
           0.50056267,
           0.24783386,
           0.032204498,
           0.38150427,
           0.1412581,
           0.64564013,
           0.01434503,
           0.059029654,
           0.42965505,
           0.48413214,
           0.43026158,
           0.64717853,
           0.19976611,
           0.4951768,
           0.34997535,
           0.31147265,
           0.045531858,
           0.47269398,
           0.23220201,
           0.5066604,
           0.30371258,
           0.46954596,
           0.4830647,
           0.3632873,
           0.43116468,
           0.31951833,
           0.29095864,
           0.3106541,
           0.21106684,
           0.6318215,
           0.1776154,
           0.22193575,
           0.21106039,
           0.24442463,
           0.39794922,
           0.44246635,
           0.44010442,
           1,
           0.15062048,
           0.2678622,
           0.58671707,
           0.2340923,
           0.08852976,
           0.588916
          ],
          [
           0.41458988,
           0.48034602,
           0.3177487,
           0.14028722,
           0.53682894,
           0.44027832,
           0.40186167,
           0.56044805,
           0.3148841,
           0.4617615,
           0.4839296,
           0.46613458,
           0.22403322,
           0.5368296,
           0.32012728,
           0.40427572,
           0.24518456,
           0.42998022,
           0.48023134,
           0.61632407,
           0.42634517,
           0.23767167,
           0.5646697,
           0.29846108,
           0.49716237,
           0.2094554,
           0.25430998,
           0.26342165,
           0.22602636,
           0.25795785,
           0.75757754,
           0.44646567,
           0.20200738,
           0.17008993,
           0.43020368,
           0.16594335,
           0.3747959,
           0.2485086,
           0.3311509,
           0.23391873,
           0.48807877,
           0.23729932,
           0.039834034,
           0.3470375,
           0.7157248,
           0.47904485,
           0.26561075,
           0.2243652,
           0.3765611,
           0.50818187,
           0.24780196,
           0.4227093,
           0.37167853,
           0.19778197,
           0.3051764,
           0.5765928,
           0.31546462,
           0.22477663,
           0.36273855,
           0.47038066,
           0.44585013,
           0.29332513,
           0.6135709,
           0.4400743,
           0.53408104,
           0.1875605,
           0.4660904,
           0.43147328,
           0.22289152,
           0.15062048,
           0.9999999,
           0.48473835,
           0.42949742,
           0.29113713,
           0.3572982,
           0.31180063
          ],
          [
           0.10086088,
           0.39876753,
           0.41845256,
           0.11036882,
           0.62247944,
           0.72730744,
           0.50203145,
           0.504418,
           0.34169728,
           0.3266872,
           0.5758313,
           0.5672029,
           0.45718265,
           0.43256947,
           0.3582857,
           0.6000842,
           0.59126925,
           0.60768265,
           0.505375,
           0.3552805,
           0.38671607,
           0.4009906,
           0.69857806,
           0.39100075,
           0.41202563,
           0.26513702,
           0.52721155,
           0.30235976,
           0.21341144,
           0.4419979,
           0.43906105,
           0.2790044,
           0.5316307,
           0.18819784,
           0.10569686,
           0.41401857,
           0.8007505,
           0.35735524,
           0.18982041,
           0.46886098,
           0.46814686,
           0.36630607,
           0.06562365,
           0.38330257,
           0.5043088,
           0.5320432,
           0.6156856,
           0.47449192,
           0.1984641,
           0.5012045,
           0.39609146,
           0.44437236,
           0.5372644,
           0.4984269,
           0.38464677,
           0.52168036,
           0.6074764,
           0.06504501,
           0.5617432,
           0.5632713,
           0.26955622,
           0.30120146,
           0.68337446,
           0.300355,
           0.45781446,
           0.23823088,
           0.36268508,
           0.34253561,
           0.30488232,
           0.2678622,
           0.48473835,
           0.99999976,
           0.4040168,
           0.54820395,
           0.14611858,
           0.39053687
          ],
          [
           0.21978997,
           0.6765905,
           0.57373345,
           0.45366684,
           0.70528305,
           0.675863,
           0.6657206,
           0.53998995,
           0.5310446,
           0.6353307,
           0.70793015,
           0.60164595,
           0.6342977,
           0.6955608,
           0.69433796,
           0.61848766,
           0.45143864,
           0.5889057,
           0.75789344,
           0.3393423,
           0.6908771,
           0.533132,
           0.6194102,
           0.5366581,
           0.41629732,
           0.6665116,
           0.5550761,
           0.39739168,
           0.6289892,
           0.48599926,
           0.55652815,
           0.3373701,
           0.5153706,
           0.41693678,
           0.16084579,
           0.27617097,
           0.25860083,
           0.5654141,
           0.28608668,
           0.10798713,
           0.7209524,
           0.553933,
           0.26230973,
           0.73148173,
           0.52677447,
           0.7365668,
           0.39356095,
           0.38001648,
           0.27703565,
           0.7628654,
           0.27557835,
           0.74396354,
           0.6193569,
           0.3764382,
           0.67791784,
           0.6156113,
           0.5173289,
           0.39383444,
           0.58697325,
           0.59496135,
           0.49001008,
           0.70962334,
           0.35711074,
           0.45200968,
           0.42995203,
           0.39373004,
           0.59617615,
           0.6361113,
           0.5283457,
           0.58671707,
           0.42949742,
           0.4040168,
           1,
           0.39422664,
           0.30891287,
           0.7170274
          ],
          [
           0.20896237,
           0.47350827,
           0.5326069,
           0.41248342,
           0.47125036,
           0.43486795,
           0.49402738,
           0.43761986,
           0.40281606,
           0.46320087,
           0.58102703,
           0.47492856,
           0.2674052,
           0.376272,
           0.41245887,
           0.6081903,
           0.62714887,
           0.67751455,
           0.45073217,
           0.20465314,
           0.420116,
           0.5835444,
           0.5051565,
           0.5960351,
           0.7688309,
           0.29369807,
           0.37665433,
           0.24447039,
           0.30518645,
           0.59585315,
           0.47372794,
           0.2865951,
           0.37068775,
           0.47004136,
           0.07738396,
           0.16778988,
           0.4350878,
           0.42262644,
           0.3350593,
           0.280141,
           0.5215195,
           0.45760193,
           0.14688833,
           0.42412496,
           0.45848352,
           0.53126943,
           0.35165575,
           0.75485027,
           0.2392836,
           0.516321,
           0.2579401,
           0.47131124,
           0.6409944,
           0.21176751,
           0.43766475,
           0.5525853,
           0.3649522,
           0.35796058,
           0.62244475,
           0.6137836,
           0.27174127,
           0.29191917,
           0.36458445,
           0.5972856,
           0.36092734,
           0.4180625,
           0.46870667,
           0.32268196,
           0.41175652,
           0.2340923,
           0.29113713,
           0.54820395,
           0.39422664,
           0.99999976,
           0.3123006,
           0.4853195
          ],
          [
           0.80309767,
           0.34191045,
           0.3364722,
           0.41339225,
           0.2727454,
           0.1630773,
           0.3141365,
           0.48031706,
           0.26111823,
           0.6177729,
           0.3957225,
           0.30311173,
           0.1011242,
           0.37213257,
           0.27874053,
           0.37131685,
           0.16148537,
           0.315341,
           0.29116738,
           0.4019571,
           0.2978469,
           0.21523044,
           0.3454734,
           0.36881757,
           0.55611837,
           0.22425818,
           0.12935565,
           0.3089768,
           0.21267605,
           0.22198631,
           0.5309708,
           0.69627297,
           0.14468488,
           0.36082557,
           0.49399137,
           0.04552248,
           0.051237836,
           0.15342729,
           0.57041156,
           0.06363537,
           0.33594376,
           0.262381,
           -0.011307172,
           0.23057483,
           0.45953757,
           0.35267466,
           0.19513932,
           0.24477811,
           0.55078614,
           0.38805705,
           0.16114359,
           0.3240606,
           0.39548382,
           0.039357092,
           0.269589,
           0.38564792,
           0.1708761,
           0.5109043,
           0.34355134,
           0.4405852,
           0.47654274,
           0.21219528,
           0.110935465,
           0.41009384,
           0.452132,
           0.15911904,
           0.39736164,
           0.40707183,
           0.2270003,
           0.08852976,
           0.3572982,
           0.14611858,
           0.30891287,
           0.3123006,
           1,
           0.28962713
          ],
          [
           0.19172727,
           0.74534744,
           0.675098,
           0.6801667,
           0.5667244,
           0.69953,
           0.7436501,
           0.56807953,
           0.4978754,
           0.6034175,
           0.67045397,
           0.6134572,
           0.7263181,
           0.62835395,
           0.7269757,
           0.6027076,
           0.51665664,
           0.5911232,
           0.7233943,
           0.2171666,
           0.5545346,
           0.7098983,
           0.61852103,
           0.5991821,
           0.4367326,
           0.6369157,
           0.68862677,
           0.5350443,
           0.721626,
           0.6352123,
           0.49547088,
           0.31459355,
           0.6820221,
           0.5339324,
           0.08131992,
           0.4323024,
           0.21449298,
           0.62227553,
           0.2561099,
           0.13636301,
           0.6428611,
           0.7252711,
           0.48638573,
           0.68061197,
           0.42807245,
           0.7069427,
           0.5207057,
           0.52805674,
           0.31513467,
           0.7054722,
           0.43935668,
           0.6792282,
           0.62588084,
           0.5104016,
           0.7394287,
           0.6119303,
           0.5922406,
           0.5875238,
           0.5275327,
           0.58985126,
           0.40058595,
           0.7531352,
           0.3317477,
           0.43259948,
           0.33781207,
           0.3748598,
           0.5466872,
           0.63355494,
           0.5821589,
           0.588916,
           0.31180063,
           0.39053687,
           0.7170274,
           0.4853195,
           0.28962713,
           1
          ]
         ]
        }
       ],
       "layout": {
        "coloraxis": {
         "colorbar": {
          "title": {
           "text": "Similarity Score"
          }
         },
         "colorscale": [
          [
           0,
           "rgb(247,252,240)"
          ],
          [
           0.125,
           "rgb(224,243,219)"
          ],
          [
           0.25,
           "rgb(204,235,197)"
          ],
          [
           0.375,
           "rgb(168,221,181)"
          ],
          [
           0.5,
           "rgb(123,204,196)"
          ],
          [
           0.625,
           "rgb(78,179,211)"
          ],
          [
           0.75,
           "rgb(43,140,190)"
          ],
          [
           0.875,
           "rgb(8,104,172)"
          ],
          [
           1,
           "rgb(8,64,129)"
          ]
         ]
        },
        "height": 800,
        "hoverlabel": {
         "bgcolor": "white",
         "font": {
          "family": "Rockwell",
          "size": 16
         }
        },
        "legend": {
         "title": {
          "text": "Trend"
         }
        },
        "margin": {
         "t": 60
        },
        "showlegend": true,
        "template": {
         "data": {
          "bar": [
           {
            "error_x": {
             "color": "#2a3f5f"
            },
            "error_y": {
             "color": "#2a3f5f"
            },
            "marker": {
             "line": {
              "color": "#E5ECF6",
              "width": 0.5
             },
             "pattern": {
              "fillmode": "overlay",
              "size": 10,
              "solidity": 0.2
             }
            },
            "type": "bar"
           }
          ],
          "barpolar": [
           {
            "marker": {
             "line": {
              "color": "#E5ECF6",
              "width": 0.5
             },
             "pattern": {
              "fillmode": "overlay",
              "size": 10,
              "solidity": 0.2
             }
            },
            "type": "barpolar"
           }
          ],
          "carpet": [
           {
            "aaxis": {
             "endlinecolor": "#2a3f5f",
             "gridcolor": "white",
             "linecolor": "white",
             "minorgridcolor": "white",
             "startlinecolor": "#2a3f5f"
            },
            "baxis": {
             "endlinecolor": "#2a3f5f",
             "gridcolor": "white",
             "linecolor": "white",
             "minorgridcolor": "white",
             "startlinecolor": "#2a3f5f"
            },
            "type": "carpet"
           }
          ],
          "choropleth": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "type": "choropleth"
           }
          ],
          "contour": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "contour"
           }
          ],
          "contourcarpet": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "type": "contourcarpet"
           }
          ],
          "heatmap": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "heatmap"
           }
          ],
          "heatmapgl": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "heatmapgl"
           }
          ],
          "histogram": [
           {
            "marker": {
             "pattern": {
              "fillmode": "overlay",
              "size": 10,
              "solidity": 0.2
             }
            },
            "type": "histogram"
           }
          ],
          "histogram2d": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "histogram2d"
           }
          ],
          "histogram2dcontour": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "histogram2dcontour"
           }
          ],
          "mesh3d": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "type": "mesh3d"
           }
          ],
          "parcoords": [
           {
            "line": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "parcoords"
           }
          ],
          "pie": [
           {
            "automargin": true,
            "type": "pie"
           }
          ],
          "scatter": [
           {
            "fillpattern": {
             "fillmode": "overlay",
             "size": 10,
             "solidity": 0.2
            },
            "type": "scatter"
           }
          ],
          "scatter3d": [
           {
            "line": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scatter3d"
           }
          ],
          "scattercarpet": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scattercarpet"
           }
          ],
          "scattergeo": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scattergeo"
           }
          ],
          "scattergl": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scattergl"
           }
          ],
          "scattermapbox": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scattermapbox"
           }
          ],
          "scatterpolar": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scatterpolar"
           }
          ],
          "scatterpolargl": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scatterpolargl"
           }
          ],
          "scatterternary": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 0,
              "ticks": ""
             }
            },
            "type": "scatterternary"
           }
          ],
          "surface": [
           {
            "colorbar": {
             "outlinewidth": 0,
             "ticks": ""
            },
            "colorscale": [
             [
              0,
              "#0d0887"
             ],
             [
              0.1111111111111111,
              "#46039f"
             ],
             [
              0.2222222222222222,
              "#7201a8"
             ],
             [
              0.3333333333333333,
              "#9c179e"
             ],
             [
              0.4444444444444444,
              "#bd3786"
             ],
             [
              0.5555555555555556,
              "#d8576b"
             ],
             [
              0.6666666666666666,
              "#ed7953"
             ],
             [
              0.7777777777777778,
              "#fb9f3a"
             ],
             [
              0.8888888888888888,
              "#fdca26"
             ],
             [
              1,
              "#f0f921"
             ]
            ],
            "type": "surface"
           }
          ],
          "table": [
           {
            "cells": {
             "fill": {
              "color": "#EBF0F8"
             },
             "line": {
              "color": "white"
             }
            },
            "header": {
             "fill": {
              "color": "#C8D4E3"
             },
             "line": {
              "color": "white"
             }
            },
            "type": "table"
           }
          ]
         },
         "layout": {
          "annotationdefaults": {
           "arrowcolor": "#2a3f5f",
           "arrowhead": 0,
           "arrowwidth": 1
          },
          "autotypenumbers": "strict",
          "coloraxis": {
           "colorbar": {
            "outlinewidth": 0,
            "ticks": ""
           }
          },
          "colorscale": {
           "diverging": [
            [
             0,
             "#8e0152"
            ],
            [
             0.1,
             "#c51b7d"
            ],
            [
             0.2,
             "#de77ae"
            ],
            [
             0.3,
             "#f1b6da"
            ],
            [
             0.4,
             "#fde0ef"
            ],
            [
             0.5,
             "#f7f7f7"
            ],
            [
             0.6,
             "#e6f5d0"
            ],
            [
             0.7,
             "#b8e186"
            ],
            [
             0.8,
             "#7fbc41"
            ],
            [
             0.9,
             "#4d9221"
            ],
            [
             1,
             "#276419"
            ]
           ],
           "sequential": [
            [
             0,
             "#0d0887"
            ],
            [
             0.1111111111111111,
             "#46039f"
            ],
            [
             0.2222222222222222,
             "#7201a8"
            ],
            [
             0.3333333333333333,
             "#9c179e"
            ],
            [
             0.4444444444444444,
             "#bd3786"
            ],
            [
             0.5555555555555556,
             "#d8576b"
            ],
            [
             0.6666666666666666,
             "#ed7953"
            ],
            [
             0.7777777777777778,
             "#fb9f3a"
            ],
            [
             0.8888888888888888,
             "#fdca26"
            ],
            [
             1,
             "#f0f921"
            ]
           ],
           "sequentialminus": [
            [
             0,
             "#0d0887"
            ],
            [
             0.1111111111111111,
             "#46039f"
            ],
            [
             0.2222222222222222,
             "#7201a8"
            ],
            [
             0.3333333333333333,
             "#9c179e"
            ],
            [
             0.4444444444444444,
             "#bd3786"
            ],
            [
             0.5555555555555556,
             "#d8576b"
            ],
            [
             0.6666666666666666,
             "#ed7953"
            ],
            [
             0.7777777777777778,
             "#fb9f3a"
            ],
            [
             0.8888888888888888,
             "#fdca26"
            ],
            [
             1,
             "#f0f921"
            ]
           ]
          },
          "colorway": [
           "#636efa",
           "#EF553B",
           "#00cc96",
           "#ab63fa",
           "#FFA15A",
           "#19d3f3",
           "#FF6692",
           "#B6E880",
           "#FF97FF",
           "#FECB52"
          ],
          "font": {
           "color": "#2a3f5f"
          },
          "geo": {
           "bgcolor": "white",
           "lakecolor": "white",
           "landcolor": "#E5ECF6",
           "showlakes": true,
           "showland": true,
           "subunitcolor": "white"
          },
          "hoverlabel": {
           "align": "left"
          },
          "hovermode": "closest",
          "mapbox": {
           "style": "light"
          },
          "paper_bgcolor": "white",
          "plot_bgcolor": "#E5ECF6",
          "polar": {
           "angularaxis": {
            "gridcolor": "white",
            "linecolor": "white",
            "ticks": ""
           },
           "bgcolor": "#E5ECF6",
           "radialaxis": {
            "gridcolor": "white",
            "linecolor": "white",
            "ticks": ""
           }
          },
          "scene": {
           "xaxis": {
            "backgroundcolor": "#E5ECF6",
            "gridcolor": "white",
            "gridwidth": 2,
            "linecolor": "white",
            "showbackground": true,
            "ticks": "",
            "zerolinecolor": "white"
           },
           "yaxis": {
            "backgroundcolor": "#E5ECF6",
            "gridcolor": "white",
            "gridwidth": 2,
            "linecolor": "white",
            "showbackground": true,
            "ticks": "",
            "zerolinecolor": "white"
           },
           "zaxis": {
            "backgroundcolor": "#E5ECF6",
            "gridcolor": "white",
            "gridwidth": 2,
            "linecolor": "white",
            "showbackground": true,
            "ticks": "",
            "zerolinecolor": "white"
           }
          },
          "shapedefaults": {
           "line": {
            "color": "#2a3f5f"
           }
          },
          "ternary": {
           "aaxis": {
            "gridcolor": "white",
            "linecolor": "white",
            "ticks": ""
           },
           "baxis": {
            "gridcolor": "white",
            "linecolor": "white",
            "ticks": ""
           },
           "bgcolor": "#E5ECF6",
           "caxis": {
            "gridcolor": "white",
            "linecolor": "white",
            "ticks": ""
           }
          },
          "title": {
           "x": 0.05
          },
          "xaxis": {
           "automargin": true,
           "gridcolor": "white",
           "linecolor": "white",
           "ticks": "",
           "title": {
            "standoff": 15
           },
           "zerolinecolor": "white",
           "zerolinewidth": 2
          },
          "yaxis": {
           "automargin": true,
           "gridcolor": "white",
           "linecolor": "white",
           "ticks": "",
           "title": {
            "standoff": 15
           },
           "zerolinecolor": "white",
           "zerolinewidth": 2
          }
         }
        },
        "title": {
         "font": {
          "color": "Black",
          "size": 22
         },
         "text": "<b>Similarity Matrix</b>",
         "x": 0.55,
         "xanchor": "center",
         "y": 0.95,
         "yanchor": "top"
        },
        "width": 800,
        "xaxis": {
         "anchor": "y",
         "constrain": "domain",
         "domain": [
          0,
          1
         ],
         "scaleanchor": "y"
        },
        "yaxis": {
         "anchor": "x",
         "autorange": "reversed",
         "constrain": "domain",
         "domain": [
          0,
          1
         ]
        }
       }
      }
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "topic_model.visualize_heatmap()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66497484",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "59da680a",
   "metadata": {},
   "outputs": [],
   "source": [
    "x = topic_model.get_document_info(docs)\n",
    "y = x.Name\n",
    "labels =y.values\n",
    "labels.shape\n",
    "labels\n",
    "\n",
    "excluded_topic = str(topic_info.Name[0]) \n",
    "clean_labels= [item.replace(excluded_topic, \"Unlabelled\") for item in labels] # To exclude -1 topic which is typically noise. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "d400fee3",
   "metadata": {},
   "outputs": [],
   "source": [
    "excluded_topic = str(topic_info.Name[0]) \n",
    "clean_labels= [item.replace(excluded_topic, \"Unlabelled\") for item in labels] # To exclude -1 topic which is typically noise. \n",
    "clean_labels\n",
    "clean_labels = pd.DataFrame(clean_labels, columns = ['Name'])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "c2d3df27",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'claude_labels' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[145], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m final_labels \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mmerge(clean_labels, \u001b[43mclaude_labels\u001b[49m, on\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mName\u001b[39m\u001b[38;5;124m'\u001b[39m, how\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mleft\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m      2\u001b[0m final_labels[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtopics\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m      3\u001b[0m final_labels \u001b[38;5;241m=\u001b[39m final_labels\u001b[38;5;241m.\u001b[39mfillna(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnlabelled\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'claude_labels' is not defined"
     ]
    }
   ],
   "source": [
    "final_labels = pd.merge(clean_labels, claude_labels, on='Name', how='left')\n",
    "final_labels['topics']\n",
    "final_labels = final_labels.fillna(\"Unlabelled\")\n",
    "final_labels = pd.array(final_labels['topics'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b920efe",
   "metadata": {},
   "source": [
    "The code below is to add additional information like the citation count of each papers and also their keywords.\n",
    "\n",
    "For more informations about how to integrate and style this information, consult https://datamapplot.readthedocs.io/en/latest/ and the according *Github*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "9ad22bb8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "citedby_count",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "keywords",
         "rawType": "object",
         "type": "string"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "fe8162a8-f932-469f-8a87-c54a6351b3b0",
       "rows": [
        [
         "0",
         "0",
         "Drift | Driftability | Evolutionary causation | Niche construction | Selection"
        ],
        [
         "1",
         "0",
         "Evolutionary theory | Explanatory gap | Natural selection | Selection explanation | The active role of the organism | The extended synthesis"
        ],
        [
         "2",
         "0",
         "Genetic prediction | GWAS | Idealisations | Polygenic scores | Predictive accuracy | Quantitative genetics"
        ],
        [
         "3",
         "0",
         "Collective cell migration | Generalization | Model organisms | Model systems | Phenomena | Representation"
        ],
        [
         "4",
         "1",
         "Alan M. Turing | Darwinian evolution | D‚ÄôArcy Wendworth Thompson | explanations in science | mechanism | modern synthesis | morphogenesis | organicism"
        ],
        [
         "5",
         "9",
         "Definition of obesity | Developmental canalization | Epigenetics of obesity | Genetics of obesity | Obesity | Obesity and public health"
        ],
        [
         "7",
         "16",
         "Character states | Genetic liability | Mendelian diseases | Qualitative traits | Quantitative traits | Thresholds"
        ],
        [
         "8",
         "5",
         "Cancer model | Feedback control | Set-valued analysis | Viability"
        ],
        [
         "9",
         "0",
         "B. subtilis | Bacteria | Cyanobacteria | Individuality | Multicellularity | Myxobacteria"
        ],
        [
         "10",
         "26",
         "archaea | bacteria | divergence | eukaryotes | evolution | horizontal gene transfer | last universal common ancestor (LUCA)"
        ],
        [
         "12",
         "34",
         "extramedullary hematopoiesis | hematopoietic stem cells | myelopoiesis | spleen | stem cell niche | tissue regeneration | transplantation"
        ],
        [
         "13",
         "33",
         "Adaptation | Aging | Apoptosis | Price equation | Programmed cell death | Selection | Unicellular organisms"
        ],
        [
         "19",
         "54",
         "Design | Engineering | Genetic program | Machine | Metaphor | Organism"
        ],
        [
         "20",
         "52",
         "angiogenesis | arteries | blood flow | branching morphogenesis | coronary vasculature | cxcr4 | notch | vein"
        ],
        [
         "31",
         "24",
         "development | hydrogen peroxide | mitochondria | neurodegeneration | superoxide | synapse pruning"
        ],
        [
         "33",
         "190",
         "autophagy | lysosome | macroautophagy | microautophagy | vacuole"
        ],
        [
         "34",
         "12",
         "in vitro expression | ion channels | mitochondria | outer membrane | planar lipid bilayer | Œ≤-barrel proteins"
        ],
        [
         "35",
         "12",
         "endothelial cells | fatty acid Œ≤-oxidation | glycolysis | growth factor signaling | lymphangiogenesis | lymphatics | metabolism"
        ],
        [
         "39",
         "2",
         "Linnaean nomenclature | Morphology | Shape invariants | Taxonomic categorization"
        ],
        [
         "40",
         "0",
         "Discrete mesh modelling | Gastrulation | Morphogenesis | Ventral furrow"
        ],
        [
         "42",
         "15",
         "NULL"
        ],
        [
         "43",
         "1",
         "NULL"
        ],
        [
         "46",
         "37",
         "basal bodies | cartwheel | centrioles | cilia | SAS-6"
        ],
        [
         "47",
         "9",
         "bacteria | germline | holobiont | influence | inheritance | soma | transgenerational"
        ],
        [
         "48",
         "24",
         "indicator model | mitochondria | OXPHOS | reinforcement | sexual selection"
        ],
        [
         "54",
         "14",
         "iridophore | melanocyte | neural crest | pigment pattern formation | stem cell | zebrafish"
        ],
        [
         "55",
         "31",
         "force measurement | magnetic tweezers | microtubule | pronuclear migration and rotation | spindle | spindle positioning"
        ],
        [
         "57",
         "30",
         "eukaryotic evolution | FADH /NADH ratio 2 | genome reduction | LECA | mitochondria | oxygen radicals | symbiogenesis"
        ],
        [
         "58",
         "15",
         "autophagy | Legionella pneumophila | mitochondria | mitochondria-associated ER membranes"
        ],
        [
         "59",
         "101",
         "adhesions | cadherin | cell forces | integrin | membrane signaling"
        ],
        [
         "60",
         "67",
         "genotoxic stress sensing | immunity | programmed cell death | virus-host coevolution"
        ],
        [
         "61",
         "10",
         "B. subtilis | Cell decision | Competence | Dynamics | Phenotypic diversity | Phosphorelay | Sporulation"
        ],
        [
         "62",
         "4",
         "Developmental biology | Instructive-permissive distinction | Normal development | Standards | Variation"
        ],
        [
         "64",
         "12",
         "Cambium | Cell | Coagulation | Matter | Primordial utricle | Protoplasm"
        ],
        [
         "66",
         "16",
         "bloom syndrome helicase | crossover patterning | double-strand break repair | meiosis | meiotic recombination | model organism genetics | resolution"
        ],
        [
         "68",
         "26",
         "beta-oxidation | FADH /NADH ratio 2 | peroxisome evolution | reverse electron transport (RET) | symbiogenesis"
        ],
        [
         "69",
         "52",
         "cell size control | fitness | metabolism | mevalonate pathway | mitochondria | polyploidy | statin"
        ],
        [
         "70",
         "1",
         "Dynamic systems | Embodied cognition | Enactivism | Gene expression | Merleau-Ponty | Motor development"
        ],
        [
         "79",
         "6",
         "Common ancestor | Complexity | Eukaryotes | Organic codes | Primary kingdoms | Prokaryotes | Sequence codes"
        ],
        [
         "80",
         "79",
         "cell competition | coloration | multipotent stem cells | neural crest | pattern formation | peripheral nervous system | zebrafish"
        ],
        [
         "81",
         "40",
         "chromosome architecture | gene regulation | insulators | loop topology | looped domain | TADs"
        ],
        [
         "82",
         "29",
         "actin cytoskeleton | centriole/MTOC localization | cilia | epithelial polarity | Frizzled-Vangl signaling | microtubules | planar cell polarity"
        ],
        [
         "84",
         "12",
         "antiviral defense | exaptation | mRNA decay | nonsense-mediated decay | regnase | roquin | tristetraprolin"
        ],
        [
         "89",
         "5",
         "Cancer | Reductionism | TOFT"
        ],
        [
         "90",
         "29",
         "22q11.2 deletion syndrome | ATP | calcium | mitochondria | schizophrenia | synapse | synaptic plasticity"
        ],
        [
         "93",
         "37",
         "antibiotic resistance | fitness | high-throughput sequencing | transposon sequencing | virulence"
        ],
        [
         "95",
         "81",
         "epithelial polarity | Hippo pathway | mechanosensing | mechanotransduction | TAZ | wound healing | YAP"
        ],
        [
         "96",
         "24",
         "Aging | Apoptosis | Bottleneck | Mitochondria | Mitophagy | MtDNA | Public goods"
        ],
        [
         "97",
         "88",
         "Control | Dynamic stability | Dynamical decoupling | Feedback | Organisation | Regulation"
        ],
        [
         "98",
         "53",
         "Bet-hedging | Evolution of variation | Fractional response | Functional variation | Single cell transcriptome | Single cell variation"
        ]
       ],
       "shape": {
        "columns": 2,
        "rows": 6272
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>citedby_count</th>\n",
       "      <th>keywords</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>Drift | Driftability | Evolutionary causation ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>Evolutionary theory | Explanatory gap | Natura...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>Genetic prediction | GWAS | Idealisations | Po...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>Collective cell migration | Generalization | M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3159</th>\n",
       "      <td>7</td>\n",
       "      <td>economics | ecosystem management | philosophy ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3160</th>\n",
       "      <td>15</td>\n",
       "      <td>Commons | Game theory | Global environment | H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3161</th>\n",
       "      <td>66</td>\n",
       "      <td>Africa | Biodiversity | Conservation areas | L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3162</th>\n",
       "      <td>55</td>\n",
       "      <td>Exotics | Native | Nativism | Naturalisation |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3163</th>\n",
       "      <td>12</td>\n",
       "      <td>Biological diversity | Commercialisation | Eco...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6272 rows √ó 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      citedby_count                                           keywords\n",
       "0                 0  Drift | Driftability | Evolutionary causation ...\n",
       "1                 0  Evolutionary theory | Explanatory gap | Natura...\n",
       "2                 0  Genetic prediction | GWAS | Idealisations | Po...\n",
       "3                 0  Collective cell migration | Generalization | M...\n",
       "4                 1  Alan M. Turing | Darwinian evolution | D‚ÄôArcy ...\n",
       "...             ...                                                ...\n",
       "3159              7  economics | ecosystem management | philosophy ...\n",
       "3160             15  Commons | Game theory | Global environment | H...\n",
       "3161             66  Africa | Biodiversity | Conservation areas | L...\n",
       "3162             55  Exotics | Native | Nativism | Naturalisation |...\n",
       "3163             12  Biological diversity | Commercialisation | Eco...\n",
       "\n",
       "[6272 rows x 2 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "add_info = pd.DataFrame(\n",
    "    {\"citedby_count\":df_cleaned.citedby_count, \"keywords\":df_cleaned.authkeywords.fillna('No_value')}\n",
    ")\n",
    "\n",
    "marker_size_array = np.log(1 + df_cleaned.citedby_count.values) # log for visibility\n",
    "add_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78ac1c8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "hover_text_template = \"\"\"\n",
    "<div>\n",
    "    <p> <strong> Title </strong>: {hover_text}</p>\n",
    "    <p> <strong> Citation Count </strong>: {citedby_count}</p>\n",
    "    <p> <strong> Keywords </strong>: {keywords}</p>\n",
    "</div>\n",
    "</div>\n",
    "\"\"\"\n",
    "badge_css = \"\"\"\n",
    "    border-radius:6px; \n",
    "    width:fit-content; \n",
    "    max-width:75%; \n",
    "    margin:2px; \n",
    "    padding: 2px 10px 2px 10px; \n",
    "    font-size: 10pt;\n",
    "\"\"\"\n",
    "hover_text_template = f\"\"\"\n",
    "<div>\n",
    "    <div style=\"font-size:12pt;padding:2px;\">{{hover_text}}</div>\n",
    "    <div style=\"background-color:#525356;color:#fff;{badge_css}\">{{keywords}}</div>\n",
    "    <div style=\"background-color:#eeeeeeff;{badge_css}\">Citation count: {{citedby_count}}</div>\n",
    "</div>\n",
    "\"\"\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "6677549e",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "2",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[170], line 6\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mglasbey\u001b[39;00m\n\u001b[0;32m      4\u001b[0m palette \u001b[38;5;241m=\u001b[39m glasbey\u001b[38;5;241m.\u001b[39mcreate_palette(\u001b[38;5;241m20\u001b[39m, chroma_bounds\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m20\u001b[39m,\u001b[38;5;241m75\u001b[39m), lightness_bounds\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m20\u001b[39m,\u001b[38;5;241m60\u001b[39m))\n\u001b[1;32m----> 6\u001b[0m plot \u001b[38;5;241m=\u001b[39m \u001b[43mdatamapplot\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcreate_interactive_plot\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m      7\u001b[0m \u001b[43m    \u001b[49m\u001b[43mreduced_embeddings\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m      8\u001b[0m \u001b[43m    \u001b[49m\u001b[43mclean_labels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m      9\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#hover_text= node_text,\u001b[39;49;00m\n\u001b[0;32m     10\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfont_family\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mCinzel\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     11\u001b[0m \u001b[43m    \u001b[49m\u001b[43menable_search\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m     12\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#inline_data=False,\u001b[39;49;00m\n\u001b[0;32m     13\u001b[0m \u001b[43m    \u001b[49m\u001b[43minitial_zoom_fraction\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0.9\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     14\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#offline_data_prefix=\"cord-large-cmaps-1\",\u001b[39;49;00m\n\u001b[0;32m     15\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#extra_point_data= add_info,\u001b[39;49;00m\n\u001b[0;32m     16\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#hover_text_html_template = hover_text_template,\u001b[39;49;00m\n\u001b[0;32m     17\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#marker_size_array=marker_size_array,\u001b[39;49;00m\n\u001b[0;32m     18\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#selection_handler=datamapplot.selection_handlers.DisplaySample(n_samples=25),\u001b[39;49;00m\n\u001b[0;32m     19\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#colormaps={\"Type\": df_cleaned.subtypeDescription, \"Journal\": df_cleaned.publicationName},\u001b[39;49;00m\n\u001b[0;32m     20\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#colormap_rawdata=[df_cleaned.subtypeDescription, df_cleaned.publicationName], #final_labels_legend.Name_Claude],\u001b[39;49;00m\n\u001b[0;32m     21\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#colormap_metadata=[\u001b[39;49;00m\n\u001b[0;32m     22\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;66;43;03m#{\"field\": \"Type\", \"description\": \"Type\", \"cmap\": \"Accent\", \"kind\": \"datetime\"},\u001b[39;49;00m\n\u001b[0;32m     23\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;66;43;03m#{\"field\": \"Journal\", \"description\": \"Journal\", \"cmap\": \"Dark2_r\", \"kind\": \"continuous\"},\u001b[39;49;00m\n\u001b[0;32m     24\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#],\u001b[39;49;00m\n\u001b[0;32m     25\u001b[0m \u001b[43m    \u001b[49m\u001b[43moffline_mode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m     26\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     27\u001b[0m plot\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\config.py:121\u001b[0m, in \u001b[0;36mConfigManager.complete.<locals>.decorator.<locals>.fn_with_config\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    114\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m ConfigError(\n\u001b[0;32m    115\u001b[0m                 \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mParameter \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mparam\u001b[38;5;241m.\u001b[39mname\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m is deliberately listed as \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    116\u001b[0m                 \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mforbidden from being defined through the DataMapPlot \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    117\u001b[0m                 \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mconfiguration file.\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m    118\u001b[0m                 param\n\u001b[0;32m    119\u001b[0m             )\n\u001b[0;32m    120\u001b[0m         from_config[name] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m[name]\n\u001b[1;32m--> 121\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mbound_args\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mbound_args\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m|\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mfrom_config\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\create_plots.py:489\u001b[0m, in \u001b[0;36mcreate_interactive_plot\u001b[1;34m(data_map_coords, hover_text, inline_data, noise_label, noise_color, color_label_text, label_wrap_width, label_color_map, width, height, darkmode, palette_hue_shift, palette_hue_radius_dependence, palette_theta_range, cmap, marker_size_array, marker_color_array, marker_alpha_array, use_medoids, cluster_boundary_polygons, color_cluster_boundaries, polygon_alpha, cvd_safer, jupyterhub_api_token, *label_layers, **render_html_kwds)\u001b[0m\n\u001b[0;32m    479\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(\n\u001b[0;32m    480\u001b[0m         {\n\u001b[0;32m    481\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m\"\u001b[39m: [data_map_coords\u001b[38;5;241m.\u001b[39mT[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmean()],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    485\u001b[0m         }\n\u001b[0;32m    486\u001b[0m     )\n\u001b[0;32m    487\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    488\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\n\u001b[1;32m--> 489\u001b[0m         \u001b[43m[\u001b[49m\n\u001b[0;32m    490\u001b[0m \u001b[43m            \u001b[49m\u001b[43mlabel_text_and_polygon_dataframes\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m                \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m                \u001b[49m\u001b[43mdata_map_coords\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m                \u001b[49m\u001b[43mnoise_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnoise_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m                \u001b[49m\u001b[43muse_medoids\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muse_medoids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m                \u001b[49m\u001b[43mcluster_polygons\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcluster_boundary_polygons\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m                \u001b[49m\u001b[43malpha\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpolygon_alpha\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m            \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mlabel_layers\u001b[49m\n\u001b[0;32m    499\u001b[0m \u001b[43m        \u001b[49m\u001b[43m]\u001b[49m\n\u001b[0;32m    500\u001b[0m     )\n\u001b[0;32m    502\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m cvd_safer:\n\u001b[0;32m    503\u001b[0m     cmap \u001b[38;5;241m=\u001b[39m colorcet\u001b[38;5;241m.\u001b[39mcm\u001b[38;5;241m.\u001b[39mCET_C2s\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\create_plots.py:490\u001b[0m, in \u001b[0;36m<listcomp>\u001b[1;34m(.0)\u001b[0m\n\u001b[0;32m    479\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(\n\u001b[0;32m    480\u001b[0m         {\n\u001b[0;32m    481\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m\"\u001b[39m: [data_map_coords\u001b[38;5;241m.\u001b[39mT[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmean()],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    485\u001b[0m         }\n\u001b[0;32m    486\u001b[0m     )\n\u001b[0;32m    487\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    488\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\n\u001b[0;32m    489\u001b[0m         [\n\u001b[1;32m--> 490\u001b[0m             \u001b[43mlabel_text_and_polygon_dataframes\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m                \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m                \u001b[49m\u001b[43mdata_map_coords\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m                \u001b[49m\u001b[43mnoise_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnoise_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m                \u001b[49m\u001b[43muse_medoids\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muse_medoids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m                \u001b[49m\u001b[43mcluster_polygons\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcluster_boundary_polygons\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m                \u001b[49m\u001b[43malpha\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpolygon_alpha\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m            \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m             \u001b[38;5;28;01mfor\u001b[39;00m labels \u001b[38;5;129;01min\u001b[39;00m label_layers\n\u001b[0;32m    499\u001b[0m         ]\n\u001b[0;32m    500\u001b[0m     )\n\u001b[0;32m    502\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m cvd_safer:\n\u001b[0;32m    503\u001b[0m     cmap \u001b[38;5;241m=\u001b[39m colorcet\u001b[38;5;241m.\u001b[39mcm\u001b[38;5;241m.\u001b[39mCET_C2s\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\interactive_rendering.py:849\u001b[0m, in \u001b[0;36mlabel_text_and_polygon_dataframes\u001b[1;34m(labels, data_map_coords, noise_label, use_medoids, cluster_polygons, alpha)\u001b[0m\n\u001b[0;32m    847\u001b[0m label_map \u001b[38;5;241m=\u001b[39m {n: i \u001b[38;5;28;01mfor\u001b[39;00m i, n \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(unique_non_noise_labels)}\n\u001b[0;32m    848\u001b[0m label_map[noise_label] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m\n\u001b[1;32m--> 849\u001b[0m cluster_idx_vector \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(\u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mSeries\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcluster_label_vector\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mmap(label_map))\n\u001b[0;32m    851\u001b[0m label_locations \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m    852\u001b[0m cluster_sizes \u001b[38;5;241m=\u001b[39m []\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\series.py:584\u001b[0m, in \u001b[0;36mSeries.__init__\u001b[1;34m(self, data, index, dtype, name, copy, fastpath)\u001b[0m\n\u001b[0;32m    582\u001b[0m         data \u001b[38;5;241m=\u001b[39m data\u001b[38;5;241m.\u001b[39mcopy()\n\u001b[0;32m    583\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 584\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[43msanitize_array\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    586\u001b[0m     manager \u001b[38;5;241m=\u001b[39m _get_option(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmode.data_manager\u001b[39m\u001b[38;5;124m\"\u001b[39m, silent\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m    587\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m manager \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mblock\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\construction.py:606\u001b[0m, in \u001b[0;36msanitize_array\u001b[1;34m(data, index, dtype, copy, allow_2d)\u001b[0m\n\u001b[0;32m    604\u001b[0m subarr \u001b[38;5;241m=\u001b[39m data\n\u001b[0;32m    605\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m data\u001b[38;5;241m.\u001b[39mdtype \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mobject\u001b[39m:\n\u001b[1;32m--> 606\u001b[0m     subarr \u001b[38;5;241m=\u001b[39m \u001b[43mmaybe_infer_to_datetimelike\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    607\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[0;32m    608\u001b[0m         object_index\n\u001b[0;32m    609\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m using_pyarrow_string_dtype()\n\u001b[0;32m    610\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m is_string_dtype(subarr)\n\u001b[0;32m    611\u001b[0m     ):\n\u001b[0;32m    612\u001b[0m         \u001b[38;5;66;03m# Avoid inference when string option is set\u001b[39;00m\n\u001b[0;32m    613\u001b[0m         subarr \u001b[38;5;241m=\u001b[39m data\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\dtypes\\cast.py:1181\u001b[0m, in \u001b[0;36mmaybe_infer_to_datetimelike\u001b[1;34m(value)\u001b[0m\n\u001b[0;32m   1178\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;28mtype\u001b[39m(value))  \u001b[38;5;66;03m# pragma: no cover\u001b[39;00m\n\u001b[0;32m   1179\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m value\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m   1180\u001b[0m     \u001b[38;5;66;03m# Caller is responsible\u001b[39;00m\n\u001b[1;32m-> 1181\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(value\u001b[38;5;241m.\u001b[39mndim)  \u001b[38;5;66;03m# pragma: no cover\u001b[39;00m\n\u001b[0;32m   1183\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(value):\n\u001b[0;32m   1184\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m value\n",
      "\u001b[1;31mValueError\u001b[0m: 2"
     ]
    }
   ],
   "source": [
    "import datamapplot.selection_handlers\n",
    "#df_cleaned['year'] = pd.to_datetime(df_cleaned['citing_year'].astype(str), format='%Y')\n",
    "import glasbey\n",
    "palette = glasbey.create_palette(20, chroma_bounds=(20,75), lightness_bounds=(20,60))\n",
    "\n",
    "plot = datamapplot.create_interactive_plot(\n",
    "    reduced_embeddings,\n",
    "    clean_labels,\n",
    "    #hover_text= node_text,\n",
    "    font_family=\"Cinzel\",\n",
    "    enable_search=True,\n",
    "    #inline_data=False,\n",
    "    initial_zoom_fraction=0.9,\n",
    "    #offline_data_prefix=\"cord-large-cmaps-1\",\n",
    "    #extra_point_data= add_info,\n",
    "    #hover_text_html_template = hover_text_template,\n",
    "    #marker_size_array=marker_size_array,\n",
    "    #selection_handler=datamapplot.selection_handlers.DisplaySample(n_samples=25),\n",
    "    #colormaps={\"Type\": df_cleaned.subtypeDescription, \"Journal\": df_cleaned.publicationName},\n",
    "    #colormap_rawdata=[df_cleaned.subtypeDescription, df_cleaned.publicationName], #final_labels_legend.Name_Claude],\n",
    "    #colormap_metadata=[\n",
    "        #{\"field\": \"Type\", \"description\": \"Type\", \"cmap\": \"Accent\", \"kind\": \"datetime\"},\n",
    "        #{\"field\": \"Journal\", \"description\": \"Journal\", \"cmap\": \"Dark2_r\", \"kind\": \"continuous\"},\n",
    "    #],\n",
    "    offline_mode=True,\n",
    "    )\n",
    "plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8cbe416e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             Original Countries      Continent\n",
      "0                       Belgium         Europe\n",
      "1                Belgium;Norway  Collaboration\n",
      "2                  Poland;Italy  Collaboration\n",
      "3                 United States  North America\n",
      "4                        Poland         Europe\n",
      "...                         ...            ...\n",
      "6267  France;Switzerland;France  Collaboration\n",
      "6268              United States  North America\n",
      "6269                    Finland         Europe\n",
      "6270              United States  North America\n",
      "6271                     Canada  North America\n",
      "\n",
      "[6272 rows x 2 columns]\n",
      "\n",
      "Total entries: 6272\n",
      "Breakdown by continent:\n",
      "Continent\n",
      "North America    2352\n",
      "Europe           2132\n",
      "Collaboration     872\n",
      "Asia              284\n",
      "Unknown           265\n",
      "Oceania           265\n",
      "South America      81\n",
      "Africa             21\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Use the countries list from the input document\n",
    "countries = np.array(df_cleaned.affiliation_country)\n",
    "\n",
    "def get_continent(country):\n",
    "    continents = {\n",
    "        'North America': ['United States', 'Canada', 'Mexico'],\n",
    "        'South America': ['Brazil', 'Argentina', 'Chile', 'Colombia', 'Uruguay', 'Venezuela'],\n",
    "        'Europe': ['United Kingdom', 'Germany', 'France', 'Spain', 'Italy', 'Netherlands', 'Poland', \n",
    "                   'Croatia', 'Czech Republic', 'Denmark', 'Romania', 'Austria', 'Belgium', 'Greece', \n",
    "                   'Ireland', 'Norway', 'Portugal', 'Russian Federation', 'Slovakia', 'Sweden', 'Switzerland', \n",
    "                   'Hungary', 'Finland', 'Luxembourg', 'Estonia', 'Lithuania', 'Slovenia', 'Iceland', 'Serbia', 'Cyprus'],\n",
    "        'Asia': ['China', 'Israel', 'United Arab Emirates', 'Japan', 'Turkey', 'Kazakhstan', \n",
    "                 'Bangladesh', 'Malaysia', 'India', 'Pakistan', 'Taiwan', 'Singapore', 'South Korea', \n",
    "                 'Hong Kong', 'Indonesia', 'Philippines'],\n",
    "        'Africa': ['Sierra Leone', 'South Africa', 'Kenya', 'Egypt'],\n",
    "        'Oceania': ['Australia', 'New Zealand']\n",
    "    }\n",
    "    \n",
    "    # If multiple countries are listed, check for unique countries\n",
    "    unique_countries = set(country.split(';'))\n",
    "    \n",
    "    # Handle NULL or empty cases\n",
    "    if 'NULL' in unique_countries or len(unique_countries) == 0:\n",
    "        return 'Unknown'\n",
    "    \n",
    "    # If multiple unique countries\n",
    "    if len(unique_countries) > 1:\n",
    "        return 'Collaboration'\n",
    "    \n",
    "    # If only one unique country\n",
    "    country_name = list(unique_countries)[0]\n",
    "    for continent, countries in continents.items():\n",
    "        if country_name in countries:\n",
    "            return continent\n",
    "    return 'Unknown'\n",
    "\n",
    "# Create original DataFrame\n",
    "df_countries = pd.DataFrame({'Original Countries': countries})\n",
    "\n",
    "# Add Continent column\n",
    "df_countries['Continent'] = df_countries['Original Countries'].apply(get_continent)\n",
    "\n",
    "print(df_countries)\n",
    "print(f\"\\nTotal entries: {len(df_countries)}\")\n",
    "print(f\"Breakdown by continent:\\n{df_countries['Continent'].value_counts()}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2290f47",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cleaned[\"coverDate\"] = pd.to_datetime(df_cleaned[\"coverDate\"])\n",
    "date = pd.array(df_cleaned.coverDate)\n",
    "\n",
    "\n",
    "journal_top = df_cleaned\n",
    "top_10_journals = journal_top[\"publicationName\"].value_counts().nlargest(10).index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf4cda82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace journal names not in the top 10 with \"Not in top 10\"\n",
    "journal_top[\"publicationName\"] = journal_top[\"publicationName\"].where(journal_top[\"publicationName\"].isin(top_10_journals), \"Not in top 10\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "946adf9e",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "2",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[162], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m plot \u001b[38;5;241m=\u001b[39m \u001b[43mdatamapplot\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcreate_interactive_plot\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m      2\u001b[0m \u001b[43m    \u001b[49m\u001b[43mreduced_embeddings\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m      3\u001b[0m \u001b[43m    \u001b[49m\u001b[43mclean_labels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m      4\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#hover_text= node_text,\u001b[39;49;00m\n\u001b[0;32m      5\u001b[0m \u001b[43m    \u001b[49m\u001b[43mfont_family\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mCinzel\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m      6\u001b[0m \u001b[43m    \u001b[49m\u001b[43menable_search\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m      7\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#inline_data=False,\u001b[39;49;00m\n\u001b[0;32m      8\u001b[0m \u001b[43m    \u001b[49m\u001b[43minitial_zoom_fraction\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0.9\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m      9\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#offline_data_prefix=\"cord-large-cmaps-1\",\u001b[39;49;00m\n\u001b[0;32m     10\u001b[0m \u001b[43m    \u001b[49m\u001b[43mextra_point_data\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43madd_info\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     11\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhover_text_html_template\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mhover_text_template\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     12\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmarker_size_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmarker_size_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     13\u001b[0m \u001b[43m    \u001b[49m\u001b[43mselection_handler\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdatamapplot\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mselection_handlers\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDisplaySample\u001b[49m\u001b[43m(\u001b[49m\u001b[43mn_samples\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m25\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     14\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;66;43;03m#colormaps={\"Collaboration\": df_countries.Continent, \"Type\": df_cleaned.subtypeDescription, \"Journal\": journal_top.publicationName},\u001b[39;49;00m\n\u001b[0;32m     15\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhistogram_data\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mdf_cleaned\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcoverDate\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     16\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtitle\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mPhilosophy of Biology\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m  \u001b[49m\n\u001b[0;32m     17\u001b[0m \u001b[43m    \u001b[49m\u001b[43msub_title\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mFrom <i>PhilPapers</i> (n = 6,272)\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     18\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhistogram_n_bins\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m35\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     19\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhistogram_settings\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m{\u001b[49m\n\u001b[0;32m     20\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_log_scale\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m     21\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_title\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mPublication Year\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     22\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_bin_fill_color\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m#282a36\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     23\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_bin_unselected_fill_color\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m#b5b5b9\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     24\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_bin_selected_fill_color\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m#f68571\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     25\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_width\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;241;43m400\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     26\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mhistogram_height\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[38;5;241;43m75\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m     27\u001b[0m \u001b[43m    \u001b[49m\u001b[43m}\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m     28\u001b[0m \u001b[43m    \u001b[49m\n\u001b[0;32m     29\u001b[0m \u001b[43m)\u001b[49m\n\u001b[0;32m     30\u001b[0m plot\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\config.py:121\u001b[0m, in \u001b[0;36mConfigManager.complete.<locals>.decorator.<locals>.fn_with_config\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    114\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m ConfigError(\n\u001b[0;32m    115\u001b[0m                 \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mParameter \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mparam\u001b[38;5;241m.\u001b[39mname\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m is deliberately listed as \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    116\u001b[0m                 \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mforbidden from being defined through the DataMapPlot \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    117\u001b[0m                 \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mconfiguration file.\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m    118\u001b[0m                 param\n\u001b[0;32m    119\u001b[0m             )\n\u001b[0;32m    120\u001b[0m         from_config[name] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m[name]\n\u001b[1;32m--> 121\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfn\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mbound_args\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mbound_args\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m|\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mfrom_config\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\create_plots.py:489\u001b[0m, in \u001b[0;36mcreate_interactive_plot\u001b[1;34m(data_map_coords, hover_text, inline_data, noise_label, noise_color, color_label_text, label_wrap_width, label_color_map, width, height, darkmode, palette_hue_shift, palette_hue_radius_dependence, palette_theta_range, cmap, marker_size_array, marker_color_array, marker_alpha_array, use_medoids, cluster_boundary_polygons, color_cluster_boundaries, polygon_alpha, cvd_safer, jupyterhub_api_token, *label_layers, **render_html_kwds)\u001b[0m\n\u001b[0;32m    479\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(\n\u001b[0;32m    480\u001b[0m         {\n\u001b[0;32m    481\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m\"\u001b[39m: [data_map_coords\u001b[38;5;241m.\u001b[39mT[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmean()],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    485\u001b[0m         }\n\u001b[0;32m    486\u001b[0m     )\n\u001b[0;32m    487\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    488\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\n\u001b[1;32m--> 489\u001b[0m         \u001b[43m[\u001b[49m\n\u001b[0;32m    490\u001b[0m \u001b[43m            \u001b[49m\u001b[43mlabel_text_and_polygon_dataframes\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m                \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m                \u001b[49m\u001b[43mdata_map_coords\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m                \u001b[49m\u001b[43mnoise_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnoise_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m                \u001b[49m\u001b[43muse_medoids\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muse_medoids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m                \u001b[49m\u001b[43mcluster_polygons\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcluster_boundary_polygons\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m                \u001b[49m\u001b[43malpha\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpolygon_alpha\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m            \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m \u001b[43m            \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mlabel_layers\u001b[49m\n\u001b[0;32m    499\u001b[0m \u001b[43m        \u001b[49m\u001b[43m]\u001b[49m\n\u001b[0;32m    500\u001b[0m     )\n\u001b[0;32m    502\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m cvd_safer:\n\u001b[0;32m    503\u001b[0m     cmap \u001b[38;5;241m=\u001b[39m colorcet\u001b[38;5;241m.\u001b[39mcm\u001b[38;5;241m.\u001b[39mCET_C2s\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\create_plots.py:490\u001b[0m, in \u001b[0;36m<listcomp>\u001b[1;34m(.0)\u001b[0m\n\u001b[0;32m    479\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(\n\u001b[0;32m    480\u001b[0m         {\n\u001b[0;32m    481\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m\"\u001b[39m: [data_map_coords\u001b[38;5;241m.\u001b[39mT[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m.\u001b[39mmean()],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    485\u001b[0m         }\n\u001b[0;32m    486\u001b[0m     )\n\u001b[0;32m    487\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    488\u001b[0m     label_dataframe \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\n\u001b[0;32m    489\u001b[0m         [\n\u001b[1;32m--> 490\u001b[0m             \u001b[43mlabel_text_and_polygon_dataframes\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m                \u001b[49m\u001b[43mlabels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m                \u001b[49m\u001b[43mdata_map_coords\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m                \u001b[49m\u001b[43mnoise_label\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnoise_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m                \u001b[49m\u001b[43muse_medoids\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muse_medoids\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m                \u001b[49m\u001b[43mcluster_polygons\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcluster_boundary_polygons\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m                \u001b[49m\u001b[43malpha\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpolygon_alpha\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m            \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m             \u001b[38;5;28;01mfor\u001b[39;00m labels \u001b[38;5;129;01min\u001b[39;00m label_layers\n\u001b[0;32m    499\u001b[0m         ]\n\u001b[0;32m    500\u001b[0m     )\n\u001b[0;32m    502\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m cvd_safer:\n\u001b[0;32m    503\u001b[0m     cmap \u001b[38;5;241m=\u001b[39m colorcet\u001b[38;5;241m.\u001b[39mcm\u001b[38;5;241m.\u001b[39mCET_C2s\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\datamapplot\\interactive_rendering.py:849\u001b[0m, in \u001b[0;36mlabel_text_and_polygon_dataframes\u001b[1;34m(labels, data_map_coords, noise_label, use_medoids, cluster_polygons, alpha)\u001b[0m\n\u001b[0;32m    847\u001b[0m label_map \u001b[38;5;241m=\u001b[39m {n: i \u001b[38;5;28;01mfor\u001b[39;00m i, n \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(unique_non_noise_labels)}\n\u001b[0;32m    848\u001b[0m label_map[noise_label] \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m\n\u001b[1;32m--> 849\u001b[0m cluster_idx_vector \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(\u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mSeries\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcluster_label_vector\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mmap(label_map))\n\u001b[0;32m    851\u001b[0m label_locations \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m    852\u001b[0m cluster_sizes \u001b[38;5;241m=\u001b[39m []\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\series.py:584\u001b[0m, in \u001b[0;36mSeries.__init__\u001b[1;34m(self, data, index, dtype, name, copy, fastpath)\u001b[0m\n\u001b[0;32m    582\u001b[0m         data \u001b[38;5;241m=\u001b[39m data\u001b[38;5;241m.\u001b[39mcopy()\n\u001b[0;32m    583\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 584\u001b[0m     data \u001b[38;5;241m=\u001b[39m \u001b[43msanitize_array\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    586\u001b[0m     manager \u001b[38;5;241m=\u001b[39m _get_option(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmode.data_manager\u001b[39m\u001b[38;5;124m\"\u001b[39m, silent\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m    587\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m manager \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mblock\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\construction.py:606\u001b[0m, in \u001b[0;36msanitize_array\u001b[1;34m(data, index, dtype, copy, allow_2d)\u001b[0m\n\u001b[0;32m    604\u001b[0m subarr \u001b[38;5;241m=\u001b[39m data\n\u001b[0;32m    605\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m data\u001b[38;5;241m.\u001b[39mdtype \u001b[38;5;241m==\u001b[39m \u001b[38;5;28mobject\u001b[39m:\n\u001b[1;32m--> 606\u001b[0m     subarr \u001b[38;5;241m=\u001b[39m \u001b[43mmaybe_infer_to_datetimelike\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    607\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[0;32m    608\u001b[0m         object_index\n\u001b[0;32m    609\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m using_pyarrow_string_dtype()\n\u001b[0;32m    610\u001b[0m         \u001b[38;5;129;01mand\u001b[39;00m is_string_dtype(subarr)\n\u001b[0;32m    611\u001b[0m     ):\n\u001b[0;32m    612\u001b[0m         \u001b[38;5;66;03m# Avoid inference when string option is set\u001b[39;00m\n\u001b[0;32m    613\u001b[0m         subarr \u001b[38;5;241m=\u001b[39m data\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\pandas\\core\\dtypes\\cast.py:1181\u001b[0m, in \u001b[0;36mmaybe_infer_to_datetimelike\u001b[1;34m(value)\u001b[0m\n\u001b[0;32m   1178\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;28mtype\u001b[39m(value))  \u001b[38;5;66;03m# pragma: no cover\u001b[39;00m\n\u001b[0;32m   1179\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m value\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m!=\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m   1180\u001b[0m     \u001b[38;5;66;03m# Caller is responsible\u001b[39;00m\n\u001b[1;32m-> 1181\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(value\u001b[38;5;241m.\u001b[39mndim)  \u001b[38;5;66;03m# pragma: no cover\u001b[39;00m\n\u001b[0;32m   1183\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(value):\n\u001b[0;32m   1184\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m value\n",
      "\u001b[1;31mValueError\u001b[0m: 2"
     ]
    }
   ],
   "source": [
    "plot = datamapplot.create_interactive_plot(\n",
    "    reduced_embeddings,\n",
    "    clean_labels,\n",
    "    #hover_text= node_text,\n",
    "    font_family=\"Cinzel\",\n",
    "    enable_search=True,\n",
    "    #inline_data=False,\n",
    "    initial_zoom_fraction=0.9,\n",
    "    #offline_data_prefix=\"cord-large-cmaps-1\",\n",
    "    extra_point_data= add_info,\n",
    "    hover_text_html_template = hover_text_template,\n",
    "    marker_size_array=marker_size_array,\n",
    "    selection_handler=datamapplot.selection_handlers.DisplaySample(n_samples=25),\n",
    "    #colormaps={\"Collaboration\": df_countries.Continent, \"Type\": df_cleaned.subtypeDescription, \"Journal\": journal_top.publicationName},\n",
    "    histogram_data = df_cleaned.coverDate,\n",
    "    title = f\"Philosophy of Biology\",  \n",
    "    sub_title= f\"From <i>PhilPapers</i> (n = 6,272)\",\n",
    "    histogram_n_bins = 35,\n",
    "    histogram_settings={\n",
    "    \"histogram_log_scale\":False,\n",
    "        \"histogram_title\":\"Publication Year\",\n",
    "        \"histogram_bin_fill_color\":\"#282a36\",\n",
    "        \"histogram_bin_unselected_fill_color\":\"#b5b5b9\",\n",
    "        \"histogram_bin_selected_fill_color\":\"#f68571\",\n",
    "        \"histogram_width\":400,\n",
    "        \"histogram_height\":75,\n",
    "    },\n",
    "    \n",
    ")\n",
    "plot\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce9dd99f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "73it [00:04, 14.78it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.plotly.v1+json": {
       "config": {
        "plotlyServerURL": "https://plot.ly"
       },
       "data": [
        {
         "hoverinfo": "text",
         "hovertext": [
          "<b>Topic 0</b><br>Words: cell, elementary, particles, division, mitogenetic",
          "<b>Topic 0</b><br>Words: repair, transcription, uv, dna, pyrimidine",
          "<b>Topic 0</b><br>Words: tgfŒ≤, negative, inhibitory, growth, receptor",
          "<b>Topic 0</b><br>Words: subsystems, genome, constrained, genomic, turnover",
          "<b>Topic 0</b><br>Words: apc, protein, notch, transcription, deltex",
          "<b>Topic 0</b><br>Words: atm, ir, strand, kinase, dna",
          "<b>Topic 0</b><br>Words: synonymous, mutations, modification, pausing, nail",
          "<b>Topic 0</b><br>Words: svs, binding, enhancers, shadow, rna",
          "<b>Topic 0</b><br>Words: introns, snornas, dna, rnas, hipk2",
          "<b>Topic 0</b><br>Words: dna, chromatin, rna, histone, modifications",
          "<b>Topic 0</b><br>Words: dna, transcription, rna, binding, cell",
          "<b>Topic 0</b><br>Words: transcription, dna, histone, rna, chromatin",
          "<b>Topic 0</b><br>Words: cell, dna, blm, meiotic, rna",
          "<b>Topic 0</b><br>Words: transcription, dna, rna, cell, genome",
          "<b>Topic 0</b><br>Words: nucleosome, histone, chromatin, cell, cells",
          "<b>Topic 0</b><br>Words: dna, methyltransferases, methylcytosine, marks, regulatory",
          "<b>Topic 0</b><br>Words: replication, origins, chromatin, dna, prereplication"
         ],
         "marker": {
          "color": "#E69F00"
         },
         "mode": "lines",
         "name": "0_dna_transcription_rna_cell",
         "type": "scatter",
         "x": [
          "1936-01-16T13:26:24",
          "1992-11-15T00:00:00",
          "1994-03-01T18:14:24",
          "1995-06-16T12:28:48",
          "2001-11-30T07:40:48",
          "2003-03-17T01:55:12",
          "2007-01-30T08:38:24",
          "2010-12-15T15:21:36",
          "2012-03-31T09:36:00",
          "2013-07-16T03:50:24",
          "2014-10-30T22:04:48",
          "2016-02-14T16:19:12",
          "2017-05-31T10:33:36",
          "2018-09-15T04:48:00",
          "2019-12-30T23:02:24",
          "2022-07-31T11:31:12",
          "2023-11-15T05:45:36"
         ],
         "y": [
          2,
          3,
          1,
          3,
          3,
          1,
          1,
          5,
          10,
          13,
          42,
          42,
          14,
          17,
          12,
          1,
          1
         ]
        },
        {
         "hoverinfo": "text",
         "hovertext": [
          "<b>Topic 1</b><br>Words: leeuwenhoek, observations, population, important, investigations",
          "<b>Topic 1</b><br>Words: vries, mendel, correns, papers, bailey",
          "<b>Topic 1</b><br>Words: brooks, students, marine, teacher, experimentalism",
          "<b>Topic 1</b><br>Words: hensen, sampling, goldschmidt, randomness, reproducibility",
          "<b>Topic 1</b><br>Words: mendel, units, did, blurred, paired",
          "<b>Topic 1</b><br>Words: sages, talmudic, embryology, month, jewish",
          "<b>Topic 1</b><br>Words: mendel, hieracium, pisum, work, neglect",
          "<b>Topic 1</b><br>Words: mendelian, program, chromosome, solving, tension",
          "<b>Topic 1</b><br>Words: protozoology, protozoa, cytology, unicellular, discipline",
          "<b>Topic 1</b><br>Words: empiricist, empiricism, pavlov, inheritance, acquired",
          "<b>Topic 1</b><br>Words: mendelians, biometricians, bateson, weimar, engineering",
          "<b>Topic 1</b><br>Words: molecular, 1993, biology, japan, kluwer",
          "<b>Topic 1</b><br>Words: 1995, polysemy, liposome, smear, perplexities",
          "<b>Topic 1</b><br>Words: mcclintock, transposition, pangenes, maize, work",
          "<b>Topic 1</b><br>Words: twins, galton, patent, elements, monozygotic",
          "<b>Topic 1</b><br>Words: darbishire, ciliates, mendelism, bateson, maupas",
          "<b>Topic 1</b><br>Words: haeckel, fisher, morphology, pearson, embryology",
          "<b>Topic 1</b><br>Words: mendel, breeding, brno, heredity, genetics",
          "<b>Topic 1</b><br>Words: haeckel, illustrations, pictures, contested, alternatives",
          "<b>Topic 1</b><br>Words: loeb, mendel, godlewski, resistance, century",
          "<b>Topic 1</b><br>Words: johannsen, laboratory, molecular, field, sciences",
          "<b>Topic 1</b><br>Words: avery, mendel, lucas, work, schleiden",
          "<b>Topic 1</b><br>Words: bonner, dictyostelium, ornithology, thienemann, civic",
          "<b>Topic 1</b><br>Words: genetics, heredity, mendel, population, wff",
          "<b>Topic 1</b><br>Words: redi, insects, generation, spontaneous, atomistic",
          "<b>Topic 1</b><br>Words: goldschmidt, dobzhansky, biology, new, chromosomal",
          "<b>Topic 1</b><br>Words: kluyver, fisher, leonard, manton, dichotomy",
          "<b>Topic 1</b><br>Words: protoplasm, embryology, theory, 1960s, genetics",
          "<b>Topic 1</b><br>Words: haeckel, century, galton, miescher, mendel",
          "<b>Topic 1</b><br>Words: bonnevie, biometrician, fingerprints, mendelian, mendel",
          "<b>Topic 1</b><br>Words: bernard, heredity, century, wool, biology",
          "<b>Topic 1</b><br>Words: tower, mcclintock, monod, jacob, lewontin"
         ],
         "marker": {
          "color": "#56B4E9"
         },
         "mode": "lines",
         "name": "1_mendel_vries_century_genetics",
         "type": "scatter",
         "x": [
          "1967-01-15T19:12:00",
          "1968-05-01T13:26:24",
          "1969-08-16T07:40:48",
          "1973-07-01T14:24:00",
          "1979-12-16T09:36:00",
          "1981-04-01T03:50:24",
          "1985-02-14T10:33:36",
          "1987-09-15T23:02:24",
          "1988-12-30T17:16:48",
          "1990-04-16T11:31:12",
          "1991-08-01T05:45:36",
          "1992-11-15T00:00:00",
          "1994-03-01T18:14:24",
          "1998-01-15T00:57:36",
          "1999-05-01T19:12:00",
          "2000-08-15T13:26:24",
          "2001-11-30T07:40:48",
          "2004-06-30T20:09:36",
          "2005-10-15T14:24:00",
          "2007-01-30T08:38:24",
          "2008-05-16T02:52:48",
          "2009-08-30T21:07:12",
          "2010-12-15T15:21:36",
          "2012-03-31T09:36:00",
          "2013-07-16T03:50:24",
          "2014-10-30T22:04:48",
          "2016-02-14T16:19:12",
          "2017-05-31T10:33:36",
          "2019-12-30T23:02:24",
          "2021-04-15T17:16:48",
          "2022-07-31T11:31:12",
          "2023-11-15T05:45:36"
         ],
         "y": [
          2,
          3,
          1,
          3,
          1,
          2,
          1,
          2,
          2,
          2,
          4,
          6,
          2,
          5,
          6,
          3,
          8,
          4,
          3,
          11,
          7,
          12,
          10,
          3,
          2,
          10,
          6,
          8,
          12,
          3,
          14,
          5
         ]
        },
        {
         "hoverinfo": "text",
         "hovertext": [
          "<b>Topic 2</b><br>Words: men, bronze, tools, cultural, iron",
          "<b>Topic 2</b><br>Words: culture, wilson, lumsden, chaired, maddox",
          "<b>Topic 2</b><br>Words: culture, cultural, elaborating, biological, covered",
          "<b>Topic 2</b><br>Words: culture, cultural, human, hypotheses, costly",
          "<b>Topic 2</b><br>Words: culture, cultural, shortcoming, egalitarian, human",
          "<b>Topic 2</b><br>Words: transmission, genji, cultural, culture, traits",
          "<b>Topic 2</b><br>Words: cultural, history, social, sociocultural, theory",
          "<b>Topic 2</b><br>Words: cultural, literal, extension, culture, clone",
          "<b>Topic 2</b><br>Words: cultural, modular, recur, endlessly, multimodular",
          "<b>Topic 2</b><br>Words: cultural, blackmore, meme, imitation, dawkins",
          "<b>Topic 2</b><br>Words: polymers, idea, worldviews, cultural, worldview",
          "<b>Topic 2</b><br>Words: cultural, culture, learning, imitative, learner",
          "<b>Topic 2</b><br>Words: cultural, culture, 2006, cambridge, university",
          "<b>Topic 2</b><br>Words: cultural, culture, evolution, representations, darwinian",
          "<b>Topic 2</b><br>Words: cultural, imitating, memetics, evolution, culture",
          "<b>Topic 2</b><br>Words: cultural, social, psychology, conformity, biases",
          "<b>Topic 2</b><br>Words: cultural, attractors, sperber, cumulative, traits3",
          "<b>Topic 2</b><br>Words: cultural, culture, information, dual, transmission",
          "<b>Topic 2</b><br>Words: cultural, transmission, culture, evolution, disciplines",
          "<b>Topic 2</b><br>Words: cultural, transmission, culture, preservative, cumulative",
          "<b>Topic 2</b><br>Words: cultural, culture, attraction, learning, transmission",
          "<b>Topic 2</b><br>Words: cultural, culture, transmission, lorenz, cumulative",
          "<b>Topic 2</b><br>Words: teaching, cultural, learning, transmission, assessor",
          "<b>Topic 2</b><br>Words: cultural, culture, fidelity, cumulative, transmission",
          "<b>Topic 2</b><br>Words: cultural, culture, transmission, fidelity, cumulative",
          "<b>Topic 2</b><br>Words: cultural, culture, imitation, informational, reliability",
          "<b>Topic 2</b><br>Words: cultural, phylogenies, culture, reinforcement, evolution"
         ],
         "marker": {
          "color": "#009E73"
         },
         "mode": "lines",
         "name": "2_cultural_culture_transmission_evolutio...",
         "type": "scatter",
         "x": [
          "1974-10-16T08:38:24",
          "1983-10-31T16:19:12",
          "1986-06-01T04:48:00",
          "1991-08-01T05:45:36",
          "1992-11-15T00:00:00",
          "1996-09-30T06:43:12",
          "1998-01-15T00:57:36",
          "1999-05-01T19:12:00",
          "2000-08-15T13:26:24",
          "2001-11-30T07:40:48",
          "2003-03-17T01:55:12",
          "2004-06-30T20:09:36",
          "2005-10-15T14:24:00",
          "2007-01-30T08:38:24",
          "2008-05-16T02:52:48",
          "2009-08-30T21:07:12",
          "2010-12-15T15:21:36",
          "2012-03-31T09:36:00",
          "2013-07-16T03:50:24",
          "2014-10-30T22:04:48",
          "2016-02-14T16:19:12",
          "2017-05-31T10:33:36",
          "2018-09-15T04:48:00",
          "2019-12-30T23:02:24",
          "2021-04-15T17:16:48",
          "2022-07-31T11:31:12",
          "2023-11-15T05:45:36"
         ],
         "y": [
          1,
          1,
          1,
          3,
          3,
          2,
          2,
          2,
          1,
          3,
          1,
          3,
          15,
          12,
          3,
          1,
          2,
          8,
          6,
          4,
          8,
          10,
          3,
          7,
          12,
          5,
          4
         ]
        },
        {
         "hoverinfo": "text",
         "hovertext": [
          "<b>Topic 3</b><br>Words: genetics, reductionism, excessive, holism, 1981",
          "<b>Topic 3</b><br>Words: entrapment, guidelines, police, chorea, paramedical",
          "<b>Topic 3</b><br>Words: project, hgp, issues, section, consent",
          "<b>Topic 3</b><br>Words: rawls, eugenics, eugenic, principles, liberties",
          "<b>Topic 3</b><br>Words: project, genome, transforming, human, story",
          "<b>Topic 3</b><br>Words: insurers, testing, proposer, emptive, insurer",
          "<b>Topic 3</b><br>Words: hgp, eugenic, project, insurance, ceph",
          "<b>Topic 3</b><br>Words: bioethics, disability, medical, era, oppression",
          "<b>Topic 3</b><br>Words: disability, genetics, genetic, personal, ethical",
          "<b>Topic 3</b><br>Words: providers, consumer, team, clinical, consumers",
          "<b>Topic 3</b><br>Words: nursing, genetic, health, human, nurses",
          "<b>Topic 3</b><br>Words: diets, genetics, geneticists, bioethicists, computer",
          "<b>Topic 3</b><br>Words: consent, genetic, bioethicists, therapeutic, medical",
          "<b>Topic 3</b><br>Words: pharmacogenetics, genetic, consent, ethics, ethical",
          "<b>Topic 3</b><br>Words: gonadex, ethnic, leo, stigmatization, information",
          "<b>Topic 3</b><br>Words: research, genetics, genetic, ethical, researchers",
          "<b>Topic 3</b><br>Words: research, consent, genetics, genetic, family",
          "<b>Topic 3</b><br>Words: consent, ethical, enhancement, global, hype",
          "<b>Topic 3</b><br>Words: pr√ºm, health, dohad, parenthood, gestation",
          "<b>Topic 3</b><br>Words: bioethics, health, gene, communitarian, public",
          "<b>Topic 3</b><br>Words: project, hgap, sequencing, archival, wes",
          "<b>Topic 3</b><br>Words: forensic, recontact, ests, duty, 23andme",
          "<b>Topic 3</b><br>Words: donor, cytogeneticists, conceived, human, sequencing",
          "<b>Topic 3</b><br>Words: bgi, donor, experiences, interpretations, blood",
          "<b>Topic 3</b><br>Words: genomics, results, return, disability, research",
          "<b>Topic 3</b><br>Words: maid, patient, medical, deaf, genetic",
          "<b>Topic 3</b><br>Words: pgss, idealisations, exceptionalism, epigenetics, genomics"
         ],
         "marker": {
          "color": "#F0E442"
         },
         "mode": "lines",
         "name": "3_genetic_genetics_ethical_research",
         "type": "scatter",
         "x": [
          "1979-12-16T09:36:00",
          "1983-10-31T16:19:12",
          "1991-08-01T05:45:36",
          "1992-11-15T00:00:00",
          "1994-03-01T18:14:24",
          "1995-06-16T12:28:48",
          "1996-09-30T06:43:12",
          "1998-01-15T00:57:36",
          "1999-05-01T19:12:00",
          "2000-08-15T13:26:24",
          "2001-11-30T07:40:48",
          "2003-03-17T01:55:12",
          "2004-06-30T20:09:36",
          "2005-10-15T14:24:00",
          "2007-01-30T08:38:24",
          "2008-05-16T02:52:48",
          "2009-08-30T21:07:12",
          "2010-12-15T15:21:36",
          "2012-03-31T09:36:00",
          "2013-07-16T03:50:24",
          "2014-10-30T22:04:48",
          "2017-05-31T10:33:36",
          "2018-09-15T04:48:00",
          "2019-12-30T23:02:24",
          "2021-04-15T17:16:48",
          "2022-07-31T11:31:12",
          "2023-11-15T05:45:36"
         ],
         "y": [
          2,
          2,
          3,
          1,
          2,
          1,
          7,
          2,
          6,
          2,
          4,
          5,
          5,
          8,
          3,
          6,
          10,
          2,
          7,
          6,
          3,
          5,
          4,
          3,
          7,
          8,
          6
         ]
        },
        {
         "hoverinfo": "text",
         "hovertext": [
          "<b>Topic 4</b><br>Words: energetic, organism, increase, efficiency, economy",
          "<b>Topic 4</b><br>Words: miles, hour, gradual, justly, walk",
          "<b>Topic 4</b><br>Words: thermodynamic, systems, reproducing, combine, thermodynamics",
          "<b>Topic 4</b><br>Words: economy, optimality, economization, phylogenetic, economical",
          "<b>Topic 4</b><br>Words: brooks, information, abused, occur, theory",
          "<b>Topic 4</b><br>Words: atlan, organizing, potential, self, conceived",
          "<b>Topic 4</b><br>Words: systems, energy, biological, ecological, genealogical",
          "<b>Topic 4</b><br>Words: railroads, railroad, perfected, empirical, systems",
          "<b>Topic 4</b><br>Words: energy, dissipation, maximum, thermodynamic, action",
          "<b>Topic 4</b><br>Words: production, energy, complexity, physical, entropy",
          "<b>Topic 4</b><br>Words: detection, chaos, dynamics, thermodynamics, revolution",
          "<b>Topic 4</b><br>Words: mst, organizing, self, systems, complementary",
          "<b>Topic 4</b><br>Words: maximum, optimizing, uniformity, thermodynamic, consumption",
          "<b>Topic 4</b><br>Words: programming, complexity, biochemical, simulations, computer",
          "<b>Topic 4</b><br>Words: organization, self, operators, incompleteness, cognitive",
          "<b>Topic 4</b><br>Words: computation, neurotransmitters, hormones, cognitive, intentional",
          "<b>Topic 4</b><br>Words: economics, neoclassical, methodology, universities, stiglitz",
          "<b>Topic 4</b><br>Words: igus, protein, algorithms, folding, neurally",
          "<b>Topic 4</b><br>Words: systems, aggregativity, complex, purpose, design",
          "<b>Topic 4</b><br>Words: self, organization, evolution, systems, organizing",
          "<b>Topic 4</b><br>Words: simulation, emergence, algorithmic, simondon, technical",
          "<b>Topic 4</b><br>Words: universe, energy, complexity, quantum, ns",
          "<b>Topic 4</b><br>Words: cloning, emergent, complexity, emergence, phenomenon",
          "<b>Topic 4</b><br>Words: physical, reductive, constraint, evolution, technological",
          "<b>Topic 4</b><br>Words: simulations, processes, physics, organism, experiments",
          "<b>Topic 4</b><br>Words: complexity, ld, highest, time, needed",
          "<b>Topic 4</b><br>Words: regulation, constitutive, regime, self, organism",
          "<b>Topic 4</b><br>Words: barrier, cooperation, organization, overcome, managed",
          "<b>Topic 4</b><br>Words: quantum, sadra, mulla, motion, particles",
          "<b>Topic 4</b><br>Words: realization, transmission, multiple, physics, principle",
          "<b>Topic 4</b><br>Words: gravitational, directed, goal, energy, trends",
          "<b>Topic 4</b><br>Words: physics, energy, systems, thermodynamic, matter"
         ],
         "marker": {
          "color": "#D55E00"
         },
         "mode": "lines",
         "name": "4_systems_energy_complexity_self",
         "type": "scatter",
         "x": [
          "1969-08-16T07:40:48",
          "1970-12-01T01:55:12",
          "1976-01-31T02:52:48",
          "1977-05-16T21:07:12",
          "1985-02-14T10:33:36",
          "1986-06-01T04:48:00",
          "1988-12-30T17:16:48",
          "1990-04-16T11:31:12",
          "1991-08-01T05:45:36",
          "1992-11-15T00:00:00",
          "1994-03-01T18:14:24",
          "1995-06-16T12:28:48",
          "1996-09-30T06:43:12",
          "1998-01-15T00:57:36",
          "2000-08-15T13:26:24",
          "2001-11-30T07:40:48",
          "2003-03-17T01:55:12",
          "2004-06-30T20:09:36",
          "2005-10-15T14:24:00",
          "2007-01-30T08:38:24",
          "2008-05-16T02:52:48",
          "2009-08-30T21:07:12",
          "2010-12-15T15:21:36",
          "2012-03-31T09:36:00",
          "2013-07-16T03:50:24",
          "2014-10-30T22:04:48",
          "2016-02-14T16:19:12",
          "2018-09-15T04:48:00",
          "2019-12-30T23:02:24",
          "2021-04-15T17:16:48",
          "2022-07-31T11:31:12",
          "2023-11-15T05:45:36"
         ],
         "y": [
          1,
          1,
          1,
          1,
          1,
          2,
          3,
          4,
          3,
          6,
          4,
          2,
          2,
          3,
          2,
          3,
          1,
          3,
          3,
          4,
          7,
          7,
          6,
          4,
          7,
          2,
          3,
          2,
          7,
          4,
          7,
          2
         ]
        }
       ],
       "layout": {
        "height": 450,
        "hoverlabel": {
         "bgcolor": "white",
         "font": {
          "family": "Rockwell",
          "size": 16
         }
        },
        "legend": {
         "title": {
          "text": "<b>Global Topic Representation"
         }
        },
        "template": {
         "data": {
          "bar": [
           {
            "error_x": {
             "color": "rgb(36,36,36)"
            },
            "error_y": {
             "color": "rgb(36,36,36)"
            },
            "marker": {
             "line": {
              "color": "white",
              "width": 0.5
             },
             "pattern": {
              "fillmode": "overlay",
              "size": 10,
              "solidity": 0.2
             }
            },
            "type": "bar"
           }
          ],
          "barpolar": [
           {
            "marker": {
             "line": {
              "color": "white",
              "width": 0.5
             },
             "pattern": {
              "fillmode": "overlay",
              "size": 10,
              "solidity": 0.2
             }
            },
            "type": "barpolar"
           }
          ],
          "carpet": [
           {
            "aaxis": {
             "endlinecolor": "rgb(36,36,36)",
             "gridcolor": "white",
             "linecolor": "white",
             "minorgridcolor": "white",
             "startlinecolor": "rgb(36,36,36)"
            },
            "baxis": {
             "endlinecolor": "rgb(36,36,36)",
             "gridcolor": "white",
             "linecolor": "white",
             "minorgridcolor": "white",
             "startlinecolor": "rgb(36,36,36)"
            },
            "type": "carpet"
           }
          ],
          "choropleth": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "type": "choropleth"
           }
          ],
          "contour": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "contour"
           }
          ],
          "contourcarpet": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "type": "contourcarpet"
           }
          ],
          "heatmap": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "heatmap"
           }
          ],
          "heatmapgl": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "heatmapgl"
           }
          ],
          "histogram": [
           {
            "marker": {
             "line": {
              "color": "white",
              "width": 0.6
             }
            },
            "type": "histogram"
           }
          ],
          "histogram2d": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "histogram2d"
           }
          ],
          "histogram2dcontour": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "histogram2dcontour"
           }
          ],
          "mesh3d": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "type": "mesh3d"
           }
          ],
          "parcoords": [
           {
            "line": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "parcoords"
           }
          ],
          "pie": [
           {
            "automargin": true,
            "type": "pie"
           }
          ],
          "scatter": [
           {
            "fillpattern": {
             "fillmode": "overlay",
             "size": 10,
             "solidity": 0.2
            },
            "type": "scatter"
           }
          ],
          "scatter3d": [
           {
            "line": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scatter3d"
           }
          ],
          "scattercarpet": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scattercarpet"
           }
          ],
          "scattergeo": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scattergeo"
           }
          ],
          "scattergl": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scattergl"
           }
          ],
          "scattermapbox": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scattermapbox"
           }
          ],
          "scatterpolar": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scatterpolar"
           }
          ],
          "scatterpolargl": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scatterpolargl"
           }
          ],
          "scatterternary": [
           {
            "marker": {
             "colorbar": {
              "outlinewidth": 1,
              "tickcolor": "rgb(36,36,36)",
              "ticks": "outside"
             }
            },
            "type": "scatterternary"
           }
          ],
          "surface": [
           {
            "colorbar": {
             "outlinewidth": 1,
             "tickcolor": "rgb(36,36,36)",
             "ticks": "outside"
            },
            "colorscale": [
             [
              0,
              "#440154"
             ],
             [
              0.1111111111111111,
              "#482878"
             ],
             [
              0.2222222222222222,
              "#3e4989"
             ],
             [
              0.3333333333333333,
              "#31688e"
             ],
             [
              0.4444444444444444,
              "#26828e"
             ],
             [
              0.5555555555555556,
              "#1f9e89"
             ],
             [
              0.6666666666666666,
              "#35b779"
             ],
             [
              0.7777777777777778,
              "#6ece58"
             ],
             [
              0.8888888888888888,
              "#b5de2b"
             ],
             [
              1,
              "#fde725"
             ]
            ],
            "type": "surface"
           }
          ],
          "table": [
           {
            "cells": {
             "fill": {
              "color": "rgb(237,237,237)"
             },
             "line": {
              "color": "white"
             }
            },
            "header": {
             "fill": {
              "color": "rgb(217,217,217)"
             },
             "line": {
              "color": "white"
             }
            },
            "type": "table"
           }
          ]
         },
         "layout": {
          "annotationdefaults": {
           "arrowhead": 0,
           "arrowwidth": 1
          },
          "autotypenumbers": "strict",
          "coloraxis": {
           "colorbar": {
            "outlinewidth": 1,
            "tickcolor": "rgb(36,36,36)",
            "ticks": "outside"
           }
          },
          "colorscale": {
           "diverging": [
            [
             0,
             "rgb(103,0,31)"
            ],
            [
             0.1,
             "rgb(178,24,43)"
            ],
            [
             0.2,
             "rgb(214,96,77)"
            ],
            [
             0.3,
             "rgb(244,165,130)"
            ],
            [
             0.4,
             "rgb(253,219,199)"
            ],
            [
             0.5,
             "rgb(247,247,247)"
            ],
            [
             0.6,
             "rgb(209,229,240)"
            ],
            [
             0.7,
             "rgb(146,197,222)"
            ],
            [
             0.8,
             "rgb(67,147,195)"
            ],
            [
             0.9,
             "rgb(33,102,172)"
            ],
            [
             1,
             "rgb(5,48,97)"
            ]
           ],
           "sequential": [
            [
             0,
             "#440154"
            ],
            [
             0.1111111111111111,
             "#482878"
            ],
            [
             0.2222222222222222,
             "#3e4989"
            ],
            [
             0.3333333333333333,
             "#31688e"
            ],
            [
             0.4444444444444444,
             "#26828e"
            ],
            [
             0.5555555555555556,
             "#1f9e89"
            ],
            [
             0.6666666666666666,
             "#35b779"
            ],
            [
             0.7777777777777778,
             "#6ece58"
            ],
            [
             0.8888888888888888,
             "#b5de2b"
            ],
            [
             1,
             "#fde725"
            ]
           ],
           "sequentialminus": [
            [
             0,
             "#440154"
            ],
            [
             0.1111111111111111,
             "#482878"
            ],
            [
             0.2222222222222222,
             "#3e4989"
            ],
            [
             0.3333333333333333,
             "#31688e"
            ],
            [
             0.4444444444444444,
             "#26828e"
            ],
            [
             0.5555555555555556,
             "#1f9e89"
            ],
            [
             0.6666666666666666,
             "#35b779"
            ],
            [
             0.7777777777777778,
             "#6ece58"
            ],
            [
             0.8888888888888888,
             "#b5de2b"
            ],
            [
             1,
             "#fde725"
            ]
           ]
          },
          "colorway": [
           "#1F77B4",
           "#FF7F0E",
           "#2CA02C",
           "#D62728",
           "#9467BD",
           "#8C564B",
           "#E377C2",
           "#7F7F7F",
           "#BCBD22",
           "#17BECF"
          ],
          "font": {
           "color": "rgb(36,36,36)"
          },
          "geo": {
           "bgcolor": "white",
           "lakecolor": "white",
           "landcolor": "white",
           "showlakes": true,
           "showland": true,
           "subunitcolor": "white"
          },
          "hoverlabel": {
           "align": "left"
          },
          "hovermode": "closest",
          "mapbox": {
           "style": "light"
          },
          "paper_bgcolor": "white",
          "plot_bgcolor": "white",
          "polar": {
           "angularaxis": {
            "gridcolor": "rgb(232,232,232)",
            "linecolor": "rgb(36,36,36)",
            "showgrid": false,
            "showline": true,
            "ticks": "outside"
           },
           "bgcolor": "white",
           "radialaxis": {
            "gridcolor": "rgb(232,232,232)",
            "linecolor": "rgb(36,36,36)",
            "showgrid": false,
            "showline": true,
            "ticks": "outside"
           }
          },
          "scene": {
           "xaxis": {
            "backgroundcolor": "white",
            "gridcolor": "rgb(232,232,232)",
            "gridwidth": 2,
            "linecolor": "rgb(36,36,36)",
            "showbackground": true,
            "showgrid": false,
            "showline": true,
            "ticks": "outside",
            "zeroline": false,
            "zerolinecolor": "rgb(36,36,36)"
           },
           "yaxis": {
            "backgroundcolor": "white",
            "gridcolor": "rgb(232,232,232)",
            "gridwidth": 2,
            "linecolor": "rgb(36,36,36)",
            "showbackground": true,
            "showgrid": false,
            "showline": true,
            "ticks": "outside",
            "zeroline": false,
            "zerolinecolor": "rgb(36,36,36)"
           },
           "zaxis": {
            "backgroundcolor": "white",
            "gridcolor": "rgb(232,232,232)",
            "gridwidth": 2,
            "linecolor": "rgb(36,36,36)",
            "showbackground": true,
            "showgrid": false,
            "showline": true,
            "ticks": "outside",
            "zeroline": false,
            "zerolinecolor": "rgb(36,36,36)"
           }
          },
          "shapedefaults": {
           "fillcolor": "black",
           "line": {
            "width": 0
           },
           "opacity": 0.3
          },
          "ternary": {
           "aaxis": {
            "gridcolor": "rgb(232,232,232)",
            "linecolor": "rgb(36,36,36)",
            "showgrid": false,
            "showline": true,
            "ticks": "outside"
           },
           "baxis": {
            "gridcolor": "rgb(232,232,232)",
            "linecolor": "rgb(36,36,36)",
            "showgrid": false,
            "showline": true,
            "ticks": "outside"
           },
           "bgcolor": "white",
           "caxis": {
            "gridcolor": "rgb(232,232,232)",
            "linecolor": "rgb(36,36,36)",
            "showgrid": false,
            "showline": true,
            "ticks": "outside"
           }
          },
          "title": {
           "x": 0.05
          },
          "xaxis": {
           "automargin": true,
           "gridcolor": "rgb(232,232,232)",
           "linecolor": "rgb(36,36,36)",
           "showgrid": false,
           "showline": true,
           "ticks": "outside",
           "title": {
            "standoff": 15
           },
           "zeroline": false,
           "zerolinecolor": "rgb(36,36,36)"
          },
          "yaxis": {
           "automargin": true,
           "gridcolor": "rgb(232,232,232)",
           "linecolor": "rgb(36,36,36)",
           "showgrid": false,
           "showline": true,
           "ticks": "outside",
           "title": {
            "standoff": 15
           },
           "zeroline": false,
           "zerolinecolor": "rgb(36,36,36)"
          }
         }
        },
        "title": {
         "font": {
          "color": "Black",
          "size": 22
         },
         "text": "<b>Topics over Time</b>",
         "x": 0.4,
         "xanchor": "center",
         "y": 0.95,
         "yanchor": "top"
        },
        "width": 1250,
        "xaxis": {
         "showgrid": true
        },
        "yaxis": {
         "showgrid": true,
         "title": {
          "text": "Frequency"
         }
        }
       }
      }
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "timestamps = df_cleaned.coverDate.to_list()\n",
    "topics_over_time = topic_model.topics_over_time(docs, timestamps, nr_bins= 100)\n",
    "topics_over_time\n",
    "\n",
    "\n",
    "topic_model.visualize_topics_over_time(topics_over_time, top_n_topics=5)\n",
    "topic_model.visualize_topics_over_time(topics_over_time, top_n_topics=5)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f026db8",
   "metadata": {},
   "source": [
    "## Save the map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f284505",
   "metadata": {},
   "outputs": [],
   "source": [
    "plot.save(PATH_TO_VIZ + \"PHILPAPERS_all_philo_of_biology_articles_BERTopic.html\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14e523a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "llama_model_loader: loaded meta data with 20 key-value pairs and 291 tensors from C:/Users/jacob/OneDrive/Bureau/openhermes-2.5-mistral-7b.Q4_K_M.gguf (version GGUF V3 (latest))\n",
      "llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.\n",
      "llama_model_loader: - kv   0:                       general.architecture str              = llama\n",
      "llama_model_loader: - kv   1:                               general.name str              = teknium_openhermes-2.5-mistral-7b\n",
      "llama_model_loader: - kv   2:                       llama.context_length u32              = 32768\n",
      "llama_model_loader: - kv   3:                     llama.embedding_length u32              = 4096\n",
      "llama_model_loader: - kv   4:                          llama.block_count u32              = 32\n",
      "llama_model_loader: - kv   5:                  llama.feed_forward_length u32              = 14336\n",
      "llama_model_loader: - kv   6:                 llama.rope.dimension_count u32              = 128\n",
      "llama_model_loader: - kv   7:                 llama.attention.head_count u32              = 32\n",
      "llama_model_loader: - kv   8:              llama.attention.head_count_kv u32              = 8\n",
      "llama_model_loader: - kv   9:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010\n",
      "llama_model_loader: - kv  10:                       llama.rope.freq_base f32              = 10000.000000\n",
      "llama_model_loader: - kv  11:                          general.file_type u32              = 15\n",
      "llama_model_loader: - kv  12:                       tokenizer.ggml.model str              = llama\n",
      "llama_model_loader: - kv  13:                      tokenizer.ggml.tokens arr[str,32002]   = [\"<unk>\", \"<s>\", \"</s>\", \"<0x00>\", \"<...\n",
      "llama_model_loader: - kv  14:                      tokenizer.ggml.scores arr[f32,32002]   = [0.000000, 0.000000, 0.000000, 0.0000...\n",
      "llama_model_loader: - kv  15:                  tokenizer.ggml.token_type arr[i32,32002]   = [2, 3, 3, 6, 6, 6, 6, 6, 6, 6, 6, 6, ...\n",
      "llama_model_loader: - kv  16:                tokenizer.ggml.bos_token_id u32              = 1\n",
      "llama_model_loader: - kv  17:                tokenizer.ggml.eos_token_id u32              = 32000\n",
      "llama_model_loader: - kv  18:            tokenizer.ggml.padding_token_id u32              = 0\n",
      "llama_model_loader: - kv  19:               general.quantization_version u32              = 2\n",
      "llama_model_loader: - type  f32:   65 tensors\n",
      "llama_model_loader: - type q4_K:  193 tensors\n",
      "llama_model_loader: - type q6_K:   33 tensors\n",
      "llm_load_vocab: control-looking token:  32000 '<|im_end|>' was not control-type; this is probably a bug in the model. its type will be overridden\n",
      "llm_load_vocab: control token:      2 '</s>' is not marked as EOG\n",
      "llm_load_vocab: control token:      1 '<s>' is not marked as EOG\n",
      "llm_load_vocab: special tokens cache size = 5\n",
      "llm_load_vocab: token to piece cache size = 0.1637 MB\n",
      "llm_load_print_meta: format           = GGUF V3 (latest)\n",
      "llm_load_print_meta: arch             = llama\n",
      "llm_load_print_meta: vocab type       = SPM\n",
      "llm_load_print_meta: n_vocab          = 32002\n",
      "llm_load_print_meta: n_merges         = 0\n",
      "llm_load_print_meta: vocab_only       = 0\n",
      "llm_load_print_meta: n_ctx_train      = 32768\n",
      "llm_load_print_meta: n_embd           = 4096\n",
      "llm_load_print_meta: n_layer          = 32\n",
      "llm_load_print_meta: n_head           = 32\n",
      "llm_load_print_meta: n_head_kv        = 8\n",
      "llm_load_print_meta: n_rot            = 128\n",
      "llm_load_print_meta: n_swa            = 0\n",
      "llm_load_print_meta: n_embd_head_k    = 128\n",
      "llm_load_print_meta: n_embd_head_v    = 128\n",
      "llm_load_print_meta: n_gqa            = 4\n",
      "llm_load_print_meta: n_embd_k_gqa     = 1024\n",
      "llm_load_print_meta: n_embd_v_gqa     = 1024\n",
      "llm_load_print_meta: f_norm_eps       = 0.0e+00\n",
      "llm_load_print_meta: f_norm_rms_eps   = 1.0e-05\n",
      "llm_load_print_meta: f_clamp_kqv      = 0.0e+00\n",
      "llm_load_print_meta: f_max_alibi_bias = 0.0e+00\n",
      "llm_load_print_meta: f_logit_scale    = 0.0e+00\n",
      "llm_load_print_meta: n_ff             = 14336\n",
      "llm_load_print_meta: n_expert         = 0\n",
      "llm_load_print_meta: n_expert_used    = 0\n",
      "llm_load_print_meta: causal attn      = 1\n",
      "llm_load_print_meta: pooling type     = 0\n",
      "llm_load_print_meta: rope type        = 0\n",
      "llm_load_print_meta: rope scaling     = linear\n",
      "llm_load_print_meta: freq_base_train  = 10000.0\n",
      "llm_load_print_meta: freq_scale_train = 1\n",
      "llm_load_print_meta: n_ctx_orig_yarn  = 32768\n",
      "llm_load_print_meta: rope_finetuned   = unknown\n",
      "llm_load_print_meta: ssm_d_conv       = 0\n",
      "llm_load_print_meta: ssm_d_inner      = 0\n",
      "llm_load_print_meta: ssm_d_state      = 0\n",
      "llm_load_print_meta: ssm_dt_rank      = 0\n",
      "llm_load_print_meta: ssm_dt_b_c_rms   = 0\n",
      "llm_load_print_meta: model type       = 7B\n",
      "llm_load_print_meta: model ftype      = Q4_K - Medium\n",
      "llm_load_print_meta: model params     = 7.24 B\n",
      "llm_load_print_meta: model size       = 4.07 GiB (4.83 BPW) \n",
      "llm_load_print_meta: general.name     = teknium_openhermes-2.5-mistral-7b\n",
      "llm_load_print_meta: BOS token        = 1 '<s>'\n",
      "llm_load_print_meta: EOS token        = 32000 '<|im_end|>'\n",
      "llm_load_print_meta: EOT token        = 32000 '<|im_end|>'\n",
      "llm_load_print_meta: UNK token        = 0 '<unk>'\n",
      "llm_load_print_meta: PAD token        = 0 '<unk>'\n",
      "llm_load_print_meta: LF token         = 13 '<0x0A>'\n",
      "llm_load_print_meta: EOG token        = 32000 '<|im_end|>'\n",
      "llm_load_print_meta: max token length = 48\n",
      "llm_load_tensors: tensor 'token_embd.weight' (q4_K) (and 290 others) cannot be used with preferred buffer type CPU_AARCH64, using CPU instead\n",
      "llm_load_tensors:   CPU_Mapped model buffer size =  4165.38 MiB\n",
      ".................................................................................................\n",
      "llama_new_context_with_model: n_seq_max     = 1\n",
      "llama_new_context_with_model: n_ctx         = 4096\n",
      "llama_new_context_with_model: n_ctx_per_seq = 4096\n",
      "llama_new_context_with_model: n_batch       = 512\n",
      "llama_new_context_with_model: n_ubatch      = 512\n",
      "llama_new_context_with_model: flash_attn    = 0\n",
      "llama_new_context_with_model: freq_base     = 10000.0\n",
      "llama_new_context_with_model: freq_scale    = 1\n",
      "llama_new_context_with_model: n_ctx_per_seq (4096) < n_ctx_train (32768) -- the full capacity of the model will not be utilized\n",
      "llama_kv_cache_init:        CPU KV buffer size =   512.00 MiB\n",
      "llama_new_context_with_model: KV self size  =  512.00 MiB, K (f16):  256.00 MiB, V (f16):  256.00 MiB\n",
      "llama_new_context_with_model:        CPU  output buffer size =     0.12 MiB\n",
      "llama_new_context_with_model:        CPU compute buffer size =   296.01 MiB\n",
      "llama_new_context_with_model: graph nodes  = 1030\n",
      "llama_new_context_with_model: graph splits = 1\n",
      "AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | AMX_INT8 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | RISCV_VECT = 0 | WASM_SIMD = 0 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | \n",
      "Model metadata: {'general.name': 'teknium_openhermes-2.5-mistral-7b', 'general.architecture': 'llama', 'llama.context_length': '32768', 'llama.rope.dimension_count': '128', 'llama.embedding_length': '4096', 'llama.block_count': '32', 'llama.feed_forward_length': '14336', 'llama.attention.head_count': '32', 'tokenizer.ggml.eos_token_id': '32000', 'general.file_type': '15', 'llama.attention.head_count_kv': '8', 'llama.attention.layer_norm_rms_epsilon': '0.000010', 'llama.rope.freq_base': '10000.000000', 'tokenizer.ggml.model': 'llama', 'general.quantization_version': '2', 'tokenizer.ggml.bos_token_id': '1', 'tokenizer.ggml.padding_token_id': '0'}\n",
      "Using fallback chat format: llama-2\n"
     ]
    }
   ],
   "source": [
    "from llama_cpp import Llama\n",
    "\n",
    "# Use llama.cpp to load in a Quantized LLM\n",
    "llm = Llama(model_path=\"C:/Users/jacob/OneDrive/Bureau/openhermes-2.5-mistral-7b.Q4_K_M.gguf\", n_gpu_layers=-1, n_ctx=4096, stop=[\"Q:\", \"\\n\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8426902",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'llm' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[1], line 12\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01mbertopic\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mrepresentation\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m KeyBERTInspired, LlamaCPP\n\u001b[0;32m      3\u001b[0m prompt \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\"\"\u001b[39m\u001b[38;5;124m Q:\u001b[39m\n\u001b[0;32m      4\u001b[0m \u001b[38;5;124mI have a topic that contains the following keywords: \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m[KEYWORDS]\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m.\u001b[39m\n\u001b[0;32m      5\u001b[0m \n\u001b[0;32m      6\u001b[0m \u001b[38;5;124mBased on the above information, can you give a short label of the topic of at most 5 words?\u001b[39m\n\u001b[0;32m      7\u001b[0m \u001b[38;5;124mA:\u001b[39m\n\u001b[0;32m      8\u001b[0m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[0;32m     10\u001b[0m representation_model \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m     11\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mKeyBERT\u001b[39m\u001b[38;5;124m\"\u001b[39m: KeyBERTInspired(),\n\u001b[1;32m---> 12\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mLLM\u001b[39m\u001b[38;5;124m\"\u001b[39m: LlamaCPP(\u001b[43mllm\u001b[49m, prompt\u001b[38;5;241m=\u001b[39mprompt),\n\u001b[0;32m     13\u001b[0m }\n",
      "\u001b[1;31mNameError\u001b[0m: name 'llm' is not defined"
     ]
    }
   ],
   "source": [
    "from bertopic.representation import KeyBERTInspired, LlamaCPP\n",
    "\n",
    "prompt = \"\"\" Q:\n",
    "I have a topic that contains the following keywords: '[KEYWORDS]'.\n",
    "\n",
    "Based on the above information, can you give a short label of the topic of at most 5 words?\n",
    "A:\n",
    "\"\"\"\n",
    "\n",
    "representation_model = {\n",
    "    \"KeyBERT\": KeyBERTInspired(),\n",
    "    \"LLM\": LlamaCPP(llm, prompt=prompt),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c52f1042",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'docs' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[5], line 4\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01msentence_transformers\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m SentenceTransformer\n\u001b[0;32m      3\u001b[0m embedding_model \u001b[38;5;241m=\u001b[39m SentenceTransformer(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mBAAI/bge-small-en\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m----> 4\u001b[0m embeddings \u001b[38;5;241m=\u001b[39m embedding_model\u001b[38;5;241m.\u001b[39mencode(\u001b[43mdocs\u001b[49m, show_progress_bar\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m      6\u001b[0m \u001b[38;5;66;03m# Pre-reduce embeddings for visualization purposes\u001b[39;00m\n\u001b[0;32m      7\u001b[0m reduced_embeddings \u001b[38;5;241m=\u001b[39m UMAP(n_neighbors\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m150\u001b[39m, n_components\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m2\u001b[39m, min_dist\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0.0\u001b[39m, metric\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mcosine\u001b[39m\u001b[38;5;124m'\u001b[39m, random_state\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m42\u001b[39m)\u001b[38;5;241m.\u001b[39mfit_transform(embeddings)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'docs' is not defined"
     ]
    }
   ],
   "source": [
    "# Pre-calculate embeddings\n",
    "from sentence_transformers import SentenceTransformer\n",
    "embedding_model = SentenceTransformer('BAAI/bge-small-en')\n",
    "embeddings = embedding_model.encode(docs, show_progress_bar=True)\n",
    "\n",
    "# Pre-reduce embeddings for visualization purposes\n",
    "reduced_embeddings = UMAP(n_neighbors=150, n_components=2, min_dist=0.0, metric='cosine', random_state=42).fit_transform(embeddings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94e72edc",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'embedding_model' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[4], line 14\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[38;5;66;03m# Create a CountVectorizer with the custom stopwords\u001b[39;00m\n\u001b[0;32m      8\u001b[0m vectorizer \u001b[38;5;241m=\u001b[39m CountVectorizer(stop_words\u001b[38;5;241m=\u001b[39mcustom_stopwords)\n\u001b[0;32m     12\u001b[0m topic_model \u001b[38;5;241m=\u001b[39m BERTopic(\n\u001b[0;32m     13\u001b[0m   \u001b[38;5;66;03m# Sub-models\u001b[39;00m\n\u001b[1;32m---> 14\u001b[0m   embedding_model\u001b[38;5;241m=\u001b[39m\u001b[43membedding_model\u001b[49m,\n\u001b[0;32m     15\u001b[0m   umap_model\u001b[38;5;241m=\u001b[39mumap_model,\n\u001b[0;32m     16\u001b[0m   hdbscan_model\u001b[38;5;241m=\u001b[39mhdbscan_model,\n\u001b[0;32m     17\u001b[0m   representation_model\u001b[38;5;241m=\u001b[39mrepresentation_model,\n\u001b[0;32m     18\u001b[0m   vectorizer_model\u001b[38;5;241m=\u001b[39mvectorizer,\n\u001b[0;32m     19\u001b[0m   \n\u001b[0;32m     20\u001b[0m   \u001b[38;5;66;03m# Hyperparameters\u001b[39;00m\n\u001b[0;32m     21\u001b[0m   top_n_words\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m10\u001b[39m,\n\u001b[0;32m     22\u001b[0m   verbose\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m     23\u001b[0m )\n\u001b[0;32m     25\u001b[0m \u001b[38;5;66;03m# Train model\u001b[39;00m\n\u001b[0;32m     26\u001b[0m topics, probs \u001b[38;5;241m=\u001b[39m topic_model\u001b[38;5;241m.\u001b[39mfit_transform(docs, embeddings)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'embedding_model' is not defined"
     ]
    }
   ],
   "source": [
    "from bertopic import BERTopic\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "# Define a custom list of stopwords (or use an extended one)\n",
    "custom_stopwords = [\"the\", \"and\", \"or\", \"in\", \"on\", \"at\", \"of\", \"is\", \"to\", \"be\", \"with\", \"are\", \"that\", \"this\", \"by\", \"for\"]\n",
    "\n",
    "# Create a CountVectorizer with the custom stopwords\n",
    "vectorizer = CountVectorizer(stop_words=custom_stopwords)\n",
    "\n",
    "\n",
    "\n",
    "topic_model = BERTopic(\n",
    "  # Sub-models\n",
    "  embedding_model=embedding_model,\n",
    "  umap_model=umap_model,\n",
    "  hdbscan_model=hdbscan_model,\n",
    "  representation_model=representation_model,\n",
    "  vectorizer_model=vectorizer,\n",
    "  \n",
    "  # Hyperparameters\n",
    "  top_n_words=10,\n",
    "  verbose=True\n",
    ")\n",
    "\n",
    "# Train model\n",
    "topics, probs = topic_model.fit_transform(docs, embeddings)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "744751b7",
   "metadata": {},
   "source": [
    "# Biology from BioArxiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba511e57",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Directory\n",
    "PATH_TO_DATA = r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\DATA\\\\\"\n",
    "PATH_TO_VIZ = r\"C:\\Users\\jacob\\OneDrive - Universit√© Laval\\biophilo\\Visualisation\\\\\"\n",
    "# Packages to import. \n",
    "import pandas as pd \n",
    "import numpy as np\n",
    "import datamapplot\n",
    "\n",
    "\n",
    "from sentence_transformers import SentenceTransformer\n",
    "from bertopic import BERTopic\n",
    "from umap import UMAP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b864e694",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = pd.read_csv(\"C:\\\\Users\\\\jacob\\\\OneDrive - Universit√© Laval\\\\biophilo\\\\Data\\\\PHILOSOPHY_OF_BIOLOGY_ALL.csv\")\n",
    "\n",
    "\n",
    "#test\n",
    "df = pd.read_csv(\"C:/Users/jacob/OneDrive - Universit√© Laval/biophilo/Data/BIO_ARXIV_DATA/all_bio_arxiv.csv\")\n",
    "df = df.fillna(\"NULL\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7514dcd8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>preprint_doi</th>\n",
       "      <th>published_doi</th>\n",
       "      <th>published_journal</th>\n",
       "      <th>preprint_platform</th>\n",
       "      <th>preprint_title</th>\n",
       "      <th>preprint_authors</th>\n",
       "      <th>preprint_category</th>\n",
       "      <th>preprint_date</th>\n",
       "      <th>published_date</th>\n",
       "      <th>preprint_abstract</th>\n",
       "      <th>preprint_author_corresponding</th>\n",
       "      <th>preprint_author_corresponding_institution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1101/001081</td>\n",
       "      <td>10.1093/bioinformatics/btu121</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>PyRAD: assembly of de novo RADseq loci for phy...</td>\n",
       "      <td>Deren A. R. Eaton;</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>2013-12-03</td>\n",
       "      <td>2014-03-05</td>\n",
       "      <td>Restriction-site associated genomic markers ar...</td>\n",
       "      <td>Deren A. R. Eaton</td>\n",
       "      <td>University of Chicago</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1101/001297</td>\n",
       "      <td>10.1371/journal.pone.0085203</td>\n",
       "      <td>PLOS ONE</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Aerodynamic characteristics of a feathered din...</td>\n",
       "      <td>Dennis Evangelista;Griselda Cardona;Eric Guent...</td>\n",
       "      <td>Biophysics</td>\n",
       "      <td>2013-12-10</td>\n",
       "      <td>2014-01-15</td>\n",
       "      <td>We report the effects of posture and morpholog...</td>\n",
       "      <td>Dennis  Evangelista</td>\n",
       "      <td>UC Berkeley</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1101/000422</td>\n",
       "      <td>10.3389/fgene.2014.00013</td>\n",
       "      <td>Frontiers in Genetics</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>On the optimal trimming of high-throughput mRN...</td>\n",
       "      <td>Matthew D MacManes;</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>2013-11-14</td>\n",
       "      <td>2014-01-31</td>\n",
       "      <td>The widespread and rapid adoption of high-thro...</td>\n",
       "      <td>Matthew D MacManes</td>\n",
       "      <td>University of New Hampshire</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1101/001396</td>\n",
       "      <td>10.1162/NECO_a_00568</td>\n",
       "      <td>Neural Computation</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Parametric inference in the large data limit u...</td>\n",
       "      <td>Justin B. Kinney;Gurinder S. Atwal;</td>\n",
       "      <td>Biophysics</td>\n",
       "      <td>2013-12-13</td>\n",
       "      <td>2014-03-10</td>\n",
       "      <td>Motivated by data-rich experiments in transcri...</td>\n",
       "      <td>Justin B. Kinney</td>\n",
       "      <td>Cold Spring Harbor Laboratory</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1101/002980</td>\n",
       "      <td>10.1016/j.bpj.2014.01.012</td>\n",
       "      <td>Biophysical Journal</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Genetic drift suppresses bacterial conjugation...</td>\n",
       "      <td>Peter D. Freese;Kirill S. Korolev;Jose I Jimen...</td>\n",
       "      <td>Biophysics</td>\n",
       "      <td>2014-02-24</td>\n",
       "      <td>2014-02-18</td>\n",
       "      <td>Conjugation is the primary mechanism of horizo...</td>\n",
       "      <td>Irene A. Chen</td>\n",
       "      <td>Univ. of California - Santa Barbara</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256833</th>\n",
       "      <td>10.1101/2023.05.11.540369</td>\n",
       "      <td>10.1073/pnas.2309306120</td>\n",
       "      <td>Proceedings of the National Academy of Sciences</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>RAD51-mediated R-loop formation acts to repair...</td>\n",
       "      <td>Girasol, M. J.; Krasilnikova, M.; Marques, C. ...</td>\n",
       "      <td>microbiology</td>\n",
       "      <td>2023-05-11</td>\n",
       "      <td>2023-11-21</td>\n",
       "      <td>RNA-DNA hybrids are epigenetic features of all...</td>\n",
       "      <td>Richard McCulloch</td>\n",
       "      <td>University of Glasgow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256834</th>\n",
       "      <td>10.1101/2023.06.09.544423</td>\n",
       "      <td>10.1002/jez.b.23236</td>\n",
       "      <td>Journal of Experimental Zoology Part B: Molecu...</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>3D spheroid culturing of Astyanax mexicanus li...</td>\n",
       "      <td>Biswas, T.; Rajendran, N.; Hassan, H.; Zhao, C...</td>\n",
       "      <td>evolutionary biology</td>\n",
       "      <td>2023-06-10</td>\n",
       "      <td>2024-01-08</td>\n",
       "      <td>In vitro assays are crucial tools for gaining ...</td>\n",
       "      <td>Nicolas Rohner</td>\n",
       "      <td>Stowers Institute for Medical Research, Univer...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256835</th>\n",
       "      <td>10.1101/2024.01.23.576905</td>\n",
       "      <td>10.1016/j.vaccine.2024.04.073</td>\n",
       "      <td>Vaccine</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Evaluation of Precision of the Plasmodium know...</td>\n",
       "      <td>Mertens, J. E.; Rigby, C. A.; Bardelli, M.; Qu...</td>\n",
       "      <td>immunology</td>\n",
       "      <td>2024-01-24</td>\n",
       "      <td>2024-05-03</td>\n",
       "      <td>Recent data indicate increasing disease burden...</td>\n",
       "      <td>Kazutoyo Miura</td>\n",
       "      <td>NIAID/NIH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256836</th>\n",
       "      <td>10.1101/2023.09.27.557722</td>\n",
       "      <td>10.1371/journal.pbio.3002711</td>\n",
       "      <td>PLOS Biology</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Working together to control mutation: how coll...</td>\n",
       "      <td>Green, R.; Wang, H.; Botchey, C.; Zhang, N.; W...</td>\n",
       "      <td>evolutionary biology</td>\n",
       "      <td>2023-09-27</td>\n",
       "      <td>2024-07-15</td>\n",
       "      <td>Mutagenesis is responsive to many environmenta...</td>\n",
       "      <td>Christopher G. Knight</td>\n",
       "      <td>University of Manchester</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256837</th>\n",
       "      <td>10.1101/2024.07.16.603773</td>\n",
       "      <td>10.3390/molecules29235576</td>\n",
       "      <td>Molecules</td>\n",
       "      <td>bioRxiv</td>\n",
       "      <td>Identification and partial reconstitution of t...</td>\n",
       "      <td>Iacovelli, R.; Poon, F.; Haslinger, K.</td>\n",
       "      <td>microbiology</td>\n",
       "      <td>2024-07-16</td>\n",
       "      <td>2024-11-26</td>\n",
       "      <td>Hericium erinaceus (Lions Mane mushroom) is wi...</td>\n",
       "      <td>Kristina Haslinger</td>\n",
       "      <td>University of Groningen</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>256838 rows √ó 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     preprint_doi                  published_doi  \\\n",
       "0                  10.1101/001081  10.1093/bioinformatics/btu121   \n",
       "1                  10.1101/001297   10.1371/journal.pone.0085203   \n",
       "2                  10.1101/000422       10.3389/fgene.2014.00013   \n",
       "3                  10.1101/001396           10.1162/NECO_a_00568   \n",
       "4                  10.1101/002980      10.1016/j.bpj.2014.01.012   \n",
       "...                           ...                            ...   \n",
       "256833  10.1101/2023.05.11.540369        10.1073/pnas.2309306120   \n",
       "256834  10.1101/2023.06.09.544423            10.1002/jez.b.23236   \n",
       "256835  10.1101/2024.01.23.576905  10.1016/j.vaccine.2024.04.073   \n",
       "256836  10.1101/2023.09.27.557722   10.1371/journal.pbio.3002711   \n",
       "256837  10.1101/2024.07.16.603773      10.3390/molecules29235576   \n",
       "\n",
       "                                        published_journal preprint_platform  \\\n",
       "0                                          Bioinformatics           bioRxiv   \n",
       "1                                                PLOS ONE           bioRxiv   \n",
       "2                                   Frontiers in Genetics           bioRxiv   \n",
       "3                                      Neural Computation           bioRxiv   \n",
       "4                                     Biophysical Journal           bioRxiv   \n",
       "...                                                   ...               ...   \n",
       "256833    Proceedings of the National Academy of Sciences           bioRxiv   \n",
       "256834  Journal of Experimental Zoology Part B: Molecu...           bioRxiv   \n",
       "256835                                            Vaccine           bioRxiv   \n",
       "256836                                       PLOS Biology           bioRxiv   \n",
       "256837                                          Molecules           bioRxiv   \n",
       "\n",
       "                                           preprint_title  \\\n",
       "0       PyRAD: assembly of de novo RADseq loci for phy...   \n",
       "1       Aerodynamic characteristics of a feathered din...   \n",
       "2       On the optimal trimming of high-throughput mRN...   \n",
       "3       Parametric inference in the large data limit u...   \n",
       "4       Genetic drift suppresses bacterial conjugation...   \n",
       "...                                                   ...   \n",
       "256833  RAD51-mediated R-loop formation acts to repair...   \n",
       "256834  3D spheroid culturing of Astyanax mexicanus li...   \n",
       "256835  Evaluation of Precision of the Plasmodium know...   \n",
       "256836  Working together to control mutation: how coll...   \n",
       "256837  Identification and partial reconstitution of t...   \n",
       "\n",
       "                                         preprint_authors  \\\n",
       "0                                      Deren A. R. Eaton;   \n",
       "1       Dennis Evangelista;Griselda Cardona;Eric Guent...   \n",
       "2                                     Matthew D MacManes;   \n",
       "3                     Justin B. Kinney;Gurinder S. Atwal;   \n",
       "4       Peter D. Freese;Kirill S. Korolev;Jose I Jimen...   \n",
       "...                                                   ...   \n",
       "256833  Girasol, M. J.; Krasilnikova, M.; Marques, C. ...   \n",
       "256834  Biswas, T.; Rajendran, N.; Hassan, H.; Zhao, C...   \n",
       "256835  Mertens, J. E.; Rigby, C. A.; Bardelli, M.; Qu...   \n",
       "256836  Green, R.; Wang, H.; Botchey, C.; Zhang, N.; W...   \n",
       "256837             Iacovelli, R.; Poon, F.; Haslinger, K.   \n",
       "\n",
       "           preprint_category preprint_date published_date  \\\n",
       "0             Bioinformatics    2013-12-03     2014-03-05   \n",
       "1                 Biophysics    2013-12-10     2014-01-15   \n",
       "2             Bioinformatics    2013-11-14     2014-01-31   \n",
       "3                 Biophysics    2013-12-13     2014-03-10   \n",
       "4                 Biophysics    2014-02-24     2014-02-18   \n",
       "...                      ...           ...            ...   \n",
       "256833          microbiology    2023-05-11     2023-11-21   \n",
       "256834  evolutionary biology    2023-06-10     2024-01-08   \n",
       "256835            immunology    2024-01-24     2024-05-03   \n",
       "256836  evolutionary biology    2023-09-27     2024-07-15   \n",
       "256837          microbiology    2024-07-16     2024-11-26   \n",
       "\n",
       "                                        preprint_abstract  \\\n",
       "0       Restriction-site associated genomic markers ar...   \n",
       "1       We report the effects of posture and morpholog...   \n",
       "2       The widespread and rapid adoption of high-thro...   \n",
       "3       Motivated by data-rich experiments in transcri...   \n",
       "4       Conjugation is the primary mechanism of horizo...   \n",
       "...                                                   ...   \n",
       "256833  RNA-DNA hybrids are epigenetic features of all...   \n",
       "256834  In vitro assays are crucial tools for gaining ...   \n",
       "256835  Recent data indicate increasing disease burden...   \n",
       "256836  Mutagenesis is responsive to many environmenta...   \n",
       "256837  Hericium erinaceus (Lions Mane mushroom) is wi...   \n",
       "\n",
       "       preprint_author_corresponding  \\\n",
       "0                  Deren A. R. Eaton   \n",
       "1                Dennis  Evangelista   \n",
       "2                 Matthew D MacManes   \n",
       "3                   Justin B. Kinney   \n",
       "4                      Irene A. Chen   \n",
       "...                              ...   \n",
       "256833             Richard McCulloch   \n",
       "256834                Nicolas Rohner   \n",
       "256835                Kazutoyo Miura   \n",
       "256836         Christopher G. Knight   \n",
       "256837            Kristina Haslinger   \n",
       "\n",
       "                preprint_author_corresponding_institution  \n",
       "0                                   University of Chicago  \n",
       "1                                             UC Berkeley  \n",
       "2                             University of New Hampshire  \n",
       "3                           Cold Spring Harbor Laboratory  \n",
       "4                     Univ. of California - Santa Barbara  \n",
       "...                                                   ...  \n",
       "256833                              University of Glasgow  \n",
       "256834  Stowers Institute for Medical Research, Univer...  \n",
       "256835                                          NIAID/NIH  \n",
       "256836                           University of Manchester  \n",
       "256837                            University of Groningen  \n",
       "\n",
       "[256838 rows x 12 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cleaned  = df[df['preprint_abstract'] != 'NULL']\n",
    "#df_cleaned  = df_cleaned[df_cleaned['subtypeDescription'] == 'Article']\n",
    "df_cleaned = df_cleaned.reset_index(drop=True)\n",
    "df_cleaned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2973224d",
   "metadata": {},
   "outputs": [],
   "source": [
    "docs = df_cleaned.preprint_abstract.to_list()\n",
    "\n",
    "df_cleaned['preprint_date'] = pd.to_datetime(df_cleaned['preprint_date'])\n",
    "df_cleaned['year'] = df_cleaned['preprint_date'].dt.year\n",
    "\n",
    "columns_to_combine = ['preprint_authors', 'year', 'preprint_title']\n",
    "df_cleaned['combined'] = df_cleaned[columns_to_combine].apply(lambda row: ', '.join(map(str, row)), axis=1)\n",
    "node_text = df_cleaned.combined.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2aa5f0f7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5f59f270ac8a46e586d10ea316aaa951",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Batches:   0%|          | 0/8027 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[14], line 4\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21;01msentence_transformers\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m SentenceTransformer\n\u001b[0;32m      3\u001b[0m embedding_model \u001b[38;5;241m=\u001b[39m SentenceTransformer(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mBAAI/bge-small-en\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m----> 4\u001b[0m embeddings \u001b[38;5;241m=\u001b[39m \u001b[43membedding_model\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencode\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdocs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshow_progress_bar\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m      6\u001b[0m \u001b[38;5;66;03m# Pre-reduce embeddings for visualization purposes\u001b[39;00m\n\u001b[0;32m      7\u001b[0m \u001b[38;5;66;03m#reduced_embeddings = UMAP(n_neighbors=150, n_components=2, min_dist=0.0, metric='cosine', random_state=42).fit_transform(embeddings)\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\sentence_transformers\\SentenceTransformer.py:623\u001b[0m, in \u001b[0;36mSentenceTransformer.encode\u001b[1;34m(self, sentences, prompt_name, prompt, batch_size, show_progress_bar, output_value, precision, convert_to_numpy, convert_to_tensor, device, normalize_embeddings, **kwargs)\u001b[0m\n\u001b[0;32m    620\u001b[0m features\u001b[38;5;241m.\u001b[39mupdate(extra_features)\n\u001b[0;32m    622\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m torch\u001b[38;5;241m.\u001b[39mno_grad():\n\u001b[1;32m--> 623\u001b[0m     out_features \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mforward\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfeatures\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    624\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdevice\u001b[38;5;241m.\u001b[39mtype \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhpu\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    625\u001b[0m         out_features \u001b[38;5;241m=\u001b[39m copy\u001b[38;5;241m.\u001b[39mdeepcopy(out_features)\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\sentence_transformers\\SentenceTransformer.py:690\u001b[0m, in \u001b[0;36mSentenceTransformer.forward\u001b[1;34m(self, input, **kwargs)\u001b[0m\n\u001b[0;32m    688\u001b[0m     module_kwarg_keys \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodule_kwargs\u001b[38;5;241m.\u001b[39mget(module_name, [])\n\u001b[0;32m    689\u001b[0m     module_kwargs \u001b[38;5;241m=\u001b[39m {key: value \u001b[38;5;28;01mfor\u001b[39;00m key, value \u001b[38;5;129;01min\u001b[39;00m kwargs\u001b[38;5;241m.\u001b[39mitems() \u001b[38;5;28;01mif\u001b[39;00m key \u001b[38;5;129;01min\u001b[39;00m module_kwarg_keys}\n\u001b[1;32m--> 690\u001b[0m     \u001b[38;5;28minput\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[43mmodule\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mmodule_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    691\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28minput\u001b[39m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\sentence_transformers\\models\\Transformer.py:393\u001b[0m, in \u001b[0;36mTransformer.forward\u001b[1;34m(self, features, **kwargs)\u001b[0m\n\u001b[0;32m    390\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtoken_type_ids\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01min\u001b[39;00m features:\n\u001b[0;32m    391\u001b[0m     trans_features[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtoken_type_ids\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m features[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtoken_type_ids\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m--> 393\u001b[0m output_states \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mauto_model\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mtrans_features\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mreturn_dict\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m    394\u001b[0m output_tokens \u001b[38;5;241m=\u001b[39m output_states[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m    396\u001b[0m \u001b[38;5;66;03m# If the AutoModel is wrapped with a PeftModelForFeatureExtraction, then it may have added virtual tokens\u001b[39;00m\n\u001b[0;32m    397\u001b[0m \u001b[38;5;66;03m# We need to extend the attention mask to include these virtual tokens, or the pooling will fail\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\transformers\\models\\bert\\modeling_bert.py:1142\u001b[0m, in \u001b[0;36mBertModel.forward\u001b[1;34m(self, input_ids, attention_mask, token_type_ids, position_ids, head_mask, inputs_embeds, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[0;32m   1135\u001b[0m \u001b[38;5;66;03m# Prepare head mask if needed\u001b[39;00m\n\u001b[0;32m   1136\u001b[0m \u001b[38;5;66;03m# 1.0 in head_mask indicate we keep the head\u001b[39;00m\n\u001b[0;32m   1137\u001b[0m \u001b[38;5;66;03m# attention_probs has shape bsz x n_heads x N x N\u001b[39;00m\n\u001b[0;32m   1138\u001b[0m \u001b[38;5;66;03m# input head_mask has shape [num_heads] or [num_hidden_layers x num_heads]\u001b[39;00m\n\u001b[0;32m   1139\u001b[0m \u001b[38;5;66;03m# and head_mask is converted to shape [num_hidden_layers x batch x num_heads x seq_length x seq_length]\u001b[39;00m\n\u001b[0;32m   1140\u001b[0m head_mask \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mget_head_mask(head_mask, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mconfig\u001b[38;5;241m.\u001b[39mnum_hidden_layers)\n\u001b[1;32m-> 1142\u001b[0m encoder_outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoder\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   1143\u001b[0m \u001b[43m    \u001b[49m\u001b[43membedding_output\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1144\u001b[0m \u001b[43m    \u001b[49m\u001b[43mattention_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mextended_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1145\u001b[0m \u001b[43m    \u001b[49m\u001b[43mhead_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhead_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1146\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoder_hidden_states\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencoder_hidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1147\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoder_attention_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencoder_extended_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1148\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpast_key_values\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpast_key_values\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1149\u001b[0m \u001b[43m    \u001b[49m\u001b[43muse_cache\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43muse_cache\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1150\u001b[0m \u001b[43m    \u001b[49m\u001b[43moutput_attentions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moutput_attentions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1151\u001b[0m \u001b[43m    \u001b[49m\u001b[43moutput_hidden_states\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moutput_hidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1152\u001b[0m \u001b[43m    \u001b[49m\u001b[43mreturn_dict\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mreturn_dict\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1153\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1154\u001b[0m sequence_output \u001b[38;5;241m=\u001b[39m encoder_outputs[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m   1155\u001b[0m pooled_output \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpooler(sequence_output) \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpooler \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\transformers\\models\\bert\\modeling_bert.py:695\u001b[0m, in \u001b[0;36mBertEncoder.forward\u001b[1;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_values, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[0;32m    684\u001b[0m     layer_outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_gradient_checkpointing_func(\n\u001b[0;32m    685\u001b[0m         layer_module\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__call__\u001b[39m,\n\u001b[0;32m    686\u001b[0m         hidden_states,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    692\u001b[0m         output_attentions,\n\u001b[0;32m    693\u001b[0m     )\n\u001b[0;32m    694\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 695\u001b[0m     layer_outputs \u001b[38;5;241m=\u001b[39m \u001b[43mlayer_module\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    696\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    697\u001b[0m \u001b[43m        \u001b[49m\u001b[43mattention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    698\u001b[0m \u001b[43m        \u001b[49m\u001b[43mlayer_head_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    699\u001b[0m \u001b[43m        \u001b[49m\u001b[43mencoder_hidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    700\u001b[0m \u001b[43m        \u001b[49m\u001b[43mencoder_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    701\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpast_key_value\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    702\u001b[0m \u001b[43m        \u001b[49m\u001b[43moutput_attentions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    703\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    705\u001b[0m hidden_states \u001b[38;5;241m=\u001b[39m layer_outputs[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m    706\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m use_cache:\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\transformers\\models\\bert\\modeling_bert.py:585\u001b[0m, in \u001b[0;36mBertLayer.forward\u001b[1;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[0;32m    573\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mforward\u001b[39m(\n\u001b[0;32m    574\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m    575\u001b[0m     hidden_states: torch\u001b[38;5;241m.\u001b[39mTensor,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    582\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Tuple[torch\u001b[38;5;241m.\u001b[39mTensor]:\n\u001b[0;32m    583\u001b[0m     \u001b[38;5;66;03m# decoder uni-directional self-attention cached key/values tuple is at positions 1,2\u001b[39;00m\n\u001b[0;32m    584\u001b[0m     self_attn_past_key_value \u001b[38;5;241m=\u001b[39m past_key_value[:\u001b[38;5;241m2\u001b[39m] \u001b[38;5;28;01mif\u001b[39;00m past_key_value \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m--> 585\u001b[0m     self_attention_outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mattention\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    586\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    587\u001b[0m \u001b[43m        \u001b[49m\u001b[43mattention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    588\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhead_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    589\u001b[0m \u001b[43m        \u001b[49m\u001b[43moutput_attentions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43moutput_attentions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    590\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpast_key_value\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mself_attn_past_key_value\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    591\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    592\u001b[0m     attention_output \u001b[38;5;241m=\u001b[39m self_attention_outputs[\u001b[38;5;241m0\u001b[39m]\n\u001b[0;32m    594\u001b[0m     \u001b[38;5;66;03m# if decoder, the last output is tuple of self-attn cache\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\transformers\\models\\bert\\modeling_bert.py:515\u001b[0m, in \u001b[0;36mBertAttention.forward\u001b[1;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[0;32m    505\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;21mforward\u001b[39m(\n\u001b[0;32m    506\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m    507\u001b[0m     hidden_states: torch\u001b[38;5;241m.\u001b[39mTensor,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    513\u001b[0m     output_attentions: Optional[\u001b[38;5;28mbool\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    514\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Tuple[torch\u001b[38;5;241m.\u001b[39mTensor]:\n\u001b[1;32m--> 515\u001b[0m     self_outputs \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mself\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    516\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    517\u001b[0m \u001b[43m        \u001b[49m\u001b[43mattention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    518\u001b[0m \u001b[43m        \u001b[49m\u001b[43mhead_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    519\u001b[0m \u001b[43m        \u001b[49m\u001b[43mencoder_hidden_states\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    520\u001b[0m \u001b[43m        \u001b[49m\u001b[43mencoder_attention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    521\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpast_key_value\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    522\u001b[0m \u001b[43m        \u001b[49m\u001b[43moutput_attentions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    523\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    524\u001b[0m     attention_output \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moutput(self_outputs[\u001b[38;5;241m0\u001b[39m], hidden_states)\n\u001b[0;32m    525\u001b[0m     outputs \u001b[38;5;241m=\u001b[39m (attention_output,) \u001b[38;5;241m+\u001b[39m self_outputs[\u001b[38;5;241m1\u001b[39m:]  \u001b[38;5;66;03m# add attentions if we output them\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1736\u001b[0m, in \u001b[0;36mModule._wrapped_call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1734\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_compiled_call_impl(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)  \u001b[38;5;66;03m# type: ignore[misc]\u001b[39;00m\n\u001b[0;32m   1735\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m-> 1736\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_call_impl\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\torch\\nn\\modules\\module.py:1747\u001b[0m, in \u001b[0;36mModule._call_impl\u001b[1;34m(self, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1742\u001b[0m \u001b[38;5;66;03m# If we don't have any hooks, we want to skip the rest of the logic in\u001b[39;00m\n\u001b[0;32m   1743\u001b[0m \u001b[38;5;66;03m# this function, and just call forward.\u001b[39;00m\n\u001b[0;32m   1744\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_forward_pre_hooks\n\u001b[0;32m   1745\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_backward_pre_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_backward_hooks\n\u001b[0;32m   1746\u001b[0m         \u001b[38;5;129;01mor\u001b[39;00m _global_forward_hooks \u001b[38;5;129;01mor\u001b[39;00m _global_forward_pre_hooks):\n\u001b[1;32m-> 1747\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mforward_call\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1749\u001b[0m result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1750\u001b[0m called_always_called_hooks \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n",
      "File \u001b[1;32mc:\\Users\\jacob\\anaconda3\\envs\\test\\Lib\\site-packages\\transformers\\models\\bert\\modeling_bert.py:440\u001b[0m, in \u001b[0;36mBertSdpaSelfAttention.forward\u001b[1;34m(self, hidden_states, attention_mask, head_mask, encoder_hidden_states, encoder_attention_mask, past_key_value, output_attentions)\u001b[0m\n\u001b[0;32m    432\u001b[0m \u001b[38;5;66;03m# We dispatch to SDPA's Flash Attention or Efficient kernels via this `is_causal` if statement instead of an inline conditional assignment\u001b[39;00m\n\u001b[0;32m    433\u001b[0m \u001b[38;5;66;03m# in SDPA to support both torch.compile's dynamic shapes and full graph options. An inline conditional prevents dynamic shapes from compiling.\u001b[39;00m\n\u001b[0;32m    434\u001b[0m \u001b[38;5;66;03m# The tgt_len > 1 is necessary to match with AttentionMaskConverter.to_causal_4d that does not create\u001b[39;00m\n\u001b[0;32m    435\u001b[0m \u001b[38;5;66;03m# a causal mask in case tgt_len == 1.\u001b[39;00m\n\u001b[0;32m    436\u001b[0m is_causal \u001b[38;5;241m=\u001b[39m (\n\u001b[0;32m    437\u001b[0m     \u001b[38;5;28;01mTrue\u001b[39;00m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mis_decoder \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m is_cross_attention \u001b[38;5;129;01mand\u001b[39;00m attention_mask \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mand\u001b[39;00m tgt_len \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[0;32m    438\u001b[0m )\n\u001b[1;32m--> 440\u001b[0m attn_output \u001b[38;5;241m=\u001b[39m \u001b[43mtorch\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfunctional\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mscaled_dot_product_attention\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    441\u001b[0m \u001b[43m    \u001b[49m\u001b[43mquery_layer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    442\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkey_layer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    443\u001b[0m \u001b[43m    \u001b[49m\u001b[43mvalue_layer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    444\u001b[0m \u001b[43m    \u001b[49m\u001b[43mattn_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mattention_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    445\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdropout_p\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdropout_prob\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtraining\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;241;43m0.0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    446\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_causal\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_causal\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    447\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    449\u001b[0m attn_output \u001b[38;5;241m=\u001b[39m attn_output\u001b[38;5;241m.\u001b[39mtranspose(\u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m2\u001b[39m)\n\u001b[0;32m    450\u001b[0m attn_output \u001b[38;5;241m=\u001b[39m attn_output\u001b[38;5;241m.\u001b[39mreshape(bsz, tgt_len, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mall_head_size)\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Pre-calculate embeddings\n",
    "from sentence_transformers import SentenceTransformer\n",
    "embedding_model = SentenceTransformer('BAAI/bge-small-en')\n",
    "embeddings = embedding_model.encode(docs, show_progress_bar=True)\n",
    "\n",
    "# Pre-reduce embeddings for visualization purposes\n",
    "#reduced_embeddings = UMAP(n_neighbors=150, n_components=2, min_dist=0.0, metric='cosine', random_state=42).fit_transform(embeddings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8be9bd8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define sub-models\n",
    "from hdbscan import HDBSCAN\n",
    "from umap import UMAP\n",
    "umap_model = UMAP(n_neighbors=100, n_components=2, min_dist=0.0, metric='cosine', random_state=42)\n",
    "hdbscan_model = HDBSCAN(min_cluster_size=100, metric='euclidean', cluster_selection_method='eom', prediction_data=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8ea5001",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'embeddings' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[13], line 26\u001b[0m\n\u001b[0;32m     12\u001b[0m topic_model \u001b[38;5;241m=\u001b[39m BERTopic(\n\u001b[0;32m     13\u001b[0m   \u001b[38;5;66;03m# Sub-models\u001b[39;00m\n\u001b[0;32m     14\u001b[0m   embedding_model\u001b[38;5;241m=\u001b[39membedding_model,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     22\u001b[0m   verbose\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m     23\u001b[0m )\n\u001b[0;32m     25\u001b[0m \u001b[38;5;66;03m# Train model\u001b[39;00m\n\u001b[1;32m---> 26\u001b[0m topics, probs \u001b[38;5;241m=\u001b[39m topic_model\u001b[38;5;241m.\u001b[39mfit_transform(docs, \u001b[43membeddings\u001b[49m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'embeddings' is not defined"
     ]
    }
   ],
   "source": [
    "from bertopic import BERTopic\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "\n",
    "# Define a custom list of stopwords (or use an extended one)\n",
    "custom_stopwords = [\"the\", \"and\", \"or\", \"in\", \"on\", \"at\", \"of\", \"is\", \"to\", \"be\", \"with\", \"are\", \"that\", \"this\", \"by\", \"for\"]\n",
    "\n",
    "# Create a CountVectorizer with the custom stopwords\n",
    "vectorizer = CountVectorizer(stop_words=custom_stopwords)\n",
    "\n",
    "\n",
    "\n",
    "topic_model = BERTopic(\n",
    "  # Sub-models\n",
    "  embedding_model=embedding_model,\n",
    "  umap_model=umap_model,\n",
    "  hdbscan_model=hdbscan_model,\n",
    "  #representation_model=representation_model,\n",
    "  vectorizer_model=vectorizer,\n",
    "  \n",
    "  # Hyperparameters\n",
    "  top_n_words=10,\n",
    "  verbose=True\n",
    ")\n",
    "\n",
    "# Train model\n",
    "topics, probs = topic_model.fit_transform(docs, embeddings)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "test",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
